diff -Nur fbreader-0.8.2a.orig/makefiles/arch/openzaurus.mk fbreader-0.8.2a/makefiles/arch/openzaurus.mk
--- fbreader-0.8.2a.orig/makefiles/arch/openzaurus.mk	2007-04-17 18:53:46.000000000 +0000
+++ fbreader-0.8.2a/makefiles/arch/openzaurus.mk	2007-04-24 00:57:17.000000000 +0000
@@ -17,7 +17,7 @@
 RM = rm -rvf
 RM_QUIET = rm -rf
 QTINCLUDE = -I $(INCPATH)/qt2
-GTKINCLUDE = -I$(LIBPATH)/glib-2.0/include -I$(LIBPATH)/gtk-2.0/include -I$(INCPATH)/glib-2.0 -I$(INCPATH)/gtk-2.0 -I$(INCPATH)/pango-1.0 -I$(INCPATH)/atk-1.0
+GTKINCLUDE = -I$(LIBPATH)/glib-2.0/include -I$(LIBPATH)/gtk-2.0/include -I$(INCPATH)/cairo -I$(INCPATH)/glib-2.0 -I$(INCPATH)/gtk-2.0 -I$(INCPATH)/pango-1.0 -I$(INCPATH)/atk-1.0
 MOC = $(TOOLSPATH)/moc
 CC = $(TOOLSPATH)/arm-linux-gcc
 AR = $(TOOLSPATH)/arm-linux-ar rsu
@@ -28,10 +28,10 @@
 ARCHIVER_LIB = -lbz2
 
 ifeq "$(UI_TYPE)" "opie"
-	UILIBS = -lqte -lqpe
+	UILIBS = -lqt-mt
 else
-	UILIBS = -lgpewidget -lgtk-x11-2.0 -lgdk-x11-2.0 -lgdk_pixbuf-2.0
+	UILIBS = $(shell pkg-config --libs gtk+-2.0) -lgpewidget
 endif
 
-CFLAGS = -pipe -DOPIE_NO_DEBUG -DQT_NO_DEBUG -DQWS -fno-exceptions -fno-rtti -march=armv4 -mtune=xscale --param large-function-growth=2000 --param inline-unit-growth=200 -Wall -Wno-ctor-dtor-privacy -W -Winline
-LDFLAGS = -Wl,-rpath,$(LIBDIR)
+CFLAGS = -pipe -DOPIE_NO_DEBUG -DQT_NO_DEBUG -DQWS -fno-exceptions -fno-rtti $(OE_CFLAGS)
+LDFLAGS = -Wl,-rpath,$(LIBDIR) -Wl,-rpath,$(LIBPATH)
diff -Nur fbreader-0.8.2a.orig/makefiles/target.mk fbreader-0.8.2a/makefiles/target.mk
--- fbreader-0.8.2a.orig/makefiles/target.mk	2007-04-17 18:53:46.000000000 +0000
+++ fbreader-0.8.2a/makefiles/target.mk	2007-04-24 00:58:30.000000000 +0000
@@ -1,6 +1,5 @@
-#TARGET_ARCH = openzaurus
-#UI_TYPE = opie
-#UI_TYPE = gpe
+TARGET_ARCH = openzaurus
+UI_TYPE = gpe
 
 #TARGET_ARCH = opensimpad-0.9.0
 #UI_TYPE = opie
@@ -17,8 +16,8 @@
 #UI_TYPE = qt
 #UI_TYPE = qt4
 
-TARGET_ARCH = win32
-UI_TYPE = win32
+#TARGET_ARCH = win32
+#UI_TYPE = win32
 
 #TARGET_ARCH = pocketpc
 #UI_TYPE = win32
