--- original/drivers/i2c/chips/x1205-rtc.c	Sat Mar  5 23:41:12 2005
+++ linux-2.6.11/drivers/i2c/chips/x1205-rtc.c	Sat Mar  5 23:41:25 2005
@@ -139,7 +139,6 @@
 };
 
 static struct i2c_client x1205_i2c_client = {
-	.id		=	I2C_DRIVERID_X1205,
 	.flags		=	0,
 	.addr		=	X1205_I2C_BUS_ADDR,	// chip address - NOTE: 7bit
 	.adapter	=	NULL,			// the adapter we sit on assigned in attach
@@ -301,7 +300,6 @@
 	int errno;
 		
 	x1205_i2c_client.adapter = adapter;
-	x1205_i2c_client.id++;
 
 	if ((x1205_get_datetime(&x1205_i2c_client, &tm, X1205_CCR_BASE)) != NOERR)	//test for functional driver 
 		return -EIO;
