Index: fuse-2.7.1/util/Makefile.am
===================================================================
--- fuse-2.7.1.orig/util/Makefile.am	2007-04-25 18:06:47.000000000 +0200
+++ fuse-2.7.1/util/Makefile.am	2007-11-16 16:57:29.000000000 +0100
@@ -33,10 +33,6 @@
 	$(INSTALL_PROGRAM) $(srcdir)/mount.fuse $(DESTDIR)$(MOUNT_FUSE_PATH)/mount.fuse
 	$(mkdir_p) $(DESTDIR)$(INIT_D_PATH)
 	$(INSTALL_PROGRAM) $(srcdir)/init_script $(DESTDIR)$(INIT_D_PATH)/fuse
-	@if test -x /usr/sbin/update-rc.d; then \
-		echo "/usr/sbin/update-rc.d fuse start 34 S . start 41 0 6 . || true"; \
-		/usr/sbin/update-rc.d fuse start 34 S . start 41 0 6 . || true; \
-	fi
 
 install-data-local:
 	$(mkdir_p) $(DESTDIR)$(UDEV_RULES_PATH)
@@ -46,7 +43,3 @@
 	rm -f $(DESTDIR)$(MOUNT_FUSE_PATH)/mount.fuse
 	rm -f $(DESTDIR)$(UDEV_RULES_PATH)/99-fuse.rules
 	rm -f $(DESTDIR)$(INIT_D_PATH)/fuse
-	@if test -x /usr/sbin/update-rc.d; then \
-		echo "/usr/sbin/update-rc.d fuse remove || true"; \
-		/usr/sbin/update-rc.d fuse remove || true; \
-	fi
