
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- qt-x11-opensource-src-4.1.1/src/src.pro~no-tools
+++ qt-x11-opensource-src-4.1.1/src/src.pro
@@ -3,7 +3,7 @@
 
 # this order is important
 win32:SUBDIRS += winmain
-SUBDIRS += tools/moc tools/rcc tools/uic corelib xml gui sql network svg
+SUBDIRS += corelib xml gui sql network svg
 !embedded:contains(QT_CONFIG, opengl): SUBDIRS += opengl
 contains(QT_CONFIG, qt3support): SUBDIRS += qt3support
 !cross_compile {
