--- /tmp/configure.in	2008-09-05 14:01:25.000000000 +0200
+++ eel-2.22.2/configure.in	2008-09-05 14:01:58.693198000 +0200
@@ -98,26 +98,6 @@
 
 
 
-dnl ===========================================================================
-
-dnl strftime checks
-
-AC_TRY_RUN([#include <time.h>
-                int main ()
-                {
-                  char buf[100];
-                  struct tm tm = {0};
-                  tm.tm_year = 99;
-                  if (strftime(buf, 100, "%EY", &tm) == 4 &&
-		      strcmp (buf, "1999")==0)
-                    return 0;
-                  return 1;
-                }
-            ],
-	    AC_DEFINE(HAVE_STRFTIME_EXTENSION, 1, [Define if strftime supports %E and %O modifiers.])
-            )
-
-dnl ===========================================================================
 
 dnl X development libraries check
 
