diff -Nur esmart-0.9.0.004~/src/bin/esmart_file_dialog_test.c esmart-0.9.0.004/src/bin/esmart_file_dialog_test.c
--- esmart-0.9.0.004~/src/bin/esmart_file_dialog_test.c	2005-09-29 14:33:09.000000000 -0700
+++ esmart-0.9.0.004/src/bin/esmart_file_dialog_test.c	2005-09-29 14:33:48.000000000 -0700
@@ -8,6 +8,7 @@
  * $ esmart_file_dialog_test /path/to/my_custom/theme.eet 
  * to test a custom theme you've written for it
  */
+#include <stdio.h>
 #include <Ecore.h>
 #include <Edje.h>
 #include <Ecore_Evas.h>
diff -Nur esmart-0.9.0.004~/src/lib/esmart_trans_x11/esmart_trans_x11.c esmart-0.9.0.004/src/lib/esmart_trans_x11/esmart_trans_x11.c
--- esmart-0.9.0.004~/src/lib/esmart_trans_x11/esmart_trans_x11.c	2005-09-29 14:33:09.000000000 -0700
+++ esmart-0.9.0.004/src/lib/esmart_trans_x11/esmart_trans_x11.c	2005-09-29 14:33:19.000000000 -0700
@@ -4,6 +4,7 @@
  *
  ***************************************************************************/
 
+#include <stdio.h>
 #include <stdlib.h>
 #include "config.h"
 #include <X11/Xlib.h>
