diff -Naur blueprobe-0.15_orig/blueprobe.init blueprobe-0.15_h6300_fix/blueprobe.init
--- blueprobe-0.15_orig/blueprobe.init	2005-10-05 22:37:35.706302041 +0300
+++ blueprobe-0.15_h6300_fix/blueprobe.init	2005-10-05 22:54:51.840676832 +0300
@@ -57,6 +57,13 @@
 	SCRIPT=/etc/bluetooth/TIInit_3.2.26.bts
 	PROBE=no
 	;;
+    "HP iPAQ H6300")
+	BLUETOOTH=yes
+	PORT=/dev/ttyS0
+	SPEED=115200
+	PROTO=texas
+	PROBE=no
+	;;
     *)
 	BLUETOOTH=no
         ;;
