--- galculator-1.2.3/src/Makefile.am	2004-03-20 19:58:18.000000000 +0100
+++ galculator-1.2.3-oe/src/Makefile.am	2004-05-26 15:35:46.000000000 +0200
@@ -1,9 +1,8 @@
 ## Process this file with automake to produce Makefile.in
 #	-DPACKAGE_DATA_DIR=\""$(datadir)"\" \
-#	\
+#	
 
-INCLUDES = \
-	-DPACKAGE_LOCALE_DIR=\""$(datadir)/locale"\" \
+INCLUDES = -DPACKAGE_LOCALE_DIR=\""$(datadir)/locale"\" \
 	-DPACKAGE_GLADE_DIR=\""$(datadir)/galculator/glade"\" \
 	@PACKAGE_CFLAGS@
 
