diff -Naur plugins/configure.ac blasel/configure.ac
--- plugins/configure.ac	2005-02-04 18:27:58.000000000 +0100
+++ blasel/configure.ac	2005-03-02 21:31:09.000000000 +0100
@@ -27,7 +27,7 @@
 TUXBOX_APPS_LIB_PKGCONFIG(XMLTREE,tuxbox-xmltree)
 TUXBOX_APPS_LIB_PKGCONFIG(CONFIGFILE,tuxbox-configfile)
 
-CPPFLAGS="$CPPFLAGS"
+CPPFLAGS="$CPPFLAGS -DHAVE_DREAMBOX_HARDWARE -DDREAMBOX"
 
 AC_OUTPUT([
 Makefile
