Index: fx2/master/Makefile.am
===================================================================
RCS file: /cvs/tuxbox/apps/tuxbox/plugins/fx2/master/Makefile.am,v
retrieving revision 1.5
diff -u -b -B -w -p -r1.5 Makefile.am
--- a/fx2/master/Makefile.am	4 Sep 2004 14:49:40 -0000	1.5
+++ b/fx2/master/Makefile.am	1 Nov 2004 14:47:03 -0000
@@ -6,7 +6,7 @@ master_la_SOURCES = board.c somain.c
 
 master_la_LIBADD = ../lib/libfx2.la
 
-master_la_LDFLAGS = -rpath $(PLUGINDIR) -R$(TARGET_PLUGINDIR) -module -avoid-version
+master_la_LDFLAGS = -rpath $(PLUGINDIR) -module -avoid-version
 
 install-exec-local:
 	install -d $(DESTDIR)$(PLUGINDIR)
Index: fx2/mines/Makefile.am
===================================================================
RCS file: /cvs/tuxbox/apps/tuxbox/plugins/fx2/mines/Makefile.am,v
retrieving revision 1.15
diff -u -b -B -w -p -r1.15 Makefile.am
--- a/fx2/mines/Makefile.am	4 Sep 2004 14:49:40 -0000	1.15
+++ b/fx2/mines/Makefile.am	1 Nov 2004 14:47:03 -0000
@@ -6,7 +6,7 @@ mines_la_SOURCES = board.c somain.c
 
 mines_la_LIBADD = ../lib/libfx2.la
 
-mines_la_LDFLAGS = -rpath $(PLUGINDIR) -R$(TARGET_PLUGINDIR) -module -avoid-version
+mines_la_LDFLAGS = -rpath $(PLUGINDIR) -module -avoid-version
 
 install-exec-local:
 	install -d $(DESTDIR)$(PLUGINDIR)
Index: fx2/satfind/Makefile.am
===================================================================
RCS file: /cvs/tuxbox/apps/tuxbox/plugins/fx2/satfind/Makefile.am,v
retrieving revision 1.4
diff -u -b -B -w -p -r1.4 Makefile.am
--- a/fx2/satfind/Makefile.am	4 Sep 2004 14:49:40 -0000	1.4
+++ b/fx2/satfind/Makefile.am	1 Nov 2004 14:47:03 -0000
@@ -6,7 +6,7 @@ satfind_la_SOURCES = satfind.c
 
 satfind_la_LIBADD = ../lib/libfx2.la
 
-satfind_la_LDFLAGS = -rpath $(PLUGINDIR) -R$(TARGET_PLUGINDIR) -module -avoid-version
+satfind_la_LDFLAGS = -rpath $(PLUGINDIR) -module -avoid-version
 
 install-exec-local:
 	install -d $(DESTDIR)$(PLUGINDIR)
Index: fx2/snake/Makefile.am
===================================================================
RCS file: /cvs/tuxbox/apps/tuxbox/plugins/fx2/snake/Makefile.am,v
retrieving revision 1.8
diff -u -b -B -w -p -r1.8 Makefile.am
--- a/fx2/snake/Makefile.am	4 Sep 2004 14:49:40 -0000	1.8
+++ b/fx2/snake/Makefile.am	1 Nov 2004 14:47:03 -0000
@@ -6,7 +6,7 @@ snake_la_SOURCES = snake.c somain.c
 
 snake_la_LIBADD = ../lib/libfx2.la
 
-snake_la_LDFLAGS = -rpath $(PLUGINDIR) -R$(TARGET_PLUGINDIR) -module -avoid-version
+snake_la_LDFLAGS = -rpath $(PLUGINDIR) -module -avoid-version
 
 install-exec-local:
 	install -d $(DESTDIR)$(PLUGINDIR)
Index: fx2/sokoban/Makefile.am
===================================================================
RCS file: /cvs/tuxbox/apps/tuxbox/plugins/fx2/sokoban/Makefile.am,v
retrieving revision 1.4
diff -u -b -B -w -p -r1.4 Makefile.am
--- a/fx2/sokoban/Makefile.am	4 Sep 2004 14:49:41 -0000	1.4
+++ b/fx2/sokoban/Makefile.am	1 Nov 2004 14:47:03 -0000
@@ -6,7 +6,7 @@ soko_la_SOURCES = board.c somain.c
 
 soko_la_LIBADD = ../lib/libfx2.la
 
-soko_la_LDFLAGS = -rpath $(PLUGINDIR) -R$(TARGET_PLUGINDIR) -module -avoid-version
+soko_la_LDFLAGS = -rpath $(PLUGINDIR) -module -avoid-version
 
 install-exec-local:
 	install -d $(DESTDIR)$(PLUGINDIR)
Index: fx2/tetris/Makefile.am
===================================================================
RCS file: /cvs/tuxbox/apps/tuxbox/plugins/fx2/tetris/Makefile.am,v
retrieving revision 1.24
diff -u -b -B -w -p -r1.24 Makefile.am
--- a/fx2/tetris/Makefile.am	4 Sep 2004 14:49:41 -0000	1.24
+++ b/fx2/tetris/Makefile.am	1 Nov 2004 14:47:03 -0000
@@ -9,7 +9,7 @@ tetris_la_LIBADD =       \
 	@CURL_LIBS@
 
 
-tetris_la_LDFLAGS = -rpath $(PLUGINDIR) -R$(TARGET_PLUGINDIR) -module -avoid-version
+tetris_la_LDFLAGS = -rpath $(PLUGINDIR) -module -avoid-version
 
 install-exec-local:
 	install -d $(DESTDIR)$(PLUGINDIR)
Index: fx2/vierg/Makefile.am
===================================================================
RCS file: /cvs/tuxbox/apps/tuxbox/plugins/fx2/vierg/Makefile.am,v
retrieving revision 1.5
diff -u -b -B -w -p -r1.5 Makefile.am
--- a/fx2/vierg/Makefile.am	4 Sep 2004 14:49:41 -0000	1.5
+++ b/fx2/vierg/Makefile.am	1 Nov 2004 14:47:03 -0000
@@ -6,7 +6,7 @@ vierg_la_SOURCES = board.c somain.c
 
 vierg_la_LIBADD = ../lib/libfx2.la
 
-vierg_la_LDFLAGS = -rpath $(PLUGINDIR) -R$(TARGET_PLUGINDIR) -module -avoid-version
+vierg_la_LDFLAGS = -rpath $(PLUGINDIR) -module -avoid-version
 
 install-exec-local:
 	install -d $(DESTDIR)$(PLUGINDIR)
diff -Naur plugins/Makefile.am patched/Makefile.am
--- a/Makefile.am	2005-02-04 18:27:58.000000000 +0100
+++ b/Makefile.am	2005-03-02 21:35:47.000000000 +0100
@@ -2,7 +2,7 @@
 
 if BOXTYPE_DREAMBOX
 SUBDIRS = \
-	include tuxmail tuxtxt tuxcom fx2 enigma
+	include tuxmail tuxtxt tuxcom fx2
 else
 SUBDIRS = \
 	include tuxmail tuxtxt tuxcom fx2 vncviewer enigma
