diff -Nurd atmelwlandriver-3.4.1.0/src/usb/vnetusba.c atmelwlandriver-3.4.1.0-hh/src/usb/vnetusba.c
--- atmelwlandriver-3.4.1.0/src/usb/vnetusba.c	2005-12-02 10:25:21.946561154 -0800
+++ atmelwlandriver-3.4.1.0-hh/src/usb/vnetusba.c	2005-12-19 11:34:38.070346125 -0800
@@ -72,7 +72,7 @@
 #endif
 
 struct usb_driver vnet_driver = {
-#if LINUX_VERSION_CODE < 0x020415
+#if (LINUX_VERSION_CODE < 0x020415) && !defined(CONFIG_IPAQ_HANDHELD)
 owner : THIS_MODULE,
 #endif
 #if defined(RFMD)
@@ -2174,7 +2174,7 @@
 		return NULL;
 	}
   */
-#if (LINUX_VERSION_CODE < 0x020413)
+#if (LINUX_VERSION_CODE < 0x020413) && !defined(CONFIG_IPAQ_HANDHELD)
 	if (!register_netdev(Adapter->net)) {
 		err("register_netdev failed");
 #else
