Index: FS/fs_trans.h
===================================================================
--- FS.orig/fs_trans.h	2011-02-25 23:00:34.025908119 +0100
+++ FS/fs_trans.h	2011-02-25 23:01:19.165133120 +0100
@@ -24,4 +24,4 @@
 
 #define FS_t
 #define TRANS_CLIENT
-#include <Xtrans.h>
+#include <X11/Xtrans/Xtrans.h>
Index: FS/fs_trans.c
===================================================================
--- FS.orig/fs_trans.c	2011-02-25 23:08:51.177693119 +0100
+++ FS/fs_trans.c	2011-02-25 23:09:00.378958118 +0100
@@ -28,5 +28,5 @@
 
 #define FS_t
 #define TRANS_CLIENT
-#include <transport.c>
+#include <X11/Xtrans/transport.c>
 
