Index: launcher/launcher.cpp
===================================================================
--- launcher.orig/launcher.cpp	2009-12-23 22:29:17.396221157 +0100
+++ launcher/launcher.cpp	2009-12-23 22:31:45.849117517 +0100
@@ -646,10 +646,10 @@
     // Not supported: flat is simpler for the user
     } else {
 /* ### libqtopia FIXME also moving docLnks... */
-        LnkProperties prop(appLnk,0 );
+    /*    LnkProperties prop(appLnk,0 );
 
         if (QPEApplication::execDialog( &prop )==QDialog::Accepted && tabs->currentView()==tabs->docView()) {
-        }
+        }*/
     }
 }
 
