Index: linux-2.6.27/arch/powerpc/boot/dts/mpc8313erdb.dts
===================================================================
--- linux-2.6.27.orig/arch/powerpc/boot/dts/mpc8313erdb.dts	2008-12-19 09:40:44.000000000 +0100
+++ linux-2.6.27/arch/powerpc/boot/dts/mpc8313erdb.dts	2008-12-19 09:42:47.000000000 +0100
@@ -18,10 +18,8 @@
 	#size-cells = <1>;
 
 	aliases {
-		ethernet0 = &enet0;
 		ethernet1 = &enet1;
 		serial0 = &serial0;
-		serial1 = &serial1;
 		pci0 = &pci0;
 	};
 
@@ -79,18 +77,18 @@
 			             "fsl,elbc-fcm-nand";
 			reg = <0x1 0x0 0x2000>;
 
-			u-boot@0 {
-				reg = <0x0 0x100000>;
+			kernel@0 {
+				reg = <0x0 0x400000>;
 				read-only;
 			};
 
-			kernel@100000 {
-				reg = <0x100000 0x300000>;
-			};
-
 			fs@400000 {
-				reg = <0x400000 0x1c00000>;
+				reg = <0x400000 0x4000000>;
 			};
+
+			appli@4400000 {
+				reg = <0x4400000 0x3c00000>;
+ 			};
 		};
 	};
 
@@ -125,9 +123,13 @@
 				interrupts = <14 0x8>;
 				interrupt-parent = <&ipic>;
 				dfsrr;
-				rtc@68 {
-					compatible = "dallas,ds1339";
-					reg = <0x68>;
+				rtc@6f {
+					compatible = "isl12024";
+					reg = <0x6f>;
+				};
+				at24@50 {
+					compatible = "at24,24c32";
+					reg = <0x50>;
 				};
 			};
 
@@ -164,40 +166,6 @@
 			mode = "cpu";
 		};
 
-		dma@82a8 {
-			#address-cells = <1>;
-			#size-cells = <1>;
-			compatible = "fsl,mpc8313-dma", "fsl,elo-dma";
-			reg = <0x82a8 4>;
-			ranges = <0 0x8100 0x1a8>;
-			interrupt-parent = <&ipic>;
-			interrupts = <71 8>;
-			cell-index = <0>;
-			dma-channel@0 {
-				compatible = "fsl,mpc8313-dma-channel", "fsl,elo-dma-channel";
-				reg = <0 0x80>;
-				interrupt-parent = <&ipic>;
-				interrupts = <71 8>;
-			};
-			dma-channel@80 {
-				compatible = "fsl,mpc8313-dma-channel", "fsl,elo-dma-channel";
-				reg = <0x80 0x80>;
-				interrupt-parent = <&ipic>;
-				interrupts = <71 8>;
-			};
-			dma-channel@100 {
-				compatible = "fsl,mpc8313-dma-channel", "fsl,elo-dma-channel";
-				reg = <0x100 0x80>;
-				interrupt-parent = <&ipic>;
-				interrupts = <71 8>;
-			};
-			dma-channel@180 {
-				compatible = "fsl,mpc8313-dma-channel", "fsl,elo-dma-channel";
-				reg = <0x180 0x28>;
-				interrupt-parent = <&ipic>;
-				interrupts = <71 8>;
-			};
-		};
 
 		/* phy type (ULPI, UTMI, UTMI_WIDE, SERIAL) */
 		usb@23000 {
@@ -211,42 +179,21 @@
 			sleep = <&pmc 0x00300000>;
 		};
 
-		enet0: ethernet@24000 {
-			#address-cells = <1>;
-			#size-cells = <1>;
-			sleep = <&pmc 0x20000000>;
-			ranges;
-
-			cell-index = <0>;
-			device_type = "network";
-			model = "eTSEC";
-			compatible = "gianfar", "simple-bus";
-			reg = <0x24000 0x1000>;
-			local-mac-address = [ 00 00 00 00 00 00 ];
-			interrupts = <37 0x8 36 0x8 35 0x8>;
-			interrupt-parent = <&ipic>;
-			phy-handle = < &phy1 >;
-			fsl,magic-packet;
+		
 
-			mdio@24520 {
-				#address-cells = <1>;
-				#size-cells = <0>;
-				compatible = "fsl,gianfar-mdio";
-				reg = <0x24520 0x20>;
-				phy1: ethernet-phy@1 {
-					interrupt-parent = <&ipic>;
-					interrupts = <19 0x8>;
-					reg = <0x1>;
-					device_type = "ethernet-phy";
-				};
-				phy4: ethernet-phy@4 {
-					interrupt-parent = <&ipic>;
-					interrupts = <20 0x8>;
-					reg = <0x4>;
-					device_type = "ethernet-phy";
-				};
+		mdio@24520 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			compatible = "fsl,gianfar-mdio";
+			reg = <0x24520 0x20>;				
+			phy4: ethernet-phy@4 {
+				interrupt-parent = <&ipic>;
+				interrupts = <20 0x8>;
+				reg = <0x4>;
+				device_type = "ethernet-phy";
 			};
 		};
+		
 
 		enet1: ethernet@25000 {
 			cell-index = <1>;
@@ -262,20 +209,10 @@
 			fsl,magic-packet;
 		};
 
-		serial0: serial@4500 {
+		serial0: serial@4600 {
 			cell-index = <0>;
 			device_type = "serial";
 			compatible = "ns16550";
-			reg = <0x4500 0x100>;
-			clock-frequency = <0>;
-			interrupts = <9 0x8>;
-			interrupt-parent = <&ipic>;
-		};
-
-		serial1: serial@4600 {
-			cell-index = <1>;
-			device_type = "serial";
-			compatible = "ns16550";
 			reg = <0x4600 0x100>;
 			clock-frequency = <0>;
 			interrupts = <10 0x8>;
@@ -309,7 +246,7 @@
 			 * manual, or if you are adapting this device tree
 			 * to a different board.
 			 */
-			status = "fail";
+			status = "okay";
 		};
 
 		gtm1: timer@500 {
@@ -337,18 +274,12 @@
 		pci0: pci@e0008500 {
 			cell-index = <1>;
 			interrupt-map-mask = <0xf800 0x0 0x0 0x7>;
-			interrupt-map = <
-					/* IDSEL 0x0E -mini PCI */
-					 0x7000 0x0 0x0 0x1 &ipic 18 0x8
-					 0x7000 0x0 0x0 0x2 &ipic 18 0x8
-					 0x7000 0x0 0x0 0x3 &ipic 18 0x8
-					 0x7000 0x0 0x0 0x4 &ipic 18 0x8
-
+			interrupt-map = <				
 					/* IDSEL 0x0F - PCI slot */
-					 0x7800 0x0 0x0 0x1 &ipic 17 0x8
-					 0x7800 0x0 0x0 0x2 &ipic 18 0x8
-					 0x7800 0x0 0x0 0x3 &ipic 17 0x8
-					 0x7800 0x0 0x0 0x4 &ipic 18 0x8>;
+					 0x7800 0x0 0x0 0x1 &ipic 48 0x8
+					 0x7800 0x0 0x0 0x2 &ipic 48 0x8
+					 0x7800 0x0 0x0 0x3 &ipic 48 0x8
+					 0x7800 0x0 0x0 0x4 &ipic 48 0x8>;
 			interrupt-parent = <&ipic>;
 			interrupts = <66 0x8>;
 			bus-range = <0x0 0x0>;
