--- linux-2.6.14.1/fs/jffs2/nodelist.h.orig	2005-11-09 23:12:29.641449728 -0800
+++ linux-2.6.14.1/fs/jffs2/nodelist.h	2005-11-09 23:13:19.228570049 -0800
@@ -44,7 +44,9 @@
 #define D2(x)
 #endif
 
-#define JFFS2_NATIVE_ENDIAN
+#undef JFFS2_NATIVE_ENDIAN
+#define JFFS2_BIG_ENDIAN 1
+#undef JFFS2_LITTLE_ENDIAN
 
 /* Note we handle mode bits conversion from JFFS2 (i.e. Linux) to/from
    whatever OS we're actually running on here too. */
