*** linux-2.6.13/drivers/usb/net/pegasus.h.orig	Mon Aug 29 01:41:01 2005
--- linux-2.6.13/drivers/usb/net/pegasus.h	Sat Oct 22 20:51:29 2005
***************
*** 181,186 ****
--- 181,188 ----
  		DEFAULT_GPIO_RESET )
  PEGASUS_DEV( "SpeedStream USB 10/100 Ethernet", VENDOR_ACCTON, 0x5046,
  		DEFAULT_GPIO_RESET | PEGASUS_II )
+ PEGASUS_DEV( "Philips CPWUE01/00 USB 10/100 Ethernet", VENDOR_ACCTON, 0xb004,
+ 		DEFAULT_GPIO_RESET | PEGASUS_II )
  PEGASUS_DEV( "ADMtek ADM8511 \"Pegasus II\" USB Ethernet",
  		VENDOR_ADMTEK, 0x8511,
  		DEFAULT_GPIO_RESET | PEGASUS_II | HAS_HOME_PNA )
