Index: setuptools-0.6c8/setuptools/command/sdist.py
===================================================================
--- setuptools-0.6c8.orig/setuptools/command/sdist.py
+++ setuptools-0.6c8/setuptools/command/sdist.py
@@ -95,7 +95,7 @@ def entries_finder(dirname, filename):
         for match in entries_pattern.finditer(data):
             yield joinpath(dirname,unescape(match.group(1)))
     else:
-        log.warn("unrecognized .svn/entries format in %s", dirname)
+        print ("unrecognized .svn/entries format in %s", dirname)
 
 
 finders = [
