diff -uNr eve.orig//data/themes/default.edc eve/data/themes/default.edc
--- eve.orig//data/themes/default.edc	2010-08-13 15:12:14.000000000 +0200
+++ eve/data/themes/default.edc	2010-08-20 20:15:10.000000000 +0200
@@ -202,7 +202,6 @@
                 type: RECT;
                 description { state: "default" 0.0;
                     color: 0 0 0 255;
-                    min: 480 800;
                 }
             }
         }
@@ -335,7 +334,6 @@
                 type: RECT;
                 description { state: "default" 0.0;
                     color: 0 0 0 255;
-                    min: 480 800;
                 }
             }
             part { name: "view";
