From 1c402a0db04c355a8414af4871b77bb62f72391f Mon Sep 17 00:00:00 2001
From: Michael 'Mickey' Lauer <mickey@vanille-media.de>
Date: Thu, 1 Oct 2009 13:52:44 +0200
Subject: [PATCH] glib-2.0: Fix g_regex_get_pattern binding

---
 vapi/glib-2.0.vapi |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vapi/glib-2.0.vapi b/vapi/glib-2.0.vapi
index 4837bc6..3682e7f 100644
--- a/vapi/glib-2.0.vapi
+++ b/vapi/glib-2.0.vapi
@@ -2707,7 +2707,7 @@ namespace GLib {
 	[CCode (ref_function = "g_regex_ref", unref_function = "g_regex_unref")]
 	public class Regex {
 		public Regex (string pattern, RegexCompileFlags compile_options = 0, RegexMatchFlags match_options = 0) throws RegexError;
-		public string get_pattern ();
+		public unowned string get_pattern ();
 		public int get_max_backref ();
 		public int get_capture_count ();
 		public int get_string_number (string name);
-- 
1.6.0.4

