*** usr/link.cc.orig	Sun Dec 23 20:31:49 2007
--- usr/link.cc	Sun Dec 23 20:32:46 2007
***************
*** 278,285 ****
     fixed-up env_link structure, scanning for the environment variables
     and their defaults, and opening the flash instance of the
     environment.  It first looks for the "Loader" partition.  If there
!    is none, it uses the first partition with the assumption that APEX
!    may be the primary boot loader.
  
  */
  
--- 278,286 ----
     fixed-up env_link structure, scanning for the environment variables
     and their defaults, and opening the flash instance of the
     environment.  It first looks for the "Loader" partition.  If there
!    is none, it then looks for the "apex" partition.  If there is none,
!    it uses the first partition with the assumption that APEX  may be
!    the primary boot loader.
  
  */
  
***************
*** 289,294 ****
--- 290,298 ----
    MTDPartition mtd = MTDPartition::find ("Loader");
  
    if (!mtd.is ())
+     mtd = MTDPartition::find ("apex");
+ 
+   if (!mtd.is ())
      mtd = MTDPartition::first ();
  
    bool fFound = mtd.is () && open_apex (mtd);
