Index: MySQL-python-1.2.1_p2/site.cfg
===================================================================
--- MySQL-python-1.2.1_p2.orig/site.cfg	2006-05-31 12:13:45.000000000 +0200
+++ MySQL-python-1.2.1_p2/site.cfg	2006-05-31 12:13:55.000000000 +0200
@@ -6,7 +6,7 @@
 
 [options]
 embedded = False
-threadsafe = True
+threadsafe = False
 static = False
 
 # Use the compiler section to add additional options for the extension build.
