--- a/src/Makefile.am~	2011-12-22 22:14:55.829779823 +0100
+++ b/src/Makefile.am	2011-12-22 22:15:14.353679720 +0100
@@ -52,7 +52,6 @@
           atomic_ops/sysdeps/gcc/arm.h \
           atomic_ops/sysdeps/gcc/avr32.h \
           atomic_ops/sysdeps/gcc/cris.h \
-          atomic_ops/sysdeps/gcc/hexagon.h \
           atomic_ops/sysdeps/gcc/hppa.h \
           atomic_ops/sysdeps/gcc/ia64.h \
           atomic_ops/sysdeps/gcc/m68k.h \
