Index: stblinux-2.6.12/drivers/char/brcmserial.c
===================================================================
--- stblinux-2.6.12.orig/drivers/char/brcmserial.c	2007-12-16 22:49:40.000000000 +0100
+++ stblinux-2.6.12/drivers/char/brcmserial.c	2007-12-16 22:56:00.000000000 +0100
@@ -1208,6 +1208,7 @@
 	600, 1200, 1800, 2400, 4800, 9600, 19200,
 	38400, 57600, 115200, 230400, 460800, 0 };
 
+#if 0
 int tty_get_baud_rate(struct tty_struct *tty)
 {
 	struct async_struct * info = (struct async_struct *)tty->driver_data;
@@ -1234,6 +1235,8 @@
 }
 #endif
 
+#endif
+
 /*
  * This routine is called to set the UART divisor registers to match
  * the specified baud rate for a serial port.
