From 5e60ef39027890ffc15ba308bb0ca76e9cc9f55a Mon Sep 17 00:00:00 2001
From: Ulf Samuelsson <ulf_samuelsson@telia.com>
Date: Fri, 28 Oct 2011 12:31:08 +0200
Subject: [PATCH 26/39] .gitignore: Update

---
 .gitignore |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/.gitignore b/.gitignore
index 18bf9dc..e8399ba 100644
--- a/.gitignore
+++ b/.gitignore
@@ -24,4 +24,5 @@ tags
 install/*
 *.rej
 *.patch
+patches/*
 
-- 
1.7.5.4

