From f702f4fdc32e79402e24e2523fe1aba19f6b8daa Mon Sep 17 00:00:00 2001
From: Ulf Samuelsson <ulf_samuelsson@telia.com>
Date: Fri, 28 Oct 2011 15:18:28 +0200
Subject: [PATCH] Fix EXTRA_INSTALL - add missing /

---
 Makefile |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 6dcd9c1..b6ef661 100644
--- a/Makefile
+++ b/Makefile
@@ -214,8 +214,8 @@ EXTRA_INSTALL+=files/NAND-empty-1MB.jffs2.bz2
 EXTRA_INSTALL+=files/SD-card-tools.tar.bz2
 
 ifeq ($(CONFIG_RAW_AT91),y)
-RAW_AT91=$(BINDIR)raw-at91
-EXTRA_INSTALL+=$(BINDIR)raw-at91
+RAW_AT91=$(BINDIR)/raw-at91
+EXTRA_INSTALL+=$(BINDIR)/raw-at91
 endif
 ifeq ($(CONFIG_SX_AT91),y)
 SX_AT91=$(BINDIR)/sx-at91
-- 
1.7.5.4

