From e3acb9616bce282f82a26e3a191e760c2176f59e Mon Sep 17 00:00:00 2001
From: Gennady Kupava <gb@bsdmn.com>
Date: Mon, 23 Aug 2010 21:28:02 +0200
Subject: [PATCH] glamo-display: fix WSOD for 242 timming

---
 drivers/mfd/glamo/glamo-display.c |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/drivers/mfd/glamo/glamo-display.c b/drivers/mfd/glamo/glamo-display.c
index 57292ff..da6f9b3 100644
--- a/drivers/mfd/glamo/glamo-display.c
+++ b/drivers/mfd/glamo/glamo-display.c
@@ -172,15 +172,17 @@ int glamo_lcd_cmd_mode(struct glamodrm_handle *gdrm, int on)
 	} else {
 		/* RGB interface needs vsync/hsync */
 		int mode;
+		//mdelay(1);
 		mode = reg_read_lcd(gdrm, GLAMO_REG_LCD_MODE3);
 		if ( mode & GLAMO_LCD_MODE3_RGB)
 			reg_write_lcd(gdrm, GLAMO_REG_LCD_COMMAND1,
 				  GLAMO_LCD_CMD_TYPE_DISP |
 				  GLAMO_LCD_CMD_DATA_DISP_SYNC);
-
+		mdelay(1);
 		reg_write_lcd(gdrm, GLAMO_REG_LCD_COMMAND1,
 			  GLAMO_LCD_CMD_TYPE_DISP |
 			  GLAMO_LCD_CMD_DATA_DISP_FIRE);
+		//mdelay(1);
 	}
 
 	return 0;
-- 
1.7.2

