diff -urN konqueror-old/configure konqueror-embedded-snapshot-20030705/configure
--- konqueror-old/configure	2003-08-03 17:49:31.000000000 -0700
+++ konqueror-embedded-snapshot-20030705/configure	2003-08-03 17:50:02.000000000 -0700
@@ -24232,14 +24232,14 @@
 if test "x$want_qpe" = "xyes"; then
 
 
-  if test -z "1.5.0"; then
+  if test -z "1.0.0"; then
     qtopia_minver_maj=1
     qtopia_minver_min=5
     qtopia_minver_pat=0
   else
-    qtopia_minver_maj=`echo "1.5.0" | sed -e "s/^\(.*\)\..*\..*$/\1/"`
-    qtopia_minver_min=`echo "1.5.0" | sed -e "s/^.*\.\(.*\)\..*$/\1/"`
-    qtopia_minver_pat=`echo "1.5.0" | sed -e "s/^.*\..*\.\(.*\)$/\1/"`
+    qtopia_minver_maj=`echo "1.0.0" | sed -e "s/^\(.*\)\..*\..*$/\1/"`
+    qtopia_minver_min=`echo "1.0.0" | sed -e "s/^.*\.\(.*\)\..*$/\1/"`
+    qtopia_minver_pat=`echo "1.0.0" | sed -e "s/^.*\..*\.\(.*\)$/\1/"`
   fi
 
   qtopia_minver="$qtopia_minver_maj$qtopia_minver_min$qtopia_minver_pat"
diff -urN konqueror-old/configure.in konqueror-embedded-snapshot-20030705/configure.in
--- konqueror-old/configure.in	2003-08-03 17:49:31.000000000 -0700
+++ konqueror-embedded-snapshot-20030705/configure.in	2003-08-03 17:49:47.000000000 -0700
@@ -53,7 +53,7 @@
               want_qpe=$enableval,want_qpe=no)
 
 if test "x$want_qpe" = "xyes"; then
-    AC_PATH_QTOPIA(1.5.0)
+    AC_PATH_QTOPIA(1.0.0)
     dnl hack
     CXXFLAGS="$CXXFLAGS -D_QT_QPE_"
 fi
