*** omniORB-4.0.7.old/include/omniORB4/CORBA_sysdep_auto.h	2004-10-17 16:14:28.000000000 -0400
--- omniORB-4.0.7/include/omniORB4/CORBA_sysdep_auto.h	2006-08-29 11:29:08.000000000 -0400
***************
*** 85,91 ****
  #endif
  
  
! #if defined(SIZEOF_LONG_DOUBLE) && (SIZEOF_LONG_DOUBLE == 16)
  #  define HAS_LongDouble
  #  define _CORBA_LONGDOUBLE_DECL long double
  #endif
--- 85,91 ----
  #endif
  
  
! #if defined(SIZEOF_LONG_DOUBLE)
  #  define HAS_LongDouble
  #  define _CORBA_LONGDOUBLE_DECL long double
  #endif
