diff -ur glider_V1.0.0.orig/glider.desktop glider_V1.0.0/glider.desktop
--- glider_V1.0.0.orig/glider.desktop	2004-06-17 00:18:09.000000000 +0200
+++ glider_V1.0.0/glider.desktop	2004-06-17 00:18:19.000000000 +0200
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Name=Glider
 Name[de]=Glider
-Icon=glider
+Icon=glider/glider
 Exec=glider
 Comment=Fight evil aliens using your space glider.
 Comment[de]=Fliege einen Raumgleiter gegen Ausserirdische.
