--- /tmp/Makefile.am	2008-04-01 10:50:58.848945668 +0200
+++ gnuradio-3.1.2/usrp/Makefile.am	2008-04-01 10:51:06.095612812 +0200
@@ -24,7 +24,7 @@
 	usrp.iss.in \
 	usrp.inf
 
-SUBDIRS = host firmware fpga doc
+SUBDIRS = host firmware fpga 
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = \
--- /tmp/Makefile.am	2008-04-01 10:52:10.615605452 +0200
+++ gnuradio-3.1.2/gr-trellis/Makefile.am	2008-04-01 10:52:19.807923018 +0200
@@ -21,5 +21,5 @@
 
 include $(top_srcdir)/Makefile.common
 
-SUBDIRS = src doc
+SUBDIRS = src 
 
