Index: git/Makefile.in
===================================================================
--- git.orig/Makefile.in	2009-12-04 12:53:00.000000000 -0800
+++ git/Makefile.in	2009-12-04 16:25:28.000000000 -0800
@@ -300,7 +300,7 @@ else
 endif
 
 # Installs development library links.
-install_dev: install_headers all
+install_dev: install_headers install_runtime
 	$(INSTALL) -d $(PREFIX)$(DEVEL_PREFIX)lib
 	-$(INSTALL) -m 644 lib/*.[ao] $(PREFIX)$(DEVEL_PREFIX)lib/
 ifeq ($(HAVE_SHARED),y)
