diff -Nurd cmake-2.8.2.orig/Source/CursesDialog/CMakeLists.txt cmake-2.8.2/Source/CursesDialog/CMakeLists.txt
--- cmake-2.8.2.orig/Source/CursesDialog/CMakeLists.txt	2010-06-28 11:16:00.000000000 -0400
+++ cmake-2.8.2/Source/CursesDialog/CMakeLists.txt	2010-08-23 10:36:31.936455461 -0400
@@ -33,5 +33,6 @@
 ADD_EXECUTABLE(ccmake ${CURSES_SRCS} )
 TARGET_LINK_LIBRARIES(ccmake CMakeLib)
 TARGET_LINK_LIBRARIES(ccmake cmForm)
+TARGET_LINK_LIBRARIES(ccmake tinfo)
 
 INSTALL_TARGETS(/bin ccmake)
