--- configure.in.org	2008-05-01 12:40:58.000000000 +0200
+++ configure.in	2008-05-01 12:39:39.000000000 +0200
@@ -796,13 +796,14 @@
     # pointer types. GCC may generate bad code as a result of that,
     # so use -fno-strict-aliasing if supported.
     AC_MSG_CHECKING(whether $CC accepts -fno-strict-aliasing)
-     ac_save_cc="$CC"
+    AC_CACHE_VAL(ac_cv_no_strict_aliasing_ok,
+    [ac_save_cc="$CC"
      CC="$CC -fno-strict-aliasing"
      AC_TRY_RUN([int main() { return 0; }],
      ac_cv_no_strict_aliasing_ok=yes,
      ac_cv_no_strict_aliasing_ok=no,
      ac_cv_no_strict_aliasing_ok=no)
-     CC="$ac_save_cc"
+     CC="$ac_save_cc"])
     AC_MSG_RESULT($ac_cv_no_strict_aliasing_ok)
     if test $ac_cv_no_strict_aliasing_ok = yes
     then
