Index: cardinfo.c
===================================================================
RCS file: /cvs/gpe/base/gpe-conf/cardinfo.c,v
retrieving revision 1.27
diff -u -r1.27 cardinfo.c
--- cardinfo.c	4 Apr 2004 09:52:46 -0000	1.27
+++ cardinfo.c	4 Jun 2004 09:29:41 -0000
@@ -34,11 +34,11 @@
 #include <poll.h>
 #include <time.h>
 
-#include <pcmcia/version.h>
-#include <pcmcia/cs_types.h>
-#include <pcmcia/cs.h>
-#include <pcmcia/cistpl.h>
-#include <pcmcia/ds.h>
+#include "pcmcia/version.h"
+#include "pcmcia/cs_types.h"
+#include "pcmcia/cs.h"
+#include "pcmcia/cistpl.h"
+#include "pcmcia/ds.h"
 
 #include <gtk/gtk.h>
 
