--- qtopia-free-1.6.2/src/3rdparty/applications/embeddedkonsole/qpe-embeddedkonsole.control~	Mon Jul  7 19:43:19 2003
+++ qtopia-free-1.7.0/src/3rdparty/applications/embeddedkonsole/qpe-embeddedkonsole.control	Mon Jul  7 20:01:34 2003
@@ -5,6 +5,6 @@
 Architecture: $CPU_ARCH
 Arch: $DEVICE_ARCH
 Version: $QPE_VERSION-13
-Depends: qpe-libqtopia, ptydevs
+Depends: qpe-libqtopia
 Description: KDE's konsole (shell terminal)
  Ported to the Qtopia environment.
