--- qtopia-free-1.7.0/src/plugins/fontfactories/freetype/fontfactoryttf_qws.cpp	Wed Jun  4 02:59:21 2003
+++ qtopia-free-1.7.0/src/plugins/fontfactories/freetype/fontfactoryttf_qws.cpp.new	Wed Jan 14 20:10:54 2004
@@ -31,7 +31,8 @@
 
 
 extern "C" {
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include <freetype/ftglyph.h>
 #include <freetype/ftoutln.h>
 #include <freetype/ftbbox.h>
--- qtopia-free-1.7.0/src/plugins/fontfactories/freetype/fontfactoryttf_qws.h	Wed Jun  4 02:59:21 2003
+++ qtopia-free-1.7.0/src/plugins/fontfactories/freetype/fontfactoryttf_qws.h.new	Wed Jan 14 20:11:13 2004
@@ -31,7 +31,8 @@
 
 
 extern "C" {
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 }
 
 // ascent, descent, width(ch), width(string), maxwidth?
