--- Makefile.linux.~1~	2003-07-08 16:05:24.000000000 +1000
+++ qtopia-free-1.7.0/src/Makefile.linux	2003-09-19 13:02:30.000000000 +1000
@@ -71,7 +71,6 @@
 	plugins/decorations/flat
 
 SUPPORTEDAPPS=\
-	3rdparty/tools/atd \
 	applications/addressbook \
 	applications/calculator \
 	applications/clock \
@@ -107,7 +106,8 @@
 	settings/language \
 	tools/quicklauncher \
 	tools/qcop \
-	tools/symlinker
+	tools/symlinker 
+        
 
 $(SUPPORTEDLIBS): $(CORELIBS)
 
