diff -Naru qemu-snapshot-2007-06-24_05.orig/sdl.c qemu-snapshot-2007-06-24_05/sdl.c
--- qemu-snapshot-2007-06-24_05.orig/sdl.c	2007-06-21 23:08:02.000000000 +0200
+++ qemu-snapshot-2007-06-24_05/sdl.c	2007-06-24 16:33:20.000000000 +0200
@@ -245,7 +245,7 @@
 
     if (kbd_mouse_is_absolute()) {
         SDL_ShowCursor(1);
-        SDL_SetCursor(sdl_cursor_hidden);
+        /* SDL_SetCursor(sdl_cursor_hidden); */
     } else {
         SDL_ShowCursor(0);
     }
