Index: openjdk/jdk/make/com/sun/java/pack/Makefile
===================================================================
--- openjdk/jdk/make/com/sun/java/pack/Makefile	2010-06-13 15:25:21.548426411 +0200
+++ openjdk/jdk/make/com/sun/java/pack/Makefile	2010-06-13 15:25:43.858007543 +0200
@@ -103,6 +103,7 @@
   OTHER_LDLIBS += $(LIBCXX) -lc
 # setup the list of libraries to link in...
 ifeq ($(PLATFORM), linux)
+  OTHER_LDLIBS  += -lpthread
 ifeq ("$(CC_VER_MAJOR)", "3")
   OTHER_LDLIBS  += -Wl,-Bstatic -lgcc_eh -Wl,-Bdynamic
 endif
