--- sylpheed-2.0.0beta6/config.h.in.old	2005-07-18 11:57:23.000000000 +0200
+++ sylpheed-2.0.0beta6/config.h.in	2005-07-18 11:59:57.000000000 +0200
@@ -263,6 +263,9 @@
 /* Define to 1 if your <sys/time.h> declares `struct tm'. */
 #undef TM_IN_SYS_TIME
 
+/* Define if you use GnuTLS to support SSL. */
+#undef USE_GNUTLS
+
 /* Define if you use GPGME to support OpenPGP. */
 #undef USE_GPGME
 
@@ -273,6 +276,9 @@
 #undef USE_LDAP
 
 /* Define if you use OpenSSL to support SSL. */
+#undef USE_OPENSSL
+
+/* Define if SSL is supported. */
 #undef USE_SSL
 
 /* Whether to use multithread or not */
--- sylpheed-2.0.0beta6/configure.in.old	2005-07-18 12:00:24.000000000 +0200
+++ sylpheed-2.0.0beta6/configure.in	2005-07-18 12:04:33.000000000 +0200
@@ -202,7 +202,8 @@
 #include <openssl/opensslv.h>
 ],		[ return OPENSSL_VERSION_NUMBER; ],
 		[ AC_MSG_RESULT(yes)
-		  AC_DEFINE(USE_SSL, 1, Define if you use OpenSSL to support SSL.) ],
+	  	  AC_DEFINE(USE_SSL, 1, Define if SSL is supported.)
+	  	  AC_DEFINE(USE_OPENSSL, 1, Define if you use OpenSSL to support SSL.) ],
 		[ AC_MSG_RESULT(no)
 		  LIBS="$ac_save_LIBS"
 		  ac_cv_enable_ssl=no ])
@@ -211,6 +212,27 @@
 	AC_MSG_RESULT(no)
 fi
 
+AC_ARG_ENABLE(gnutls,
+	[  --enable-gnutls         Enable SSL support using GnuTLS [default=no]],
+	[ac_cv_enable_gnutls=$enableval], [ac_cv_enable_gnutls=no])
+AC_MSG_CHECKING([whether to use GnuTLS])
+if test $ac_cv_enable_gnutls = yes; then
+	AC_MSG_RESULT(yes)
+	AC_MSG_CHECKING([if GnuTLS is available])
+	LIBS="$LIBS -lgnutls-openssl"
+	AC_TRY_LINK([
+#include <gnutls/openssl.h>
+],	[ return OPENSSL_VERSION_NUMBER; ],
+	[ AC_MSG_RESULT(yes)
+	  AC_DEFINE(USE_SSL, 1, Define if SSL is supported.)
+	  AC_DEFINE(USE_GNUTLS, 1, Define if you use GnuTLS to support SSL.) ],
+	[ AC_MSG_RESULT(no)
+	  LIBS="$ac_save_LIBS"
+	  ac_cv_enable_gnutls=no ])
+else
+	AC_MSG_RESULT(no)
+fi
+
 dnl Check for X-Face support
 AC_ARG_ENABLE(compface,
 	[  --disable-compface      Do not use compface (X-Face)],
@@ -350,6 +372,7 @@
 echo "JPilot        : $ac_cv_enable_jpilot"
 echo "LDAP          : $ac_cv_enable_ldap"
 echo "OpenSSL       : $ac_cv_enable_ssl"
+echo "GnuTLS        : $ac_cv_enable_gnutls"
 echo "iconv         : $am_cv_func_iconv"
 echo "compface      : $ac_cv_enable_compface"
 echo "IPv6          : $ac_cv_enable_ipv6"
--- sylpheed-2.0.0beta6/src/ssl.h.old	2005-07-18 12:09:35.000000000 +0200
+++ sylpheed-2.0.0beta6/src/ssl.h	2005-07-18 12:11:15.000000000 +0200
@@ -27,11 +27,19 @@
 #if USE_SSL
 
 #include <glib.h>
+#if USE_OPENSSL
 #include <openssl/crypto.h>
 #include <openssl/x509.h>
 #include <openssl/pem.h>
 #include <openssl/ssl.h>
 #include <openssl/err.h>
+#else
+#if USE_GNUTLS
+#include <gnutls/openssl.h>
+#else
+#error Must select either GnuTLS or OpenSSL
+#endif
+#endif
 
 #include "socket.h"
 
