--- sylpheed-2.0.0beta6/src/ssl.c.orig	2005-07-20 15:45:12.882201824 +0100
+++ sylpheed-2.0.0beta6/src/ssl.c	2005-07-20 16:10:04.006516472 +0100
@@ -54,9 +54,9 @@
 		debug_print(_("SSLv23 not available\n"));
 	} else {
 		debug_print(_("SSLv23 available\n"));
-		if (certs_dir &&
+		if (certs_dir /*&&
 		    !SSL_CTX_load_verify_locations(ssl_ctx_SSLv23, NULL,
-						   certs_dir))
+						   certs_dir)*/)
 			g_warning("SSLv23 SSL_CTX_load_verify_locations failed.\n");
 	}
 
@@ -65,9 +65,9 @@
 		debug_print(_("TLSv1 not available\n"));
 	} else {
 		debug_print(_("TLSv1 available\n"));
-		if (certs_dir &&
+		if (certs_dir /*&&
 		    !SSL_CTX_load_verify_locations(ssl_ctx_TLSv1, NULL,
-						   certs_dir))
+						   certs_dir)*/)
 			g_warning("TLSv1 SSL_CTX_load_verify_locations failed.\n");
 	}
 
@@ -151,8 +151,8 @@
 			g_free(str);
 		}
 
-		verify_result = SSL_get_verify_result(sockinfo->ssl);
-		if (verify_result == X509_V_OK)
+		verify_result = 0; /*SSL_get_verify_result(sockinfo->ssl);*/
+		if (verify_result == 1)
 			debug_print("SSL verify OK\n");
 		else
 			g_warning("%s: SSL certificate verify failed (%ld: %s)\n",
