--- xffm-4.0.5/src/Makefile.am	2003-05-29 12:13:17.000000000 -0700
+++ xffm-4.0.5.new/src/Makefile.am	2004-06-11 10:26:06.509143832 -0700
@@ -94,16 +94,16 @@
 install-data-local:
 	$(mkinstalldirs) $(DESTDIR)$(bindir)
 	( cd $(DESTDIR)$(bindir); \
-	  rm -f xffstab; ln xffm xffstab; \
-	  rm -f xfbook; ln xffm xfbook; \
-	  rm -f xfapps; ln xffm xfapps; \
-	  rm -f xftree4; ln xffm xftree4; \
-	  rm -f xfsamba4; ln xffm xfsamba4; \
-	  rm -f xfbook4; ln xffm xfbook4; \
-	  rm -f xfapps4; ln xffm xfapps4; \
-	  rm -f xfglob4; ln xffm xfglob4; \
-	  rm -f xftrash4; ln xffm xftrash4; \
-	  rm -f xffstab4; ln xffm xffstab4; \
+	  rm -f xffstab; ln -s xffm xffstab; \
+	  rm -f xfbook; ln -s xffm xfbook; \
+	  rm -f xfapps; ln -s xffm xfapps; \
+	  rm -f xftree4; ln -s xffm xftree4; \
+	  rm -f xfsamba4; ln -s xffm xfsamba4; \
+	  rm -f xfbook4; ln -s xffm xfbook4; \
+	  rm -f xfapps4; ln -s xffm xfapps4; \
+	  rm -f xfglob4; ln -s xffm xfglob4; \
+	  rm -f xftrash4; ln -s xffm xftrash4; \
+	  rm -f xffstab4; ln -s xffm xffstab4; \
 	)
 
 pkgconfigdir = $(libdir)/pkgconfig
