--- src/Makefile.am.old	2005-06-09 19:25:34.000000000 +0200
+++ src/Makefile.am	2005-06-09 19:27:14.000000000 +0200
@@ -19,12 +19,14 @@
 
 libsapwood_la_LDFLAGS = -avoid-version -module -Wl,-z,defs
 libsapwood_la_LIBADD = $(GTK_LIBS)
+libsapwood_la_CFLAGS = $(AM_CFLAGS)
 
 daemondir = $(libdir)/sapwood
 daemon_PROGRAMS = sapwood-server
 
 sapwood_server_SOURCES = sapwood-server.c sapwood-proto.c sapwood-proto.h
 sapwood_server_LDADD = $(GDK_LIBS)
+sapwood_server_CFLAGS = $(AM_CFLAGS)
 
 MAINTAINERCLEANFILES = Makefile.in
 
