diff -urN fbvncserver-0.9.4.old/Makefile fbvncserver-0.9.4/Makefile
--- fbvncserver-0.9.4.old/Makefile	2003-05-19 22:52:24.000000000 -0400
+++ fbvncserver-0.9.4/Makefile	2004-02-27 10:49:27.000000000 -0500
@@ -86,7 +86,7 @@
 		${STRIP} $@
 
 zaurus_kbdsim.o:		kbdsim.c
-		${CC}  -Wall -D__KERNEL__ -c -o $@ $<
+		${CC} ${KERNEL_INCLUDES} -Wall -D__KERNEL__ -c -o $@ $<
 
 ipaq_kbdsim.o:			kbdsim.c
 		${CC} -DIPAQ -Wall -D__KERNEL__ -c -o $@ $<
