--- slang-1.4.9/autoconf/Makefile.in.orig	2004-09-24 15:51:57.054180737 +0100
+++ slang-1.4.9/autoconf/Makefile.in	2004-09-24 15:52:47.065068195 +0100
@@ -13,12 +13,12 @@
 	@echo Use make install-elf to install it.
 runtests:
 	cd src; $(MAKE) runtests
-demos:
-	cd demo; $(MAKE)
+#demos:
+#	cd demo; $(MAKE)
 clean:
 	/bin/rm -f *~
 	cd src; $(MAKE) clean
-	cd demo; $(MAKE) clean
+#	cd demo; $(MAKE) clean
 install:
 	cd src; $(MAKE) install
 install-elf:
@@ -32,6 +32,6 @@
 distclean:
 	/bin/rm -f *~ Makefile config.status config.log config.cache files.pck
 	cd src; $(MAKE) distclean
-	cd demo; $(MAKE) distclean
+#	cd demo; $(MAKE) distclean
 #
 
