
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- openswan-2.1.2/lib/libdes/Makefile~flags.patch
+++ openswan-2.1.2/lib/libdes/Makefile
@@ -58,9 +58,10 @@
 #CC=gcc
 #CFLAG= -O4 -funroll-loops -fomit-frame-pointer
 # normally overridden by FreeS/WAN Makefiles anyway
-CFLAG= -O3 -fomit-frame-pointer -I${KLIPSD}/include -I${SRCDIR}
+CFLAG= -O3 -fomit-frame-pointer
 
 CFLAGS=$(OPTS) $(CFLAG)
+override CFLAGS+=-I${KLIPSD}/include -I${SRCDIR}
 CPP=$(CC) -E
 
 # Assember version of des_encrypt*().
--- openswan-2.1.2/lib/libfreeswan/Makefile~flags.patch
+++ openswan-2.1.2/lib/libfreeswan/Makefile
@@ -45,7 +45,8 @@
 
 LIB=libfreeswan.a
 # Original flags
-CFLAGS=-I. -I${SRCDIR} -I${KLIPSD} -I${FREESWANSRCDIR} $(USERCOMPILE)
+override CFLAGS=-I. -I${SRCDIR} -I${KLIPSD} -I${FREESWANSRCDIR}
+CFLAGS+= $(USERCOMPILE)
 CFLAGS+= -Wall
 #CFLAGS+= -Wconversion
 #CFLAGS+= -Wmissing-prototypes
@@ -59,7 +60,7 @@
 CFLAGS+= -Wbad-function-cast 
 
 ifeq ($(USE_NAT_TRAVERSAL),true)
-CFLAGS+= -DNAT_TRAVERSAL
+override CFLAGS+= -DNAT_TRAVERSAL
 endif
 
 
--- openswan-2.1.2/lib/libipsecpolicy/Makefile~flags.patch
+++ openswan-2.1.2/lib/libipsecpolicy/Makefile
@@ -30,7 +30,8 @@
 
 LIB=libipsecpolicy.a
 # Original flags
-CFLAGS=-I. -I${KLIPSD} -I${FREESWANSRCDIR} $(USERCOMPILE)
+override CFLAGS+= -I. -I${KLIPSD} -I${FREESWANSRCDIR}
+CFLAGS+= $(USERCOMPILE)
 CFLAGS+= -Wall
 CFLAGS+= -Wpointer-arith
 CFLAGS+= -Wcast-qual
--- openswan-2.1.2/lib/liblwres/Makefile~flags.patch
+++ openswan-2.1.2/lib/liblwres/Makefile
@@ -24,7 +24,8 @@
 CDEFINES = -g
 CWARNINGS = -Werror
 
-CFLAGS=${CINCLUDES} ${CDEFINES} ${CWARNINGS}
+override CFLAGS=${CINCLUDES}
+CFLAGS+=${CDEFINES} ${CWARNINGS}
 
 VERSION="@(\#) freeswan-hacking-9.2.1-for-fs2"
 LIBINTERFACE=2
--- openswan-2.1.2/lib/libopenswan/Makefile~flags.patch
+++ openswan-2.1.2/lib/libopenswan/Makefile
@@ -46,7 +46,8 @@
 
 LIB=libopenswan.a
 # Original flags
-CFLAGS=-I. -I${SRCDIR} -I${KLIPSD} -I${OPENSWANSRCDIR} $(USERCOMPILE)
+override CFLAGS=-I. -I${SRCDIR} -I${KLIPSD} -I${OPENSWANSRCDIR}
+CFLAGS+= $(USERCOMPILE)
 CFLAGS+= -Wall
 #CFLAGS+= -Wconversion
 #CFLAGS+= -Wmissing-prototypes
@@ -60,7 +61,7 @@
 CFLAGS+= -Wbad-function-cast 
 
 ifeq ($(USE_NAT_TRAVERSAL),true)
-CFLAGS+= -DNAT_TRAVERSAL
+override CFLAGS+= -DNAT_TRAVERSAL
 endif
 
 
--- openswan-2.1.2/programs/Makefile.program~flags.patch
+++ openswan-2.1.2/programs/Makefile.program
@@ -1,7 +1,7 @@
 
 include ${FREESWANSRCDIR}/Makefile.ver
 
-CFLAGS+=$(USERCOMPILE) -I${KLIPSINC}
+override CFLAGS+=-I${KLIPSINC}
 
 CFLAGS+= $(USERCOMPILE)
 CFLAGS+= -Wall
@@ -15,10 +15,9 @@
 #CFLAGS+= -W
 #CFLAGS+= -Wwrite-strings
 CFLAGS+= -Wbad-function-cast 
-CFLAGS+= -DNAT_TRAVERSAL
 
 ifeq ($(USE_NAT_TRAVERSAL),true)
-CFLAGS+= -DNAT_TRAVERSAL
+override CFLAGS+= -DNAT_TRAVERSAL
 endif
 
 # die if there are any warnings
--- openswan-2.1.2/programs/lwdnsq/Makefile~flags.patch
+++ openswan-2.1.2/programs/lwdnsq/Makefile
@@ -29,7 +29,7 @@
 CFLAGS+=-I${BIND9INCDIR}
 endif
 
-CFLAGS+=-I${LWRESINCL}
+override CFLAGS+=-I${LWRESINCL}
 
 #USERCOMPILE=-g
 
--- openswan-2.1.2/programs/pluto/Makefile~flags.patch
+++ openswan-2.1.2/programs/pluto/Makefile
@@ -203,7 +203,7 @@
 DEFINES+= -DSMARTCARD
 endif
 
-CPPFLAGS = $(HDRDIRS) $(DEFINES) \
+override CPPFLAGS = $(HDRDIRS) $(DEFINES) \
 	-DSHARED_SECRETS_FILE=\"${FINALCONFDIR}/ipsec.secrets\" \
 	-DPOLICYGROUPSDIR=\"${FINALCONFDDIR}/policies\" \
 	-DPERPEERLOGDIR=\"${FINALLOGDIR}/pluto/peer\"
--- openswan-2.1.2/testing/utils/ikeping/Makefile~flags.patch
+++ openswan-2.1.2/testing/utils/ikeping/Makefile
@@ -35,7 +35,7 @@
 
 HDRDIRS =  $(FREESWANINCLS)
 
-CPPFLAGS = $(HDRDIRS) $(BYTE_ORDER) -DDEBUG 
+override CPPFLAGS = $(HDRDIRS) $(BYTE_ORDER) -DDEBUG 
 
 ALLFLAGS = $(CPPFLAGS) $(CFLAGS)
 
--- openswan-2.1.2/testing/utils/uml_netjig/Makefile~flags.patch
+++ openswan-2.1.2/testing/utils/uml_netjig/Makefile
@@ -43,7 +43,8 @@
 #PCAPH=-I/home/mcr/src/libpcap
 #PCAPLIB=-L/home/mcr/src/linux/i386/libpcap -lpcap
 
-CFLAGS = -g -Wall ${PCAPH} ${NETDISSECTH} ${OPTIONS}
+CFLAGS = -g -Wall ${PCAPH} ${NETDISSECTH}
+override CFLAGS += ${OPTIONS}
 LIBS = ${NETDISSECTLIB} ${PCAPLIB} ${OPENSSL}
 
 
