--- a/tools/Makefile	2007-03-11 00:55:44.000000000 +0000
+++ b/tools/Makefile	2007-03-11 00:55:52.000000000 +0000
@@ -194,14 +194,14 @@
 
 $(obj)environment.c:
 		@rm -f $(obj)environment.c
-		ln -s $(src)../common/environment.c $(obj)environment.c
+		ln -sf $(src)../common/environment.c $(obj)environment.c
 
 $(obj)environment.o:	$(obj)environment.c
 		$(CC) -g $(HOST_ENVIRO_CFLAGS) $(CPPFLAGS) -c -o $@ $<
 
 $(obj)crc32.c:
 		@rm -f $(obj)crc32.c
-		ln -s $(src)../lib_generic/crc32.c $(obj)crc32.c
+		ln -sf $(src)../lib_generic/crc32.c $(obj)crc32.c
 
 $(LOGO_H):	$(obj)bmp_logo $(LOGO_BMP)
 		$(obj)./bmp_logo $(LOGO_BMP) >$@
