Index: automake-1.10.2/Makefile.am
===================================================================
--- automake-1.10.2.orig/Makefile.am
+++ automake-1.10.2/Makefile.am
@@ -75,7 +75,8 @@
   -e 's,[@]SHELL[@],$(SHELL),g' \
   -e 's,[@]VERSION[@],$(VERSION),g' \
   -e 's,[@]configure_input[@],Generated from $@.in; do not edit by hand.,g' \
-  -e 's,[@]datadir[@],$(datadir),g'
+  -e 's,[@]datadir[@],$(datadir),g' \
+  -e 's,[@]bindir[@],$(bindir),g'
 
 ## These files depend on Makefile so they are rebuilt if $(VERSION),
 ## $(datadir) or other do_subst'ituted variables change.
Index: automake-1.10.2/Makefile.in
===================================================================
--- automake-1.10.2.orig/Makefile.in
+++ automake-1.10.2/Makefile.in
@@ -186,7 +186,8 @@
   -e 's,[@]SHELL[@],$(SHELL),g' \
   -e 's,[@]VERSION[@],$(VERSION),g' \
   -e 's,[@]configure_input[@],Generated from $@.in; do not edit by hand.,g' \
-  -e 's,[@]datadir[@],$(datadir),g'
+  -e 's,[@]datadir[@],$(datadir),g' \
+  -e 's,[@]bindir[@],$(bindir),g'
 
 WGET = wget
 WGET_SV_CVS = $(WGET) http://savannah.gnu.org/cgi-bin/viewcvs/~checkout~/
Index: automake-1.10.2/automake.in
===================================================================
--- automake-1.10.2.orig/automake.in
+++ automake-1.10.2/automake.in
@@ -1,8 +1,10 @@
-#!@PERL@ -w
+#!@bindir@/env perl
 # -*- perl -*-
 # @configure_input@
 
-eval 'case $# in 0) exec @PERL@ -S "$0";; *) exec @PERL@ -S "$0" "$@";; esac'
+use warnings;
+
+eval 'case $# in 0) exec @bindir@/env perl -S "$0";; *) exec @bindir@/env perl -S "$0" "$@";; esac'
     if 0;
 
 # automake - create Makefile.in from Makefile.am
Index: automake-1.10.2/aclocal.in
===================================================================
--- automake-1.10.2.orig/aclocal.in
+++ automake-1.10.2/aclocal.in
@@ -1,4 +1,4 @@
-#!@PERL@ -w
+#!@bindir@/env perl -w
 # -*- perl -*-
 # @configure_input@
 
