upstream: not applicable upstream
origin: unknown contributor, sorry I forgot your name, your work is highly appreciated

--- scim-1.4.6/configure.ac.old	2007-05-26 19:08:41.000000000 +0200
+++ scim-1.4.6/configure.ac	2007-05-26 21:15:00.000000000 +0200
@@ -112,20 +112,8 @@
 AM_CONDITIONAL(HAVE_XSLTPROC, test x$XSLTPROC != xno)
 AC_SUBST(XSLTPROC)

-# Checks if docbook-style-xsl is available
-AC_CHECK_FILE(
- [/usr/share/sgml/docbook/xsl-stylesheets/html/tldp-html.xsl],
- [DOCBOOK_XSL=/usr/share/sgml/docbook/xsl-stylesheets/html/tldp-html.xsl],
- [AC_CHECK_FILE(
-  [/usr/share/sgml/docbook/xsl-stylesheets/html/docbook.xsl],
-  [DOCBOOK_XSL=/usr/share/sgml/docbook/xsl-stylesheets/html/docbook.xsl],
-   [AC_CHECK_FILE(
-    [/usr/share/xml/docbook/stylesheet/nwalsh/current/html/docbook.xsl],
-    [DOCBOOK_XSL=/usr/share/xml/docbook/stylesheet/nwalsh/current/html/docbook.xsl],
-    [DOCBOOK_XSL=no]
-    )]
-  )]
-)
+# Not needed
+DOCBOOK_XSL=no

 AC_SUBST(DOCBOOK_XSL)

@@ -156,16 +144,6 @@
 AC_CHECK_SIZEOF(unsigned long long int)
 AC_CHECK_HEADERS([stdint.h],[has_stdint=yes],[has_stdint=no])

-if test x$ac_cv_sizeof_unsigned_short_int = x2 && test x$ac_cv_sizeof_char = x1 ; then
-  :
-else
-  AC_MSG_ERROR([
-*** SCIM requires
-***      sizeof (unsigned short int)  == 2
-***      sizeof (char)                == 1
-*** You might want to consider using the GNU C compiler.
-])
-fi

 if test x$ac_cv_sizeof_wchar_t = x0 ; then
   AC_MSG_ERROR([
