From 3227bd5c412e7eb0d4370b2834e71723f6b4be48 Mon Sep 17 00:00:00 2001
From: Tuomas Kulve <tuomas.kulve@movial.fi>
Date: Mon, 27 Oct 2008 18:55:59 +0200
Subject: [PATCH] Removed resolution check that prevents scaling when output resolution doesn't match the original resolution.

---
 drivers/video/omap/dispc.c |    3 ---
 1 files changed, 0 insertions(+), 3 deletions(-)

diff --git a/drivers/video/omap/dispc.c b/drivers/video/omap/dispc.c
index 0f0b2e5..1df0c1e 100644
--- a/drivers/video/omap/dispc.c
+++ b/drivers/video/omap/dispc.c
@@ -579,9 +579,6 @@ static int omap_dispc_set_scale(int plane,
 	if ((unsigned)plane > OMAPFB_PLANE_NUM)
 		return -ENODEV;
 
-	if (out_width != orig_width || out_height != orig_height)
-		return -EINVAL;
-
 	enable_lcd_clocks(1);
 	if (orig_width < out_width) {
 		/*
-- 
1.5.6.5

