diff -urNd ../gpe-nmf-0.17-r0/gpe-nmf-0.17/backend/Makefile gpe-nmf-0.17/backend/Makefile
--- ../gpe-nmf-0.17-r0/gpe-nmf-0.17/backend/Makefile	2004-07-06 22:02:57.000000000 +0100
+++ gpe-nmf-0.17/backend/Makefile	2004-08-03 12:42:34.000000000 +0100
@@ -48,7 +48,7 @@
 clean:
 	rm -f $(DEPS) $(OBJS) $(OGG_OBJS) $(MAD_OBJS) $(BACKEND) d-vorbis.so d-mad.so
 
-install_program: all
+install-program: all
 	mkdir -p $(DESTDIR)$(PREFIX)/lib
 	install -s $(BACKEND) $(DESTDIR)$(PREFIX)/lib/$(BACKEND)
 	for i in $(CODECS); do install -s $$i $(DESTDIR)$(PREFIX)/lib/gpe-nmf/decoders/$$i; done
diff -urNd ../gpe-nmf-0.17-r0/gpe-nmf-0.17/Makefile gpe-nmf-0.17/Makefile
--- ../gpe-nmf-0.17-r0/gpe-nmf-0.17/Makefile	2004-08-03 03:31:46.000000000 +0100
+++ gpe-nmf-0.17/Makefile	2004-08-03 12:37:33.000000000 +0100
@@ -43,7 +43,7 @@
 	make -C backend PREFIX=$(PREFIX) GST=$(GST) GST_VERSION=$(GST_VERSION) CVSBUILD=$(CVSBUILD)
 
 $(PACKAGE): $(OBJS) $(LIB)
-	$(CC) -o $@ $^ $(PACKAGE_LDFLAGS) -Lbackend -lgpenmf
+	$(CC) -o $@ $^ $(LDFLAGS) $(PACKAGE_LDFLAGS) -Lbackend -lgpenmf
 
 install-program: all $(PACKAGE).desktop
 	make -C backend PREFIX=$(PREFIX) DESTDIR=$(DESTDIR) install
