--- zd1211-driver-r83.orig/Makefile	2006-07-09 04:59:42.000000000 -0500
+++ zd1211-driver-r83/Makefile	2006-10-21 16:27:41.000000000 -0500
@@ -4,25 +4,25 @@
 #
 #
 
-CC=gcc
-CPP=g++
-LD=ld
+CC ?= gcc
+CPP ?= g++
+LD ?= ld
 rM=rm -f -r
 
 VERSION := $(shell uname -r)
-MODPATH := /lib/modules/$(VERSION)
+MODPATH ?= /lib/modules/$(VERSION)
 
 # if the kernel is 2.6.x, turn on this
-KERN_26=y
+#KERN_26=y
 
-KERNEL_SOURCE=$(MODPATH)/source
+KERNEL_SOURCE ?= $(MODPATH)/source
 #KERNEL_SOURCE=/usr/src/linux
 
 # set to 1 for zd1211b
 ZD1211REV_B=0
 
 SRC_DIR=src
-DEFINES=-D__KERNEL__ -DMODULE=1
+DEFINES ?= -D__KERNEL__ -DMODULE=1
 
 
 
@@ -228,9 +228,9 @@
 	depmod -a $(VERSION)
 
 #for apdbg
-	gcc -o apdbg apdbg.c
-	chmod +x apdbg
-	cp ./apdbg /sbin/apdbg   
+#	gcc -o apdbg apdbg.c
+#	chmod +x apdbg
+#	cp ./apdbg /sbin/apdbg   
 
 clean:
 	rm -rf .tmp_versions .*.cmd *.ko *.mod.c *.mod.o *.o $(SRC_DIR)/*.o  $(SRC_DIR)/.*.o.cmd
