--- linux-2.6.16.20/arch/sh/cchips/hd6446x/hd64461/setup.c_orig	2006-06-09 00:49:01.000000000 +0000
+++ linux-2.6.16.20/arch/sh/cchips/hd6446x/hd64461/setup.c	2006-06-09 00:49:16.000000000 +0000
@@ -79,7 +79,7 @@
 	.end		= end_hd64461_irq,
 };
 
-static irqreturn_t hd64461_interrupt(int irq, void *dev_id, struct pt_regs *regs)
+static irqreturn_t hd64461_interrupt(int irq, void *dev_id, volatile struct pt_regs *regs)
 {
 	printk(KERN_INFO
 	       "HD64461: spurious interrupt, nirr: 0x%x nimr: 0x%x\n",
