--- gpe-conf-0.1.30/sleep/main.c.orig	2006-06-05 12:23:34.458994776 +0200
+++ gpe-conf-0.1.30/sleep/main.c	2006-06-05 12:23:40.216119560 +0200
@@ -23,7 +23,7 @@
   g_spawn_command_line_sync (cmd, NULL, NULL, NULL, NULL);
   if(save_ISconf(ISconf, ISconf->confName)) {
     char homeConf[MAXPATHLEN];
-    sprintf(homeConf, "%s/ipaq-sleep.conf", getenv("HOME"));
+    sprintf(homeConf, "%s/.sleep.conf", getenv("HOME"));
     if(!save_ISconf(ISconf, homeConf))
       strcpy(ISconf->confName, homeConf);
   }
@@ -42,7 +42,7 @@
   char cname[MAXPATHLEN];
   GtkWidget *GPE_Config_Sleep;
 
-  sprintf(cname,"%s/ipaq-sleep.conf",g_get_home_dir());
+  sprintf(cname,"%s/.sleep.conf",g_get_home_dir());
   ISconf = load_ISconf(cname);
   if(ISconf == NULL) {
     strcpy(cname, "/etc/ipaq-sleep.conf");
