Index: linux-2.6.21/arch/arm/mach-pxa/ezx-a780.c
===================================================================
--- linux-2.6.21.orig/arch/arm/mach-pxa/ezx-a780.c	2007-08-31 23:58:17.000000000 -0300
+++ linux-2.6.21/arch/arm/mach-pxa/ezx-a780.c	2007-08-31 23:59:03.000000000 -0300
@@ -17,6 +17,7 @@
 #include <linux/mmc/host.h>
 #include <linux/irq.h>
 #include <linux/input.h>
+#include <linux/gpio_keys.h>
 
 #include <asm/mach-types.h>
 #include <asm/mach/arch.h>
@@ -311,10 +312,33 @@
 	.resource       = pcap_ts_resources,
 };
 
+static struct gpio_keys_button a780flip_buttons[] = {
+       [0] = {
+               .keycode        = KEY_SLEEP,
+               .gpio           = GPIO_FLIP_PIN,
+               .desc           = "A780 flip",
+       },
+};
+
+static struct gpio_keys_platform_data a780flip_platform_data = {
+       .buttons        = a780flip_buttons,
+       .nbuttons       = 1,
+};
+
+static struct platform_device a780flip_device = {
+       .name           = "gpio-keys",
+       .id             = -1,
+       .dev            = {
+               .platform_data  = &a780flip_platform_data,
+       },
+};
+
+
 static struct platform_device *devices[] __initdata = {
 	&a780_pcap_device,
 	&a780_emu_device,
 	&pcap_ts_device,
+	&a780flip_device,
 };
 
 static void __init a780_init(void)
