Index: linux-2.6.21/arch/arm/mach-pxa/ezx-e6.c
===================================================================
--- linux-2.6.21.orig/arch/arm/mach-pxa/ezx-e6.c	2007-09-24 20:14:54.000000000 -0300
+++ linux-2.6.21/arch/arm/mach-pxa/ezx-e6.c	2007-09-24 20:15:18.000000000 -0300
@@ -203,8 +203,15 @@
 	},
 };
 
+
+struct platform_device e6_eoc_device = {
+	.name		= "ezx-eoc",
+	.id		= -1,
+};
+
 static struct platform_device *devices[] __initdata = {
 	&e6_pcap_device,
+	&e6_eoc_device,
 };
 
 static void __init e6_init(void)
