Index: linux-2.6.21/arch/arm/mach-pxa/ezx-e6.c
===================================================================
--- linux-2.6.21.orig/arch/arm/mach-pxa/ezx-e6.c	2007-09-25 00:03:01.000000000 -0300
+++ linux-2.6.21/arch/arm/mach-pxa/ezx-e6.c	2007-09-25 02:00:58.000000000 -0300
@@ -209,9 +209,34 @@
 	.id		= -1,
 };
 
+/* PCAP_TS */
+struct resource pcap_ts_resources[] = {
+	[0] = {
+		.start          = EZX_IRQ_ADCDONE,
+		.end            = EZX_IRQ_ADCDONE,
+		.flags          = IORESOURCE_IRQ,
+	},
+	[1] = {
+		.start          = EZX_IRQ_TS,
+		.end            = EZX_IRQ_TS,
+		.flags          = IORESOURCE_IRQ,
+	}
+};
+
+struct platform_device pcap_ts_device = {
+	.name           = "pcap-ts",
+	.id             = -1,
+	.dev		= {
+		.parent = &e6_pcap_device.dev,
+	},
+	.num_resources  = ARRAY_SIZE(pcap_ts_resources),
+	.resource       = pcap_ts_resources,
+};
+
 static struct platform_device *devices[] __initdata = {
 	&e6_pcap_device,
 	&e6_eoc_device,
+	&pcap_ts_device,
 };
 
 static void __init e6_init(void)
