--- ion3-20041104.orig/etc/cfg_bindings.lua	2004-10-05 09:03:31.000000000 +0000
+++ ion3-20041104/etc/cfg_bindings.lua	2004-11-27 23:06:00.000000000 +0000
@@ -52,7 +52,7 @@
     kpress(MOD1.."F1", "ioncore.show_manual()"),
     
     bdoc("Run a terminal emulator."),
-    kpress(MOD2.."F2", "ioncore.exec_on(_, 'xterm')"),
+    kpress(MOD2.."F2", "ioncore.exec_on(_, 'x-terminal-emulator')"),
     
     bdoc("Create a new workspace of chosen default type."),
     kpress(MOD1.."F9", "ioncore.create_ws(_)"),
