--- ltrace-0.4.orig/configure.ac
+++ ltrace-0.4/configure.ac
@@ -41,13 +41,6 @@
 dnl   AC_CHECK_FUNCS(gettimeofday strdup strerror strspn)
 AC_CHECK_FUNCS(getopt getopt_long)
 
-for path in /usr/include/elfutils /usr/local/include/elfutils \
-	    /usr/include/libelf /usr/local/include/libelf; do
-	if test -f ${path}/gelf.h; then
-		CPPFLAGS="$CPPFLAGS -I ${path}"
-	fi
-done
-
 AC_CHECK_HEADER(gelf.h,,AC_MSG_ERROR([***** gelf.h not found *****]))
 		 
 AC_CHECK_LIB(elf,gelf_getdyn)
