Index: src/e_mod_win.c
===================================================================
--- src/e_mod_win.c	(revision 14)
+++ src/e_mod_win.c	(working copy)
@@ -1,4 +1,4 @@
-#include <e.h>
+#include <enlightenment/e.h>
 #include "e_mod_win.h"
 #include "e_slipshelf.h"
 #include "e_slipwin.h"
Index: src/e_kbd.c
===================================================================
--- src/e_kbd.c	(revision 14)
+++ src/e_kbd.c	(working copy)
@@ -1,4 +1,4 @@
-#include <e.h>
+#include <enlightenment/e.h>
 #include "e_kbd.h"
 
 /* internal calls */
Index: src/e_slip.c
===================================================================
--- src/e_slip.c	(revision 14)
+++ src/e_slip.c	(working copy)
@@ -1,4 +1,4 @@
-#include <e.h>
+#include <enlightenment/e.h>
 #include "e_slip.h"
 
 EAPI int E_EVENT_SLIP_DEL = 0;
Index: src/e_mod_layout.c
===================================================================
--- src/e_mod_layout.c	(revision 14)
+++ src/e_mod_layout.c	(working copy)
@@ -1,4 +1,4 @@
-#include <e.h>
+#include <enlightenment/e.h>
 #include "e_mod_main.h"
 #include "e_mod_layout.h"
 
Index: src/e_mod_main.c
===================================================================
--- src/e_mod_main.c	(revision 14)
+++ src/e_mod_main.c	(working copy)
@@ -1,4 +1,4 @@
-#include <e.h>
+#include <enlightenment/e.h>
 
 #include "e_kbd.h"
 #include "e_slipshelf.h"
Index: src/e_slipwin.c
===================================================================
--- src/e_slipwin.c	(revision 14)
+++ src/e_slipwin.c	(working copy)
@@ -1,4 +1,4 @@
-#include <e.h>
+#include <enlightenment/e.h>
 #include "e_slipwin.h"
 
 EAPI int E_EVENT_SLIPWIN_DEL = 0;
Index: src/e_slipshelf.c
===================================================================
--- src/e_slipshelf.c	(revision 14)
+++ src/e_slipshelf.c	(working copy)
@@ -1,4 +1,4 @@
-#include <e.h>
+#include <enlightenment/e.h>
 #include "e_slipshelf.h"
 
 /* FIXME: break out gadcons for 2 swallow areas */
