Index: configure.in
===================================================================
--- configure.in	(Revision 17)
+++ configure.in	(Arbeitskopie)
@@ -28,6 +28,9 @@
 AC_SUBST(LOCALEDIR, "${localedir}")
 AC_DEFINE_UNQUOTED(LOCALEDIR, "${localedir}", "Module Locale Directory")
 
+keyboarddir="${datarootdir}/${PACKAGE}/keyboard"
+AC_SUBST(keyboarddir)
+
 #
 # Check EFL Libs
 #
Index: src/e_kbd.c
===================================================================
--- src/e_kbd.c	(Revision 17)
+++ src/e_kbd.c	(Arbeitskopie)
@@ -57,6 +57,7 @@
 
 static int _e_kbd_config_parse(E_Kbd *kbd, const char *config);
 static Evas_Object *_theme_obj_new(Evas *e, const char *custom_dir, const char *group);
+static char* _e_kbd_get_default_path(void);
 
 /* state */
 
@@ -470,7 +471,7 @@
    edje_object_part_swallow(kbd->base_obj, "e.swallow.content", o);
    evas_object_show(o);
    kbd->layout_obj = o;
-   _e_kbd_config_parse(kbd, "/home/raster/work/illume/default.kbd");
+   _e_kbd_config_parse(kbd, _e_kbd_get_default_path());
    _e_kbd_layout_build(kbd);
    
    edje_object_size_min_calc(kbd->base_obj, &mw, &mh);
@@ -744,3 +745,26 @@
      }
    return o;
 }
+
+/*
+ * Return the to be used keyboard file
+ */
+static char *
+_e_kbd_get_default_path()
+{
+   static char *buf = 0;
+
+   if (getenv("ILLUME_KEYBOARD_DIR"))
+     {
+        if (!buf)
+           {
+              buf = (char*)malloc(sizeof(char)*PATH_MAX);
+           }
+
+        snprintf(buf, PATH_MAX, "%s/default.kbd", getenv("ILLUME_KEYBOARD_DIR"));
+        return buf;
+     }
+
+
+    return KEYBOARDDIR"/default.kbd";
+}
Index: src/Makefile.am
===================================================================
--- src/Makefile.am	(Revision 17)
+++ src/Makefile.am	(Arbeitskopie)
@@ -1,6 +1,8 @@
 ACLOCAL_AMFLAGS = -I m4
 MAINTAINERCLEANFILES = Makefile.in
 
+AM_CPPFLAGS = -DKEYBOARDDIR=\"$(keyboarddir)\"
+
 INCLUDES = -I. \
 	 -I$(top_srcdir) \
 	 @e_cflags@
Index: Makefile.am
===================================================================
--- Makefile.am	(Revision 17)
+++ Makefile.am	(Arbeitskopie)
@@ -14,6 +14,9 @@
 files_DATA = module.desktop \
 	     e-module-illume.edj \
 	     illume.edj
+
+keybddir = $(keyboarddir)
+keybd_DATA = default.kbd
 	     
 EXTRA_DIST = $(files_DATA) \
 	     illume.edc \
