Index: diversity-xmpp-backend/Makefile.am
===================================================================
--- a/diversity-xmpp-backend/Makefile.am	(revision 276)
+++ a/diversity-xmpp-backend/Makefile.am	(working copy)
@@ -1,3 +1,4 @@
+if HAVE_XMPP
 INCLUDES = -I${top_srcdir}/include
 
 AM_CPPFLAGS =			\
@@ -4,9 +5,7 @@
 	$(GLIB_CFLAGS)		\
 	$(LM_CFLAGS)
 
-if HAVE_XMPP
 noinst_LTLIBRARIES = libdiversity-xmpp-backend.la
-endif
 
 noinst_HEADERS = diversity-xmpp-backend.h
 
@@ -48,3 +47,4 @@
 
 xmpp-marshal.h: xmpp-marshal.list
 	$(GLIB_GENMARSHAL) --prefix xmpp_marshal --header $< > $@
+endif
