
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- library/widget_showing.cpp~nomax
+++ library/widget_showing.cpp
@@ -51,9 +51,12 @@
         wg->show();
         return;
     }
-
+#ifndef OPIE_NO_WINDOWED
+    if ( TRUE ) {
+#else
     if ( !nomax
          && ( qApp->desktop()->width() <= 320 ) ){
+#endif
         wg->showMaximized();
     } else {
 #ifdef Q_WS_QWS
--- library/qpeapplication.cpp~nomax
+++ library/qpeapplication.cpp
@@ -251,6 +251,7 @@
 
     static bool read_widget_rect(const QString &app, bool &maximized, QPoint &p, QSize &s)
     {
+#ifndef OPIE_NO_WINDOWED
     maximized = TRUE;
     // 350 is the trigger in qwsdefaultdecoration for providing a resize button
     if ( qApp->desktop()->width() <= 350 )
@@ -272,7 +273,7 @@
 
         return TRUE;
     }
-
+#endif
     return FALSE;
     }
 
@@ -320,6 +321,7 @@
 
     static void store_widget_rect(QWidget *w, QString &app)
     {
+#ifndef OPIE_NO_WINDOWED
     // 350 is the trigger in qwsdefaultdecoration for providing a resize button
     if ( qApp->desktop()->width() <= 350 )
         return;
@@ -340,6 +342,7 @@
     QString s;
     s.sprintf("%d,%d,%d,%d,%d", r.left() + offsetX, r.top() + offsetY, r.width(), r.height(), w->isMaximized() );
     cfg.writeEntry( app, s );
+#endif
     }
 
     static bool setWidgetCaptionFromAppName( QWidget* /*mw*/, const QString& /*appName*/, const QString& /*appsPath*/ )
