--- configure.in.orig	2004-03-14 23:30:31.000000000 +0000
+++ configure.in	2004-03-15 00:00:51.000000000 +0000
@@ -340,7 +340,7 @@
 AC_MSG_CHECKING([Whether to write dependencies into .pc files])
 case $enable_explicit_deps in
   auto)
-    deplib_check_method=`(./libtool --config; echo eval echo \\$deplib_check_method) | sh`
+    deplib_check_method=`($host_alias-libtool --config; echo eval echo \\$deplib_check_method) | sh`
     if test "X$deplib_check_method" = Xnone || test "x$enable_static" = xyes ; then
       enable_explicit_deps=yes  
     else
@@ -664,7 +664,7 @@
     dnl Now we check to see if our libtool supports shared lib deps
     dnl (in a rather ugly way even)
     if $dynworks; then
-        pixbuf_libtool_config="${CONFIG_SHELL-/bin/sh} ./libtool --config"
+        pixbuf_libtool_config="${CONFIG_SHELL-/bin/sh} $host_alias-libtool --config"
         pixbuf_deplibs_check=`$pixbuf_libtool_config | \
             grep '^[[a-z_]]*check[[a-z_]]*_method=[['\''"]]' | \
             sed 's/.*[['\''"]]\(.*\)[['\''"]]$/\1/'`
