--- prelink-0.0.20030902.orig/src/dso.c	2003-08-11 11:55:11.000000000 +0000
+++ prelink-0.0.20030902/src/dso.c	2003-09-16 23:26:16.000000000 +0000
@@ -334,7 +334,13 @@
       }
     else
       sections[--k] = i;
+/* this fails for the statically linked upx executable
   assert (j == k);
+*/
+  if (j != k) {
+      printf("assert(j == k) at dso.c:311 has failed for file %s\n", name);
+      goto error_out;
+  }
 
   section_cmp_dso = dso;
   qsort (sections + k, dso->ehdr.e_shnum - k, sizeof (*sections), section_cmp);
