Index: gcc/ChangeLog
from  Alexandre Oliva  <aoliva@redhat.com>

	PR middle-end/15666
	* c-decl.c (finish_decl): Use change_decl_assembler_name for the
	builtin decl as well.

Index: gcc/c-decl.c
===================================================================
RCS file: /cvs/gcc/gcc/gcc/c-decl.c,v
retrieving revision 1.504
diff -u -p -r1.504 c-decl.c
--- gcc/gcc/c-decl.c 31 May 2004 22:06:27 -0000 1.504
+++ gcc/gcc/c-decl.c 1 Jun 2004 04:03:41 -0000
@@ -2886,7 +2886,7 @@ finish_decl (tree decl, tree init, tree 
 	{
 	  tree builtin = built_in_decls [DECL_FUNCTION_CODE (decl)];
 	  SET_DECL_RTL (builtin, NULL_RTX);
-	  SET_DECL_ASSEMBLER_NAME (builtin, get_identifier (starred));
+	  change_decl_assembler_name (builtin, get_identifier (starred));
 #ifdef TARGET_MEM_FUNCTIONS
 	  if (DECL_FUNCTION_CODE (decl) == BUILT_IN_MEMCPY)
 	    init_block_move_fn (starred);
