--- gcc-3.4.0/gcc/config/arm/arm.c~	2004-04-12 17:17:05.000000000 +0100
+++ gcc-3.4.0/gcc/config/arm/arm.c	2004-04-25 21:27:08.000000000 +0100
@@ -2739,7 +2739,7 @@
 	   && GET_MODE_SIZE (mode) <= 4
 	   && arm_address_register_rtx_p (XEXP (x, 0), strict_p)
 	   && GET_CODE (XEXP (x, 1)) == PLUS
-	   && XEXP (XEXP (x, 1), 0) == XEXP (x, 0))
+	   && rtx_equal_p (XEXP (XEXP (x, 1), 0), XEXP (x, 0)))
     return arm_legitimate_index_p (mode, XEXP (XEXP (x, 1), 1), strict_p);
 
   /* After reload constants split into minipools will have addresses
