Index: qemu-0.10.3/net.c
===================================================================
--- qemu-0.10.3.orig/net.c	2009-05-19 17:46:54.650870195 +0400
+++ qemu-0.10.3/net.c	2009-05-19 17:47:05.612854521 +0400
@@ -955,7 +955,7 @@
         return -1;
     }
     memset(&ifr, 0, sizeof(ifr));
-    ifr.ifr_flags = IFF_TAP | IFF_NO_PI;
+    ifr.ifr_flags = IFF_TAP | IFF_NO_PI | IFF_ONE_QUEUE;
     if (ifname[0] != '\0')
         pstrcpy(ifr.ifr_name, IFNAMSIZ, ifname);
     else
