--- /scratch/obj.x86_64/oe/build/tmp/work/qemux86-oe-linux-uclibc/linux-2.6.38-r4/linux-2.6.38/arch/x86/include/asm/ptrace.h.old	2011-03-15 02:20:32.000000000 +0100
+++ /scratch/obj.x86_64/oe/build/tmp/work/qemux86-oe-linux-uclibc/linux-2.6.38-r4/linux-2.6.38/arch/x86/include/asm/ptrace.h	2011-05-04 20:21:37.755073517 +0200
@@ -130,6 +130,7 @@ struct pt_regs {
 
 #ifdef __KERNEL__
 
+#include <asm/linkage.h>
 #include <linux/init.h>
 
 struct cpuinfo_x86;
@@ -143,8 +144,8 @@ extern void send_sigtrap(struct task_str
 			 int error_code, int si_code);
 void signal_fault(struct pt_regs *regs, void __user *frame, char *where);
 
-extern long syscall_trace_enter(struct pt_regs *);
-extern void syscall_trace_leave(struct pt_regs *);
+extern asmregparm long syscall_trace_enter(struct pt_regs *);
+extern asmregparm void syscall_trace_leave(struct pt_regs *);
 
 static inline unsigned long regs_return_value(struct pt_regs *regs)
 {
