--- pulseaudio-0.9.15/configure.ac-orig	2010-03-19 07:06:56.000000000 -0700
+++ pulseaudio-0.9.15/configure.ac	2010-03-22 14:06:06.000000000 -0700
@@ -1294,7 +1294,7 @@ fi
 AC_ARG_WITH(
         [module-dir],
         AS_HELP_STRING([--with-module-dir],[Directory where to install the modules to (defaults to ${libdir}/pulse-${PA_MAJORMINORMICRO}/modules/]),
-        [modlibexecdir=$withval], [modlibexecdir="${libdir}/pulse-${PA_MAJORMINORMICRO}/modules/"])
+        [modlibexecdir=$withval], [modlibexecdir="${libdir}/pulse-${PA_MAJORMINORMICRO}/modules"])
 
 AC_SUBST(modlibexecdir)
 
