Index: liboil-0.3.15/configure.ac
===================================================================
--- liboil-0.3.15.orig/configure.ac	2008-07-01 06:45:53.000000000 +0200
+++ liboil-0.3.15/configure.ac	2008-09-25 13:44:18.633758105 +0200
@@ -187,16 +187,6 @@
     true)
 fi
 
-if test x$HAVE_GCC_ASM = xyes -a x$HAVE_ARM = xyes ; then
-  AS_COMPILER_FLAG(["-Wa,-mfpu=vfp"],
-    [VFP_CFLAGS="$VFP_CFLAGS -mfpu=vfp"],
-    true)
-  #AS_COMPILER_FLAG(["-Wa,-mfloat-abi=softfp"],
-  #  [VFP_CFLAGS="$VFP_CFLAGS -mfloat-abi=softfp"],
-  #  true)
-fi
-AC_SUBST(VFP_CFLAGS)
-
 #AS_COMPILER_FLAG(["-fasm-blocks"], HAVE_ASM_BLOCKS=yes, HAVE_ASM_BLOCKS=no)
 HAVE_ASM_BLOCKS=no
 if test "x$HAVE_ASM_BLOCKS" = "xyes"
