
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- qt-embedded-opensource-4.0.0-b1/qmake/generators/makefile.cpp~old-moc-compatibility.patch
+++ qt-embedded-opensource-4.0.0-b1/qmake/generators/makefile.cpp
@@ -950,7 +950,7 @@
                 deps += "$(MOC) ";
             deps += (*it) + " " + findDependencies((m)).join(" ");
             t << m << ": " << deps << "\n\t"
-              << "$(MOC)" << " $(DEFINES) $(INCPATH) " << varGlue("QMAKE_COMPILER_DEFINES","-D"," -D"," ")
+              << "$(MOC) " /* << " $(DEFINES) $(INCPATH) " << varGlue("QMAKE_COMPILER_DEFINES","-D"," -D"," ") */
               << (*it) << " -o " << m << endl << endl;
         }
     }
