
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- qt-embedded-opensource-4.0.0-b1/mkspecs/features/uic.prf~old-uic-compatibility.patch
+++ qt-embedded-opensource-4.0.0-b1/mkspecs/features/uic.prf
@@ -13,8 +13,8 @@
 isEmpty(UI_SOURCES_DIR):UI_SOURCES_DIR = $$UI_DIR
 isEmpty(UI_HEADERS_DIR):UI_HEADERS_DIR = $$UI_DIR
 
-uic.commands = $$QMAKE_UIC -3 $$QMAKE_UIC3 ${QMAKE_FILE_IN} -o ${QMAKE_FILE_OUT}
-uic.output = $$UI_HEADERS_DIR/ui_${QMAKE_FILE_BASE}$${first(QMAKE_EXT_H)}
+uic.commands = $$QMAKE_UIC ${QMAKE_FILE_IN} -o ${QMAKE_FILE_OUT}
+uic.output = $$UI_HEADERS_DIR/${QMAKE_FILE_BASE}$${first(QMAKE_EXT_H)}
 uic.input = FORMS
 uic.variable_out = HEADERS
 uic.CONFIG += no_link
@@ -24,9 +24,9 @@
 INCLUDEPATH += $$UI_HEADERS_DIR
 
 uic3 {
-    uic3_decl.commands = $$QMAKE_UIC3 -decl ui_${QMAKE_FILE_BASE}$${first(QMAKE_EXT_H)} ${QMAKE_FILE_IN} -o ${QMAKE_FILE_OUT}
+    uic3_decl.commands = $$QMAKE_UIC3 -decl ${QMAKE_FILE_BASE}$${first(QMAKE_EXT_H)} ${QMAKE_FILE_IN} -o ${QMAKE_FILE_OUT}
     uic3_decl.output = $$UI_HEADERS_DIR/${QMAKE_FILE_BASE}$${first(QMAKE_EXT_H)}
-    uic3_decl.depends = $$QMAKE_UIC3 $$UI_HEADERS_DIR/ui_${QMAKE_FILE_BASE}$${first(QMAKE_EXT_H)}
+    uic3_decl.depends = $$QMAKE_UIC3 $$UI_HEADERS_DIR/${QMAKE_FILE_BASE}$${first(QMAKE_EXT_H)}
     uic3_decl.input = FORMS
     uic3_decl.CONFIG += no_link
     uic3_decl.variable_out = HEADERS
