Index: xorg-server-1.3.0.0/hw/kdrive/glamo/glamo.c
===================================================================
--- xorg-server-1.3.0.0.orig/hw/kdrive/glamo/glamo.c	2007-11-30 10:18:43.000000000 +0100
+++ xorg-server-1.3.0.0/hw/kdrive/glamo/glamo.c	2007-11-30 10:35:27.000000000 +0100
@@ -38,9 +38,7 @@
 	if (glamoc == NULL)
 		return FALSE;
 
-#ifdef KDRIVEFBDEV
 	if (!initialized && fbdevInitialize(card, &glamoc->backend_priv.fbdev)) {
-		glamoc->use_fbdev = TRUE;
 		initialized = TRUE;
 		glamoc->backend_funcs.cardfini = fbdevCardFini;
 		glamoc->backend_funcs.scrfini = fbdevScreenFini;
@@ -58,29 +56,6 @@
 		glamoc->backend_funcs.randrSetConfig = fbdevRandRSetConfig;
 #endif
 	}
-#endif
-#ifdef KDRIVEVESA
-	if (!initialized && vesaInitialize(card, &glamoc->backend_priv.vesa)) {
-		glamoc->use_vesa = TRUE;
-		initialized = TRUE;
-		glamoc->backend_funcs.cardfini = vesaCardFini;
-		glamoc->backend_funcs.scrfini = vesaScreenFini;
-		glamoc->backend_funcs.initScreen = vesaInitScreen;
-		glamoc->backend_funcs.finishInitScreen = vesaFinishInitScreen;
-		glamoc->backend_funcs.createRes = vesaCreateResources;
-		glamoc->backend_funcs.preserve = vesaPreserve;
-		glamoc->backend_funcs.restore = vesaRestore;
-		glamoc->backend_funcs.dpms = vesaDPMS;
-		glamoc->backend_funcs.enable = vesaEnable;
-		glamoc->backend_funcs.disable = vesaDisable;
-		glamoc->backend_funcs.getColors = vesaGetColors;
-		glamoc->backend_funcs.putColors = vesaPutColors;
-#ifdef RANDR
-		glamoc->backend_funcs.randrSetConfig = vesaRandRSetConfig;
-#endif
-	}
-#endif
-
 	if (!initialized || !GLAMOMapReg(card, glamoc)) {
 		xfree(glamoc);
 		return FALSE;
@@ -158,18 +133,8 @@
 
 	if (screen->fb[0].depth == 0)
 		screen->fb[0].depth = 16;
-#ifdef KDRIVEFBDEV
-	if (glamoc->use_fbdev) {
-		success = fbdevScreenInitialize(screen,
-						&glamos->backend_priv.fbdev);
-	}
-#endif
-#ifdef KDRIVEVESA
-	if (glamoc->use_vesa) {
-		success = vesaScreenInitialize(screen,
-					       &glamos->backend_priv.vesa);
-	}
-#endif
+
+	success = fbdevScreenInitialize(screen, &glamos->backend_priv.fbdev);
 
 	if (!success) {
 		screen->driver = NULL;
Index: xorg-server-1.3.0.0/hw/kdrive/glamo/glamo.h
===================================================================
--- xorg-server-1.3.0.0.orig/hw/kdrive/glamo/glamo.h	2007-11-30 10:25:06.000000000 +0100
+++ xorg-server-1.3.0.0/hw/kdrive/glamo/glamo.h	2007-11-30 10:35:17.000000000 +0100
@@ -30,13 +30,7 @@
 #include <kdrive-config.h>
 #endif
 
-#ifdef KDRIVEFBDEV
 #include <fbdev.h>
-#endif
-#ifdef KDRIVEVESA
-#include <vesa.h>
-#endif
-
 #include "kxv.h"
 
 #define GLAMO_REG_BASE(c)		((c)->attr.address[0])
@@ -99,12 +93,7 @@
 
 typedef struct _GLAMOCardInfo {
 	union {
-#ifdef KDRIVEFBDEV
 		FbdevPriv fbdev;
-#endif
-#ifdef KDRIVEVESA
-		VesaCardPrivRec vesa;
-#endif
 	} backend_priv;
 	struct backend_funcs backend_funcs;
 
@@ -112,7 +101,6 @@
 	Bool is_3362;
 	CARD32 crtc_pitch;
 	CARD32 crtc2_pitch;
-	Bool use_fbdev, use_vesa;
 } GLAMOCardInfo;
 
 #define getGLAMOCardInfo(kd)	((GLAMOCardInfo *) ((kd)->card->driver))
@@ -155,12 +143,7 @@
 
 typedef struct _GLAMOScreenInfo {
 	union {
-#ifdef KDRIVEFBDEV
 		FbdevScrPriv fbdev;
-#endif
-#ifdef KDRIVEVESA
-		VesaScreenPrivRec vesa;
-#endif
 	} backend_priv;
 	KaaScreenInfoRec kaa;
 
Index: xorg-server-1.3.0.0/hw/kdrive/glamo/glamo_stub.c
===================================================================
--- xorg-server-1.3.0.0.orig/hw/kdrive/glamo/glamo_stub.c	2007-11-28 18:44:27.000000000 +0100
+++ xorg-server-1.3.0.0/hw/kdrive/glamo/glamo_stub.c	2007-11-30 10:34:10.000000000 +0100
@@ -59,9 +59,6 @@
 ddxUseMsg (void)
 {
 	KdUseMsg();
-#ifdef KDRIVEVESA
-	vesaUseMsg();
-#endif
 }
 
 int
@@ -69,9 +66,6 @@
 {
 	int	ret;
 
-#ifdef KDRIVEVESA
-	if (!(ret = vesaProcessArgument (argc, argv, i)))
-#endif
 		ret = KdProcessArgument(argc, argv, i);
 
 	return ret;
