
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- qt-x11-opensource-src-4.0.1/configure~cross-compile
+++ qt-x11-opensource-src-4.0.1/configure
@@ -477,7 +477,7 @@
             UNKNOWN_ARG=yes
         fi
         ;;
-    -prefix|-docdir|-headerdir|-plugindir|-datadir|-libdir|-bindir|-translationdir|-sysconfdir|-examplesdir|-demosdir|-depths|-make|-nomake|-platform|-xplatform|-buildkey)
+    -prefix|-docdir|-headerdir|-plugindir|-datadir|-libdir|-bindir|-translationdir|-sysconfdir|-examplesdir|-demosdir|-depths|-make|-nomake|-platform|-xplatform|-buildkey|-crossarch)
         VAR=`echo $1 | sed "s,^-\(.*\),\1,"`
         shift
         VAL=$1
@@ -768,6 +768,9 @@
     xplatform)
         XPLATFORM="$VAL"
         ;;
+    crossarch)
+        CROSSARCH="$VAL"
+        ;;
     debug-and-release)
         if [ "$VAL" = "yes" ] || [ "$VAL" = "no" ]; then
             CFG_DEBUG_RELEASE="$VAL"
@@ -1649,100 +1652,7 @@
     echo "Determining system architecture... ($UNAME_SYSTEM:$UNAME_RELEASE:$UNAME_MACHINE)"
 fi
 
-if [ "$PLATFORM" != "$XPLATFORM" -a "$CFG_EMBEDDED" != "no" ]; then
-    case "$CFG_EMBEDDED" in
-    x86)
-        ARCH=i386
-        ;;
-    ipaq|sharp)
-        ARCH=arm
-        ;;
-    *)
-        ARCH="$CFG_EMBEDDED"
-        ;;
-    esac
-else
-    case "$UNAME_SYSTEM:$UNAME_RELEASE:$UNAME_MACHINE" in
-    IRIX*:*:*)
-        ARCH=`uname -p`
-        if [ "$OPT_VERBOSE" = "yes" ]; then
-            echo "    SGI ($ARCH)"
-        fi
-        ;;
-    SunOS:5*:sun4*)
-        if [ "$OPT_VERBOSE" = "yes" ]; then
-            echo "    Sun SPARC (sparc)"
-        fi
-        ARCH=sparc
-        ;;
-    Darwin:*:Power?Macintosh)
-        if [ "$OPT_VERBOSE" = "yes" ]; then
-            echo "    32-bit Apple PowerPC (powerpc)"
-        fi
-        ARCH=powerpc
-        ;;
-    Darwin:*:x86)
-        if [ "$OPT_VERBOSE" = "yes" ]; then
-            echo "    32-bit Apple Intel (x86)"
-        fi
-        ARCH=i386
-        ;;
-    AIX:*:00????????00)
-        if [ "$OPT_VERBOSE" = "yes" ]; then
-        echo "    64-bit IBM PowerPC (powerpc)"
-        fi
-        ARCH=powerpc
-        ;;
-    HP-UX:*:9000*)
-        if [ "$OPT_VERBOSE" = "yes" ]; then
-            echo "    HP PA-RISC (parisc)"
-        fi
-        ARCH=parisc
-        ;;
-    *:*:i?86)
-        if [ "$OPT_VERBOSE" = "yes" ]; then
-            echo "    32-bit Intel 80x86 (i386)"
-        fi
-        ARCH=i386
-        ;;
-    *:*:x86_64)
-        if [ "$PLATFORM" = "linux-g++-32" ]; then
-            if [ "$OPT_VERBOSE" = "yes" ]; then
-                echo "    32 bit on 64-bit AMD 80x86 (i386)"
-            fi
-            ARCH=i386
-        else
-            if [ "$OPT_VERBOSE" = "yes" ]; then
-                echo "    64-bit AMD 80x86 (x86_64)"
-            fi
-            ARCH=x86_64
-        fi
-        ;;
-    *:*:ppc)
-        if [ "$OPT_VERBOSE" = "yes" ]; then
-            echo "    32-bit PowerPC (powerpc)"
-        fi
-        ARCH=powerpc
-        ;;
-    *:*:*)
-        if [ "$OPT_VERBOSE" = "yes" ]; then
-            echo "    Trying '$UNAME_MACHINE'..."
-        fi
-        ARCH="$UNAME_MACHINE"
-        ;;
-    esac
-fi
-
-if [ -d "$relpath/src/corelib/arch/$ARCH" ]; then
-    if [ "$OPT_VERBOSE" = "yes" ]; then
-        echo "    '$ARCH' is supported"
-    fi
-else
-    if [ "$OPT_VERBOSE" = "yes" ]; then
-        echo "    '$ARCH' is unsupported, using 'generic'"
-    fi
-    ARCH=generic
-fi
+ARCH="$CROSSARCH"
 
 if [ "$OPT_VERBOSE" = "yes" ]; then
         echo "System architecture: '$ARCH'"
