--- enlightenment-0.16.999.015/src/bin/e_config.c~	2005-09-22 20:50:16.000000000 -0700
+++ enlightenment-0.16.999.015/src/bin/e_config.c	2005-09-30 10:51:02.000000000 -0700
@@ -377,7 +377,7 @@
 	/* DEFAULT CONFIG */
 	e_config = E_NEW(E_Config, 1);
 	e_config->config_version = E_CONFIG_FILE_VERSION;
-	e_config->show_splash = 1;
+	e_config->show_splash = 0;
 	e_config->desktop_default_background = strdup("");
 	e_config->desktop_default_name = strdup("Desktop %i, %i");
 	e_config->menus_scroll_speed = 1000.0;
