From 35f013c24fb78d6ae7257abb4b4ed5f356cac36b Mon Sep 17 00:00:00 2001
From: Brijesh Singh <bksingh@ti.com>
Date: Sun, 9 May 2010 05:27:41 -0500
Subject: [PATCH] gstreamer ti dm6467 usesinglecsserver

---
 src/gstticodecplugin_dm6467.cfg |   24 +++++++-----------------
 1 files changed, 7 insertions(+), 17 deletions(-)

diff --git a/src/gstticodecplugin_dm6467.cfg b/src/gstticodecplugin_dm6467.cfg
index a141f03..9e64775 100644
--- a/src/gstticodecplugin_dm6467.cfg
+++ b/src/gstticodecplugin_dm6467.cfg
@@ -29,25 +29,15 @@ TraceUtil.attrs = TraceUtil.SOCRATES_TRACING;
  *  ======== Engine Configuration ========
  */
 var Engine = xdc.useModule('ti.sdo.ce.Engine');
-var decodeEngine = Engine.createFromServer(
-    "decode",
-    "./decodeCombo.x64P",
-    "ti.sdo.servers.decode"
+var demoEngine = Engine.createFromServer(
+    "codecServer",
+    "./bin/cs.x64P",
+    "ti.sdo.server.cs"
     );
-var encodeEngine = Engine.createFromServer(
-    "encode",
-    "./encodeCombo.x64P",
-    "ti.sdo.servers.encode"
-    );
-
-var encpath = "" + java.lang.System.getenv("ENCODE_COMBO");
-if (encpath != "" && encpath != "null") {
-	encodeEngine.server = java.lang.System.getenv("ENCODE_COMBO");
-}
 
-var decpath = "" + java.lang.System.getenv("DECODE_COMBO");
-if (decpath != "" && decpath != "null") {
-	decodeEngine.server = java.lang.System.getenv("DECODE_COMBO"); 
+var combopath = "" + java.lang.System.getenv("CODEC_SERVER");
+if (combopath != "" && combopath != "null") {
+        demoEngine.server = java.lang.System.getenv("CODEC_SERVER");
 }
 
 /* Load support for the 'Davinci Multimedia Application Interface' modules */
-- 
1.5.4.3

