Index: gdal-1.5.2/apps/GNUmakefile
===================================================================
--- gdal-1.5.2.orig/apps/GNUmakefile	2007-12-21 04:20:10.000000000 +0100
+++ gdal-1.5.2/apps/GNUmakefile	2008-11-09 20:58:31.506906134 +0100
@@ -21,7 +21,7 @@
 		ogrtindex$(EXE)
 endif
 
-default:	gdal-config-inst gdal-config $(BIN_LIST)
+default:	gdal-config $(BIN_LIST)
 
 lib-depend:
 	(cd ../gcore ; $(MAKE) )
@@ -108,41 +108,26 @@
 		 -o test_ogrsf$(EXE)
 
 clean:
-	$(RM) *.o $(BIN_LIST) core gdal-config gdal-config-inst
+	$(RM) *.o $(BIN_LIST) core gdal-config
 
 $(DEP_LIBS):
 
-
 gdal-config:	gdal-config.in ../GDALmake.opt ./GNUmakefile ../VERSION
 	rm -f gdal-config
 	echo '#!/bin/sh' > gdal-config
-	echo 'CONFIG_LIBS="$(CONFIG_LIBS)"' >> gdal-config
+	echo 'CONFIG_LIBS="$(CONFIG_LIBS_INS)"' >> gdal-config
 	echo 'CONFIG_DEP_LIBS="$(LIBS)"' >> gdal-config
-	echo 'CONFIG_PREFIX="$(GDAL_ROOT)"' >> gdal-config
-	echo 'CONFIG_CFLAGS="-I$(GDAL_ROOT)/port -I$(GDAL_ROOT)/gcore -I$(GDAL_ROOT)/alg -I$(GDAL_ROOT)/ogr -I$(GDAL_ROOT)/ogr/ogrsf_frmts"' >> gdal-config
-	echo 'CONFIG_DATA="$(GDAL_ROOT)/data"' >> gdal-config
+	echo 'CONFIG_PREFIX="$(INST_PREFIX)"' >> gdal-config
+	echo 'CONFIG_CFLAGS="-I$(INST_INCLUDE)"' >> gdal-config
+	echo 'CONFIG_DATA="$(INST_DATA)"' >> gdal-config
 	echo 'CONFIG_VERSION="'`cat ../VERSION`'"' >> gdal-config
 	echo 'CONFIG_OGR_ENABLED=$(OGR_ENABLED)' >> gdal-config
 	echo 'CONFIG_FORMATS="$(GDAL_FORMATS)"' >> gdal-config
 	cat gdal-config.in >> gdal-config
 	chmod a+x gdal-config
 
-gdal-config-inst:	gdal-config.in ../GDALmake.opt ./GNUmakefile ../VERSION
-	rm -f gdal-config-inst
-	echo '#!/bin/sh' > gdal-config-inst
-	echo 'CONFIG_LIBS="$(CONFIG_LIBS_INS)"' >> gdal-config-inst
-	echo 'CONFIG_DEP_LIBS="$(LIBS)"' >> gdal-config-inst
-	echo 'CONFIG_PREFIX="$(INST_PREFIX)"' >> gdal-config-inst
-	echo 'CONFIG_CFLAGS="-I$(INST_INCLUDE)"' >> gdal-config-inst
-	echo 'CONFIG_DATA="$(INST_DATA)"' >> gdal-config-inst
-	echo 'CONFIG_VERSION="'`cat ../VERSION`'"' >> gdal-config-inst
-	echo 'CONFIG_OGR_ENABLED=$(OGR_ENABLED)' >> gdal-config-inst
-	echo 'CONFIG_FORMATS="$(GDAL_FORMATS)"' >> gdal-config-inst
-	cat gdal-config.in >> gdal-config-inst
-	chmod a+x gdal-config-inst
-
 
 install: default
 	for f in $(BIN_LIST) ; do $(INSTALL) $$f $(DESTDIR)$(INST_BIN) ; done
-	$(INSTALL) gdal-config-inst $(DESTDIR)$(INST_BIN)/gdal-config
+	$(INSTALL) gdal-config $(DESTDIR)$(INST_BIN)/
 
