Index: git/arch/arm/mach-kirkwood/openrd_base-setup.c
===================================================================
--- git.orig/arch/arm/mach-kirkwood/openrd_base-setup.c	2009-09-19 19:31:45.000000000 +0200
+++ git/arch/arm/mach-kirkwood/openrd_base-setup.c	2009-09-19 19:31:54.000000000 +0200
@@ -70,8 +70,17 @@
 	kirkwood_ge00_init(&openrd_base_ge00_data);
 	kirkwood_sata_init(&openrd_base_sata_data);
 	kirkwood_sdio_init(&openrd_base_mvsdio_data);
+	kirkwood_i2c_init();
 }
 
+static int __init openrd_base_pci_init(void)
+{
+	if (machine_is_openrd_base())
+		kirkwood_pcie_init();
+	return 0;
+}
+subsys_initcall(openrd_base_pci_init);
+
 MACHINE_START(OPENRD_BASE, "Marvell OpenRD Base Board")
 	/* Maintainer: Dhaval Vasa <dhaval.vasa@einfochips.com> */
 	.phys_io	= KIRKWOOD_REGS_PHYS_BASE,
