https://bugzilla.redhat.com/attachment.cgi?id=423232

Index: libtiff/tif_ojpeg.c
===================================================================
--- libtiff/tif_ojpeg.c.orig
+++ libtiff/tif_ojpeg.c
@@ -1937,6 +1937,10 @@ OJPEGReadBufferFill(OJPEGState* sp)
 					sp->in_buffer_source=osibsEof;
 				else
 				{
+					if (sp->tif->tif_dir.td_stripoffset == 0) {
+						TIFFErrorExt(sp->tif->tif_clientdata,sp->tif->tif_name,"Strip offsets are missing");
+						return(0);
+					}
 					sp->in_buffer_file_pos=sp->tif->tif_dir.td_stripoffset[sp->in_buffer_next_strile];
 					if (sp->in_buffer_file_pos!=0)
 					{
