--- linux-2.4.26-umlhipac/arch/um/fs/hostfs/externfs.c.orig	Tue May 18 00:10:42 2004
+++ linux-2.4.26-umlhipac/arch/um/fs/hostfs/externfs.c	Thu May 20 21:12:54 2004
@@ -961,7 +961,7 @@
 	if(name == NULL) 
 		return(-ENOMEM);
 
-	err = (*ops->access_file)(name, r, w, x, current->fsuid, current->fsgid,
+	err = (*ops->access_file)(name, current->fsuid, current->fsgid, r, w, x,
 				  mount);
 	kfree(name);
 
