
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- imlib2-1.1.0/src/rgbadraw.c~gcc34
+++ imlib2-1.1.0/src/rgbadraw.c
@@ -2357,14 +2357,14 @@
                        /* last span on line and still not < ss->x */
                        if (ps)
                           ps->next = s;
-                     nospans:
+                     nospans:;
                     }
                }
              if (i == y2)
                 goto nolines;
              i += step;
           }
-      nolines:
+      nolines:;
      }
    for (i = 0; i < h; i++)
      {
