Index: binutils/bfd/elf32-arm.c
===================================================================
RCS file: /cvs/src/src/bfd/elf32-arm.c,v
retrieving revision 1.162
retrieving revision 1.163
diff -u -r1.162 -r1.163
--- binutils/bfd/elf32-arm.c	23 Dec 2008 09:01:45 -0000	1.162
+++ binutils/bfd/elf32-arm.c	23 Dec 2008 11:46:17 -0000	1.163
@@ -4608,6 +4608,10 @@
   Elf_Internal_Shdr *hdr;
   unsigned int i, localsyms;
 
+  /* PR 7093: Make sure that we are dealing with an arm elf binary.  */
+  if (! is_arm_elf (abfd))
+    return;
+
   if ((abfd->flags & DYNAMIC) != 0)
     return;
 
