--- mplayer-1.1.5/configure.orig	2005-11-08 15:07:52.000000000 +0000
+++ mplayer-1.1.5/configure	2005-11-08 15:08:40.000000000 +0000
@@ -4627,33 +4627,7 @@
 echocheck "ALSA audio"
 if test "$_alsa" != no ; then
   _alsa=no
-  cat > $TMPC << EOF
-#include <sys/asoundlib.h>
-int main(void) { return (!(SND_LIB_MAJOR==0 && SND_LIB_MINOR==5)); }
-EOF
-  cc_check -lasound $_ld_dl $_ld_pthread && $TMPO && _alsaver='0.5.x'
-
-  cat > $TMPC << EOF
-#include <sys/asoundlib.h>
-int main(void) { return (!(SND_LIB_MAJOR==0 && SND_LIB_MINOR==9)); }
-EOF
-  cc_check -lasound $_ld_dl $_ld_pthread && $TMPO && _alsaver='0.9.x-sys'
-  cat > $TMPC << EOF
-#include <alsa/asoundlib.h>
-int main(void) { return (!(SND_LIB_MAJOR==0 && SND_LIB_MINOR==9)); }
-EOF
-  cc_check -lasound $_ld_dl $_ld_pthread && $TMPO && _alsaver='0.9.x-alsa'
-
-  cat > $TMPC << EOF
-#include <sys/asoundlib.h>
-int main(void) { return (!(SND_LIB_MAJOR==1 && SND_LIB_MINOR==0)); }
-EOF
-  cc_check -lasound $_ld_dl $_ld_pthread && $TMPO && _alsaver='1.0.x-sys'
-  cat > $TMPC << EOF
-#include <alsa/asoundlib.h>
-int main(void) { return (!(SND_LIB_MAJOR==1 && SND_LIB_MINOR==0)); }
-EOF
-  cc_check -lasound $_ld_dl $_ld_pthread && $TMPO && _alsaver='1.0.x-alsa'
+  _alsaver='1.0.x-alsa'
 fi
 _def_alsa5='#undef HAVE_ALSA5'
 _def_alsa9='#undef HAVE_ALSA9'
