--- ./configure.orig	2004-09-24 19:09:56.000000000 +0930
+++ ./configure	2004-09-24 19:22:21.000000000 +0930
@@ -3554,38 +3554,6 @@
 #################
 
 
-echocheck "SDL"
-if test -z "$_sdlconfig" ; then
-  if ( sdl-config --version ) >>"$TMPLOG" 2>&1 ; then
-    _sdlconfig="sdl-config"
-  elif ( sdl11-config --version ) >>"$TMPLOG" 2>&1 ; then
-    _sdlconfig="sdl11-config"
-  else
-    _sdlconfig=false
-  fi
-fi
-if test "$_sdl" = auto || test "$_sdl" = yes ; then
-  cat > $TMPC << EOF
-#include <SDL.h>
-int main(int argc, char *argv[]) { return 0; }
-EOF
-  _sdl=no
-  if "$_sdlconfig" --version >>"$TMPLOG" 2>&1 ; then
-    if cc_check `$_sdlconfig --cflags` `$_sdlconfig --libs` >>"$TMPLOG" 2>&1 ; then
-      _sdlversion=`$_sdlconfig --version | sed 's/[^0-9]//g'`
-        if test "$_sdlversion" -gt 116 ; then
-          if test "$_sdlversion" -lt 121 ; then
-            _def_sdlbuggy='#define BUGGY_SDL'
-          else
-            _def_sdlbuggy='#undef BUGGY_SDL'
-          fi
-          _sdl=yes
-        else
-          _sdl=outdated
-        fi
-     fi
-  fi
-fi
 if test "$_sdl" = yes ; then
   _def_sdl='#define HAVE_SDL 1'
   if cygwin ; then
