Index: u-boot-1.3.2/include/configs/MPC8313ERDB.h
===================================================================
--- u-boot-1.3.2.orig/include/configs/MPC8313ERDB.h	2008-09-12 18:47:22.000000000 +0200
+++ u-boot-1.3.2/include/configs/MPC8313ERDB.h	2008-09-12 18:48:11.000000000 +0200
@@ -460,6 +460,10 @@
 	HRCWH_BIG_ENDIAN |\
 	HRCWH_LALE_NORMAL)
 
+/* Watchdog */
+#define CONFIG_WATCHDOG
+#define CFG_WATCHDOG_VALUE 0xFFFF
+
 /* System IO Config */
 #define CFG_SICRH	(SICRH_TSOBI1 | SICRH_TSOBI2)	/* RGMII */
 #define CFG_SICRL	SICRL_USBDR			/* Enable Internal USB Phy  */
