--- compcache-0.6pre3/Makefile.orig	2009-08-14 02:33:45.169249060 +0800
+++ compcache-0.6pre3/Makefile	2009-08-14 02:34:15.594938335 +0800
@@ -13,7 +13,9 @@
 		M=$(PWD)/sub-projects/allocators/xvmalloc-kmod modules
 	make -C $(KERNELDIR) M=$(PWD) modules
 	make -C sub-projects/rzscontrol
-	@ln -sf sub-projects/allocators/xvmalloc-kmod/xvmalloc.ko
+
+modules_install:
+	make -C $(KERNELDIR) M=$(PWD) modules_install
 
 doc:
 	make -C sub-projects/rzscontrol doc
