--- freeswan-2.04/lib/libdes/Makefile	Tue Apr 22 19:38:03 2003
+++ freeswan-2.04/lib/libdes/Makefile	Fri Jan  2 14:30:34 2004
@@ -58,9 +58,10 @@
 #CC=gcc
 #CFLAG= -O4 -funroll-loops -fomit-frame-pointer
 # normally overridden by FreeS/WAN Makefiles anyway
-CFLAG= -O3 -fomit-frame-pointer -I${KLIPSD}/include -I${SRCDIR}
+CFLAG= -O3 -fomit-frame-pointer
 
-CFLAGS=$(OPTS) $(CFLAG)
+CFLAGS=$(CFLAG)
+override CFLAGS += $(OPTS) -I${KLIPSD}/include -I${SRCDIR}
 CPP=$(CC) -E
 
 # Assember version of des_encrypt*().
--- freeswan-2.04/lib/libfreeswan/Makefile	Wed Jul  9 21:39:20 2003
+++ freeswan-2.04/lib/libfreeswan/Makefile	Fri Jan  2 14:31:16 2004
@@ -43,7 +43,7 @@
 
 LIB=libfreeswan.a
 # Original flags
-CFLAGS=-I. -I${SRCDIR} -I${KLIPSD} -I${FREESWANSRCDIR} $(USERCOMPILE)
+CFLAGS=$(USERCOMPILE)
 CFLAGS+= -Wall
 #CFLAGS+= -Wconversion
 #CFLAGS+= -Wmissing-prototypes
@@ -55,6 +55,7 @@
 #CFLAGS+= -W
 #CFLAGS+= -Wwrite-strings
 CFLAGS+= -Wbad-function-cast 
+override CFLAGS += -I. -I${SRCDIR} -I${KLIPSD} -I${FREESWANSRCDIR}
 
 
 
--- freeswan-2.04/lib/libipsecpolicy/Makefile	Tue May 13 21:12:12 2003
+++ freeswan-2.04/lib/libipsecpolicy/Makefile	Fri Jan  2 14:31:42 2004
@@ -30,12 +30,13 @@
 
 LIB=libipsecpolicy.a
 # Original flags
-CFLAGS=-I. -I${KLIPSD} -I${FREESWANSRCDIR} $(USERCOMPILE)
+CFLAGS = $(USERCOMPILE)
 CFLAGS+= -Wall
 CFLAGS+= -Wpointer-arith
 CFLAGS+= -Wcast-qual
 CFLAGS+= -Wstrict-prototypes
 CFLAGS+= -Wbad-function-cast 
+override CFLAGS += -I. -I${KLIPSD} -I${FREESWANSRCDIR} $(USERCOMPILE)
 
 MANS=
 
--- freeswan-2.04/lib/liblwres/Makefile	Fri Apr 11 12:15:40 2003
+++ freeswan-2.04/lib/liblwres/Makefile	Fri Jan  2 14:32:14 2004
@@ -24,7 +24,8 @@
 CDEFINES = -g
 CWARNINGS = -Werror
 
-CFLAGS=${CINCLUDES} ${CDEFINES} ${CWARNINGS}
+CFLAGS=${CDEFINES} ${CWARNINGS}
+override CFLAGS+=${CINCLUDES}
 
 VERSION="@(\#) freeswan-hacking-9.2.1-for-fs2"
 LIBINTERFACE=2
--- freeswan-2.04/programs/Makefile.program	Sun Jun 29 16:34:49 2003
+++ freeswan-2.04/programs/Makefile.program	Fri Jan  2 14:54:58 2004
@@ -1,7 +1,7 @@
 
 include ${FREESWANSRCDIR}/Makefile.ver
 
-CFLAGS+=$(USERCOMPILE) -I${KLIPSINC}
+override CFLAGS+=-I${KLIPSINC}
 
 CFLAGS+= $(USERCOMPILE)
 CFLAGS+= -Wall
@@ -24,7 +24,7 @@
 CFLAGS+= ${WERROR}
 
 ifneq ($(LD_LIBRARY_PATH),)
-LDFLAGS=-L$(LD_LIBRARY_PATH)
+LDFLAGS+=-L$(LD_LIBRARY_PATH)
 endif
 
 MANDIR8=$(MANTREE)/man8
--- freeswan-2.04/programs/lwdnsq/Makefile	Tue Sep  2 20:13:24 2003
+++ freeswan-2.04/programs/lwdnsq/Makefile	Fri Jan  2 14:32:56 2004
@@ -24,7 +24,7 @@
 LWRESINCL=${LWRESDIR}/include
 
 LIBS:=${FREESWANLIB} ${LWRESLIB} ${BIND9STATICLIBDIR}/libdns.a ${BIND9STATICLIBDIR}/libisc.a
-CFLAGS+=-I${LWRESINCL}
+override CFLAGS+=-I${LWRESINCL}
 #USERCOMPILE=-g
 
 
--- freeswan-2.04/programs/pluto/Makefile	Mon Aug 25 17:12:34 2003
+++ freeswan-2.04/programs/pluto/Makefile	Fri Jan  2 14:45:56 2004
@@ -107,7 +107,7 @@
 	-DGCC_LINT \
 	# -DLEAK_DETECTIVE
 
-CPPFLAGS = $(HDRDIRS) $(DEFINES) \
+override CPPFLAGS += $(HDRDIRS) $(DEFINES) \
 	-DSHARED_SECRETS_FILE=\"${FINALCONFDIR}/ipsec.secrets\" \
 	-DPOLICYGROUPSDIR=\"${FINALCONFDDIR}/policies\" \
 	-DPERPEERLOGDIR=\"${FINALLOGDIR}/pluto/peer\"
@@ -120,7 +120,7 @@
 LIBSPLUTO+= -lgmp -lresolv # -lefence
 
 ifneq ($(LD_LIBRARY_PATH),)
-LDFLAGS=-L$(LD_LIBRARY_PATH)
+LDFLAGS+=-L$(LD_LIBRARY_PATH)
 endif
 
 LIBSADNS = $(FREESWANLIB)
--- freeswan-2.04/testing/utils/ikeping/Makefile	Wed Mar 27 17:28:25 2002
+++ freeswan-2.04/testing/utils/ikeping/Makefile	Fri Jan  2 14:33:43 2004
@@ -35,7 +35,7 @@
 
 HDRDIRS =  $(FREESWANINCLS)
 
-CPPFLAGS = $(HDRDIRS) $(BYTE_ORDER) -DDEBUG 
+override CPPFLAGS += $(HDRDIRS) $(BYTE_ORDER) -DDEBUG 
 
 ALLFLAGS = $(CPPFLAGS) $(CFLAGS)
 
--- freeswan-2.04/testing/utils/uml_netjig/Makefile	Thu Nov  7 16:17:49 2002
+++ freeswan-2.04/testing/utils/uml_netjig/Makefile	Fri Jan  2 14:34:19 2004
@@ -43,7 +43,8 @@
 #PCAPH=-I/home/mcr/src/libpcap
 #PCAPLIB=-L/home/mcr/src/linux/i386/libpcap -lpcap
 
-CFLAGS = -g -Wall ${PCAPH} ${NETDISSECTH} ${OPTIONS}
+CFLAGS = -g -Wall
+override CFLAGS += ${PCAPH} ${NETDISSECTH} ${OPTIONS}
 LIBS = ${NETDISSECTLIB} ${PCAPLIB} ${OPENSSL}
 
 
