--- mjpegtools-1.9.0/lavtools/Makefile.am.orig	2010-09-29 19:36:46.000000000 +0000
+++ mjpegtools-1.9.0/lavtools/Makefile.am	2010-09-29 19:37:55.000000000 +0000
@@ -58,9 +58,6 @@
 liblavplay_la_CPPFLAGS = $(AM_CPPFLAGS) $(LIBDV_CFLAGS) $(X_CFLAGS)
 liblavplay_la_LDFLAGS = $(LAV_ALL_LIB_OPTS)
 liblavplay_la_LIBADD = liblavfile.la $(SDL_LIBS) liblavjpeg.la $(LIBDV_LIBS) $(LIBMJPEGUTILS)
-if HAVE_V4L
-   liblavplay_la_LIBADD += ${X_LIBS} -lX11
-endif
 liblavplay_la_DEPENDENCIES = liblavfile.la liblavjpeg.la
 
 # *********************************************************************
