--- linux-mips-2.4.24-pre2+mtd-2004-01-27/fs/jffs2/erase.c	2004-11-17 18:17:59.000000000 +0100
+++ linux/fs/jffs2/erase.c	2004-11-17 18:44:52.260067088 +0100
@@ -365,11 +365,13 @@
 		jeb->dirty_size = 0;
 		jeb->wasted_size = 0;
 	} else {
-		struct jffs2_unknown_node marker = {
-			.magic =	cpu_to_je16(JFFS2_MAGIC_BITMASK),
-			.nodetype =	cpu_to_je16(JFFS2_NODETYPE_CLEANMARKER),
-			.totlen =	cpu_to_je32(c->cleanmarker_size)
-		};
+		/* compiler workaround, structure was not initialized before
+		   on mipsel cross compilers
+        	   fix by Eugene.Wisor@flukenetworks.com */
+	        struct jffs2_unknown_node marker;
+		marker.magic = cpu_to_je16(JFFS2_MAGIC_BITMASK);
+		marker.nodetype = cpu_to_je16(JFFS2_NODETYPE_CLEANMARKER);
+		marker.totlen = cpu_to_je32(c->cleanmarker_size);
 
 		marker.hdr_crc = cpu_to_je32(crc32(0, &marker, sizeof(struct jffs2_unknown_node)-4));
 
