
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- udev-058/Makefile~flags.patch
+++ udev-058/Makefile
@@ -153,16 +153,17 @@
 
 SYSFS = $(PWD)/libsysfs/sysfs.a
 
-CFLAGS +=	-I$(PWD)/libsysfs/sysfs \
+override CFLAGS +=	-I$(PWD)/libsysfs/sysfs \
 		-I$(PWD)/libsysfs
 
 ifeq ($(strip $(USE_LOG)),true)
-	CFLAGS += -DUSE_LOG
+override	CFLAGS += -DUSE_LOG
 endif
 
 # if DEBUG is enabled, then we do not strip or optimize
 ifeq ($(strip $(DEBUG)),true)
-	CFLAGS  += -O1 -g -DDEBUG
+	CFLAGS  += -O1 -g
+	override CFLAGS += -DDEBUG
 	LDFLAGS += -Wl
 	STRIPCMD = /bin/true -Since_we_are_debugging
 else
@@ -178,9 +179,9 @@
 	KLCC		= $(KLIBC_INSTALL)/bin/klcc
 	CC		= $(KLCC)
 	LD		= $(KLCC)
-	LDFLAGS		+= -static
+	override LDFLAGS		+= -static
 else
-	CFLAGS		+= -Wshadow -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations
+	override CFLAGS		+= -Wshadow -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations
 endif
 
 ifeq ($(strip $(USE_SELINUX)),true)
