--- /tmp/Makefile.am	2006-04-25 15:47:49.313035880 +0200
+++ fontconfig-2.3.94/fc-lang/Makefile.am	2006-04-25 15:48:22.807943880 +0200
@@ -31,6 +31,7 @@
 TMPL=fclang.tmpl.h
 STMPL=${top_srcdir}/fc-lang/fclang.tmpl.h
 TARG=fclang.h
+FC_LANG=./fc-lang
 
 noinst_PROGRAMS=fc-lang
 
@@ -44,6 +45,6 @@
 
 $(TARG):$(ORTH) fc-lang${EXEEXT} $(STMPL)
 	rm -f $(TARG)
-	./fc-lang${EXEEXT} -d ${srcdir} $(ORTH) < $(STMPL) > $(TARG)
+	$(FC_LANG)${EXEEXT} -d ${srcdir} $(ORTH) < $(STMPL) > $(TARG)
 
 CLEANFILES=$(TARG)
