Index: zaurusd/config/alsa/c7x0-default.state
===================================================================
--- zaurusd.orig/config/alsa/c7x0-default.state	2006-04-10 08:56:46.000000000 +0200
+++ zaurusd/config/alsa/c7x0-default.state	2006-05-01 19:00:55.748053640 +0200
@@ -50,7 +50,7 @@
 		comment.type BOOLEAN
 		comment.count 1
 		iface MIXER
-		name 'Mic Capture Switch'
+		name 'Capture Mic Switch'
 		value true
 	}
 	control.7 {
@@ -106,7 +106,7 @@
 		comment.count 1
 		iface MIXER
 		name 'Output Mixer Line Bypass Switch'
-		value false
+		value true
 	}
 	control.13 {
 		comment.access 'read write'
