--- zaurusd/config/alsa/5600-default.state.orig	2006-11-27 21:20:53.992954093 +0100
+++ zaurusd/config/alsa/5600-default.state	2006-11-27 21:17:21.000000000 +0100
@@ -155,7 +155,7 @@
 		comment.count 2
 		comment.range '0 - 127'
 		iface MIXER
-		name 'Playback Volume'
+		name 'Master Playback Volume'
 		value.0 121
 		value.1 121
 	}
@@ -164,7 +164,7 @@
 		comment.type BOOLEAN
 		comment.count 2
 		iface MIXER
-		name 'Playback ZC Switch'
+		name 'Master Playback ZC Switch'
 		value.0 false
 		value.1 false
 	}
@@ -292,6 +292,6 @@
 		comment.item.1 On
 		iface MIXER
 		name 'Speaker Function'
-		value Off
+		value On
 	}
 }
