--- ./sound/vorbis.cpp.old	2004-05-16 04:23:59.000000000 +0930
+++ ./sound/vorbis.cpp	2004-05-16 04:24:23.000000000 +0930
@@ -29,7 +29,7 @@
 #include "sound/audiostream.h"
 #include "sound/audiocd.h"
 
-#include <vorbis/vorbisfile.h>
+#include <tremor/ivorbisfile.h>
 
 
 AudioStream *makeVorbisStream(OggVorbis_File *file, int duration);
--- ./configure.old	2004-05-16 04:46:47.000000000 +0930
+++ ./configure	2004-05-16 04:50:42.000000000 +0930
@@ -560,15 +560,15 @@
 if test "$_vorbis" = auto ; then
   _vorbis=no
   cat > $TMPC << EOF
-#include <vorbis/codec.h>
+#include <tremor/ivorbiscodec.h>
 int main(void) { vorbis_packet_blocksize(0,0); return 0; }
 EOF
   cc_check $LDFLAGS $CXXFLAGS $OGG_CFLAGS $OGG_LIBS $VORBIS_CFLAGS $VORBIS_LIBS \
-  -lvorbis -logg -lm && _vorbis=yes
+  -lvorbisidec -logg -lm && _vorbis=yes
 fi
 if test "$_vorbis" = yes ; then
   _def_vorbis='#define USE_VORBIS'
-  LIBS="$LIBS $OGG_LIBS $VORBIS_LIBS -lvorbisfile -lvorbis -logg"
+  LIBS="$LIBS $OGG_LIBS $VORBIS_LIBS -lvorbisidec -logg"
   INCLUDES="$INCLUDES $OGG_CFLAGS $VORBIS_CFLAGS"
 else
   _def_vorbis='#undef USE_VORBIS'
