--- libosso-2.13/src/osso-init.c.orig	2008-09-22 12:56:49.000000000 -0700
+++ libosso-2.13/src/osso-init.c	2008-09-22 12:57:22.000000000 -0700
@@ -657,7 +657,8 @@
     return DBUS_HANDLER_RESULT_NOT_YET_HANDLED;
 }
 
-inline int __attribute__ ((visibility("hidden")))
+//inline int __attribute__ ((visibility("hidden")))
+static int
 muali_convert_msgtype(int t)
 {
     switch (t) {
--- libosso-2.13/src/osso-internal.h.orig	2008-09-22 12:57:39.000000000 -0700
+++ libosso-2.13/src/osso-internal.h	2008-09-22 12:58:13.000000000 -0700
@@ -209,7 +209,8 @@
 #  define dprint(f, a...)
 # endif /* LIBOSSO_DEBUG */
 
-inline int __attribute__ ((visibility("hidden")))
+//inline int __attribute__ ((visibility("hidden")))
+static int
 muali_convert_msgtype(int t);
 
 muali_arg_t* _get_muali_args(DBusMessageIter *iter);
@@ -297,7 +298,8 @@
 _muali_parse_id(const char *id, muali_bus_type *bus, char *sender,
                 int *serial);
 
-inline void __attribute__ ((visibility("hidden")))
+//inline void __attribute__ ((visibility("hidden")))
+static void
 _muali_make_id(muali_bus_type bus, const char *sender, int serial, char *id);
 
 #endif /* OSSO_INTERNAL_H_ */
--- libosso-2.13/src/osso-rpc.c.orig	2008-09-22 12:58:53.000000000 -0700
+++ libosso-2.13/src/osso-rpc.c	2008-09-22 12:59:42.000000000 -0700
@@ -1026,7 +1026,8 @@
         *serial = atoi(buf);
 }
 
-inline void __attribute__ ((visibility("hidden")))
+//inline void __attribute__ ((visibility("hidden")))
+static void
 _muali_make_id(muali_bus_type bus, const char *sender, int serial, char *id)
 {
         snprintf(id, MAX_MSGID_LEN, "%d,%s,%d", bus, sender, serial);

