Index: DirectFB-1.1.0/directfb-internal.pc.in
===================================================================
--- DirectFB-1.1.0.orig/directfb-internal.pc.in	2008-06-22 16:16:13.851722040 +0200
+++ DirectFB-1.1.0/directfb-internal.pc.in	2008-06-22 16:16:33.665465096 +0200
@@ -2,10 +2,10 @@
 exec_prefix=@exec_prefix@
 moduledir=@MODULEDIR@
 moduledirname=@MODULEDIRNAME@
-includedir=@INTERNALINCLUDEDIR@
+includedir=@includedir@
 
 Name: DirectFB-Internal
 Description: Third party module support package for DirectFB.
 Version: @VERSION@
 Requires: directfb = @VERSION@
-Cflags: @DFB_INTERNAL_CFLAGS@ -I@INTERNALINCLUDEDIR@
+Cflags: @DFB_INTERNAL_CFLAGS@ -I${includedir}/directfb -I${includedir}/directfb-internal
Index: DirectFB-1.1.0/directfb.pc.in
===================================================================
--- DirectFB-1.1.0.orig/directfb.pc.in	2008-06-22 16:16:13.868387652 +0200
+++ DirectFB-1.1.0/directfb.pc.in	2008-06-22 16:16:50.816223652 +0200
@@ -8,4 +8,4 @@
 Version: @VERSION@
 Requires: fusion direct
 Libs: -L${libdir} -ldirectfb @THREADLIB@ @DYNLIB@ @ZLIB_LIBS@ @OSX_LIBS@
-Cflags: @THREADFLAGS@ -I@INCLUDEDIR@
+Cflags: @THREADFLAGS@ -I${includedir}/directfb
Index: DirectFB-1.1.0/lib/direct/direct.pc.in
===================================================================
--- DirectFB-1.1.0.orig/lib/direct/direct.pc.in	2008-06-22 16:16:13.888387824 +0200
+++ DirectFB-1.1.0/lib/direct/direct.pc.in	2008-06-22 16:16:17.121721173 +0200
@@ -7,4 +7,4 @@
 Description: DirectFB base development library
 Version: @VERSION@
 Libs: -L${libdir} -ldirect @THREADLIB@ @DYNLIB@
-Cflags: @THREADFLAGS@ -I@INCLUDEDIR@
+Cflags: @THREADFLAGS@ -I${includedir}/directfb
Index: DirectFB-1.1.0/lib/fusion/fusion.pc.in
===================================================================
--- DirectFB-1.1.0.orig/lib/fusion/fusion.pc.in	2008-06-22 16:16:13.905054457 +0200
+++ DirectFB-1.1.0/lib/fusion/fusion.pc.in	2008-06-22 16:16:17.121721173 +0200
@@ -8,4 +8,4 @@
 Version: @VERSION@
 Requires: direct
 Libs: -L${libdir} -lfusion
-Cflags: -I@INCLUDEDIR@
+Cflags: -I${includedir}/directfb
Index: DirectFB-1.1.0/lib/voodoo/voodoo.pc.in
===================================================================
--- DirectFB-1.1.0.orig/lib/voodoo/voodoo.pc.in	2008-06-22 16:16:13.931720483 +0200
+++ DirectFB-1.1.0/lib/voodoo/voodoo.pc.in	2008-06-22 16:16:17.121721173 +0200
@@ -8,4 +8,4 @@
 Version: @VERSION@
 Requires: direct
 Libs: -L${libdir} -lvoodoo
-Cflags: -I@INCLUDEDIR@
+Cflags: -I${includedir}/directfb
