diff -Naur org/Makefile patched/Makefile
--- org/Makefile	2008-08-19 12:45:17.000000000 +0200
+++ patched/Makefile	2008-08-19 12:44:55.000000000 +0200
@@ -4,9 +4,11 @@
 #
 #
 
-HOST=
-CC=$(HOST)gcc
-LD=$(HOST)ld
+#HOST=
+#CC=$(HOST)gcc
+#LD=$(HOST)ld
+VERSION := $(shell uname -r)
+MODPATH := /lib/modules/$(VERSION)
 KERN_VER=$(shell uname -r | cut -b1-3;)
 KDIR := /lib/modules/$(shell uname -r)/build
 PWD := $(shell pwd)
@@ -17,15 +19,17 @@
 
 ifeq ($(KERN_VER), 2.6) 
     KERN_26=y
-    KERNEL_SOURCE=/usr/src/linux-2.6.9
+    KERNEL_SOURCE ?= $(MODPATH)/source
 else 
     KERN_24=y
     KERNEL_SOURCE=/usr/src/linux-2.4
 endif
 
 SRC_DIR=src
-DEFINES=-D__KERNEL__ -DMODULE=1
+DEFINES ?= -D__KERNEL__ -DMODULE=1
 
+# set to 1 for zd1211b
+ZD1211REV_B=1
 
 KERNRELEASE := $(shell uname -r;)
 MODPATH := /lib/modules/$(KERNRELEASE)
@@ -362,14 +366,14 @@
 	/bin/sh scripts/Menuconfig
 
 debug:
-	gcc -o apdbg apdbg.c
-	chmod +x apdbg
-	cp ./apdbg /sbin/apdbg   
-	make -C Menudbg
-	mv Menudbg/menudbg .
-	chmod +x menudbg
-	cp ./menudbg /sbin
-	gcc -DPKT_SEND_FUNCTION -o winevl_iface Winevl_iface/pkt_send.c Winevl_iface/winevl_bridge.c
+	#gcc -o apdbg apdbg.c
+	#chmod +x apdbg
+	#cp ./apdbg /sbin/apdbg   
+	#make -C Menudbg
+	#mv Menudbg/menudbg .
+	#chmod +x menudbg
+	#cp ./menudbg /sbin
+	#gcc -DPKT_SEND_FUNCTION -o winevl_iface Winevl_iface/pkt_send.c Winevl_iface/winevl_bridge.c
 
 clean:
 	rm -rf .tmp_versions .*.cmd *.ko *.mod.c *.mod.o *.o $(SRC_DIR)/*.o  $(SRC_DIR)/.*.o.cmd menudbg apdbg winevl_iface
