Index: linux-2.6.23/drivers/video/pxafb.c
===================================================================
--- linux-2.6.23.orig/drivers/video/pxafb.c	2008-01-25 16:25:21.000000000 -0800
+++ linux-2.6.23/drivers/video/pxafb.c	2008-01-25 16:32:14.000000000 -0800
@@ -1194,7 +1194,7 @@
 		if ((clkinfo->old == 13000))
 			break;
 
-		pcd = get_pcd(fbi->fb.var.pixclock);
+		pcd = get_pcd(fbi,fbi->fb.var.pixclock);
 		lccr3 = fbi->reg_lccr3;
 		set_hsync_time(fbi, pcd);
 		fbi->reg_lccr3 = (fbi->reg_lccr3 & ~0xff) | LCCR3_PixClkDiv(pcd);
