--- s/configure.ac~	2005-12-15 00:24:25.000000000 +0000
+++ s/configure.ac	2008-07-27 10:05:30.000000000 +0100
@@ -45,7 +45,7 @@
 AC_SUBST(FONTDIR)
 
 PKG_CHECK_MODULES(MAPS, [fontutil])
-MAPFILES_PATH=$(pkg-config --variable=mapdir fontutil)
+MAPFILES_PATH=$(PKG_CONFIG_PATH="${STAGING_LIBDIR_NATIVE}/pkg-config" ${PKG_CONFIG} --variable=mapdir fontutil)
 AC_SUBST(MAPFILES_PATH)
 
 XORG_FONT_CHECK_ISO8859_1
