--- linux-2.4.22/drivers/scsi/sd.c~	2005-12-13 06:04:08.000000000 +1030
+++ linux-2.4.22/drivers/scsi/sd.c	2005-12-13 10:08:13.912932696 +1030
@@ -1087,6 +1087,7 @@
 			rscsi_disks[i].capacity >>= 1;
 	}
                         // add by super, moified by pete
+#if 0
         if (rscsi_disks[i].device->removable){
                 if(0x32 == rscsi_disks[i].device->host->hostt->port){
 		                printk("is removable disk \n");
@@ -1113,6 +1114,7 @@
 	                        );
                }
         }else {
+#endif
                 printk("%x port connect!!!!!\n",rscsi_disks[i].device->host->hostt->port);
                 if(0x32 == rscsi_disks[i].device->host->hostt->port){
                         *IXP425_GPIO_GPOUTR &= 0xfffb;
@@ -1133,7 +1135,9 @@
                                         NULL
                         );
                 }
+#if 0
         }
+#endif
                 //end
 
 
@@ -1682,22 +1686,28 @@
 							port2_attached = 0;
                         if(port2_attached == 0)
 							*IXP425_GPIO_GPOUTR |= 0x8;
+#if 0
 						if(SDp->removable == 1){
 	                        sprintf(flash, "flash_sd%c", 'a' + i);
                         	remove_proc_entry(flash, NULL);
                         	remove_proc_entry(media, NULL);
 						}else{
+#endif
 							sprintf(hdd, "hdd_sd%c", 'a' + i);
 			                        	remove_proc_entry(hdd,NULL);
 			                        	remove_proc_entry(media, NULL);
+#if 0
 						}                
+#endif
                         break;
                 case 0x32:
                         *IXP425_GPIO_GPOUTR |= 0x4;
+#if 0
 						if(SDp->removable == 1){
 							remove_proc_entry("flash_sda",NULL);
 						}
 						else
+#endif
                         	remove_proc_entry("hdd_sda",NULL);
                         break;
                 default:
