--- /tmp/configure	2008-08-01 20:59:48.000000000 +0200
+++ qt-embedded-linux-opensource-src-4.4.1/configure	2008-08-01 21:02:05.000000000 +0200
@@ -765,7 +765,7 @@
             UNKNOWN_ARG=yes
         fi
         ;;
-    -prefix|-docdir|-headerdir|-plugindir|-datadir|-libdir|-bindir|-translationdir|-sysconfdir|-examplesdir|-demosdir|-depths|-make|-nomake|-platform|-xplatform|-buildkey|-sdk|-arch|-host-arch|-mysql_config)
+    -prefix|-docdir|-headerdir|-plugindir|-datadir|-libdir|-bindir|-translationdir|-sysconfdir|-examplesdir|-demosdir|-depths|-make|-nomake|-platform|-xplatform|-buildkey|-sdk|-arch|-host-arch|-mysql_config|-crossarch)
         VAR=`echo $1 | sed "s,^-\(.*\),\1,"`
         shift
         VAL="$1"
@@ -1182,7 +1182,10 @@
     xplatform)
         XPLATFORM="$VAL"
         ;;
-    debug-and-release)
+	crossarch)
+		CROSSARCH="$VAL"
+		;;
+	debug-and-release)
         if [ "$VAL" = "yes" ] || [ "$VAL" = "no" ]; then
             CFG_DEBUG_RELEASE="$VAL"
         else
@@ -2437,6 +2440,8 @@
     CFG_ARCH=$CFG_HOST_ARCH
 fi
 
+CFG_ARCH="$CROSSARCH"
+
 if [ -d "$relpath/src/corelib/arch/$CFG_ARCH" ]; then
     if [ "$OPT_VERBOSE" = "yes" ]; then
         echo "    '$CFG_ARCH' is supported"
