--- mono-1.2.5.1/mono/mini/genmdesc.pl.orig	2007-10-03 21:02:07.000000000 +0200
+++ mono-1.2.5.1/mono/mini/genmdesc.pl	2007-10-03 21:06:16.000000000 +0200
@@ -36,7 +36,9 @@
 		$i++;
 	}
 	close (OPS);
-	my $cpp = "cpp -undef ";
+	my $cpp = $ENV{"CPP"};
+	$cpp = "cpp" unless defined $cpp;
+	$cpp .= " -undef ";
 	foreach (@defines) {
 		$cpp .= " -U$_";
 		$arch_found = 1 if $arch eq $_;
@@ -44,7 +46,7 @@
 	die "$arch arch is not supported.\n" unless $arch_found;
 
 	$cpp .= " -D$arch $srcdir/mini-ops.h|";
-	#print "Running: $cpp\n";
+	print "Running: $cpp\n";
 	open (OPS, $cpp) || die "Cannot execute cpp: $!";
 	while (<OPS>) {
 		next unless /MINI_OP\s*\(\s*(\S+?)\s*,\s*"(.*?)"/;
