--- mono-1.2.20070131/mono/monoburg/Makefile.am	2006-10-10 07:00:26.000000000 +0200
+++ mono-1.2.20070131.mod/mono/monoburg/Makefile.am	2007-02-13 17:44:02.000000000 +0100
@@ -1,5 +1,3 @@
-CC=$(CC_FOR_BUILD)
-
 am_CFLAGS = $(BUILD_GLIB_CFLAGS) -I$(srcdir) $(CFLAGS_FOR_BUILD)
 
 #noinst_PROGRAMS = sample
@@ -12,10 +10,10 @@
 	bison $(srcdir)/monoburg.y -o parser.c
 
 monoburg$(BUILD_EXEEXT): $(srcdir)/monoburg.c $(srcdir)/monoburg.h parser.c
-	$(CC_FOR_BUILD) -o $@ $(srcdir)/monoburg.c parser.c $(am_CFLAGS) $(LDFLAGS) $(BUILD_GLIB_LIBS)
+	$(CC) -o $@ $(srcdir)/monoburg.c parser.c $(am_CFLAGS) $(LDFLAGS) $(BUILD_GLIB_LIBS)
 
 sample.c: monoburg$(BUILD_EXEEXT) $(srcdir)/sample.brg
-	./monoburg$(BUILD_EXEEXT) $(srcdir)/sample.brg > sample.c
+	monoburg$(BUILD_EXEEXT) $(srcdir)/sample.brg > sample.c
 
 #sample_LDADD = $(BUILD_GLIB_LIBS)
 
