--- anthy-7811/mkanthydic/Makefile.am~	2006-05-13 18:28:35.000000000 +0900
+++ anthy-7811/mkanthydic/Makefile.am	2006-06-18 11:07:19.080000000 +0900
@@ -3,12 +3,12 @@
 CLEANFILES = anthy.dic
 INCLUDES = -I$(top_srcdir)/include -DSRCDIR=\"$(srcdir)\"
 
-noinst_PROGRAMS = mkfiledic
-mkfiledic_SOURCES = mkfiledic.c
-mkfiledic_LDADD = ../src-diclib/libdiclib.la
+#noinst_PROGRAMS = mkfiledic
+#mkfiledic_SOURCES = mkfiledic.c
+#mkfiledic_LDADD = ../src-diclib/libdiclib.la
 
-anthy.dic : mkfiledic ../mkworddic/anthy.wdic ../depgraph/anthy.dep
-	./mkfiledic
+anthy.dic : ../mkworddic/anthy.wdic ../depgraph/anthy.dep
+	mkfiledic
 
 
 # To install 

--- anthy-7811/depgraph/Makefile.am~	2006-06-02 00:20:54.000000000 +0900
+++ anthy-7811/depgraph/Makefile.am	2006-06-18 11:08:13.350000000 +0900
@@ -9,11 +9,11 @@
 EXTRA_DIST = indepword.txt $(DEPWORDS)
 
 # Generate the dictionary
-noinst_PROGRAMS = mkdepgraph
-mkdepgraph_SOURCES = mkdepgraph.c
-mkdepgraph_LDADD =  ../src-main/libanthy.la ../src-worddic/libanthydic.la
+#noinst_PROGRAMS = mkdepgraph
+#mkdepgraph_SOURCES = mkdepgraph.c
+#mkdepgraph_LDADD =  ../src-main/libanthy.la ../src-worddic/libanthydic.la
 
-anthy.dep : mkdepgraph
-	./mkdepgraph
+anthy.dep : 
+	mkdepgraph
 
 noinst_DATA = anthy.dep

--- anthy-7811/mkworddic/Makefile.am~	2006-05-13 18:29:07.000000000 +0900
+++ anthy-7811/mkworddic/Makefile.am	2006-06-18 11:18:35.810000000 +0900
@@ -16,14 +16,14 @@
  udict dict.args.in
 
 # Generate the dictionary
-noinst_PROGRAMS = mkworddic
-mkworddic_SOURCES = mkdic.c writewords.c mkudic.c mkdic.h
-mkworddic_LDADD = ../src-worddic/libanthydic.la
+#noinst_PROGRAMS = mkworddic
+#mkworddic_SOURCES = mkdic.c writewords.c mkudic.c mkdic.h
+#mkworddic_LDADD = ../src-worddic/libanthydic.la
 
 noinst_DATA = anthy.wdic
 
-anthy.wdic : mkworddic
-	   ./mkworddic -f ./dict.args
+anthy.wdic : 
+	   mkworddic -f ./dict.args
 
 
 # To install
--- anthy-7811/mkmatrix/Makefile.am~	2006-06-02 00:19:41.000000000 +0900
+++ anthy-7811/mkmatrix/Makefile.am	2006-06-18 11:31:37.250000000 +0900
@@ -1,12 +1,12 @@
 EXTRA_DIST = dep.sorted
 
-noinst_PROGRAMS = mkmatrix
-INCLUDES = -I$(top_srcdir)/include
-mkmatrix_LDADD = ../src-worddic/libanthydic.la -lm
-mkmatrix_SOURCES = mkmatrix.c
+#noinst_PROGRAMS = mkmatrix
+#INCLUDES = -I$(top_srcdir)/include
+#mkmatrix_LDADD = ../src-worddic/libanthydic.la -lm
+#mkmatrix_SOURCES = mkmatrix.c
 
 noinst_DATA = matrix
 CLEANFILES = matrix
 
-matrix	: mkmatrix $(top_srcdir)/mkmatrix/dep.sorted
-	./mkmatrix $(top_srcdir)/mkmatrix/dep.sorted -o matrix
+matrix	: $(top_srcdir)/mkmatrix/dep.sorted
+	mkmatrix $(top_srcdir)/mkmatrix/dep.sorted -o matrix
