--- gal-2.1.10/configure.in~	2004-06-01 14:33:00.000000000 +0100
+++ gal-2.1.10/configure.in	2004-06-26 20:22:52.000000000 +0100
@@ -145,7 +145,7 @@
 		exit (1);
 
 	exit (0);
-}], ac_cv_libiconv_utf8=yes, [ac_cv_libiconv_utf8=no; have_iconv=no], [ac_cv_libiconv_utf8=no; have_iconv=no]))
+}], ac_cv_libiconv_utf8=yes, [ac_cv_libiconv_utf8=no; have_iconv=no], [ac_cv_libiconv_utf8=yes; have_iconv=yes]))
 fi
 
 if test "$have_iconv" = no; then
@@ -154,25 +154,6 @@
 AC_SUBST(ICONV_CFLAGS)
 AC_SUBST(ICONV_LIBS)
 
-CFLAGS="$CFLAGS -I$srcdir"
-
-AC_MSG_CHECKING(preferred charset formats for system iconv)
-AC_TRY_RUN([
-#define CONFIGURE_IN
-#include "iconv-detect.c"
-],[
-	AC_MSG_RESULT(found)
-],[
-	AC_MSG_RESULT(not found)
-	AC_WARN([
-	*** The iconv-detect program was unable to determine the
-	*** preferred charset formats recognized by your system
-	*** iconv library. It is suggested that you install a
-	*** working iconv library such as the one found at
-	*** ftp://ftp.gnu.org/pub/gnu/libiconv
-	])
-])
-
 CFLAGS="$save_CFLAGS"
 LIBS="$save_LIBS"
 
@@ -200,7 +181,7 @@
 }],[
 AC_DEFINE(HAVE_LKSTRFTIME, 1, [strftime supports use of l and k])
 ac_cv_lkstrftime=yes
-],ac_cv_lkstrftime=no,ac_cv_lkstrftime=no])
+],ac_cv_lkstrftime=no,ac_cv_lkstrftime=yes])
 AC_MSG_RESULT($ac_cv_lkstrftime)
 
 dnl Check for nl_langinfo and CODESET
