--- gal-2.1.10/gal/a11y/e-table/gal-a11y-e-table-item.c~	2004-05-19 14:44:42.000000000 +0100
+++ gal-2.1.10/gal/a11y/e-table/gal-a11y-e-table-item.c	2004-06-27 13:28:59.000000000 +0100
@@ -49,6 +49,8 @@
 						     ESelectionModel *selection);
 static gboolean gal_a11y_e_table_item_unref_selection (GalA11yETableItem *a11y);
 
+static gpointer *eti_reinit_data (AtkTable *table, ETableItem *item);
+
 #if 0
 static void
 unref_accessible (gpointer user_data, GObject *obj_loc)
