Index: php-5.2.6/ext/standard/string.c
===================================================================
--- php-5.2.6.orig/ext/standard/string.c
+++ php-5.2.6/ext/standard/string.c
@@ -592,14 +592,12 @@ PHP_FUNCTION(nl_langinfo)
 #endif
 #ifdef DECIMAL_POINT
 		case DECIMAL_POINT:
-#endif
-#ifdef RADIXCHAR
+#elif RADIXCHAR
 		case RADIXCHAR:
 #endif
 #ifdef THOUSANDS_SEP
 		case THOUSANDS_SEP:
-#endif
-#ifdef THOUSEP
+#elif THOUSEP
 		case THOUSEP:
 #endif
 #ifdef GROUPING
