Index: expedite/configure.in
===================================================================
--- expedite.orig/configure.in
+++ expedite/configure.in
@@ -47,24 +47,11 @@
 PKG_CHECK_MODULES([EVAS], evas)
 
 dnl OpenGL GLEW
-PKG_CHECK_EXISTS(evas-opengl-glew,
-  [
-    AC_DEFINE(HAVE_EVAS_OPENGL_GLEW, 1, [Evas GL Glew Engine Support])
-    have_opengl_glew="yes"
-    glew_libs="-lglew32 -lgdi32"
-  ],
-  [ have_opengl_glew="no" ]
-)
+have_opengl_glew="no"
 AM_CONDITIONAL(BUILD_OPENGL_GLEW, test "x$have_opengl_glew" = "xyes")
 
 dnl OpenGL X11
-PKG_CHECK_EXISTS(evas-opengl-x11,
-  [
-    AC_DEFINE(HAVE_EVAS_OPENGL_X11, 1, [Evas GL X11 Engine Support])
-    have_opengl_x11="yes"
-  ],
-  [ have_opengl_x11="no" ]
-)
+have_opengl_x11="no"
 AM_CONDITIONAL(BUILD_OPENGL_X11, test "x$have_opengl_x11" = "xyes")
 
 dnl Software X11
@@ -109,49 +96,23 @@
 
 # TODO: Check availability of libs
 dnl DirectDraw
-PKG_CHECK_EXISTS(evas-software-ddraw,
-  [
-    AC_DEFINE(HAVE_EVAS_SOFTWARE_DDRAW, 1, [Evas Software DirectDraw Engine Support])
-    have_software_ddraw="yes"
-    ddraw_libs="-lddraw -ldxguid"
-  ],
-  [ have_software_ddraw="no" ]
-)
+have_software_ddraw="no"
 AM_CONDITIONAL(BUILD_SOFTWARE_DDRAW, test "x$have_software_ddraw" = "xyes")
 
 # TODO: Check availability of libs
 dnl 16bit DirectDraw
-PKG_CHECK_EXISTS(evas-software-16-ddraw,
-  [
-    AC_DEFINE(HAVE_EVAS_SOFTWARE_16_DDRAW, 1, [Evas Software 16bit DirectDraw Engine Support])
-    have_software_16_ddraw="yes"
-    ddraw_libs="-lddraw -ldxguid"
-  ],
-  [ have_software_16_ddraw="no" ]
-)
+have_software_16_ddraw="no"
 AM_CONDITIONAL(BUILD_SOFTWARE_16_DDRAW, test "x$have_software_16_ddraw" = "xyes")
 
 # TODO: Check availability of libs
 dnl Direct3D
-PKG_CHECK_EXISTS(evas-direct3d,
-  [
-    AC_DEFINE(HAVE_EVAS_DIRECT3D, 1, [Evas Direct3D Engine Support])
-    have_direct3d="yes"
-    d3d_libs="-ld3d9 -ld3dx9d"
-  ],
-  [ have_direct3d="no" ]
-)
+have_direct3d="no"
 AM_CONDITIONAL(BUILD_DIRECT3D, test "x$have_direct3d" = "xyes")
 
 dnl Software SDL
-PKG_CHECK_EXISTS(evas-software-sdl,
-  [
-    AC_DEFINE(HAVE_EVAS_SOFTWARE_SDL, 1, [Evas Software SDL Engine Support])
-    have_software_sdl="yes"
-  ],
-  [ have_direct3d="no" ]
-)
-AM_CONDITIONAL(BUILD_SOFTWARE_SDL, test "x$have_software_sdl" = "xyes")
+have_software_sdl="no"
+AM_CONDITIONAL(BUILD_SOFTWARE_SDL, test "x$have_software_sdl" = "xyes")
+
 
 if test "x$have_software_x11" = "xyes" -o "x$have_software_16_x11" = "xyes" -o "x$have_xrender_x11" = "xyes" -o "x$have_opengl_x11" = "xyes"; then
   AC_PATH_XTRA
