diff -Nur elitaire-0.0.4~/src/intl.h elitaire-0.0.4/src/intl.h
--- elitaire-0.0.4~/src/intl.h	2005-08-18 19:02:30.000000000 -0700
+++ elitaire-0.0.4/src/intl.h	2005-08-18 19:06:05.000000000 -0700
@@ -13,8 +13,12 @@
 
 inline char * sgettext(const char *msgid)
 {
+#ifdef ENABLE_NLS
 	char *msgval = gettext (msgid);
 	if (msgval == msgid)
+#else
+	  char *msgval;
+#endif
 		msgval = strrchr (msgid, '|') + 1;
 	return msgval;
 }
