--- linux-2.6.9-rc3-ds1.commit/arch/arm/nwfpe/double_cpdo.c	2004-08-14 07:36:11.000000000 +0200
+++ linux-2.6.9-rc3-ds1.snap/arch/arm/nwfpe/double_cpdo.c	2004-10-10 20:29:15.514512796 +0200
@@ -75,7 +75,11 @@
 	union float64_components u;
 
 	u.f64 = rFm;
+#ifdef __ARMEB__
+	u.i[0] ^= 0x80000000;
+#else
 	u.i[1] ^= 0x80000000;
+#endif
 
 	return u.f64;
 }
@@ -85,7 +89,11 @@
 	union float64_components u;
 
 	u.f64 = rFm;
+#ifdef __ARMEB__
+	u.i[0] &= 0x7fffffff;
+#else
 	u.i[1] &= 0x7fffffff;
+#endif
 
 	return u.f64;
 }

