From 7ed8556541d29a54a6d47e5156f42002e6051f0b Mon Sep 17 00:00:00 2001
From: Philip Balister <philip@opensdr.com>
Date: Tue, 1 Nov 2011 14:16:59 -0400
Subject: [PATCH] Disable python checks so grc builds.

Signed-off-by: Philip Balister <philip@opensdr.com>
---
 config/grc_grc.m4 |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/config/grc_grc.m4 b/config/grc_grc.m4
index 0474c4e..c17df7a 100644
--- a/config/grc_grc.m4
+++ b/config/grc_grc.m4
@@ -28,10 +28,10 @@ AC_DEFUN([GRC_GRC],[
     dnl ########################################
     if test $passed = yes; then
         PYTHON_CHECK_MODULE([sys],[Python >= 2.5],[],[passed=no],[sys.version.split()[[0]] >= "2.5"])
-        PYTHON_CHECK_MODULE([Cheetah],[Python Cheetah templates >= 2.0.0],[],[passed=no],[Cheetah.Version >= "2.0.0"])
-        PYTHON_CHECK_MODULE([lxml.etree],[Python lxml wrappers >= 1.3.6],[],[passed=no],[lxml.etree.LXML_VERSION >= (1, 3, 6, 0)])
-        PYTHON_CHECK_MODULE([gtk],[Python gtk wrappers >= 2.10.0],[],[passed=no],[gtk.pygtk_version >= (2, 10, 0)])
-        PYTHON_CHECK_MODULE([numpy],[NumPy],[],[passed=no],[True])
+dnl        PYTHON_CHECK_MODULE([Cheetah],[Python Cheetah templates >= 2.0.0],[],[passed=no],[Cheetah.Version >= "2.0.0"])
+dnl        PYTHON_CHECK_MODULE([lxml.etree],[Python lxml wrappers >= 1.3.6],[],[passed=no],[lxml.etree.LXML_VERSION >= (1, 3, 6, 0)])
+dnl        PYTHON_CHECK_MODULE([gtk],[Python gtk wrappers >= 2.10.0],[],[passed=no],[gtk.pygtk_version >= (2, 10, 0)])
+dnl        PYTHON_CHECK_MODULE([numpy],[NumPy],[],[passed=no],[True])
     fi
 
     AC_CONFIG_FILES([ \
-- 
1.7.6.4

