---
 Makefile |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: uClibc-0.9.27/Makefile
===================================================================
--- uClibc-0.9.27.orig/Makefile	2008-07-23 20:05:34.000000000 -0700
+++ uClibc-0.9.27/Makefile	2008-07-23 20:06:27.000000000 -0700
@@ -243,14 +243,14 @@ endif
 .PHONY: utils
 ifeq ($(strip $(HAVE_SHARED)),y)
 utils:
-	$(MAKE) CROSS=$(CROSS) CC=$(CC) -C utils
+	$(MAKE) CROSS="$(CROSS)" CC="$(CC)" -C utils
 else
 utils: dummy
 endif
 
 # Installs helper applications, such as 'ldd' and 'ldconfig'
 install_utils: utils
-	$(MAKE) CROSS=$(CROSS) CC=$(CC) -C utils install
+	$(MAKE) CROSS="$(CROSS)" CC="$(CC)" -C utils install
 #ifeq ($(strip $(UCLIBC_HAS_LOCALE)),y)
 #	@$(MAKE) -C libc/misc/wchar iconv.target
 #	$(INSTALL) -d $(PREFIX)$(RUNTIME_PREFIX)/usr/bin;
