--- dillo-0.8.5/dpi/Makefile.am.orig	2005-12-20 15:12:15.000000000 +0000
+++ dillo-0.8.5/dpi/Makefile.am	2005-12-20 15:13:12.000000000 +0000
@@ -1,4 +1,4 @@
-AM_CFLAGS = @GLIB_CFLAGS@
+AM_CFLAGS = @GTK_CFLAGS@
 
 bookmarksdir = $(libdir)/dillo/dpi/bookmarks
 downloadsdir = $(libdir)/dillo/dpi/downloads
@@ -13,12 +13,12 @@
 hello_PROGRAMS = hello.filter.dpi
 file_PROGRAMS = file.dpi
 
-bookmarks_dpi_LDADD = @GLIB_LIBS@
-downloads_dpi_LDADD = @GLIB_LIBS@
-ftp_filter_dpi_LDADD = @GLIB_LIBS@
-https_filter_dpi_LDADD = @GLIB_LIBS@ @LIBSSL_LIBS@
-hello_filter_dpi_LDADD = @GLIB_LIBS@
-file_dpi_LDADD = @GLIB_LIBS@ @LIBPTHREAD_LIBS@
+bookmarks_dpi_LDADD = @GTK_LIBS@
+downloads_dpi_LDADD = @GTK_LIBS@
+ftp_filter_dpi_LDADD = @GTK_LIBS@
+https_filter_dpi_LDADD = @GTK_LIBS@ @LIBSSL_LIBS@
+hello_filter_dpi_LDADD = @GTK_LIBS@
+file_dpi_LDADD = @GTK_LIBS@ @LIBPTHREAD_LIBS@
 
 file_dpi_LDFLAGS = @LIBPTHREAD_LDFLAGS@
 
