diff -urN linux-2.6.27.old/arch/arm/mach-orion5x/kurobox_pro-setup.c linux-2.6.27/arch/arm/mach-orion5x/kurobox_pro-setup.c
--- linux-2.6.27.old/arch/arm/mach-orion5x/kurobox_pro-setup.c	2008-09-13 16:25:58.000000000 +0100
+++ linux-2.6.27/arch/arm/mach-orion5x/kurobox_pro-setup.c	2008-09-13 16:35:53.000000000 +0100
@@ -53,17 +53,25 @@
 
 static struct mtd_partition kurobox_pro_nand_parts[] = {
 	{
-		.name	= "uImage",
+		.name	= "foonas",
 		.offset	= 0,
-		.size	= SZ_4M,
+		.size	= SZ_16M,
 	}, {
-		.name	= "rootfs",
-		.offset	= SZ_4M,
-		.size	= SZ_64M,
+		.name	= "foonas-em",
+		.offset	= SZ_16M,
+		.size	= SZ_16M,
+	}, {
+		.name	= "foonas-iscsi",
+		.offset	= SZ_32M,
+		.size	= SZ_16M,
+	}, {
+		.name	= "foonas-config",
+		.offset	= SZ_32M + SZ_16M,
+		.size	= SZ_32M,
 	}, {
 		.name	= "extra",
-		.offset	= SZ_4M + SZ_64M,
-		.size	= SZ_256M - (SZ_4M + SZ_64M),
+		.offset = SZ_64M + SZ_16M,
+		.size	= SZ_256M - (SZ_64M + SZ_16M),
 	},
 };
 
