From 82ea3295e435b49478cf7943100b22286baf47dc Mon Sep 17 00:00:00 2001
From: Jonathan Morton <jmorton@sd070.hel.movial.fi>
Date: Thu, 4 Jun 2009 10:43:41 +0300
Subject: [PATCH] Fix filler bug.

---
 pixman/pixman-arm-neon.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/pixman/pixman-arm-neon.c b/pixman/pixman-arm-neon.c
index 18e9a16..67f0b63 100644
--- a/pixman/pixman-arm-neon.c
+++ b/pixman/pixman-arm-neon.c
@@ -1377,7 +1377,7 @@ pixman_fill_neon (uint32_t *bits,
 	"beq		2f\n"
 	"strh		%[color], [r4]!\n"
 	"2:\n"
-	"tst		r5, #1\n"
+	"tst		%[width], #1\n"
 	"beq		2f\n"
 	"strb		%[color], [r4]!\n"
 	"2:\n"
-- 
1.5.6.3

