From 5cdd43ac753780ef2cfd6adde822af5dcba6091f Mon Sep 17 00:00:00 2001
From: Jonathan Morton <jmorton@sd070.hel.movial.fi>
Date: Thu, 4 Jun 2009 15:11:40 +0300
Subject: [PATCH] Really fix filler bug.

---
 pixman/pixman-arm-neon.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/pixman/pixman-arm-neon.c b/pixman/pixman-arm-neon.c
index cfb69f4..0c05f64 100644
--- a/pixman/pixman-arm-neon.c
+++ b/pixman/pixman-arm-neon.c
@@ -1369,15 +1369,15 @@ pixman_fill_neon (uint32_t *bits,
 	"2:\n"
 	"tst		%[width], #4\n"
 	"beq		2f\n"
-	"str		%[color], [r4]!\n"
+	"str		%[color], [r4], #4\n"
 	"2:\n"
 	"tst		%[width], #2\n"
 	"beq		2f\n"
-	"strh		%[color], [r4]!\n"
+	"strh		%[color], [r4], #2\n"
 	"2:\n"
 	"tst		%[width], #1\n"
 	"beq		2f\n"
-	"strb		%[color], [r4]!\n"
+	"strb		%[color], [r4], #1\n"
 	"2:\n"
 
 	"subs		%[height], %[height], #1\n"
-- 
1.5.6.3

