Index: git/src/Makefile.am
===================================================================
--- git.orig/src/Makefile.am	2009-04-08 02:24:01.000000000 +0800
+++ git/src/Makefile.am	2009-04-08 02:28:53.000000000 +0800
@@ -2,18 +2,49 @@
 
 lib_LTLIBRARIES = libmbfly.la
 
-noinst_PROGRAMS = X_main
 MBAF_SOURCES=mbaf/mbapp.c mbaf/mbbutton.c mbaf/mbobject.c mbaf/animated_menu.c
-libmbfly_la_SOURCES = animate.c chgcolor.c coord.c event.c geo.c	\
-	observer.c paint.c redraw_man.c rotate.c shape_path.c		\
-	shape_rect.c shape_text.c shift.c subtree_free.c timer.c 	\
-	timertool.c tools.c visibility.c X_supp.c prop.c sprite.c	\
-	mouse.c shape_image.c img_ldr.c $(MBAF_SOURCES)
+libmbfly_la_SOURCES =	animate.c \
+		      	chgcolor.c \
+			coord.c  \
+			event.c \
+			geo.c	\
+			observer.c \
+			paint.c \
+			redraw_man.c \
+			rotate.c \
+			shape_path.c \
+			shape_rect.c \
+			shape_text.c \
+			shift.c \
+			subtree_free.c \
+			timer.c \
+			timertool.c \
+			tools.c \
+			visibility.c \
+			X_supp.c \
+			prop.c \
+			sprite.c \
+			mouse.c \
+			shape_image.c \
+			img_ldr.c \
+			$(MBAF_SOURCES)
+
+libmbfly_la_CFLAGS = @cairo_CFLAGS@ \
+		     @pangocairo_CFLAGS@
+libmbfly_la_LDFLAGS = @cairo_LIBS@ \
+		      @pangocairo_LIBS@
 
-libmbfly_la_CPPFLAGS = @cairo_CFLAGS@ @pangocairo_CFLAGS@
-libmbfly_la_LDFLAGS = @cairo_LIBS@ @pangocairo_LIBS@
+bin_PROGRAMS = X_main
+noinst_PROGRAMS = X_main
 
 X_main_SOURCES = X_main.c
-X_main_LDADD = $(top_builddir)/src/libmbfly.la
-X_main_CPPFLAGS = @pangocairo_CFLAGS@ -I$(top_builddir)/src
-X_main_LDFLAGS = @pangocairo_LIBS@
+X_main_LDADD = ${top_builddir}/src/libmbfly.la
+X_main_CFLAGS = @pangocairo_CFLAGS@ \
+                 @cairo_CFLAGS@ \
+                 -I${top_builddir}/include
+X_main_LDFLAGS = @pangocairo_LIBS@ \
+                @cairo_LIBS@
+
+X_main: libmbfly.la
+	$(LIBTOOL) --tag=CC   --mode=link $(CC) $(CFLAGS) $(LDFLAGS) $(X_main_CFLAGS) $(X_main_LIBADD) X_main.c -o $@ $<
+

