Index: cups-1.1.23/scheduler/classes.c
===================================================================
--- cups-1.1.23.orig/scheduler/classes.c
+++ cups-1.1.23/scheduler/classes.c
@@ -622,7 +622,7 @@ SaveAllClasses(void)
 
   curtime = time(NULL);
   curdate = localtime(&curtime);
-  strftime(temp, sizeof(temp) - 1, CUPS_STRFTIME_FORMAT, curdate);
+  strftime(temp, sizeof(temp) - 1, "%Y-%m-%d %H:%M", curdate);
 
   cupsFilePuts(fp, "# Class configuration file for " CUPS_SVERSION "\n");
   cupsFilePrintf(fp, "# Written by cupsd on %s\n", temp);
Index: cups-1.1.23/scheduler/printers.c
===================================================================
--- cups-1.1.23.orig/scheduler/printers.c
+++ cups-1.1.23/scheduler/printers.c
@@ -1066,7 +1066,7 @@ SaveAllPrinters(void)
 
   curtime = time(NULL);
   curdate = localtime(&curtime);
-  strftime(temp, sizeof(temp) - 1, CUPS_STRFTIME_FORMAT, curdate);
+  strftime(temp, sizeof(temp) - 1, "%Y-%m-%d %H:%M", curdate);
 
   cupsFilePuts(fp, "# Printer configuration file for " CUPS_SVERSION "\n");
   cupsFilePrintf(fp, "# Written by cupsd on %s\n", temp);
