diff -ur aMule-2.1.0_orig/src/Parser.cpp aMule-2.1.0/src/Parser.cpp
--- aMule-2.1.0_orig/src/Parser.cpp	2005-12-20 05:13:25.000000000 +0100
+++ aMule-2.1.0/src/Parser.cpp	2006-01-29 15:16:45.000000000 +0100
@@ -203,14 +203,14 @@
 #   define YYMALLOC malloc
 #   if (! defined (malloc) && ! defined (YYINCLUDED_STDLIB_H) \
 	&& (defined (__STDC__) || defined (__cplusplus)))
-void *malloc (YYSIZE_T); /* INFRINGES ON USER NAME SPACE */
+/*void *malloc (YYSIZE_T);*/ /* INFRINGES ON USER NAME SPACE */
 #   endif
 #  endif
 #  ifndef YYFREE
 #   define YYFREE free
 #   if (! defined (free) && ! defined (YYINCLUDED_STDLIB_H) \
 	&& (defined (__STDC__) || defined (__cplusplus)))
-void free (void *); /* INFRINGES ON USER NAME SPACE */
+/*void free (void *);*/ /* INFRINGES ON USER NAME SPACE */
 #   endif
 #  endif
 #  ifdef __cplusplus
