--- mdadm-1.12.0/Makefile	2005-08-22 16:53:26.000000000 -0700
+++ mdadm-1.12.0/Makefile~	2005-08-22 16:53:04.000000000 -0700
@@ -119,10 +119,10 @@
 $(OBJS) : mdadm.h
 
 install : mdadm mdadm.8 md.4 mdadm.conf.5
-	$(INSTALL) -D $(STRIP) -m 755 mdadm $(DESTDIR)$(BINDIR)/mdadm
-	$(INSTALL) -D -m 644 mdadm.8 $(DESTDIR)$(MAN8DIR)/mdadm.8
-	$(INSTALL) -D -m 644 md.4 $(DESTDIR)$(MAN4DIR)/md.4
-	$(INSTALL) -D -m 644 mdadm.conf.5 $(DESTDIR)$(MAN5DIR)/mdadm.conf.5
+	$(INSTALL) -D -s -m 755 mdadm $(DESTDIR)$(BINDIR)
+	$(INSTALL) -D -m 644 mdadm.8 $(DESTDIR)$(MAN8DIR)
+	$(INSTALL) -D -m 644 md.4 $(DESTDIR)$(MAN4DIR)
+	$(INSTALL) -D -m 644 mdadm.conf.5 $(DESTDIR)$(MAN5DIR)
 
 clean : 
 	rm -f mdadm $(OBJS) core *.man mdadm.tcc mdadm.uclibc mdadm.static *.orig *.porig *.rej *.alt \
