diff -Naur socat-1.3/xioopen.h socat-1.3_patched/xioopen.h
--- socat-1.3/xioopen.h	2003-02-15 15:26:22.000000000 +0100
+++ socat-1.3_patched/xioopen.h	2006-09-05 10:49:15.000000000 +0200
@@ -42,7 +42,6 @@
 extern const char *ddirection[];
 extern const char *filetypenames[];
 extern const struct addrname addressnames[];
-extern const char *addressgroupnames[];
 extern const struct optname optionnames[];
 
 extern int xioopen_makedual(xiofile_t *file);
