Index: epdf/configure.in
===================================================================
--- epdf.orig/configure.in
+++ epdf/configure.in
@@ -95,8 +95,8 @@
      [
       have_ewl="yes"
       requirements="$requirements ewl"
-      EWL_TEST_LIBS_DIR="`$PKG_CONFIG --variable=testsdir ewl`"
-      EWL_TEST_SOURCE_DIR="`$PKG_CONFIG --variable=testssrcdir ewl`"
+      EWL_TEST_LIBS_DIR="/usr/lib/ewl/tests/"
+      EWL_TEST_SOURCE_DIR="/usr/share/ewl/examples/"
       AC_SUBST(EWL_TEST_LIBS_DIR)
       AC_SUBST(EWL_TEST_SOURCE_DIR)
      ],
@@ -144,7 +144,7 @@
   [
    have_epsilon="yes"
    requirements="$requirements epsilon imlib2"
-   EPSILON_PLUGINS_DIR="`$PKG_CONFIG --variable=pluginsdir epsilon`"
+   EPSILON_PLUGINS_DIR="/usr/lib/epsilon/plugins/"
    AC_SUBST(EPSILON_PLUGINS_DIR)
   ],
   [
