--- libopie2/opiecore/device/odevice_zaurus.cpp	10 Aug 2008 09:07:32 -0000	1.55
+++ libopie2/opiecore/device/odevice_zaurus.cpp	20 Sep 2008 23:56:20 -0000
@@ -313,14 +313,14 @@
     }
 
     // set default qte driver
-    switch( d->m_model )
+/*    switch( d->m_model )
     {
         case Model_Zaurus_SLC7x0:
             d->m_qteDriver = "W100";
             break;
-        default:
+        default:*/
             d->m_qteDriver = "Transformed";
-    }
+//    }
 
     m_leds[0] = Led_Off;
 
