Index: gpe-conf-0.1.22/battery.c
===================================================================
--- gpe-conf-0.1.22.orig/battery.c	2005-02-17 13:13:00.000000000 +0000
+++ gpe-conf-0.1.22/battery.c	2005-02-17 13:39:43.000000000 +0000
@@ -335,7 +335,10 @@
 					break;
 				}
 			}
-			
+		
+                        if (percent > 100) percent=100;
+                        if (percent < 0 )  percent=0;
+
 			gtk_progress_bar_set_fraction (GTK_PROGRESS_BAR (batt_int.bar),
 			                               (float)percent/100.0);
 			toolbar_set_style (batt_int.bar, barstate);
