--- a/Makefile.am	2010-01-22 21:27:04.000000000 +0100
+++ b/Makefile.am	2010-01-22 21:27:43.000000000 +0100
@@ -225,9 +225,9 @@
 # Warning: This overrides autoconf's normal link-line generatyion process
 if LIBGPSMM_ENABLE
 libgps_la_SOURCES += libgpsmm.cpp
-libgps_la_LINK = /bin/sh ./libtool --tag=CXX --mode=link g++ $(libgps_la_LDFLAGS) -o $@
+libgps_la_LINK = $(LIBTOOL) --tag=CXX --mode=link g++ $(libgps_la_LDFLAGS) -o $@
 else
-libgps_la_LINK = /bin/sh ./libtool --tag=CC --mode=link gcc $(libgps_la_LDFLAGS) -o $@
+libgps_la_LINK = $(LIBTOOL) --tag=CC --mode=link gcc $(libgps_la_LDFLAGS) -o $@
 endif
 
 nodist_libgps_la_SOURCES = packet_names.h ais_json.i
