
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- linux-wlan-ng-0.2.2/src/Makefile~only-the-utils
+++ linux-wlan-ng-0.2.2/src/Makefile
@@ -44,9 +44,9 @@
 
 -include ../config.mk
 
-DIRS=p80211 prism2 shared wlanctl wland nwepgen \
+DIRS=prism2 shared wlanctl wland nwepgen \
 	mkmeta wlancfg
-ALL_DIRS=mkmeta p80211 prism2 shared wlanctl wland \
+ALL_DIRS=mkmeta prism2 shared wlanctl wland \
 	nwepgen wlancfg
 
 ifneq ($(wildcard *.addon),) 
--- linux-wlan-ng-0.2.2/src/prism2/Makefile~only-the-utils
+++ linux-wlan-ng-0.2.2/src/prism2/Makefile
@@ -44,7 +44,7 @@
 
 -include ../../config.mk
 
-DIRS=driver ridlist download
+DIRS=ridlist download
 
 ifneq ($(wildcard *.addon),) 
 DIRS+=`cat *.addon`
