--- dropbear-0.44test4/options.h.orig	2004-10-01 23:53:33.000000000 +0200
+++ dropbear-0.44test4/options.h	2004-10-01 23:53:59.000000000 +0200
@@ -161,7 +161,7 @@
 /* The command to invoke for xauth when using X11 forwarding.
  * "-q" for quiet */
 #ifndef XAUTH_COMMAND
-#define XAUTH_COMMAND "/usr/X11R6/bin/xauth -q"
+#define XAUTH_COMMAND "xauth -q"
 #endif
 
 /* if you want to enable running an sftp server (such as the one included with
