--- Makefile.orig	2005-04-07 10:19:08.392784784 +0100
+++ Makefile	2005-04-07 10:21:26.262825352 +0100
@@ -8,11 +8,6 @@
 
 PRG_CFG = codec-cfg
 
-# Do not strip the binaries at installation
-ifeq ($(STRIPBINARIES),yes)
-INSTALLSTRIP = -s
-endif
-
 # These subdirectories require installation due to binaries within them.
 ifeq ($(VIDIX),yes)
 SUBDIRS += libdha vidix
@@ -283,7 +278,7 @@
 $(MENCODER_DEP): version.h help_mp.h
 
 $(PRG_CFG): version.h codec-cfg.c codec-cfg.h
-	$(HOST_CC) $(HOST_CFLAGS) -I. -g codec-cfg.c mp_msg.c -o $(PRG_CFG) -DCODECS2HTML $(EXTRA_LIB) $(EXTRA_INC) $(I18NLIBS)
+	$(BUILD_CC) $(BUILD_CFLAGS) -I. -g codec-cfg.c mp_msg.c -o $(PRG_CFG) -DCODECS2HTML $(EXTRA_LIB) $(EXTRA_INC) $(I18NLIBS)
 
 install: $(ALL_PRG)
 ifeq ($(VIDIX),yes)
@@ -368,7 +363,6 @@
 	@for a in $(PARTS); do $(MAKE) -C $$a distclean; done
 
 strip:
-	strip -s $(ALL_PRG)
 
 dep:	depend
 
