--- rhythmbox-0.8.4/omf.make	2004-04-15 12:30:20.000000000 -0700
+++ rhythmbox-0.8.4.new/omf.make	2004-06-08 16:11:06.130981816 -0700
@@ -41,7 +41,9 @@
 install-data-hook-omf:
 	$(mkinstalldirs) $(DESTDIR)$(omf_dest_dir)
 	for file in $(omffile); do \
-		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
+		if [ -f $$file.out ]; then \
+			$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
+		fi; \
 	done
 	-scrollkeeper-update -o $(DESTDIR)$(omf_dest_dir)
 
