--- mythtv-0.18.1/libs/libavcodec/msmpeg4.c.old	2005-06-03 21:31:43.000000000 +0100
+++ mythtv-0.18.1/libs/libavcodec/msmpeg4.c	2005-06-03 21:31:51.000000000 +0100
@@ -723,7 +723,7 @@
        necessitate to modify mpegvideo.c. The problem comes from the
        fact they decided to store the quantized DC (which would lead
        to problems if Q could vary !) */
-#if (defined(ARCH_X86) || defined(ARCH_X86_64)) && !defined PIC
+#if (defined(ARCH_X86) || defined(ARCH_X86_64)) && !defined __PIC__
     asm volatile(
         "movl %3, %%eax		\n\t"
 	"shrl $1, %%eax		\n\t"
--- mythtv-0.18.1/libs/libavcodec/i386/dsputil_mmx_avg.h.old	2005-06-03 21:38:57.000000000 +0100
+++ mythtv-0.18.1/libs/libavcodec/i386/dsputil_mmx_avg.h	2005-06-03 21:39:02.000000000 +0100
@@ -90,7 +90,7 @@
 	"add	$16, %2			\n\t"
 	"subl	$4, %0			\n\t"
 	"jnz	1b			\n\t"
-#ifdef PIC //Note "+bm" and "+mb" are buggy too (with gcc 3.2.2 at least) and cant be used
+#ifdef __PIC__ //Note "+bm" and "+mb" are buggy too (with gcc 3.2.2 at least) and cant be used
 	:"+m"(h), "+a"(src1), "+c"(src2), "+d"(dst)
 #else
 	:"+b"(h), "+a"(src1), "+c"(src2), "+d"(dst)
@@ -137,7 +137,7 @@
 	"add	$32, %2			\n\t"
 	"subl	$4, %0			\n\t"
 	"jnz	1b			\n\t"
-#ifdef PIC //Note "+bm" and "+mb" are buggy too (with gcc 3.2.2 at least) and cant be used
+#ifdef __PIC__ //Note "+bm" and "+mb" are buggy too (with gcc 3.2.2 at least) and cant be used
 	:"+m"(h), "+a"(src1), "+c"(src2), "+d"(dst)
 #else
 	:"+b"(h), "+a"(src1), "+c"(src2), "+d"(dst)
@@ -207,7 +207,7 @@
 	"add	$32, %2			\n\t"
 	"subl	$4, %0			\n\t"
 	"jnz	1b			\n\t"
-#ifdef PIC //Note "+bm" and "+mb" are buggy too (with gcc 3.2.2 at least) and cant be used
+#ifdef __PIC__ //Note "+bm" and "+mb" are buggy too (with gcc 3.2.2 at least) and cant be used
 	:"+m"(h), "+a"(src1), "+c"(src2), "+d"(dst)
 #else
 	:"+b"(h), "+a"(src1), "+c"(src2), "+d"(dst)
@@ -262,7 +262,7 @@
 	"add	$16, %2			\n\t"
 	"subl	$4, %0			\n\t"
 	"jnz	1b			\n\t"
-#ifdef PIC //Note "+bm" and "+mb" are buggy too (with gcc 3.2.2 at least) and cant be used
+#ifdef __PIC__ //Note "+bm" and "+mb" are buggy too (with gcc 3.2.2 at least) and cant be used
 	:"+m"(h), "+a"(src1), "+c"(src2), "+d"(dst)
 #else
 	:"+b"(h), "+a"(src1), "+c"(src2), "+d"(dst)
@@ -314,7 +314,7 @@
 	"add	$32, %2			\n\t"
 	"subl	$4, %0			\n\t"
 	"jnz	1b			\n\t"
-#ifdef PIC //Note "+bm" and "+mb" are buggy too (with gcc 3.2.2 at least) and cant be used
+#ifdef __PIC__ //Note "+bm" and "+mb" are buggy too (with gcc 3.2.2 at least) and cant be used
 	:"+m"(h), "+a"(src1), "+c"(src2), "+d"(dst)
 #else
 	:"+b"(h), "+a"(src1), "+c"(src2), "+d"(dst)
@@ -402,7 +402,7 @@
 	"add	$32, %2			\n\t"
 	"subl	$2, %0			\n\t"
 	"jnz	1b			\n\t"
-#ifdef PIC //Note "+bm" and "+mb" are buggy too (with gcc 3.2.2 at least) and cant be used
+#ifdef __PIC__ //Note "+bm" and "+mb" are buggy too (with gcc 3.2.2 at least) and cant be used
 	:"+m"(h), "+a"(src1), "+c"(src2), "+d"(dst)
 #else
 	:"+b"(h), "+a"(src1), "+c"(src2), "+d"(dst)
@@ -456,7 +456,7 @@
 	"add	$32, %2			\n\t"
 	"subl	$2, %0			\n\t"
 	"jnz	1b			\n\t"
-#ifdef PIC //Note "+bm" and "+mb" are buggy too (with gcc 3.2.2 at least) and cant be used
+#ifdef __PIC__ //Note "+bm" and "+mb" are buggy too (with gcc 3.2.2 at least) and cant be used
 	:"+m"(h), "+a"(src1), "+c"(src2), "+d"(dst)
 #else
 	:"+b"(h), "+a"(src1), "+c"(src2), "+d"(dst)
@@ -529,7 +529,7 @@
 	"add	$32, %2			\n\t"
 	"subl	$2, %0			\n\t"
 	"jnz	1b			\n\t"
-#ifdef PIC //Note "+bm" and "+mb" are buggy too (with gcc 3.2.2 at least) and cant be used
+#ifdef __PIC__ //Note "+bm" and "+mb" are buggy too (with gcc 3.2.2 at least) and cant be used
 	:"+m"(h), "+a"(src1), "+c"(src2), "+d"(dst)
 #else
 	:"+b"(h), "+a"(src1), "+c"(src2), "+d"(dst)
