Index: viking-0.9.6/src/icons/Makefile.am
===================================================================
--- viking-0.9.6.orig/src/icons/Makefile.am	2008-07-27 20:25:34.000000000 +0000
+++ viking-0.9.6/src/icons/Makefile.am	2008-09-05 14:15:29.000000000 +0000
@@ -134,8 +134,8 @@
 
 .png.png_h:
 	gdk-pixbuf-csource --name="`basename $< .png`" --struct --extern $< > $@
-	cat $@ | sed -e 's/\(pixel_data: ..\)/\1 (unsigned char \*)/' > temp.h
-	mv temp.h $@
+	cat $@ | sed -e 's/\(pixel_data: ..\)/\1 (unsigned char \*)/' > temp$@.h
+	mv temp$@.h $@
 
 INCLUDES        = @GTK_CFLAGS@
 AM_CFLAGS       = -Wall -g -D_GNU_SOURCE
