Index: mrxvt-0.5.2/src/rxvt.h
===================================================================
--- mrxvt-0.5.2.orig/src/rxvt.h
+++ mrxvt-0.5.2/src/rxvt.h
@@ -488,15 +488,8 @@ typedef char*	    XPointer;
  * turn is unsafe without checking. Untainted defines the type of data that is
  * safe.
  */
-#ifdef __GNUC__
-# define $tainted
-# define $untainted
-# define TAINTED    $tainted
-# define UNTAINTED  $untainted
-#else
-# define TAINTED
-# define UNTAINTED
-#endif
+#define TAINTED
+#define UNTAINTED
 
 
 /*
