---
 sim/common/sim-signal.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: gdb-6.8/sim/common/sim-signal.c
===================================================================
--- gdb-6.8.orig/sim/common/sim-signal.c
+++ gdb-6.8/sim/common/sim-signal.c
@@ -26,7 +26,7 @@ along with this program.  If not, see <h
    to not think the process has died (so it can be debugged at the point of
    failure).  */
 
-#ifdef _MSC_VER
+#ifdef __WIN32
 #ifndef SIGTRAP
 #define SIGTRAP 5
 #endif
