--- thttpd-2.25b/extras/htpasswd.c.old	2010-04-28 17:23:40.000000000 +0100
+++ thttpd-2.25b/extras/htpasswd.c	2010-04-28 17:24:00.000000000 +0100
@@ -49,7 +49,7 @@
     while((line[y++] = line[x++]));
 }
 
-static int getline(char *s, int n, FILE *f) {
+static int ht_getline(char *s, int n, FILE *f) {
     register int i=0;
 
     while(1) {
@@ -189,7 +189,7 @@
     strcpy(user,argv[2]);
 
     found = 0;
-    while(!(getline(line,MAX_STRING_LEN,f))) {
+    while(!(ht_getline(line,MAX_STRING_LEN,f))) {
         if(found || (line[0] == '#') || (!line[0])) {
             putline(tfp,line);
             continue;
