Index: xboard-4.4.1-r0/xboard-4.4.1/Makefile.in
===================================================================
--- xboard-4.4.1-r0.orig/xboard-4.4.1/Makefile.in
+++ xboard-4.4.1-r0/xboard-4.4.1/Makefile.in
@@ -939,7 +939,7 @@
 
 uninstall-man: uninstall-man6
 
-.MAKE: install-am install-strip
+.MAKE: install-am
 
 .PHONY: CTAGS GTAGS all all-am am--refresh check check-am clean \
 	clean-binPROGRAMS clean-generic ctags dist dist-all dist-bzip2 \
@@ -951,7 +951,7 @@
 	install-data-am install-dvi install-dvi-am install-exec \
 	install-exec-am install-html install-html-am install-info \
 	install-info-am install-man install-man6 install-pdf \
-	install-pdf-am install-ps install-ps-am install-strip \
+	install-pdf-am install-ps install-ps-am \
 	installcheck installcheck-am installdirs maintainer-clean \
 	maintainer-clean-aminfo maintainer-clean-generic \
 	maintainer-clean-vti mostlyclean mostlyclean-aminfo \
