--- u-boot.orig/include/configs/omap3530beagle.h
+++ u-boot/include/configs/omap3530beagle.h
@@ -261,8 +261,8 @@
 #define CFG_ENV_IS_IN_NAND	1
 #define CFG_ENV_IS_IN_ONENAND	1
 #define CFG_ENV_IS_IN_FLASH	1
-#define ONENAND_ENV_OFFSET	0xc0000 /* environment starts here  */
-#define SMNAND_ENV_OFFSET	0xc0000 /* environment starts here  */
+#define ONENAND_ENV_OFFSET	0x260000 /* environment starts here  */
+#define SMNAND_ENV_OFFSET	0x260000 /* environment starts here  */
 
 #define CFG_ENV_SECT_SIZE	boot_flash_sec
 #define CFG_ENV_OFFSET		boot_flash_off
