--- MPlayer-1.0pre6/Makefile.orig	2004-12-25 23:48:23.000000000 +0100
+++ MPlayer-1.0pre6/Makefile	2004-12-25 23:49:18.000000000 +0100
@@ -8,11 +8,6 @@
 
 PRG_CFG = codec-cfg
 
-# Do not strip the binaries at installation
-ifeq ($(STRIPBINARIES),yes)
-INSTALLSTRIP = -s
-endif
-
 # These subdirectories require installation due to binaries within them.
 ifeq ($(VIDIX),yes)
 SUBDIRS += libdha vidix
@@ -234,7 +229,7 @@
     ifeq ($(TARGET_WIN32),yes)
 	windres -o osdep/mplayer-rc.o osdep/mplayer.rc
     endif
-	$(CC) $(CFLAGS) -o $(PRG) $(OBJS_MPLAYER) libvo/libvo.a libao2/libao2.a $(MENU_LIBS) $(VIDIX_LIBS) $(GUI_LIBS) $(COMMON_LIBS) $(GTK_LIBS) $(VO_LIBS) $(AO_LIBS) $(EXTRA_LIB) $(LIRC_LIB) $(LIRCC_LIB) $(STATIC_LIB) $(ARCH_LIB) $(I18NLIBS) $(MATH_LIB)
+	$(CC) $(CFLAGS) -o $(PRG) $(OBJS_MPLAYER) libvo/libvo.a libao2/libao2.a $(MENU_LIBS) $(VIDIX_LIBS) $(GUI_LIBS) $(COMMON_LIBS) $(GTK_LIBS) $(VO_LIBS) $(AO_LIBS) $(EXTRA_LIB) $(LIRC_LIB) $(LIRCC_LIB) $(STATIC_LIB) $(ARCH_LIB) $(I18NLIBS) $(MATH_LIB) -ljpeg -lpng -lts $(LDFLAGS)
 
 mplayer.exe.spec.c: libmpcodecs/libmpcodecs.a
 	winebuild -fPIC -o mplayer.exe.spec.c -exe mplayer.exe -mcui \
@@ -267,7 +262,7 @@
 $(MENCODER_DEP): version.h help_mp.h
 
 $(PRG_CFG): version.h codec-cfg.c codec-cfg.h
-	$(HOST_CC) $(CFLAGS) -g codec-cfg.c mp_msg.c -o $(PRG_CFG) -DCODECS2HTML $(EXTRA_LIB) $(I18NLIBS)
+	$(BUILD_CC) -Ilibmpdemux -Iloader -Ilibvo -I. -g codec-cfg.c mp_msg.c -o $(PRG_CFG) -DCODECS2HTML $(EXTRA_LIB) $(I18NLIBS)
 
 install: $(ALL_PRG)
 ifeq ($(VIDIX),yes)
@@ -352,7 +347,7 @@
 	@for a in $(PARTS); do $(MAKE) -C $$a distclean; done
 
 strip:
-	strip -s $(ALL_PRG)
+
 
 dep:	depend
 
