From b608f4eb854983a051d2c9171a0ed730861587f5 Mon Sep 17 00:00:00 2001
From: Thomas White <taw@bitwiz.org.uk>
Date: Sun, 20 Sep 2009 00:08:09 +0100
Subject: [PATCH] dpms.h -> dpmsconst.h

---
 src/glamo-display.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/glamo-display.c b/src/glamo-display.c
index 4562546..fede315 100644
--- a/src/glamo-display.c
+++ b/src/glamo-display.c
@@ -38,7 +38,7 @@
 #include <unistd.h>
 
 #define DPMS_SERVER
-#include <X11/extensions/dpms.h>
+#include <X11/extensions/dpmsconst.h>
 
 #include "glamo.h"
 #include "glamo-regs.h"
-- 
1.6.2.5

