diff -urN gsoap-2.7.orig/gsoap/wsdl/Makefile.am gsoap-2.7.orig/gsoap/wsdl/Makefile.am
--- gsoap-2.7.orig/gsoap/wsdl/Makefile.am	2009-03-21 22:24:42.000000000 +0100
+++ gsoap-2.7.orig/gsoap/wsdl/Makefile.am	2009-04-07 11:30:50.000000000 +0200
@@ -17,7 +17,7 @@ BUILT_SOURCES=$(SOAP_CPP_SRC)
 # WSDL2H_EXTRA_LIBS=-lssl -lcrypto -lz # defined in configure.in
 
 $(SOAP_CPP_SRC) : $(SOAPHEADER)
-	$(SOAP) $(SOAP_FLAGS) $(SOAPHEADER)
+	../src/soapcpp2 $(SOAP_FLAGS) $(SOAPHEADER)
 
 #LIBS=
 
