Index: u-boot-1.3.2/drivers/mtd/nand/nand_bbt.c
===================================================================
--- u-boot-1.3.2.orig/drivers/mtd/nand/nand_bbt.c	2008-12-11 10:15:52.000000000 +0100
+++ u-boot-1.3.2/drivers/mtd/nand/nand_bbt.c	2008-12-11 10:16:39.000000000 +0100
@@ -914,10 +914,11 @@
 	.pattern = scan_ff_pattern
 };
 
+//Samsung makes sure that either the 1st or 2nd page of every initial invalid block has non-FFh data at the column address of 2048.
 static struct nand_bbt_descr largepage_memorybased = {
 	.options = 0,
 	.offs = 0,
-	.len = 2,
+	.len = 1,
 	.pattern = scan_ff_pattern
 };
 
