Index: main.c
===================================================================
RCS file: /cvs/gpe/base/gpe-beam/main.c,v
retrieving revision 1.26
diff -u -r1.26 main.c
--- main.c	16 Oct 2004 12:17:51 -0000	1.26
+++ main.c	17 Oct 2004 12:43:16 -0000
@@ -632,7 +632,7 @@
 
 
 static void
-send_vcard (void)
+do_send_vcard (void)
 {
 	if (!radio_is_on)
 		radio_on ();
@@ -651,6 +651,16 @@
 	}
 }
 
+static void
+send_vcard (void)
+{
+	scan_thread =
+		g_thread_create ((GThreadFunc)
+				 do_send_vcard, NULL, FALSE, NULL);
+	if (scan_thread == NULL)
+		gpe_perror_box (_("Unable to start sender."));
+}
+
 
 static gboolean
 get_irstatus (void)
