diff -Nur ipkg-utils.orig/ipkg-list-fields ipkg-utils/ipkg-list-fields
--- ipkg-utils.orig/ipkg-list-fields	2009-01-17 21:52:36.000000000 +0100
+++ ipkg-utils/ipkg-list-fields	2009-01-17 21:55:08.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 
 import sys, ipkg
 
diff -Nur ipkg-utils.orig/ipkg-make-index ipkg-utils/ipkg-make-index
--- ipkg-utils.orig/ipkg-make-index	2009-01-17 21:52:36.000000000 +0100
+++ ipkg-utils/ipkg-make-index	2009-01-17 21:55:13.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 # $Id: ipkg-make-index,v 1.22 2005/03/03 16:39:04 jamey Exp $
 
 import sys, os, posixpath
diff -Nur ipkg-utils.orig/ipkg-show-deps ipkg-utils/ipkg-show-deps
--- ipkg-utils.orig/ipkg-show-deps	2009-01-17 21:52:36.000000000 +0100
+++ ipkg-utils/ipkg-show-deps	2009-01-17 21:55:21.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 # $Id: ipkg-show-deps,v 1.2 2004/03/06 13:08:33 pb Exp $
 
 import sys, os, posixpath
diff -Nur ipkg-utils.orig/ipkg-unbuild ipkg-utils/ipkg-unbuild
--- ipkg-utils.orig/ipkg-unbuild	2009-01-17 21:52:35.000000000 +0100
+++ ipkg-utils/ipkg-unbuild	2009-01-17 21:55:24.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 
 import sys, os, re
 
diff -Nur ipkg-utils.orig/ipkg-update-index ipkg-utils/ipkg-update-index
--- ipkg-utils.orig/ipkg-update-index	2009-01-17 21:52:36.000000000 +0100
+++ ipkg-utils/ipkg-update-index	2009-01-17 21:55:29.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2.1
+#!/usr/bin/env python
 
 import sys, os
 from glob import glob
diff -Nur ipkg-utils.orig/ipkg-upload ipkg-utils/ipkg-upload
--- ipkg-utils.orig/ipkg-upload	2009-01-17 21:52:36.000000000 +0100
+++ ipkg-utils/ipkg-upload	2009-01-17 21:55:33.000000000 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/env python
 
 import sys, os, os.path, re
 
