diff -Nurd gpe-plucker-0.2.old/viewer.c gpe-plucker-0.2/viewer.c > remove-render.patch
--- gpe-plucker-0.2.old/viewer.c	2005-04-23 21:10:04.000000000 +0200
+++ gpe-plucker-0.2/viewer.c	2005-04-23 21:17:59.000000000 +0200
@@ -25,7 +25,6 @@
 
 /* for GPE specific things */
 #include <gpe/init.h>
-#include <gpe/render.h>
 #include <gpe/pixmaps.h>
 #include <gpe/picturebutton.h>
 #define _(_x) gettext (_x)

