--- nslu2-linksys-ramdisk-2.3r25/etc/rc.d/rc.local.orig	2004-07-27 23:37:55.000000000 -0400
+++ nslu2-linksys-ramdisk-2.3r25/etc/rc.d/rc.local	2004-09-01 21:57:17.000000000 -0400
@@ -1,5 +1,13 @@
 #!/bin/sh
+
 HOSTNAME=`hostname`
+
+if (  [ ! -f /.sda1root ] && [ -r /mnt/sda1/unslung/rc.local ] && . /mnt/sda1/unslung/rc.local ) ; then return 0 ; fi
+if (  [ ! -f /.sda2root ] && [ -r /mnt/sda2/unslung/rc.local ] && . /mnt/sda2/unslung/rc.local ) ; then return 0 ; fi
+if (  [ ! -f /.sdb1root ] && [ -r /mnt/sdb1/unslung/rc.local ] && . /mnt/sdb1/unslung/rc.local ) ; then return 0 ; fi
+if (  [ ! -f /.sdb2root ] && [ -r /mnt/sdb2/unslung/rc.local ] && . /mnt/sdb2/unslung/rc.local ) ; then return 0 ; fi
+if ( [ -r /unslung/rc.local          ] && . /unslung/rc.local          ) ; then return 0 ; fi
+
 /usr/sbin/CheckResetButton 2>/dev/null
 /usr/sbin/CheckPowerButton 2>/dev/null
 /usr/sbin/do_umount 2>/dev/null
