--- nslu2-linksys-ramdisk-2.3r25/etc/rc.d/rc.reboot.orig	2004-07-27 23:37:55.000000000 -0400
+++ nslu2-linksys-ramdisk-2.3r25/etc/rc.d/rc.reboot	2004-09-01 21:58:19.000000000 -0400
@@ -1,4 +1,13 @@
 #!/bin/sh
+
+if (  [ ! -f /.sda1root ] && [ -r /mnt/sda1/unslung/rc.reboot ] && . /mnt/sda1/unslung/rc.reboot ) ; then return 0 ; fi
+if (  [ ! -f /.sda2root ] && [ -r /mnt/sda2/unslung/rc.reboot ] && . /mnt/sda2/unslung/rc.reboot ) ; then return 0 ; fi
+if (  [ ! -f /.sdb1root ] && [ -r /mnt/sdb1/unslung/rc.reboot ] && . /mnt/sdb1/unslung/rc.reboot ) ; then return 0 ; fi
+if (  [ ! -f /.sdb2root ] && [ -r /mnt/sdb2/unslung/rc.reboot ] && . /mnt/sdb2/unslung/rc.reboot ) ; then return 0 ; fi
+if ( [ -r /unslung/rc.reboot          ] && . /unslung/rc.reboot          ) ; then return 0 ; fi
+
+/bin/echo  "Stopping UNSLUNG Packages:"; ( . /etc/rc.d/rc.unslung-stop )
+
 /bin/killall -2 upnpd 2>/dev/null
 sleep 4
 /sbin/ifdown ixp0
