--- nslu2-linksys-ramdisk-2.3r25/etc/rc.d/rc.reset_usrgrpshare.orig	2004-07-27 23:37:55.000000000 -0400
+++ nslu2-linksys-ramdisk-2.3r25/etc/rc.d/rc.reset_usrgrpshare	2004-09-01 21:58:36.000000000 -0400
@@ -1,2 +1,9 @@
 #!/bin/sh
+
+if (  [ ! -f /.sda1root ] && [ -f /mnt/sda1/unslung/rc.reset_usrgrpshare ] && . /mnt/sda1/unslung/rc.reset_usrgrpshare ) ; then return 0 ; fi
+if (  [ ! -f /.sda2root ] && [ -f /mnt/sda2/unslung/rc.reset_usrgrpshare ] && . /mnt/sda2/unslung/rc.reset_usrgrpshare ) ; then return 0 ; fi
+if (  [ ! -f /.sdb1root ] && [ -f /mnt/sdb1/unslung/rc.reset_usrgrpshare ] && . /mnt/sdb1/unslung/rc.reset_usrgrpshare ) ; then return 0 ; fi
+if (  [ ! -f /.sdb2root ] && [ -f /mnt/sdb2/unslung/rc.reset_usrgrpshare ] && . /mnt/sdb2/unslung/rc.reset_usrgrpshare ) ; then return 0 ; fi
+if ( [ -f /unslung/rc.reset_usrgrpshare          ] && . /unslung/rc.reset_usrgrpshare          ) ; then return 0 ; fi
+
 /usr/sbin/reset_ugs;
