Index: Makefile.am
===================================================================
RCS file: /cvs/gpe/base/gpe-contacts/Makefile.am,v
retrieving revision 1.15
diff -B -b -u -r1.15 Makefile.am
--- Makefile.am	20 Jun 2005 23:18:06 -0000	1.15
+++ Makefile.am	27 Jun 2005 09:39:29 -0000
@@ -47,7 +47,7 @@
 
 gpe_contacts_SOURCES += main.c
 
-pixmaps_extradir = $(datadir)/gpe/pixmaps/default
+pixmaps_extradir = $(datadir)/gpe/pixmaps/default/gpe-contacts
 pixmaps_extra_DATA = icons/entry.png \
                icons/export.png \
                icons/frame.png \
@@ -59,11 +59,6 @@
 @INTLTOOL_DESKTOP_RULE@
 
 pixmapsdir = $(datadir)/pixmaps
-
-pixmaps_extradir = $(datadir)/pixmaps
-pixmaps_extra_DATA = icons/gpe-contacts.png \
-                     icons/edit-vcard.png
-
 metadir = $(datadir)/gpe-contacts
 
 if HILDON
