
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- xmame-0.86/src/unix/unix.mak~crosscompile
+++ xmame-0.86/src/unix/unix.mak
@@ -20,6 +20,7 @@
 CFLAGS.risc       = -DALIGN_INTS -DALIGN_SHORTS 
 CFLAGS.risc_lsb   = -DALIGN_INTS -DALIGN_SHORTS -DLSB_FIRST
 CFLAGS.mips       = -DALIGN_INTS -DALIGN_SHORTS -DSGI_FIX_MWA_NOP
+CFLAGS.arm	  = -DALIGN_INTS -DALIGH_SHORTS -DLSB_FIRST
 
 ##############################################################################
 # Architecture-dependent settings
@@ -61,7 +62,7 @@
 endif
 LIBS.svgafx     = $(X11LIB) -lvga -lvgagl -lglide3
 LIBS.openstep	= -framework AppKit
-LIBS.SDL	= $(X11LIB) `$(SDL_CONFIG) --libs`
+LIBS.SDL	= `$(SDL_CONFIG) --libs`
 LIBS.photon2	= -L/usr/lib -lph -lphrender
 
 CFLAGS.x11      = $(X11INC) $(JOY_X11_CFLAGS) $(XINPUT_DEVICES_CFLAGS)
@@ -72,7 +73,7 @@
 CFLAGS.xfx      = $(X11INC) $(JOY_X11_CFLAGS) -I/usr/include/glide3
 endif
 CFLAGS.svgafx   = -I/usr/include/glide3
-CFLAGS.SDL      = $(X11INC) `$(SDL_CONFIG) --cflags` -D_REENTRANT
+CFLAGS.SDL      = `$(SDL_CONFIG) --cflags` -D_REENTRANT
 CFLAGS.photon2	=
 
 ifdef X11_DGA
@@ -562,7 +563,7 @@
 
 $(OBJ)/%.a:
 	$(CC_COMMENT) @echo 'Archiving $@ ...'
-	$(CC_COMPILE) ar $(AR_OPTS) $@ $^
+	$(CC_COMPILE) $(AR) $(AR_OPTS) $@ $^
 	$(CC_COMPILE) $(RANLIB) $@
 
 $(OSDEPEND): $(UNIX_OBJS)
@@ -591,20 +592,20 @@
 # generate C source files for the 68000 emulator
 $(OBJ)/cpu/m68000/m68kmake: src/cpu/m68000/m68kmake.c
 	$(CC_COMMENT) @echo 'M68K make $<...'
-	$(CC_COMPILE) $(CC) $(MY_CFLAGS) -DDOS -o $(OBJ)/cpu/m68000/m68kmake $<
+	$(CC_COMPILE) $(BUILD_CC) $(BUILD_CFLAGS) $(MY_CFLAGS) -DDOS -o $(OBJ)/cpu/m68000/m68kmake $<
 	$(CC_COMMENT) @echo 'Generating M68K source files...'
 	$(CC_COMPILE) $(OBJ)/cpu/m68000/m68kmake $(OBJ)/cpu/m68000 src/cpu/m68000/m68k_in.c
 
 # generate asm source files for the 68000/68020 emulators
 $(OBJ)/cpu/m68000/68000.asm:  src/cpu/m68000/make68k.c
 	$(CC_COMMENT) @echo 'Compiling $<...'
-	$(CC_COMPILE) $(CC) $(MY_CFLAGS) -O0 -DDOS -o $(OBJ)/cpu/m68000/make68k $<
+	$(CC_COMPILE) $(BUILD_CC) $(BUILD_CFLAGS) $(MY_CFLAGS) -O0 -DDOS -o $(OBJ)/cpu/m68000/make68k $<
 	$(CC_COMMENT) @echo 'Generating $@...'
 	$(CC_COMPILE) $(OBJ)/cpu/m68000/make68k $@ $(OBJ)/cpu/m68000/68000tab.asm 00
 
 $(OBJ)/cpu/m68000/68020.asm:  src/cpu/m68000/make68k.c
 	$(CC_COMMENT) @echo 'Compiling $<...'
-	$(CC_COMPILE) $(CC) $(MY_CFLAGS) -O0 -DDOS -o $(OBJ)/cpu/m68000/make68k $<
+	$(CC_COMPILE) $(BUILD_CC) $(BUILD_CFLAGS) $(MY_CFLAGS) -O0 -DDOS -o $(OBJ)/cpu/m68000/make68k $<
 	$(CC_COMMENT) @echo 'Generating $@...'
 	$(CC_COMPILE) $(OBJ)/cpu/m68000/make68k $@ $(OBJ)/cpu/m68000/68020tab.asm 20
 
