--- kexecboot-0.3/kexecboot.c	2008-08-03 17:08:31.000000000 +0200
+++ kexecboot/kexecboot.c	2008-08-04 13:35:32.167062391 +0200
@@ -84,7 +84,7 @@
 	char command[COMMAND_LINE_SIZE + 60];
 	mount(boot->device, "/mnt", boot->fstype, MS_RDONLY, NULL);
 	if( boot->cmdline )
-		sprintf(command,"/usr/sbin/kexec --command-line=\"%s root=%s rootfstype=%s\" -l %s", 
+		sprintf(command,"/usr/sbin/kexec --command-line=\"%s root=%s rootfstype=%s rootdelay=3\" -l %s", 
 			boot->cmdline, boot->device, boot->fstype, boot->kernelpath);
 	else
 		sprintf(command,"kexec -l %s", boot->kernelpath);
