Index: iptables-1.3.3/extensions/Makefile
===================================================================
--- iptables-1.3.3.orig/extensions/Makefile
+++ iptables-1.3.3/extensions/Makefile
@@ -67,10 +67,10 @@ endif
 
 ifdef NO_SHARED_LIBS
 extensions/libext.a: $(EXT_OBJS)
-	rm -f $@; ar crv $@ $(EXT_OBJS)
+	rm -f $@; $(AR) crv $@ $(EXT_OBJS)
 
 extensions/libext6.a: $(EXT6_OBJS)
-	rm -f $@; ar crv $@ $(EXT6_OBJS)
+	rm -f $@; $(AR) crv $@ $(EXT6_OBJS)
 
 extensions/initext.o: extensions/initext.c
 extensions/initext6.o: extensions/initext6.c
