diff -uNr LVM2.2.01.15.orig//lib/filters/filter-sysfs.c LVM2.2.01.15/lib/filters/filter-sysfs.c
--- LVM2.2.01.15.orig//lib/filters/filter-sysfs.c	2005-03-10 23:34:17.000000000 +0100
+++ LVM2.2.01.15/lib/filters/filter-sysfs.c	2010-10-22 12:28:36.000000000 +0200
@@ -20,6 +20,7 @@
 #ifdef linux
 
 #include <dirent.h>
+#include <sys/stat.h>
 
 static int _locate_sysfs_blocks(const char *proc, char *path, size_t len)
 {
