diff -rpN LVM2.2.01.15-orig/lib/locking/cluster_locking.c LVM2.2.01.15/lib/locking/cluster_locking.c
*** LVM2.2.01.15-orig/lib/locking/cluster_locking.c	2010-03-16 20:16:45.806355964 +0100
--- LVM2.2.01.15/lib/locking/cluster_locking.c	2010-03-16 20:19:08.946345762 +0100
***************
*** 29,34 ****
--- 29,35 ----
  #include <sys/socket.h>
  #include <sys/un.h>
  #include <unistd.h>
+ #include <limits.h>
  
  #ifndef CLUSTER_LOCKING_INTERNAL
  int lock_resource(struct cmd_context *cmd, const char *resource, int flags);
