
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- tcpdump-3.8.3/configure~fix-paths.patch
+++ tcpdump-3.8.3/configure
@@ -8904,7 +8904,7 @@
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
-  ac_lbl_cv_pcap_debug_defined=yes
+  ac_lbl_cv_pcap_debug_defined=no
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
@@ -8963,7 +8963,7 @@
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
-  ac_lbl_cv_yydebug_defined=yes
+  ac_lbl_cv_yydebug_defined=no
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
@@ -9700,7 +9700,7 @@
 echo $ECHO_N "checking for SSLeay... $ECHO_C" >&6
 ac_cv_ssleay_path=no
 incdir=no
-for dir in /usr/${host_alias} /usr /usr/local /usr/local/ssl /usr/pkg; do
+for dir in /bla; do
 	#
 	# XXX - is there a better way to check if a given library is
 	# in a given directory than checking each of the possible
