--- GConf-2.6.4/configure.in.orig	2007-03-03 18:30:35.000000000 +0000
+++ GConf-2.6.4/configure.in	2007-03-03 18:30:50.000000000 +0000
@@ -186,7 +186,7 @@
 
 AC_ARG_WITH(ipc, [  --with-ipc=[orbit/dbus/both]   choose ipc mechanism to use in the daemon, [default=dbus]], with_ipc="$withval", with_ipc=dbus)
 
-if test x$with_ipc = xorbit -o x$with_ipc = xboth; then
+if test x$with_ipc = xorbit -o x$with_ipc = xboth -o x$with_ipc = xdbus; then
   PKG_CHECK_MODULES(GCONF_ORBIT, ORBit-2.0 >= 2.4.0 linc >= 0.5.0, have_orbit=yes, have_orbit=no)
   if test x$have_orbit = xno ; then
     AC_MSG_WARN([ORBit development libraries not found])
