Index: mozilla/minimo/components/Makefile.in
===================================================================
RCS file: /cvsroot/mozilla/minimo/components/Makefile.in,v
retrieving revision 1.1.2.2
diff -u -r1.1.2.2 Makefile.in
--- mozilla/minimo/components/Makefile.in	14 Jul 2006 16:04:33 -0000	1.1.2.2
+++ mozilla/minimo/components/Makefile.in	20 Sep 2006 12:55:08 -0000
@@ -41,10 +41,10 @@
 
 include $(DEPTH)/config/autoconf.mk
 
-DIRS = phone softkb ssr device gps
+DIRS = ssr
 
 ifdef WINCE
-DIRS += ssl
+DIRS += ssl phone softkb device gps
 endif
 
 include $(topsrcdir)/config/rules.mk
Index: mozilla/minimo/components/ssr/Makefile.in
===================================================================
RCS file: /cvsroot/mozilla/minimo/components/ssr/Makefile.in,v
retrieving revision 1.1.2.1
diff -u -r1.1.2.1 Makefile.in
--- mozilla/minimo/components/ssr/Makefile.in	30 Jan 2006 18:47:01 -0000	1.1.2.1
+++ mozilla/minimo/components/ssr/Makefile.in	20 Sep 2006 14:52:46 -0000
@@ -42,7 +42,7 @@
 
 include $(DEPTH)/config/autoconf.mk
 
-MODULE_NAME     = ssr
+MODULE_NAME     = SSRModule
 MODULE		    = ssr
 LIBRARY_NAME	= ssr
 
