--- /tmp/Makefile.old	2004-07-20 22:01:25.000000000 +0200
+++ teleport-0.33/Makefile	2009-05-02 00:16:57.000000000 +0200
@@ -40,8 +40,8 @@
 
 install-program: all
 	mkdir -p $(DESTDIR)$(PREFIX)/bin
-	install -s teleport $(DESTDIR)$(PREFIX)/bin/
-	install -s tp-keygen $(DESTDIR)$(PREFIX)/bin/
+	install teleport $(DESTDIR)$(PREFIX)/bin/
+	install tp-keygen $(DESTDIR)$(PREFIX)/bin/
 	install -d $(DESTDIR)$(PREFIX)/share/pixmaps
 	install -m 644 teleport.png $(DESTDIR)$(PREFIX)/share/pixmaps
 	install -d $(DESTDIR)$(PREFIX)/share/applications
