--- git-1.6.0.4-orig/configure.ac	2008-11-18 15:28:03.000000000 -0500
+++ git-1.6.0.4/configure.ac	2008-11-18 15:30:45.000000000 -0500
@@ -373,7 +373,8 @@
 		  if (snprintf(buf, 3, "%s", "12345") != 5
 		      || strcmp(buf, "12")) return 1]])],
 	[ac_cv_snprintf_returns_bogus=no],
-	[ac_cv_snprintf_returns_bogus=yes])
+	[ac_cv_snprintf_returns_bogus=yes],
+	[ac_cv_snprintf_returns_bogus=no])
 ])
 if test $ac_cv_snprintf_returns_bogus = yes; then
 	SNPRINTF_RETURNS_BOGUS=UnfortunatelyYes
