--- gpsd-1.10/gpsd.c.orig	2004-03-12 20:25:22.281457600 +0000
+++ gpsd-1.10/gpsd.c	2004-03-12 20:27:55.263200840 +0000
@@ -529,6 +529,26 @@
 		    ",G=%6.6s",
 		    gNMEAdata.grid);
 	    break;
+	case 'H':
+	case 'h':
+	    sprintf(reply + strlen(reply),
+		     ",H=%f",
+		    gNMEAdata.track);
+	    break; 
+	case 'X':
+	case 'x':
+	    if (!FD_ISSET(fd, fds))
+		FD_SET(fd, fds);
+	     sprintf(reply + strlen(reply),
+			 " ,R=1");
+	    break;
+	case 'C':
+	case 'c':
+	    if (FD_ISSET(fd, fds))
+		FD_CLR(fd, fds);
+	    sprintf(reply + strlen(reply),
+			 " ,R=0");
+	    break;
 	case 'R':
 	case 'r':
 	    if (FD_ISSET(fd, fds)) {
