Index: linux-2.6.24/include/linux/time.h
===================================================================
--- a/include/linux/time.h
+++ b/include/linux/time.h
@@ -172,6 +172,10 @@ static inline void timespec_add_ns(struct timespec *a, u64 ns)
 {
	ns += a->tv_nsec;
	while(unlikely(ns >= NSEC_PER_SEC)) {
+		/* The following asm() prevents the compiler from
+		 * optimising this loop into a modulo operation.  */
+		asm("" : "+r"(ns));
+
		ns -= NSEC_PER_SEC;
		a->tv_sec++;
	}
