Index: vdr-1.7.10/Makefile
===================================================================
--- vdr-1.7.10.orig/Makefile	2010-01-03 18:32:17.621590237 +0100
+++ vdr-1.7.10/Makefile	2010-01-03 18:32:35.494511992 +0100
@@ -20,7 +20,7 @@
 MANDIR   = $(PREFIX)/share/man
 BINDIR   = $(PREFIX)/bin
 LOCDIR   = ./locale
-LIBS     = -ljpeg -lpthread -ldl -lcap -lrt -lfreetype -lfontconfig -lintl
+LIBS     = -ljpeg -lpthread -ldl -lcap -lrt -lfreetype -lfontconfig -lintl ${LDFLAGS}
 INCLUDES = ${CFLAGS}
 
 PLUGINDIR= ./PLUGINS
Index: vdr-1.7.10/PLUGINS/src/hello/Makefile
===================================================================
--- vdr-1.7.10.orig/PLUGINS/src/hello/Makefile	2010-01-03 18:36:04.434906531 +0100
+++ vdr-1.7.10/PLUGINS/src/hello/Makefile	2010-01-03 18:36:53.764486295 +0100
@@ -19,6 +19,7 @@
 
 CXX      ?= g++
 CXXFLAGS ?= -fPIC -g -O2 -Wall -Woverloaded-virtual -Wno-parentheses
+LIBS     ?= ${LDFLAGS}
 
 ### The directory environment:
 
@@ -95,7 +96,7 @@
 ### Targets:
 
 libvdr-$(PLUGIN).so: $(OBJS)
-	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
+	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@ $(LIBS)
 	@cp --remove-destination $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
Index: vdr-1.7.10/PLUGINS/src/osddemo/Makefile
===================================================================
--- vdr-1.7.10.orig/PLUGINS/src/osddemo/Makefile	2010-01-03 18:37:03.264901094 +0100
+++ vdr-1.7.10/PLUGINS/src/osddemo/Makefile	2010-01-03 18:37:50.811562295 +0100
@@ -17,6 +17,7 @@
 
 CXX      ?= g++
 CXXFLAGS ?= -fPIC -g -O2 -Wall -Woverloaded-virtual -Wno-parentheses
+LIBS     ?= ${LDFLAGS}
 
 ### The directory environment:
 
@@ -68,7 +69,7 @@
 ### Targets:
 
 libvdr-$(PLUGIN).so: $(OBJS)
-	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
+	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@ $(LIBS)
 	@cp --remove-destination $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
Index: vdr-1.7.10/PLUGINS/src/pictures/Makefile
===================================================================
--- vdr-1.7.10.orig/PLUGINS/src/pictures/Makefile	2010-01-03 18:38:04.324891989 +0100
+++ vdr-1.7.10/PLUGINS/src/pictures/Makefile	2010-01-03 18:38:28.857810076 +0100
@@ -19,6 +19,7 @@
 
 CXX      ?= g++
 CXXFLAGS ?= -fPIC -g -O2 -Wall -Woverloaded-virtual -Wno-parentheses
+LIBS     ?= ${LDFLAGS}
 
 ### The directory environment:
 
@@ -95,7 +96,7 @@
 ### Targets:
 
 libvdr-$(PLUGIN).so: $(OBJS)
-	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
+	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@ $(LIBS)
 	@cp --remove-destination $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
Index: vdr-1.7.10/PLUGINS/src/servicedemo/Makefile
===================================================================
--- vdr-1.7.10.orig/PLUGINS/src/servicedemo/Makefile	2010-01-03 18:38:35.431552371 +0100
+++ vdr-1.7.10/PLUGINS/src/servicedemo/Makefile	2010-01-03 18:39:08.001139005 +0100
@@ -19,6 +19,7 @@
 
 CXX      ?= g++
 CXXFLAGS ?= -fPIC -g -O2 -Wall -Woverloaded-virtual -Wno-parentheses
+LIBS     ?= ${LDFLAGS}
 
 ### The directory environment:
 
@@ -70,11 +71,11 @@
 ### Targets:
 
 libvdr-$(PLUGIN1).so: $(PLUGIN1).o
-	$(CXX) $(CXXFLAGS) -shared $(PLUGIN1).o -o $@
+	$(CXX) $(CXXFLAGS) -shared $(PLUGIN1).o -o $@ $(LIBS)
 	@cp --remove-destination $@ $(LIBDIR)/$@.$(APIVERSION)
 
 libvdr-$(PLUGIN2).so: $(PLUGIN2).o
-	$(CXX) $(CXXFLAGS) -shared $(PLUGIN2).o -o $@
+	$(CXX) $(CXXFLAGS) -shared $(PLUGIN2).o -o $@ $(LIBS)
 	@cp --remove-destination $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
Index: vdr-1.7.10/PLUGINS/src/skincurses/Makefile
===================================================================
--- vdr-1.7.10.orig/PLUGINS/src/skincurses/Makefile	2010-01-03 18:39:17.038217922 +0100
+++ vdr-1.7.10/PLUGINS/src/skincurses/Makefile	2010-01-03 18:40:01.821133352 +0100
@@ -19,6 +19,7 @@
 
 CXX      ?= g++
 CXXFLAGS ?= -fPIC -g -O2 -Wall -Woverloaded-virtual -Wno-parentheses
+LIBS     ?= ${LDFLAGS}
 
 ### The directory environment:
 
@@ -95,7 +96,7 @@
 ### Targets:
 
 libvdr-$(PLUGIN).so: $(OBJS)
-	$(CXX) $(CXXFLAGS) -shared $(OBJS) -lncursesw -o $@
+	$(CXX) $(CXXFLAGS) -shared $(OBJS) -lncursesw -o $@ $(LIBS)
 	@cp --remove-destination $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
Index: vdr-1.7.10/PLUGINS/src/sky/Makefile
===================================================================
--- vdr-1.7.10.orig/PLUGINS/src/sky/Makefile	2010-01-03 18:40:08.738209760 +0100
+++ vdr-1.7.10/PLUGINS/src/sky/Makefile	2010-01-03 18:40:20.694465450 +0100
@@ -17,6 +17,7 @@
 
 CXX      ?= g++
 CXXFLAGS ?= -fPIC -g -O2 -Wall -Woverloaded-virtual -Wno-parentheses
+LIBS     ?= ${LDFLAGS}
 
 ### The directory environment:
 
@@ -68,7 +69,7 @@
 ### Targets:
 
 libvdr-$(PLUGIN).so: $(OBJS)
-	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
+	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@ $(LIBS)
 	@cp --remove-destination $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
Index: vdr-1.7.10/PLUGINS/src/status/Makefile
===================================================================
--- vdr-1.7.10.orig/PLUGINS/src/status/Makefile	2010-01-03 18:40:27.841126169 +0100
+++ vdr-1.7.10/PLUGINS/src/status/Makefile	2010-01-03 18:40:41.844457953 +0100
@@ -17,6 +17,7 @@
 
 CXX      ?= g++
 CXXFLAGS ?= -fPIC -g -O2 -Wall -Woverloaded-virtual -Wno-parentheses
+LIBS     ?= ${LDFLAGS}
 
 ### The directory environment:
 
@@ -68,7 +69,7 @@
 ### Targets:
 
 libvdr-$(PLUGIN).so: $(OBJS)
-	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
+	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@ $(LIBS)
 	@cp --remove-destination $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
Index: vdr-1.7.10/PLUGINS/src/svdrpdemo/Makefile
===================================================================
--- vdr-1.7.10.orig/PLUGINS/src/svdrpdemo/Makefile	2010-01-03 18:40:50.818205559 +0100
+++ vdr-1.7.10/PLUGINS/src/svdrpdemo/Makefile	2010-01-03 18:41:01.831121904 +0100
@@ -17,6 +17,7 @@
 
 CXX      ?= g++
 CXXFLAGS ?= -fPIC -g -O2 -Wall -Woverloaded-virtual -Wno-parentheses
+LIBS     ?= ${LDFLAGS}
 
 ### The directory environment:
 
@@ -68,7 +69,7 @@
 ### Targets:
 
 libvdr-$(PLUGIN).so: $(OBJS)
-	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@
+	$(CXX) $(CXXFLAGS) -shared $(OBJS) -o $@ $(LIBS)
 	@cp --remove-destination $@ $(LIBDIR)/$@.$(APIVERSION)
 
 dist: clean
