diff -ur icebloxx_V1.0.0.orig/icebloxx.desktop icebloxx_V1.0.0/icebloxx.desktop
--- icebloxx_V1.0.0.orig/icebloxx.desktop	2004-06-17 00:21:27.000000000 +0200
+++ icebloxx_V1.0.0/icebloxx.desktop	2004-06-17 00:21:44.000000000 +0200
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Name=Icebloxx
 Name[de]=Icebloxx
-Icon=Icebloxx
+Icon=icebloxx/Icebloxx
 Exec=icebloxx
 Comment=A clone of the java game Iceblox (Pengo).
 Comment[de]=Eine C++ Version von Iceblox (Pengo).
