Index: dbus.c
===================================================================
RCS file: /cvs/gpe/base/gpe-bluetooth/dbus.c,v
retrieving revision 1.7
diff -u -r1.7 dbus.c
--- gpe-bluetooth/dbus.c	7 Sep 2004 00:01:27 -0000	1.7
+++ gpe-bluetooth/dbus.c	9 Jan 2005 13:56:46 -0000
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2002, 2003, 2004 Philip Blundell <philb@gnu.org>
+ * Copyright (C) 2002, 2003, 2004, 2005 Philip Blundell <philb@gnu.org>
  *
  * This program is free software; you can redistribute it and/or
  * modify it under the terms of the GNU General Public License
@@ -27,9 +27,11 @@
 
 #define _(x) gettext(x)
 
-#define SERVICE_NAME   "org.handhelds.gpe.bluez"
-#define PIN_INTERFACE_NAME SERVICE_NAME ".PinAgent"
-#define OBEX_INTERFACE_NAME SERVICE_NAME ".OBEX"
+#define PIN_SERVICE_NAME   "org.bluez.PinAgent"
+#define PIN_INTERFACE_NAME PIN_SERVICE_NAME
+
+#define OBEX_SERVICE_NAME   "org.handhelds.gpe.bluez"
+#define OBEX_INTERFACE_NAME OBEX_SERVICE_NAME ".OBEX"
 
 extern DBusHandlerResult bluez_pin_handle_dbus_request (DBusConnection *connection, DBusMessage *message);
 extern DBusHandlerResult obex_client_handle_dbus_request (DBusConnection *connection, DBusMessage *message);
@@ -96,7 +98,7 @@
 
       dbus_connection_register_object_path (connection, object_path1, &dbus_pin_vtable, NULL);
 
-      dbus_bus_acquire_service (connection, SERVICE_NAME, 0, &error);
+      dbus_bus_acquire_service (connection, PIN_SERVICE_NAME, 0, &error);
       if (dbus_error_is_set (&error))
 	{
 	  gpe_error_box_fmt (_("Failed to acquire service: %s"), error.message);
@@ -118,7 +120,7 @@
 
       dbus_connection_register_object_path (connection, object_path2, &dbus_obex_vtable, NULL);
 
-      dbus_bus_acquire_service (connection, SERVICE_NAME, 0, &error);
+      dbus_bus_acquire_service (connection, OBEX_SERVICE_NAME, 0, &error);
       if (dbus_error_is_set (&error))
 	{
 	  gpe_error_box_fmt (_("Failed to acquire service: %s"), error.message);
