
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- wget-1.9.1/src/wget.h~prototypes
+++ wget-1.9.1/src/wget.h
@@ -45,7 +45,7 @@
 #undef DEBUG_MALLOC
 
 #ifndef PARAMS
-# if PROTOTYPES
+# ifndef ANSI2KNR
 #  define PARAMS(args) args
 # else
 #  define PARAMS(args) ()
--- tmp/base/wget-1.9.1-r0/wget-1.9.1/configure.in	2003-11-08 13:15:35.000000000 -0600
+++ wget-1.9.1/configure.in	2004-05-22 19:41:40.000000000 -0500
@@ -46,11 +46,13 @@
 dnl Get cannonical host
 dnl
 AC_CANONICAL_HOST
+AH_TEMPLATE([OS_TYPE], [Define to the operating system we're building for])
 AC_DEFINE_UNQUOTED(OS_TYPE, "$host_os")
 
 dnl
 dnl Process features.
 dnl
+AH_TEMPLATE([HAVE_SOCKS], [Define to use the socks library])
 AC_ARG_WITH(socks,
 [  --with-socks            use the socks library],
 [AC_DEFINE(HAVE_SOCKS)])
@@ -59,16 +61,19 @@
 [[  --with-ssl[=SSL-ROOT]   link with SSL support [default=auto]
   --without-ssl           disable SSL autodetection]])
 
+AH_TEMPLATE([USE_OPIE], [Define to enable support for opie or s/key FTP login])
 AC_ARG_ENABLE(opie,
 [  --disable-opie          disable support for opie or s/key FTP login],
 USE_OPIE=$enableval, USE_OPIE=yes)
 test x"${USE_OPIE}" = xyes && AC_DEFINE(USE_OPIE)
 
+AH_TEMPLATE([USE_DIGEST], [Define to enable support for HTTP digest authorization])
 AC_ARG_ENABLE(digest,
 [  --disable-digest        disable support for HTTP digest authorization],
 USE_DIGEST=$enableval, USE_DIGEST=yes)
 test x"${USE_DIGEST}" = xyes && AC_DEFINE(USE_DIGEST)
 
+AH_TEMPLATE([ENABLE_DEBUG], [Define to enable debugging output])
 AC_ARG_ENABLE(debug,
 [  --disable-debug         disable support for debugging output],
 ENABLE_DEBUG=$enableval, ENABLE_DEBUG=yes)
@@ -256,6 +261,7 @@
 dnl an AC_DEFUN so that we can reuse it for other third-party
 dnl libraries.
 
+AH_TEMPLATE([HAVE_SSL], [Define to build with SSL support])
 if test x"$with_ssl" != x"no"; then
 
   wget_force_ssl=no
@@ -413,6 +419,10 @@
 dnl Find an md5 implementation.
 dnl
 
+AH_TEMPLATE([HAVE_MD5], [Define if using an MD5 implementation])
+AH_TEMPLATE([HAVE_SOLARIS_MD5], [Define if using the Solaris MD5 implementation])
+AH_TEMPLATE([HAVE_OPENSSL_MD5], [Define if using the OpenSSL MD5 implementation])
+AH_TEMPLATE([HAVE_BUILTIN_MD5], [Define if using the builtin GNU MD5 implementation])
 if test x"$wget_need_md5" = xyes
 then
   MD5_OBJ='gen-md5$o'
