--- /m4/orig-as-gcc-inline-assembly.m4	2007-03-17 04:03:30.000000000 +0200
+++ /m4/as-gcc-inline-assembly.m4	2007-11-20 12:20:24.000000000 +0200
@@ -38,7 +38,7 @@
 [
   AC_MSG_CHECKING([if compiler supports FPU instructions on PowerPC])
 
-  AC_TRY_COMPILE([], [__asm__ ("fadd 0,0,0"::) ], [flag_ok=yes], [flag_ok=no])
+  AC_TRY_COMPILE([], [double x = 0; __asm__ ("fadd 0,0,0"::"f" (x):"32") ], [flag_ok=yes], [flag_ok=no])
 
   if test "X$flag_ok" = Xyes ; then
     $1
