--- quake1src/vid_sdl.c~	2003-09-24 02:18:50.000000000 +0100
+++ quake1src/vid_sdl.c	2005-05-08 15:30:10.663089128 +0100
@@ -59,7 +59,7 @@
     Uint32 flags;
 
     // Load the SDL library
-    if (SDL_Init(SDL_INIT_VIDEO|SDL_INIT_AUDIO|SDL_INIT_CDROM) < 0)
+    if (SDL_Init(SDL_INIT_VIDEO|SDL_INIT_AUDIO) < 0)
         Sys_Error("VID: Couldn't load SDL: %s", SDL_GetError());
 
     // Set up display mode (width and height)
