--- anthy-7811/anthy-conf.in~	2006-05-13 18:27:50.000000000 +0900
+++ anthy-7811/anthy-conf.in	2006-06-18 19:23:59.890000000 +0900
@@ -1,7 +1,7 @@
 # not /usr but should be ${prefix}/share.
-ANTHYDIR @prefix@/share/anthy/
-DIC_FILE @prefix@/share/anthy/anthy.dic
+ANTHYDIR /usr/share/anthy/
+DIC_FILE /usr/share/anthy/anthy.dic
 INDEPWORD indepword.txt
 DEPWORD master.depword
-ZIPDICT_EUC @prefix@/share/anthy/zipcode.t
+ZIPDICT_EUC /usr/share/anthy/zipcode.t
 WORDS_FILE /usr/share/dict/words
