--- linux-2.6.7/arch/arm/Makefile.orig	2004-06-15 22:19:13.000000000 -0700
+++ linux-2.6.7/arch/arm/Makefile	2004-09-19 21:47:58.000000000 -0700
@@ -55,8 +55,8 @@
 tune-$(CONFIG_CPU_V6)		:=-mtune=strongarm
 
 # Need -Uarm for gcc < 3.x
-CFLAGS		+=-mapcs-32 $(arch-y) $(tune-y) -mshort-load-bytes -msoft-float -Wa,-mno-fpu -Uarm
-AFLAGS		+=-mapcs-32 $(arch-y) $(tune-y) -msoft-float -Wa,-mno-fpu
+CFLAGS		+=-mapcs-32 $(arch-y) $(tune-y) -msoft-float -Uarm
+AFLAGS		+=-mapcs-32 $(arch-y) $(tune-y) -msoft-float
 
 #Default value
 DATAADDR	:= .
