diff -urNd glibc-2.3.2-ct25/csu/Makefile glibc-2.3.2/csu/Makefile
--- glibc-2.3.2-ct25/csu/Makefile	2004-06-25 13:47:54.000000000 +0200
+++ glibc-2.3.2/csu/Makefile	2004-06-25 13:54:33.000000000 +0200
@@ -99,7 +99,7 @@
 $(objpfx)crt%.o: $(objpfx)crt%.S $(objpfx)defs.h
 	$(compile.S) -g0 $(ASFLAGS-.os) -o $@
 
-CFLAGS-initfini.s = -g0 -fPIC -fno-inline-functions $(fno-unit-at-a-time)
+CFLAGS-initfini.s = -O1 -g0 -fPIC -fno-inline-functions $(fno-unit-at-a-time)
 
 vpath initfini.c $(full_config_sysdirs)
 
