--- linux-2.6.17-vanilla/drivers/ide/ide-io.c	2006-06-18 01:49:35.000000000 +0000
+++ linux-2.6.17/drivers/ide/ide-io.c	2006-07-06 14:18:13.000000000 +0000
@@ -1442,10 +1442,10 @@
 				++count;
 				if (time_after(jiffies, last_msgtime + HZ)) {
 					last_msgtime = jiffies;
-					printk(KERN_ERR "%s%s: unexpected interrupt, "
-						"status=0x%02x, count=%ld\n",
-						hwif->name,
-						(hwif->next==hwgroup->hwif) ? "" : "(?)", stat, count);
+					//printk(KERN_ERR "%s%s: unexpected interrupt, "
+					//	"status=0x%02x, count=%ld\n",
+					//	hwif->name,
+					//	(hwif->next==hwgroup->hwif) ? "" : "(?)", stat, count);
 				}
 			}
 		}
