
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- imlib2-1.1.0/demo/Makefile.am~cross
+++ imlib2-1.1.0/demo/Makefile.am
@@ -8,8 +8,7 @@
 LDFLAGS              = -L/usr/X11R6/lib
 INCLUDES             = -I. -I$(top_srcdir) -I$(top_srcdir)/src \
                        -I$(top_srcdir)/loaders -I$(top_srcdir)/libltdl \
-                       -I/usr/X11R6/include $(X_CFLAGS) \
-                       -I$(prefix)/include -I$(includedir)
+                       $(X_CFLAGS)
 
 bin_PROGRAMS         = imlib2_view imconvert
 imconvert_SOURCES    = imconvert.c
--- imlib2-1.1.0/filters/Makefile.am~cross
+++ imlib2-1.1.0/filters/Makefile.am
@@ -8,8 +8,7 @@
 LDFLAGS		     = -L$(top_builddir)/src -L$(top_builddir)/src/.libs -L/usr/X11R6/lib
 INCLUDES             = -I. -I$(top_srcdir) -I$(top_srcdir)/src \
                        -I$(top_srcdir)/loaders -I$(top_srcdir)/libltdl \
-                       -I/usr/X11R6/include $(X_CFLAGS) \
-                       -I$(prefix)/include -I$(includedir)
+                       $(X_CFLAGS)
 
 pkgdir		     = $(libdir)/loaders/filter
 pkg_LTLIBRARIES	     = testfilter.la bump_map.la colormod.la
--- imlib2-1.1.0/loaders/Makefile.am~cross
+++ imlib2-1.1.0/loaders/Makefile.am
@@ -8,8 +8,7 @@
 LDFLAGS		     = -L$(top_builddir)/src -L$(top_builddir)/src/.libs -L/usr/X11R6/lib 
 INCLUDES             = -I. -I$(top_srcdir) -I$(top_srcdir)/src \
                        -I$(top_srcdir)/loaders -I$(top_srcdir)/libltdl \
-                       -I/usr/X11R6/include $(X_CFLAGS) \
-                       -I$(prefix)/include -I$(includedir)
+                       $(X_CFLAGS)
 
 pkgdir		     = $(libdir)/loaders/image
 
--- imlib2-1.1.0/src/Makefile.am~cross
+++ imlib2-1.1.0/src/Makefile.am
@@ -8,8 +8,7 @@
 LDFLAGS		     = -L/usr/X11R6/lib
 INCLUDES             = -I. -I$(top_srcdir) -I$(top_srcdir)/src \
                        -I$(top_srcdir)/loaders -I$(top_srcdir)/libltdl \
-                       $(freetype_cflags) -I/usr/X11R6/include $(X_CFLAGS) \
-                       -I$(prefix)/include -I$(includedir)
+                       $(freetype_cflags) $(X_CFLAGS)
 MMX_SRCS             = asm_blend.S asm_rgba.S asm_scale.S asm_rotate.S \
 		       asm_blend_cmod.S
 MMX_OBJS             = $(MMX_SRCS:.S=.lo)
@@ -42,7 +41,7 @@
 libImlib2_la_DEPENDENCIES = $(top_builddir)/config.h $(X_OBJS)
 endif
 
-SYS_LOADERS_PATH = @pkglibdir@
+SYS_LOADERS_PATH = @loadersdir@
 image.lo: loaderpath.h
 loaderpath.h: Makefile
 	@echo "#define SYS_LOADERS_PATH \"${SYS_LOADERS_PATH}\"" > $@
--- imlib2-1.1.0/test/Makefile.am~cross
+++ imlib2-1.1.0/test/Makefile.am
@@ -8,8 +8,7 @@
 LDFLAGS              = -L/usr/X11R6/lib
 INCLUDES             = -I. -I$(top_srcdir) -I$(top_srcdir)/src \
 		       -I$(top_srcdir)/loaders -I$(top_srcdir)/libltdl \
-                       -I/usr/X11R6/include $(X_CFLAGS) \
-		       -I$(prefix)/include -I$(includedir)
+                       $(X_CFLAGS)
 
 bin_PROGRAMS         = imlib2 bumpmaptest imlib2_test polytest color_spaces
 imlib2_SOURCES       = main.c
