--- ppp.old/pppd/Makefile.linux	2004-08-22 00:17:31.000000000 +0200
+++ ppp/pppd/Makefile.linux	2004-08-22 01:07:20.000000000 +0200
@@ -5,7 +5,7 @@
 
 # Default installation locations
 BINDIR = $(DESTDIR)/usr/sbin
-MANDIR = $(DESTDIR)/usr/man
+MANDIR = $(DESTDIR)/usr/share/man
 INCDIR = $(DESTDIR)/usr/include
 
 TARGETS = pppd
@@ -76,7 +76,7 @@
 
 INCLUDE_DIRS= -I../include
 
-COMPILE_FLAGS= -DHAVE_PATHS_H -DIPX_CHANGE -DHAVE_MMAP
+COMPILE_FLAGS= -DHAVE_PATHS_H -DIPX_CHANGE 
 
 CFLAGS= $(COPTS) $(COMPILE_FLAGS) $(INCLUDE_DIRS)
 
@@ -103,7 +103,7 @@
 ifdef USE_SRP
 CFLAGS	+= -DUSE_SRP
 TARGETS	+= srp-entry
-EXTRAINSTALL = $(INSTALL) -s -c -m 555 srp-entry $(BINDIR)/srp-entry
+EXTRAINSTALL = $(INSTALL) -c -m 755 srp-entry $(BINDIR)/srp-entry
 MANPAGES += srp-entry.8
 EXTRACLEAN += srp-entry.o
 NEEDDES=y
@@ -197,7 +197,7 @@
 
 ifdef FILTER
 LIBS    += -lpcap
-CFLAGS  += -DPPP_FILTER -I/usr/include/pcap
+CFLAGS  += -DPPP_FILTER 
 endif
 
 ifdef HAVE_INET6
@@ -218,14 +218,14 @@
      CFLAGS += -DMAXOCTETS
 endif
 
-INSTALL= install -o root
+INSTALL= install 
 
 all: $(TARGETS)
 
 install: pppd
 	mkdir -p $(BINDIR) $(MANDIR)
 	$(EXTRAINSTALL)
-	$(INSTALL) -s -c -m 555 pppd $(BINDIR)/pppd
+	$(INSTALL) -c -m 755 pppd $(BINDIR)/pppd
 	if chgrp pppusers $(BINDIR)/pppd 2>/dev/null; then \
 	  chmod o-rx,u+s $(BINDIR)/pppd; fi
 	$(INSTALL) -c -m 444 pppd.8 $(MANDIR)/man8
