Index: tgt-1.0.12/usr/tgtimg.c
===================================================================
--- tgt-1.0.12.orig/usr/tgtimg.c	2011-01-15 16:28:30.488656000 -0800
+++ tgt-1.0.12/usr/tgtimg.c	2011-01-15 16:35:14.958656001 -0800
@@ -438,8 +438,13 @@
 			perror("Failed creating file");
 			exit(2);
 		}
+#ifndef __UCLIBC__
 		if (posix_fallocate(fd, 0, size*1024*1024LL) == -1) {
 			perror("posix_fallocate failed.");
+#else
+		if (ftruncate (fd, size*1024*1024LL) == -1) {
+			perror("ftruncate failed.");
+#endif
 			exit(3);
 		}
 
