--- mozilla/js/src/Makefile.in	2004-03-31 09:38:54.000000000 -0700
+++ mozilla.new/js/src/Makefile.in	2004-11-12 16:29:32.856436952 -0700
@@ -342,16 +342,10 @@
 
 jsopcode.h jsopcode.c: jsopcode.tbl
 
-ifeq (,$(CROSS_COMPILE)$(filter-out WINNT,$(OS_ARCH)))
+# OE hack - we copy this in 
 jsautocfg.h:
 	touch $@
-else
-jsautocfg.h: jscpucfg$(HOST_BIN_SUFFIX)
-	@rm -f $@ jsautocfg.tmp
-	./jscpucfg > jsautocfg.tmp
-	mv jsautocfg.tmp $@
-endif
-
+	
 # jscpucfg is a strange target
 # Needs to be built with the host compiler but needs to include
 # the mdcpucfg for the target so it needs the appropriate target defines
