Index: dosbox-0.65/configure.in
===================================================================
--- dosbox-0.65.orig/configure.in	2006-06-30 22:46:36.526446000 +0200
+++ dosbox-0.65/configure.in	2006-06-30 22:49:00.516446000 +0200
@@ -235,23 +235,6 @@
    AC_MSG_WARN([Can't find SDL_net, internal modem and ipx disabled])
 fi
 
-AH_TEMPLATE(C_OPENGL,[Define to 1 to use opengl display output support])
-AC_ARG_ENABLE(opengl,AC_HELP_STRING([--disable-opengl],[Disable opengl support]),,enable_opengl=yes)
-AC_CHECK_LIB(GL, main, have_gl_lib=yes, have_gl_lib=no , )
-AC_CHECK_LIB(opengl32, main, have_opengl32_lib=yes,have_opengl32_lib=no , )
-AC_CHECK_HEADER(GL/gl.h, have_gl_h=yes , have_gl_h=no , )
-AC_MSG_CHECKING(whether opengl display output will be enabled) 
-if test x$enable_opengl = xyes -a x$have_gl_h = xyes -a x$have_gl_lib = xyes ; then 
-  AC_MSG_RESULT(yes)
-  LIBS="$LIBS -lGL"
-  AC_DEFINE(C_OPENGL,1)
-elif test x$enable_opengl = xyes -a x$have_gl_h = xyes -a x$have_opengl32_lib = xyes ; then 
-  AC_MSG_RESULT(yes)
-  LIBS="$LIBS -lopengl32"
-  AC_DEFINE(C_OPENGL,1)
-else
-  AC_MSG_RESULT(no)
-fi
 
 AH_TEMPLATE(C_SDL_SOUND,[Define to 1 to enable SDL_sound support])
 AC_CHECK_HEADER(SDL/SDL_sound.h,have_SDL_sound_h=yes,)
