--- /tmp/prcs.c	2008-05-15 20:35:49.946802023 +0200
+++ dsplink_1_50/dsplink/gpp/src/osal/Linux/prcs.c	2008-05-15 20:36:01.386858569 +0200
@@ -17,6 +17,7 @@
 
 /*  ----------------------------------- OS Specific Headers         */
 #include <linux/autoconf.h>
+#include <linux/sched.h>
 #include <linux/spinlock.h>
 #include <linux/module.h>
 #include <linux/mm.h>

--- /tmp/mem.c	2008-06-22 16:05:16.000000000 +0200
+++ dsplink_1_50/dsplink/gpp/src/osal/Linux/2.6.18/mem.c	2008-06-22 16:05:43.000000000 +0200
@@ -16,6 +16,7 @@
 
 
 /*  ----------------------------------- OS Specific Headers         */
+
 #include <linux/autoconf.h>
 #include <linux/spinlock.h>
 #if (defined(DM642_PCI) || defined (DM64LC_PCI))
@@ -31,6 +32,7 @@
 #endif /* if (defined(DM642_PCI) || defined (DM64LC_PCI)) */
 #include <linux/vmalloc.h>
 #include <asm/io.h>
+#include <linux/sched.h>
 
 /*  ----------------------------------- DSP/BIOS Link               */
 #include <dsplink.h>
