Index: openmoko-mediaplayer2/images/Makefile.am
===================================================================
--- openmoko-mediaplayer2.orig/images/Makefile.am	2007-08-25 16:15:27.000000000 +0000
+++ openmoko-mediaplayer2/images/Makefile.am	2007-08-25 16:15:35.000000000 +0000
@@ -12,8 +12,6 @@
 	ico-repeat-all.png \
 	ico-shuffle-on.png \
 	ico-shuffle-off.png \
-	ico-balance-left.png \
-	ico-balance-right.png \
 	ico-tracktype-general.png \
 	ind-music-eq-00.png \
 	ind-music-eq-01.png \
