--- avahi-0.6.1/avahi-daemon/dbus-protocol.c.orig	2005-12-09 20:48:02 +0100
+++ avahi-0.6.1/avahi-daemon/dbus-protocol.c	2005-12-09 20:48:15 +0100
@@ -1029,7 +1029,7 @@
         goto fail;
     }
 
-    if (dbus_bus_request_name(server->bus, AVAHI_DBUS_NAME, DBUS_NAME_FLAG_PROHIBIT_REPLACEMENT, &error) != DBUS_REQUEST_NAME_REPLY_PRIMARY_OWNER) {
+    if (dbus_bus_request_name(server->bus, AVAHI_DBUS_NAME, 0, &error) != DBUS_REQUEST_NAME_REPLY_PRIMARY_OWNER) {
         if (dbus_error_is_set(&error)) {
             avahi_log_error("dbus_bus_request_name(): %s", error.message);
             goto fail;
