Index: MPlayer-1.0rc1/Makefile
===================================================================
--- MPlayer-1.0rc1.orig/Makefile
+++ MPlayer-1.0rc1/Makefile
@@ -399,7 +399,7 @@ osdep/mplayer-rc.o: osdep/mplayer.rc
 	windres -o $@ osdep/mplayer.rc
 
 codec-cfg: codec-cfg.c codec-cfg.h help_mp.h
-	$(HOST_CC) -I. -DCODECS2HTML codec-cfg.c -o $@
+	$(BUILD_CC) -I. -DCODECS2HTML codec-cfg.c -o $@
 
 codecs.conf.h: codec-cfg etc/codecs.conf
 	./codec-cfg ./etc/codecs.conf > $@
