Index: MPlayer-1.0pre8/Makefile
===================================================================
--- MPlayer-1.0pre8.orig/Makefile
+++ MPlayer-1.0pre8/Makefile
@@ -438,8 +438,8 @@ codecs2html: mp_msg.o
 	$(CC) -DCODECS2HTML codec-cfg.c mp_msg.o -o $@
 
 $(PRG_CFG): codec-cfg.c codec-cfg.h help_mp.h
-	$(HOST_CC) $(HOST_CFLAGS) -I. codec-cfg.c -o $(PRG_CFG) \
-	-DCODECS2HTML $(EXTRA_LIB) $(EXTRA_INC)
+	$(BUILD_CC) -Ilibmpdemux -Iloader -Ilibvo -I. codec-cfg.c -o $(PRG_CFG) \
+	-DCODECS2HTML 
 
 install: $(ALL_PRG)
 ifeq ($(VIDIX),yes)
