--- /tmp/Makefile	2008-04-17 21:48:32.987984542 +0200
+++ trunk/Makefile	2008-04-17 21:48:52.042750779 +0200
@@ -337,7 +337,7 @@
 	if test -f $(CONFDIR)/codecs.conf ; then mv -f $(CONFDIR)/codecs.conf $(CONFDIR)/codecs.conf.old ; fi
 
 install-mplayer: mplayer$(EXESUF)
-	$(INSTALL) -m 755 $(INSTALLSTRIP) mplayer$(EXESUF) $(BINDIR)
+	$(INSTALL) -m 755  mplayer$(EXESUF) $(BINDIR)
 
 install-mplayer-man:
 	for lang in $(MAN_LANG); do \
@@ -350,7 +350,7 @@
 	done
 
 install-mencoder: mencoder$(EXESUF)
-	$(INSTALL) -m 755 $(INSTALLSTRIP) mencoder$(EXESUF) $(BINDIR)
+	$(INSTALL) -m 755 mencoder$(EXESUF) $(BINDIR)
 	for lang in $(MAN_LANG); do \
 		if test "$$lang" = en ; then \
 			cd $(MANDIR)/man1 && ln -sf mplayer.1 mencoder.1 ; \
