--- gnuradio-trunk-work/configure.ac.orig	2008-10-06 13:38:45.000000000 -0400
+++ gnuradio-trunk-work/configure.ac	2008-10-06 13:39:02.000000000 -0400
@@ -317,9 +317,7 @@
 GRC_PMT
 GRC_MBLOCK			dnl this must come after GRC_PMT
 GRC_USRP
-GRC_USRP2
 GRC_GR_USRP			dnl this must come after GRC_USRP
-GRC_GR_USRP2
 GRC_GR_GCELL			dnl this must come after GRC_GCELL and GRC_GNURADIO_CORE
 GRC_GR_MSDD6000	
 GRC_GR_AUDIO_ALSA
