--- havp-0.82/havp/default.h.in	2006/10/05 09:05:18	1.1
+++ havp-0.82/havp/default.h.in	2006/10/05 09:05:31
@@ -51,7 +51,7 @@
  "ENABLEAVAST","AVASTSOCKET","AVASTSERVER","AVASTPORT"
 //SCANNERS
 
-#define CONFIGFILE "/usr/local/etc/havp/havp.config"
+#define CONFIGFILE "/etc/havp/havp.config"
 
 
 //##############################################################
--- havp-0.82/havp/params.cpp	2006/10/05 09:04:45	1.1
+++ havp-0.82/havp/params.cpp	2006/10/05 09:05:06
@@ -58,9 +58,9 @@
     SetConfig("KEEPBACKTIME",	"5");
     SetConfig("TRICKLING",	"30");
     SetConfig("WHITELISTFIRST",	"true");
-    SetConfig("WHITELIST",	"/usr/local/etc/havp/whitelist");
-    SetConfig("BLACKLIST",	"/usr/local/etc/havp/blacklist");
-    SetConfig("TEMPLATEPATH",	"/usr/local/etc/havp/templates/en");
+    SetConfig("WHITELIST",	"/etc/havp/whitelist");
+    SetConfig("BLACKLIST",	"/etc/havp/blacklist");
+    SetConfig("TEMPLATEPATH",	"/etc/havp/templates/en");
     SetConfig("TEMPDIR",	"/var/tmp");
     SetConfig("SCANTEMPFILE",	"/var/tmp/havp/havp-XXXXXX");
     SetConfig("PIDFILE",	"/var/run/havp/havp.pid");
