--- configure.ac.old	2005-06-09 18:11:18.000000000 +0200
+++ configure.ac	2005-06-09 18:22:01.000000000 +0200
@@ -53,9 +53,9 @@
 AC_SUBST(HILDONBASELIB_CFLAGS)
 
 PKG_CHECK_MODULES(OSSOSETTINGS, osso-af-settings >= 0.8.1)
-localedir=`pkg-config osso-af-settings --variable=localedir`
+localedir="${prefix}/share/locale"
 AC_SUBST(localedir)
-hildondesktopentrydir=`pkg-config osso-af-settings --variable=desktopentrydir`
+hildondesktopentrydir="${prefix}/share/applications/hildon"
 AC_SUBST(hildondesktopentrydir)
 
 PKG_CHECK_MODULES(LIBOSSOHELP, libossohelp >= 0.0.4)
