Index: linux-2.6.20.7/drivers/usb/gadget/pxa27x_udc.c
===================================================================
--- linux-2.6.20.7.orig/drivers/usb/gadget/pxa27x_udc.c	2007-04-21 12:30:51.000000000 -0300
+++ linux-2.6.20.7/drivers/usb/gadget/pxa27x_udc.c	2007-04-21 12:32:21.000000000 -0300
@@ -1615,8 +1615,9 @@
 
 	DMSG("Connecting\n");
 	/* RPFIXME */
-	UP2OCR = UP2OCR_HXOE | UP2OCR_DPPUE | UP2OCR_DPPUBE;
-	//dev->mach->udc_command(PXA2XX_UDC_CMD_CONNECT);
+	//UP2OCR = UP2OCR_HXOE | UP2OCR_DPPUE | UP2OCR_DPPUBE;
+	UP2OCR = 0x02000000; // temporary solution for ezx.
+	dev->mach->udc_command(PXA2XX_UDC_CMD_CONNECT);
 }
 
 
