diff -pruN linux-2.6.20.7/drivers/char/ts0710.h linux-2.6.20.7_x386/drivers/char/ts0710.h
--- linux-2.6.20.7/drivers/char/ts0710.h	2007-04-22 10:51:23.000000000 +0200
+++ linux-2.6.20.7_x386/drivers/char/ts0710.h	2007-04-22 11:16:37.000000000 +0200
@@ -45,7 +45,7 @@
  *  11/18/2002  Modified
  */
 
-#include <linux/config.h>
+//#include <linux/config.h>
 #include <linux/module.h>
 
 #include <linux/errno.h>
@@ -58,7 +58,7 @@
 #include <linux/major.h>
 #include <linux/mm.h>
 #include <linux/init.h>
-#include <linux/devfs_fs_kernel.h>
+//#include <linux/devfs_fs_kernel.h>
 
 #include <asm/uaccess.h>
 #include <asm/system.h>
diff -pruN linux-2.6.20.7/drivers/char/ts0710_mux.c linux-2.6.20.7_x386/drivers/char/ts0710_mux.c
--- linux-2.6.20.7/drivers/char/ts0710_mux.c	2007-04-22 10:53:05.000000000 +0200
+++ linux-2.6.20.7_x386/drivers/char/ts0710_mux.c	2007-04-22 11:20:07.000000000 +0200
@@ -46,7 +46,7 @@
  *  11/18/2002  Second version
  *  04/21/2004  Add GPRS PROC
  */
-#include <linux/config.h>
+//#include <linux/config.h>
 #include <linux/module.h>
 #include <linux/types.h>
 
@@ -70,7 +70,7 @@
 #include <linux/mm.h>
 #include <linux/slab.h>
 #include <linux/init.h>
-#include <linux/devfs_fs_kernel.h>
+//#include <linux/devfs_fs_kernel.h>
 //#include <syslog.h>
 
 #include <asm/uaccess.h>
@@ -1894,11 +1894,14 @@ int ts0710_recv_data(ts0710_con * ts0710
 				if (test_bit(TTY_THROTTLED, &tty->flags)) {
 					queue_data = 1;
 				} else {
+					/*
 					if (test_bit
 					    (TTY_DONT_FLIP, &tty->flags)) {
 						queue_data = 1;
 						post_recv = 1;
-					} else if (recv_info->total) {
+					} else 
+					*/
+					if (recv_info->total) {
 						queue_data = 1;
 						post_recv = 1;
 					} else if (recv_room < uih_len) {
@@ -3499,11 +3502,14 @@ static void post_recv_worker(void *priva
 		if (test_bit(TTY_THROTTLED, &tty->flags)) {
 			add_post_recv_queue(&post_recv_q, recv_info);
 			continue;
-		} else if (test_bit(TTY_DONT_FLIP, &tty->flags)) {
+		}
+		/*
+		else if (test_bit(TTY_DONT_FLIP, &tty->flags)) {
 			post_recv = 1;
 			add_post_recv_queue(&post_recv_q, recv_info);
 			continue;
 		}
+		*/
 
 		flow_control = 0;
 		recv_packet2 = recv_info->mux_packet;
@@ -3893,9 +3899,9 @@ static int __init mux_init(void)
 	}
 	post_recv_count_flag = 0;
 
-	INIT_WORK(&send_tqueue, send_worker, NULL);
-	INIT_WORK(&receive_tqueue, receive_worker, NULL);
-	INIT_WORK(&post_recv_tqueue, post_recv_worker, NULL);
+	INIT_WORK(&send_tqueue, send_worker);
+	INIT_WORK(&receive_tqueue, receive_worker);
+	INIT_WORK(&post_recv_tqueue, post_recv_worker);
 
 	mux_driver = alloc_tty_driver(NR_MUXS);
 	if (!mux_driver)
@@ -3904,7 +3910,7 @@ static int __init mux_init(void)
 	mux_driver->owner = THIS_MODULE;
 	mux_driver->driver_name = "ts0710mux";
 	mux_driver->name = "mux";
-	mux_driver->devfs_name = "mux";
+	//mux_driver->devfs_name = "mux";
 	mux_driver->major = TS0710MUX_MAJOR;
 	mux_driver->minor_start = TS0710MUX_MINOR_START;
 	mux_driver->type = TTY_DRIVER_TYPE_SERIAL;
