--- /tmp/ggz.m4	2008-09-27 09:57:44.000000000 +0200
+++ ggz-client-libs-0.0.14.1/m4/ggz.m4	2008-09-27 10:06:02.000000000 +0200
@@ -139,8 +139,8 @@
     ])
 
 if test "x${prefix}" = "xNONE"; then
-   ac_ggz_prefix_incdir="${ac_default_prefix}/include"
-   ac_ggz_prefix_libdir="${ac_default_prefix}/lib"
+   ac_ggz_prefix_incdir=""
+   ac_ggz_prefix_libdir=""
    ac_ggz_prefix_bindir="${ac_default_prefix}/bin"
    ac_ggz_prefix_etcdir="${ac_default_prefix}/etc"
 else
@@ -163,26 +163,22 @@
    ac_ggz_prefix_bindir="${unq_bindir}"
    ac_ggz_prefix_etcdir="${unq_sysconfdir}"
 fi
-ac_ggz_stdinc="$ac_ggz_prefix_incdir"
-ac_ggz_stdlib="$ac_ggz_prefix_libdir"
+ac_ggz_stdinc=""
+ac_ggz_stdlib=""
 ac_ggz_stdbin="$ac_ggz_prefix_bindir"
 ac_ggz_stdetc="$ac_ggz_prefix_etcdir/ggzd"
 if test "x$ac_ggz_prefix" != "x"; then
-  ac_ggz_stdinc="$ac_ggz_stdinc $ac_ggz_prefix/include"
-  ac_ggz_stdlib="$ac_ggz_stdlib $ac_ggz_prefix/lib $ac_ggz_prefix/lib64"
+  ac_ggz_stdinc=" "
+  ac_ggz_stdlib=" "
   ac_ggz_stdbin="$ac_ggz_stdbin $ac_ggz_prefix/bin"
   ac_ggz_stdetc="$ac_ggz_stdetc $ac_ggz_prefix/etc/ggzd"
 fi
 if test "x$1" = "xdefaults" || test "x$2" = "xdefaults"; then
-  ac_ggz_stdinc="$ac_ggz_stdinc /usr/local/include /usr/include"
-  ac_ggz_stdlib="$ac_ggz_stdlib /usr/local/lib /usr/local/lib64 /usr/lib /usr/lib64"
+  ac_ggz_stdinc=" "
+  ac_ggz_stdlib=" "
   ac_ggz_stdbin="$ac_ggz_stdbin /usr/local/bin /usr/bin"
   ac_ggz_stdetc="$ac_ggz_stdetc /usr/local/etc/ggzd /etc/ggzd"
 fi
-if test "x$1" = "xexport" || test "x$2" = "xexport"; then
-  CPPFLAGS="$CPPFLAGS -I ${ac_ggz_prefix_incdir} -I /usr/local/include"
-  LDFLAGS="$LDFLAGS -L${ac_ggz_prefix_libdir} -L/usr/local/lib"
-fi
 
 save_cflags=$CFLAGS
 save_cxxflags=$CXXFLAGS
