Index: e/src/bin/e_startup.c
===================================================================
--- e/src/bin/e_startup.c	(Revision 56372)
+++ e/src/bin/e_startup.c	(Arbeitskopie)
@@ -19,16 +19,19 @@
 EAPI void
 e_startup(E_Startup_Mode mode)
 {
+   printf("e_startup() started\n");
    char buf[PATH_MAX];
 
    if (mode == E_STARTUP_START)
      {
+	printf("e_startup() mode = start\n");
 	e_user_dir_concat_static(buf, "applications/startup/.order");
 	if (!ecore_file_exists(buf))
 	  e_prefix_data_concat_static(buf, "data/applications/startup/.order");
      }
    else if (mode == E_STARTUP_RESTART)
      {
+	printf("e_startup() mode = restart\n");
 	e_user_dir_concat_static(buf, "applications/restart/.order");
 	if (!ecore_file_exists(buf))
 	  e_prefix_data_concat_static(buf, "data/applications/restart/.order");
@@ -37,6 +40,8 @@
       ecore_event_handler_add(EFREET_EVENT_DESKTOP_CACHE_UPDATE,
                               _e_startup_event_cb,
                               strdup(buf));
+   if (!desktop_cache_update_handler)
+      printf("e_startup(): ecore_event_handler_add failed\n");
    e_init_undone();
 }
 
@@ -49,6 +54,7 @@
 
    if (!startup_apps)
      {
+        printf("e_startup() finish\n");
 	e_init_done();
 	return;
      }
@@ -56,12 +62,14 @@
    start_app_pos++;
    if (!desktop)
      {
+	printf("e_startup() no desktop: finish\n");
 	e_object_del(E_OBJECT(startup_apps));
 	startup_apps = NULL;
 	start_app_pos = -1;
 	e_init_done();
 	return;
      }
+   printf("e_startup() starting: %s\n", desktop);
    e_exec(NULL, desktop, NULL, NULL, NULL);
    snprintf(buf, sizeof(buf), "%s %s", _("Starting"), desktop->name);
    e_init_status_set(buf);   
@@ -71,6 +79,7 @@
 static void
 _e_startup_next_cb(void *data __UNUSED__)
 {
+   printf("e_startup(): _e_startup_next_cb called\n");
    _e_startup();
 }
 
@@ -78,7 +87,7 @@
 _e_startup_event_cb(void *data, int ev_type __UNUSED__, void *ev __UNUSED__)
 {
    char *buf;
-
+   printf("e_startup(): _e_startup_event_cb called\n");
    ecore_event_handler_del(desktop_cache_update_handler);
    buf = data;
    startup_apps = e_order_new(buf);
