Index: git/arch/arm/mach-s3c2440/mach-gta02.c
===================================================================
--- git.orig/arch/arm/mach-s3c2440/mach-gta02.c
+++ git/arch/arm/mach-s3c2440/mach-gta02.c
@@ -471,11 +471,9 @@ static int pmu_callback(struct device *d
 			break;
 		case PMU_EVT_CHARGER_IDLE:
 			/* printk(KERN_ERR"PMU_EVT_CHARGER_IDLE\n"); */
-			neo1973_gpb_setpin(GTA02_GPIO_AUX_LED, 0);
 			break;
 		case PMU_EVT_CHARGER_ACTIVE:
 			/* printk(KERN_ERR"PMU_EVT_CHARGER_ACTIVE\n"); */
-			neo1973_gpb_setpin(GTA02_GPIO_AUX_LED, 1);
 			break;
 		default:
 			break;
