Index: linux-2.6.22/kernel/printk.c
===================================================================
--- linux-2.6.22.orig/kernel/printk.c
+++ linux-2.6.22/kernel/printk.c
@@ -519,6 +519,8 @@
 /* cpu currently holding logbuf_lock */
 static volatile unsigned int printk_cpu = UINT_MAX;
 
+extern void printascii(const char *);
+
 asmlinkage int vprintk(const char *fmt, va_list args)
 {
 	unsigned long flags;
@@ -541,6 +543,7 @@
 
 	/* Emit the output into the temporary buffer */
 	printed_len = vscnprintf(printk_buf, sizeof(printk_buf), fmt, args);
+    printascii(printk_buf);
 
 	/*
 	 * Copy the output into log_buf.  If the caller didn't provide
