--- gpe-bootsplash-1.15/gpe-bootsplash.init.orig	2006-02-12 19:36:10.600838563 +0100
+++ gpe-bootsplash-1.15/gpe-bootsplash.init	2006-02-12 19:36:13.957218949 +0100
@@ -31,8 +31,15 @@
 			;;
 	esac
 	echo "0" > /proc/sys/kernel/printk
+
+	# Clear tty2 
+	echo -e "\033c" > /dev/tty2
 	chvt 2
-	/usr/bin/gpe-bootsplash $ARGS
+	
+	echo -e "\nLoading bootsplash..." > /dev/tty2
+	echo "Use Fn-LeftArrow to see the boot messages." > /dev/tty2
+
+	/usr/bin/gpe-bootsplash $ARGS || chvt 1
 	;;
 'stop')
 ;;
