
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

Index: udev-054/Makefile
===================================================================
--- udev-054.orig/Makefile	2005-02-25 21:18:21.000000000 -0600
+++ udev-054/Makefile	2005-02-26 01:06:10.000000000 -0600
@@ -154,20 +154,22 @@
 
 SYSFS = $(PWD)/libsysfs/sysfs.a
 
-CFLAGS +=	-I$(PWD)/libsysfs/sysfs \
+override CFLAGS +=	-I$(PWD)/libsysfs/sysfs \
 		-I$(PWD)/libsysfs
 
 ifeq ($(strip $(USE_LOG)),true)
-	CFLAGS  += -DLOG
+	override CFLAGS  += -DLOG
 endif
 
 # if DEBUG is enabled, then we do not strip or optimize
+override CFLAGS += -D_GNU_SOURCE
 ifeq ($(strip $(DEBUG)),true)
-	CFLAGS  += -O1 -g -DDEBUG -D_GNU_SOURCE
+	CFLAGS += -O1 -g
+	override CFLAGS += -DDEBUG
 	LDFLAGS += -Wl,-warn-common
 	STRIPCMD = /bin/true -Since_we_are_debugging
 else
-	CFLAGS  += $(OPTIMIZATION) -fomit-frame-pointer -D_GNU_SOURCE
+	CFLAGS  += $(OPTIMIZATION) -fomit-frame-pointer
 	LDFLAGS += -s -Wl,-warn-common
 	STRIPCMD = $(STRIP) -s --remove-section=.note --remove-section=.comment
 endif
@@ -187,8 +189,8 @@
 
 	CRT0 = $(KLIBC_DIR)/crt0.o
 	LIBC = $(ARCH_LIB_OBJS) $(LIB_OBJS) $(CRT0)
-	CFLAGS += $(WARNINGS) -nostdinc				\
-		$(OPTFLAGS)					\
+	CFLAGS += $(WARNINGS) $(OPTFLAGS)
+	override CFLAGS += -nostdinc				\
 		-D__KLIBC__ -fno-builtin-printf			\
 		-I$(KLIBC_FIXUPS_DIR)				\
 		-include $(KLIBC_FIXUPS_DIR)/klibc_fixups.h	\
@@ -198,7 +200,7 @@
 		-I$(GCCINCDIR)					\
 		-I$(LINUX_INCLUDE_DIR)
 	LIB_OBJS =
-	LDFLAGS = --static --nostdlib -nostartfiles -nodefaultlibs
+	override LDFLAGS += --static --nostdlib -nostartfiles -nodefaultlibs
 
 	HEADERS	+= \
 		klibc_fixups/klibc_fixups.h	\
@@ -213,7 +215,8 @@
 	WARNINGS += -Wshadow -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations
 	CRT0 =
 	LIBC =
-	CFLAGS += $(WARNINGS) -I$(GCCINCDIR)
+	CFLAGS += $(WARNINGS)
+	override CFLAGS += -I$(GCCINCDIR)
 	LIB_OBJS = -lc
 	LDFLAGS =
 endif
@@ -221,7 +224,7 @@
 ifeq ($(strip $(USE_SELINUX)),true)
 	UDEV_OBJS += udev_selinux.o
 	LIB_OBJS += -lselinux
-	CFLAGS += -DUSE_SELINUX
+	override CFLAGS += -DUSE_SELINUX
 endif
 
 ifeq ($(strip $(V)),false)
