diff -ur shisensho_V1.0.0.orig/shisensho.desktop shisensho_V1.0.0/shisensho.desktop
--- shisensho_V1.0.0.orig/shisensho.desktop	2004-06-16 22:12:27.000000000 +0200
+++ shisensho_V1.0.0/shisensho.desktop	2004-06-16 22:12:36.000000000 +0200
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Name=Shisen-Sho
 Name[de]=Shisen-Sho
-Icon=shisensho
+Icon=shisensho/shisensho
 Exec=shisensho
 Comment=The Shisen-Sho game ported from the KDE project
 Comment[de}=Das Spiel Shisen-Sho vom KDE Projekt.
diff -ur shisensho_V1.0.0.orig/tools/Makefile shisensho_V1.0.0/tools/Makefile
--- shisensho_V1.0.0.orig/tools/Makefile	2004-06-16 22:12:27.000000000 +0200
+++ shisensho_V1.0.0/tools/Makefile	2004-06-16 22:14:46.000000000 +0200
@@ -1,3 +1,3 @@
 bin2c:
-	gcc -O2 -Wall -o bin2c bin2c.c
+	$(BUILD_CC) -O2 -Wall -o bin2c bin2c.c
 
