Index: entrance-0.9.0.009/data/config/build_config.sh.in
===================================================================
--- entrance-0.9.0.009.orig/data/config/build_config.sh.in
+++ entrance-0.9.0.009/data/config/build_config.sh.in
@@ -44,7 +44,7 @@
 rm -f $DB || exit 1
 
 # set auth to 1 for pam, 2 for shadow
-ecore_config -c $DB  -k /entrance/auth -i @auth_mode@
+ecore_config -c $DB  -k /entrance/auth -i 0
 
 # uncomment the below to enable experimental OpenGL hardware acceleration
 #ecore_config -c $DB -k /entrance/engine -i 1
