--- entrance/src/daemon/Entranced.h.~1.7.~	2004-05-19 21:02:09.000000000 -0700
+++ entrance/src/daemon/Entranced.h	2005-08-13 02:00:02.000000000 -0700
@@ -18,7 +18,7 @@
 
 #define ENTRANCED_DEBUG 1
 
-#define X_SERVER "/usr/X11R6/bin/X -quiet"
+#define X_SERVER "/etc/X11/Xserver"
 //#define X_SERVER "/usr/X11R6/bin/Xnest -full"
 #define X_DISP ":0"             /* only used if DISPLAY variable is NOT set */
 #define ENTRANCE PREFIX "/bin/entrance_wrapper"
--- entrance/src/daemon/spawner.c.~1.25.~	2005-05-28 02:31:14.000000000 -0700
+++ entrance/src/daemon/spawner.c	2005-08-13 02:20:13.000000000 -0700
@@ -155,7 +155,7 @@
         sigaction(SIGUSR1, &_entrance_x_sa, NULL);
       /* FIXME: need to parse command and NOT go thru /bin/sh!!!! */
       /* why? some /bin/sh's wont pass on this SIGUSR1 thing... */
-        execl("/bin/sh", "/bin/sh", "-c", x_cmd, NULL);
+        execl("/bin/bash", "/bin/bash", "-c", x_cmd, NULL);
         syslog(LOG_WARNING, "Could not execute X server.");
         exit(1);
      default:
