From 302745ce6f3bab7b1a97de32339405ae3fd8eacb Mon Sep 17 00:00:00 2001
From: Dmitry Baryshkov <dbaryshkov@gmail.com>
Date: Fri, 19 Oct 2007 00:05:54 +0400
Subject: [PATCH] tmio-fb-r6.patch fixes

---
 drivers/video/tmiofb.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/video/tmiofb.c b/drivers/video/tmiofb.c
index 10b0105..72eb76c 100644
--- a/drivers/video/tmiofb.c
+++ b/drivers/video/tmiofb.c
@@ -463,8 +463,8 @@ static int tmiofb_vblank (struct fb_info *fbi, struct fb_vblank *vblank)
 #define FBIO_TMIO_ACC_WRITE	0x7C639300
 #define FBIO_TMIO_ACC_SYNC	0x7C639301
 
-static int tmiofb_ioctl (struct inode *inode, struct file *file,
-		unsigned int cmd, unsigned long arg, struct fb_info *fbi)
+static int tmiofb_ioctl (struct fb_info *fbi,
+		unsigned int cmd, unsigned long arg)
 {
 	switch (cmd) {
 		case FBIOGET_VBLANK: {
@@ -677,7 +677,7 @@ static struct fb_ops tmiofb_ops_acc = {
  * 	2000		0002	display start
  * 	2000		0004	line number match (0x1ff mask???)
  */
-static irqreturn_t tmiofb_irq (int irq, void *__fbi, struct pt_regs *r)
+static irqreturn_t tmiofb_irq (int irq, void *__fbi)
 {
 	struct fb_info*			fbi	= __fbi;
 	struct tmiofb_par*		par	= fbi->par;
@@ -762,7 +762,7 @@ static int __init tmiofb_probe (struct device *dev)
 	}
 	fbi->screen_base = par->sram;
 
-	retval = request_irq (irq->start, &tmiofb_irq, SA_INTERRUPT,
+	retval = request_irq (irq->start, &tmiofb_irq, IRQF_DISABLED,
 							TMIO_NAME_LCD, fbi);
 	if (retval)
 		goto err_request_irq;
-- 
1.4.4.4

