*** ixp400_xscale_sw/src/oslinux/IxLinuxInit.c.orig	Sun Sep 25 05:59:03 2005
--- ixp400_xscale_sw/src/oslinux/IxLinuxInit.c	Fri Jun  9 14:07:06 2006
***************
*** 91,97 ****
  
  
  /* Module parameters */
! static int livelock_dispatcher = 0;/* default: don't use livelock dispatcher*/
  
  MODULE_PARM(livelock_dispatcher, "i");
  MODULE_PARM_DESC(livelock_dispatcher, "If non-zero, use the livelock prevention qmgr dispatcher");
--- 91,97 ----
  
  
  /* Module parameters */
! int livelock_dispatcher = 0;/* default: don't use livelock dispatcher*/
  
  MODULE_PARM(livelock_dispatcher, "i");
  MODULE_PARM_DESC(livelock_dispatcher, "If non-zero, use the livelock prevention qmgr dispatcher");
