--- /dev/null	2005-09-06 00:17:15.000000000 -0700
+++ ./Makefile	2005-10-02 22:15:19.974595551 -0700
@@ -0,0 +1,30 @@
+obj-m := ixp400_eth.o
+
+CFLAGS_ixp400_eth.o = -DWall \
+          -I$(IXP4XX_CSR_DIR) \
+          -I$(OSAL_DIR)/ \
+          -I$(OSAL_DIR)/os/linux/include/ \
+          -I$(OSAL_DIR)/os/linux/include/modules/ \
+          -I$(OSAL_DIR)/os/linux/include/modules/ioMem/ \
+          -I$(OSAL_DIR)/os/linux/include/modules/core/ \
+          -I$(OSAL_DIR)/os/linux/include/modules/bufferMgt/ \
+          -I$(OSAL_DIR)/os/linux/include/core/  \
+          -I$(OSAL_DIR)/os/linux/include/platforms/ \
+          -I$(OSAL_DIR)/os/linux/include/platforms/ixp400/ \
+          -I$(OSAL_DIR)/os/linux/include/core/ \
+          -I$(OSAL_DIR)/include/ \
+          -I$(OSAL_DIR)/include/modules/ \
+          -I$(OSAL_DIR)/include/modules/bufferMgt/ \
+          -I$(OSAL_DIR)/include/modules/ioMem/ \
+          -I$(OSAL_DIR)/include/modules/core/ \
+          -I$(OSAL_DIR)/include/platforms/ \
+          -I$(OSAL_DIR)/include/platforms/ixp400/ \
+          -I$(OSAL_DIR)/include/platforms/ixp400/xp425/ \
+          -I$(OSAL_DIR)/os/linux/include/platforms/ixp400/ixp425/ \
+	  $(IX_CFLAGS)
+
+default:
+	$(MAKE) ARCH=arm CROSS_COMPILE=$(LINUX_CROSS_COMPILE) V=1 -C $(LINUX_SRC) SUBDIRS=$(PWD) modules
+
+clean:
+	rm -f ixp400_eth.ko
