
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- linux-wlan-ng-0.2.2/src/include/wlan/wlan_compat.h~might_sleep
+++ linux-wlan-ng-0.2.2/src/include/wlan/wlan_compat.h
@@ -584,7 +584,7 @@
 #endif
 
 #ifndef might_sleep
-#define might_sleep(a)   do { } while (0)
+#define might_sleep()   do { } while (0)
 #endif
 
 /* Apparently 2.4.2 ethtool is quite different, maybe newer too? */
