
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- linux-wlan-ng-0.2.1-pre26/src/Makefile~only-the-modules.patch
+++ linux-wlan-ng-0.2.1-pre26/src/Makefile
@@ -44,10 +44,8 @@
 
 include ../config.mk
 
-DIRS=p80211 prism2 shared wlanctl wland nwepgen \
-	mkmeta wlancfg
-ALL_DIRS=mkmeta p80211 prism2 shared wlanctl wland \
-	nwepgen wlancfg
+DIRS=p80211 prism2 shared mkmeta
+ALL_DIRS=mkmeta p80211 prism2 shared
 
 ifneq ($(wildcard *.addon),) 
 DIRS+=`cat *.addon`
--- linux-wlan-ng-0.2.1-pre26/src/prism2/Makefile~only-the-modules.patch
+++ linux-wlan-ng-0.2.1-pre26/src/prism2/Makefile
@@ -44,7 +44,7 @@
 
 include ../../config.mk
 
-DIRS=driver ridlist download
+DIRS=driver ridlist
 
 ifneq ($(wildcard *.addon),) 
 DIRS+=`cat *.addon`
