--- balsa-2.0.17/libmutt/configure.in~	2003-03-24 22:36:15.000000000 +0000
+++ balsa-2.0.17/libmutt/configure.in	2004-06-27 16:45:40.000000000 +0100
@@ -131,7 +131,7 @@
 	if (s.st_mode & S_IWOTH) exit (0);
 	exit (1);
 }], [mutt_cv_worldwrite=yes], [mutt_cv_worldwrite=no], 
-	[ AC_MSG_ERROR("Cannot cross compile libmutt. Yet")] ))
+	[mutt_cv_worldwrite=no] ))
 
 	mutt_cv_setgid=no
 	if test $mutt_cv_worldwrite = yes; then
@@ -149,7 +149,7 @@
 	if (s.st_mode & S_IWGRP) exit (0);
 	exit (1);
 }], [mutt_cv_groupwrite=yes], [mutt_cv_groupwrite=no],
-	[ AC_MSG_ERROR("Cannot cross compile libmutt. Yet")] ))
+	[mutt_cv_worldwrite=no] ))
 
 dnl		if test $mutt_cv_groupwrite = yes; then
 dnl			AC_DEFINE(USE_DOTLOCK)
