--- gammu-0.99.0/cfg/Makefile.glo.orig	2004-10-11 04:38:06 +0200
+++ gammu-0.99.0/cfg/Makefile.glo	2004-10-24 23:28:22 +0200
@@ -124,7 +124,7 @@
 
 $(TOPDIR)/gammu/gammu: $(COMMON) $(GAMMU)
 	@echo Linking gammu
-	@$(CC) $(COMMON) $(GAMMU) -lm $(LDLIBS) -o $(TOPDIR)/gammu/gammu
+	@$(CC) $(COMMON) $(GAMMU) -lm $(LDLIBS) -o $(TOPDIR)/gammu/gammu $(LDFLAGS)
 
 gammu: $(TOPDIR)/gammu/gammu
 
@@ -134,7 +134,7 @@
 
 $(TOPDIR)/common/libGammu.so.$(SONAME_MAJOR).$(SONAME_MINOR): $(COMMON)
 	@echo Linking libGammu.so.$(SONAME_MAJOR).$(SONAME_MINOR)
-	@$(CC) $(LDLIBS) -shared -Wl,-soname -Wl,libGammu.so.$(SONAME_MAJOR).$(SONAME_MINOR) -o $(TOPDIR)/common/libGammu.so.$(SONAME_MAJOR).$(SONAME_MINOR) $(COMMON)
+	@$(CC) $(LDLIBS) -shared -Wl,-soname -Wl,libGammu.so.$(SONAME_MAJOR).$(SONAME_MINOR) -o $(TOPDIR)/common/libGammu.so.$(SONAME_MAJOR).$(SONAME_MINOR) $(COMMON) $(LDFLAGS)
 	@$(LN) -s libGammu.so.$(SONAME_MAJOR).$(SONAME_MINOR) $(TOPDIR)/common/libGammu.so.$(SONAME_MAJOR)
 	@$(LN) -s libGammu.so.$(SONAME_MAJOR) $(TOPDIR)/common/libGammu.so
 
