Index: linphone-1.6.0/mediastreamer2/configure.ac
===================================================================
--- linphone-1.6.0.orig/mediastreamer2/configure.ac	2008-11-04 03:41:04.936798321 +0300
+++ linphone-1.6.0/mediastreamer2/configure.ac	2008-11-04 03:41:09.589036036 +0300
@@ -253,7 +253,7 @@ dnl	check for various codecs libraries
 dnl	*************************************

 dnl check for installed version of speex
-PKG_CHECK_MODULES(SPEEX, speex >= 1.1.12,
+PKG_CHECK_MODULES(SPEEX, speex >= 1.1.12 speexdsp,
	[ AC_DEFINE(HAVE_SPEEX_NOISE,1,[tells whether the noise arg of speex_echo_cancel can be used]) ],
	[try_other_speex=yes]
 )
