--- bluez-utils-2.9/pcmcia/bluetooth.conf.orig	2004-09-24 14:13:13.933487924 +0100
+++ bluez-utils-2.9/pcmcia/bluetooth.conf	2004-09-24 14:13:25.737585860 +0100
@@ -27,7 +27,7 @@
   bind "serial_cs" class "bluetooth"
 
 card "AmbiCom BT2000C Bluetooth PC/CF Card"
-  version ""AmbiCom BT2000C", "Bluetooth PC/CF Card"
+  version "AmbiCom BT2000C", "Bluetooth PC/CF Card"
   bind "serial_cs" class "bluetooth"
 
 card "COM One Platinium Bluetooth PC Card"
