Index: icedtea6-1.8/Makefile.am
===================================================================
--- icedtea6-1.8.orig/Makefile.am	2010-06-17 10:29:13.654666801 +0200
+++ icedtea6-1.8/Makefile.am	2010-06-17 10:30:50.334670771 +0200
@@ -1750,7 +1745,7 @@
 stamps/rewrite-rhino.stamp: stamps/rewriter.stamp $(RHINO_JAR)
 if WITH_RHINO
 	mkdir -p rhino/rhino.{old,new}
-	(cd rhino/rhino.old ; jar xf $(RHINO_JAR))
+	(cd rhino/rhino.old ; $(ICEDTEA_BOOT_DIR)/bin/jar xf $(RHINO_JAR))
 	$(ICEDTEA_BOOT_DIR)/bin/java -cp $(abs_top_builddir)/rewriter \
 	  com.redhat.rewriter.ClassRewriter \
 	  $(abs_top_builddir)/rhino/rhino.old $(abs_top_builddir)/rhino/rhino.new \
