From ce1ae065d5eb875990f55ede0405e061e4d4b8d3 Mon Sep 17 00:00:00 2001
From: Sebastien Jan <s-jan@ti.com>
Date: Wed, 22 Dec 2010 13:53:18 +0100
Subject: [PATCH 5/5] omap4: Make 1GHz as default MPU clock

Signed-off-by: Aneesh V <aneesh@ti.com>
Signed-off-by: Rajendra Nayak <rnayak@ti.com>
Signed-off-by: Santosh Shilimkar <santosh.shilimkar@ti.com>
Signed-off-by: Sebastien Jan <s-jan@ti.com>
Signed-off-by: Ricardo Salveti de Araujo <ricardo.salveti@canonical.com>
---
 Makefile                        |    8 ++++++++
 include/configs/omap4430panda.h |    1 -
 2 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index d134fdc..1c4fdd5 100644
--- a/Makefile
+++ b/Makefile
@@ -170,6 +170,14 @@ omap3530beagle_config :    unconfig
 #########################################################################
 omap4430panda_config :    unconfig
 	@./mkconfig $(@:_config=) arm omap4 omap4430panda
+	@[ -n "$(findstring _MPU_600MHz,$@)" ] || \
+		{ echo "#define CONFIG_MPU_1000 1"	>> $(obj)include/config.h; \
+		  echo "MPU at 1GHz revision.."; \
+		}
+	@[ -z "$(findstring _MPU_600MHz,$@)" ] || \
+		{ echo "#define CONFIG_MPU_600 1"	>> $(obj)include/config.h; \
+		  echo "MPU at 600MHz revision.."; \
+		}
 
 #########################################################################
 
diff --git a/include/configs/omap4430panda.h b/include/configs/omap4430panda.h
index 74e2f42..eae02ef 100644
--- a/include/configs/omap4430panda.h
+++ b/include/configs/omap4430panda.h
@@ -44,7 +44,6 @@
  * On Phoenix board vlotage needs to be bumped up
  * before scaling the MPU up
  */
-#define CONFIG_MPU_600		1
 #define CORE_190MHZ		1
 /* Enable the below macro if MMC boot support is required */
 #define CONFIG_MMC               1
-- 
1.6.6.1

