
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- esmart-0.9.0.004/src/lib/esmart_container/esmart_container.c~compile-fix
+++ esmart-0.9.0.004/src/lib/esmart_container/esmart_container.c
@@ -30,7 +30,7 @@
    return _sort_func(o, oo);
 }
 
-void esmart_container_sort(Evas_Object *container, int (*func)(Evas_Object *, Evas_Object *$))
+void esmart_container_sort(Evas_Object *container, int (*func)(Evas_Object *, Evas_Object *))
 {
    Container *cont;
   
