Index: mono-1.2.6/configure.in
===================================================================
--- mono-1.2.6.orig/configure.in	2007-11-08 17:07:57.000000000 -0500
+++ mono-1.2.6/configure.in	2007-12-12 15:27:52.000000000 -0500
@@ -1150,6 +1150,8 @@
 		], [
 				AC_MSG_RESULT(no)
 				with_tls=pthread
+		], [
+				AC_MSG_RESULT(yes)
 		])
 	fi
 
@@ -1252,6 +1254,9 @@
 		], [
 				with_sigaltstack=no
 				AC_MSG_RESULT(no)
+		], [
+				AC_MSG_RESULT(yes)
+				AC_DEFINE(HAVE_WORKING_SIGALTSTACK)
 		])
 	fi
 
