From 0d2821af767b7cfe1a5d868a14f8a18e01070e37 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Tue, 27 Apr 2010 13:44:16 +0200
Subject: [PATCH 32/50] OMAP3: beagle: set mpurate to 600 for revB and revC1-3

---
 board/ti/beagle/beagle.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/board/ti/beagle/beagle.c b/board/ti/beagle/beagle.c
index 0544178..6778499 100644
--- a/board/ti/beagle/beagle.c
+++ b/board/ti/beagle/beagle.c
@@ -211,10 +211,12 @@ int misc_init_r(void)
 	switch (beagle_revision) {
 	case REVISION_AXBX:
 		printf("Beagle Rev Ax/Bx\n");
+		setenv("mpurate", "600");
 		break;
 	case REVISION_CX:
 		printf("Beagle Rev C1/C2/C3\n");
 		MUX_BEAGLE_C();
+		setenv("mpurate", "600");
 		break;
 	case REVISION_C4:
 		printf("Beagle Rev C4\n");
-- 
1.6.6.1

