From f4f92e1f72e2bcab85f205c66f0725a677e192e2 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Tue, 27 Apr 2010 13:45:28 +0200
Subject: [PATCH 33/50] OMAP3: beagle: prettify expansionboard message a bit

---
 board/ti/beagle/beagle.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/board/ti/beagle/beagle.c b/board/ti/beagle/beagle.c
index 6778499..58fb7c3 100644
--- a/board/ti/beagle/beagle.c
+++ b/board/ti/beagle/beagle.c
@@ -165,7 +165,7 @@ int misc_init_r(void)
 
 	beagle_identify();
 
-	printf("\nProbing for expansion boards, if none are connected you'll see a harmless I2C error.\n");
+	printf("\nProbing for expansion boards, if none are connected you'll see a harmless I2C error.\n\n");
 
 	switch (get_expansion_id()) {
 		case TINCANTOOLS_ZIPPY:
-- 
1.6.6.1

