From 442e5d80723e2772f38fe473d32287200d802907 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Fri, 30 Apr 2010 13:25:41 +0200
Subject: [PATCH 35/50] OMAP3: Beagle: set mpurate to 1000 for xM

---
 board/ti/beagle/beagle.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/board/ti/beagle/beagle.c b/board/ti/beagle/beagle.c
index 39c53f2..9300984 100644
--- a/board/ti/beagle/beagle.c
+++ b/board/ti/beagle/beagle.c
@@ -238,6 +238,7 @@ int misc_init_r(void)
 					TWL4030_PM_RECEIVER_VAUX2_VSEL_18,
 					TWL4030_PM_RECEIVER_VAUX2_DEV_GRP,
 					TWL4030_PM_RECEIVER_DEV_GRP_P1);
+		setenv("mpurate", "1000");
 		break;
 	default:
 		printf("Beagle unknown 0x%02x\n", beagle_revision);
-- 
1.6.6.1

