
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- konqueror-embedded-snapshot-20030705/konq-embed/kdesrc/kjs/ustring.h~packing.patch
+++ konqueror-embedded-snapshot-20030705/konq-embed/kdesrc/kjs/ustring.h
@@ -154,11 +154,11 @@
     /**
      * @return Lower byte.
      */
-    unsigned char& low() const { return ref().lo; }
+    unsigned char& low() const { return *((unsigned char*)&ref().lo); }
     /**
      * @return Higher byte.
      */
-    unsigned char& high() const { return ref().hi; }
+    unsigned char& high() const { return *((unsigned char*)&ref().hi); }
     /**
      * @return Character converted to lower case.
      */
