Index: control_win.c
===================================================================
RCS file: /cvs/xstroke/xstroke/control_win.c,v
retrieving revision 1.21
diff -u -r1.21 control_win.c
--- control_win.c	28 Sep 2003 11:53:42 -0000	1.21
+++ control_win.c	24 Jul 2004 15:02:52 -0000
@@ -646,9 +646,11 @@
     if (XGetWindowAttributes(dpy, window, &attr)) {
 	if (attr.all_event_masks & (KeyPressMask | KeyReleaseMask)) {
 	    return 1;
+	} else if (attr.all_event_masks & PointerMotionMask) {
+	    return 0;
 	}
     }
-    return 0;
+    return 1;
 }
 
 /*
