--- libstroke-0.5.1/configure.in.orig	2008-06-11 12:54:29.000000000 +0400
+++ libstroke-0.5.1/configure.in	2008-06-11 12:54:35.000000000 +0400
@@ -83,14 +83,14 @@
 
 dnl check for GTK
 dnl we won't build libgstroke if it's not found
-AC_MSG_CHECKING(gtk+ library)
-AM_PATH_GTK(1.2.7, HAVE_LIBGTK=yes
-  AC_DEFINE(HAVE_LIBGTK, 1, [Define if libgtk is available]),
-  AC_MSG_WARN(
-***** NOTE: The GNOME stroke support won't be built
-)
+dnl AC_MSG_CHECKING(gtk+ library)
+dnl AM_PATH_GTK(1.2.7, HAVE_LIBGTK=yes
+dnl   AC_DEFINE(HAVE_LIBGTK, 1, [Define if libgtk is available]),
+dnl   AC_MSG_WARN(
+dnl ***** NOTE: The GNOME stroke support won't be built
+dnl )
   HAVE_LIBGTK=no,
-)
+dnl )
 AM_CONDITIONAL(HAVE_LIBGTK,        test "x$HAVE_LIBGTK" = "xyes")
 
 
