Index: stunnel-4.05/configure.ac
===================================================================
--- stunnel-4.05.orig/configure.ac	2003-12-28 15:47:49.000000000 -0500
+++ stunnel-4.05/configure.ac	2005-04-01 22:14:39.751121880 -0500
@@ -176,8 +176,30 @@
 [AC_MSG_RESULT([no])]; LIBS="$saved_LIBS")
 
 dnl Check PTY device files.
-AC_CHECK_FILE("/dev/ptmx", AC_DEFINE(HAVE_DEV_PTMX))
-AC_CHECK_FILE("/dev/ptc", AC_DEFINE(HAVE_DEV_PTS_AND_PTC))
+AC_ARG_WITH(ptmx,
+    [  --with-ptmx      /dev/ptmx exists (default: check for existance)],
+    [
+        if test x"$withval" = "xyes"; then
+           AC_DEFINE(HAVE_DEV_PTMX)
+        fi
+    ],
+    [
+        # Check for ptmx device
+        AC_CHECK_FILE("/dev/ptmx", AC_DEFINE(HAVE_DEV_PTMX))
+    ]
+)
+AC_ARG_WITH(ptc,
+    [  --with-ptc      /dev/ptc exists (default: check for existance)],
+    [
+        if test x"$withval" = "xyes"; then
+           AC_DEFINE(HAVE_DEV_PTS_AND_PTC)
+        fi
+    ],
+    [
+        # Check for ptc device
+        AC_CHECK_FILE("/dev/ptc", AC_DEFINE(HAVE_DEV_PTS_AND_PTC))
+    ]
+)
 
 dnl Checks for header files.
 # AC_HEADER_DIRENT
