Index: netatalk-2.0.3/bin/afppasswd/Makefile.am
===================================================================
--- netatalk-2.0.3.orig/bin/afppasswd/Makefile.am
+++ netatalk-2.0.3/bin/afppasswd/Makefile.am
@@ -12,7 +12,8 @@ afppasswd_SOURCES = afppasswd.c
 afppasswd_LDADD = $(top_builddir)/libatalk/libatalk.la @SSL_LIBS@
 
 CFLAGS = @CFLAGS@ @SSL_CFLAGS@ -I$(top_srcdir)/sys \
-    -D_PATH_AFPDPWFILE=\"$(pkgconfdir)/afppasswd\"
+    -D_PATH_AFPDPWFILE=\"$(pkgconfdir)/afppasswd\" \
+    @BINDNOW_FLAGS@
 
 install-exec-hook:
 if USE_DHX
