diff -ruN gpe-announce-0.13/callbacks.c gpe-announce-0.13-new/callbacks.c
--- gpe-announce-0.13/callbacks.c	2007-06-14 20:56:15.000000000 +0400
+++ gpe-announce-0.13-new/callbacks.c	2005-08-24 23:04:31.000000000 +0400
@@ -28,7 +28,7 @@
 #include <linux/soundcard.h>
 #include <pthread.h>
 
-#include <gpe/soundgen.h>
+#include <soundgen.h>
 #include <gpe/schedule.h>
 
 #include <linux/types.h>
diff -ruN gpe-announce-0.13/Makefile gpe-announce-0.13-new/Makefile
--- gpe-announce-0.13/Makefile	2007-06-14 20:30:00.000000000 +0400
+++ gpe-announce-0.13-new/Makefile	2007-06-14 20:28:30.000000000 +0400
@@ -43,7 +43,7 @@
 all: $(PACKAGE)
 
 $(PACKAGE): $(OBJS)
-	$(CC) -o $@ $^ $(LDFLAGS) $(PACKAGE_LDFLAGS) -lsoundgen -lschedule
+	$(CC) -o $@ $^ $(LDFLAGS) $(PACKAGE_LDFLAGS) -lsoundgen -lschedule -lesd
 
 install-program: all 
 	install -d $(DESTDIR)$(PREFIX)/bin
