Index: SDL-1.2.11/sdl.m4
===================================================================
--- SDL-1.2.11.orig/sdl.m4	2008-10-18 21:20:13.482478928 +0200
+++ SDL-1.2.11/sdl.m4	2008-10-18 21:20:28.979441480 +0200
@@ -32,9 +32,6 @@
     fi
   fi
 
-  if test "x$prefix" != xNONE; then
-    PATH="$prefix/bin:$prefix/usr/bin:$PATH"
-  fi
   AC_PATH_PROG(SDL_CONFIG, sdl-config, no, [$PATH])
   min_sdl_version=ifelse([$1], ,0.11.0,$1)
   AC_MSG_CHECKING(for SDL - version >= $min_sdl_version)
