diff -urNd ../libxine-1.0-r1/xine-lib-1.0/configure.ac xine-lib-1.0/configure.ac
--- ../libxine-1.0-r1/xine-lib-1.0/configure.ac	2005-04-09 02:35:05 +01:00
+++ xine-lib-1.0/configure.ac	2005-04-09 02:48:17 +01:00
@@ -693,6 +693,12 @@
 AM_CONDITIONAL(HAVE_XVMC, test x$ac_have_xvmc = "xyes")
 AC_SUBST(XVMC_LIB)
 
+dnl --------------------------------------------
+dnl Declare Xv as missing
+dnl --------------------------------------------
+
+AM_CONDITIONAL(HAVE_XV, 0)
+
 
 dnl ---------------------------------------------
 dnl Checks for Xinerama extension
