
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- xine-lib-1.1.0/src/libffmpeg/libavcodec/armv4l/Makefile.am~libxine-ffmpeg-enable-arm
+++ xine-lib-1.1.0/src/libffmpeg/libavcodec/armv4l/Makefile.am
@@ -9,9 +9,8 @@
 libavcodec_armv4l_dummy =  libavcodec_armv4l_dummy.c
 EXTRA_DIST =  $(libavcodec_armv4l_src) $(libavcodec_armv4l_dummy)
 
-#if HAVE_ARMV4L
-#armv4l_modules = $(libavcodec_armv4l_src)
-#endif
-armv4l_modules = 
+if HAVE_ARMV4L
+armv4l_modules = $(libavcodec_armv4l_src)
+endif
 
 libavcodec_armv4l_la_SOURCES = $(armv4l_modules) $(libavcodec_armv4l_dummy)
