diff -Nur ewl-0.0.4.004~/src/Makefile.am ewl-0.0.4.004/src/Makefile.am
--- ewl-0.0.4.004~/src/Makefile.am	2005-09-29 18:21:52.000000000 -0700
+++ ewl-0.0.4.004/src/Makefile.am	2005-09-29 18:22:17.000000000 -0700
@@ -1,5 +1,5 @@
 
-SUBDIRS = lib bin
+SUBDIRS = lib
 
 MAINTAINERCLEANFILES = Makefile.in
 
diff -Nur ewl-0.0.4.004~/src/Makefile.in ewl-0.0.4.004/src/Makefile.in
--- ewl-0.0.4.004~/src/Makefile.in	2005-09-29 18:21:52.000000000 -0700
+++ ewl-0.0.4.004/src/Makefile.in	2005-09-29 18:22:07.000000000 -0700
@@ -108,7 +108,7 @@
 ewl_media_include = @ewl_media_include@
 requirements = @requirements@
 
-SUBDIRS = lib bin
+SUBDIRS = lib
 
 MAINTAINERCLEANFILES = Makefile.in
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
