diff -urNd ../gpe-login-0.73-r0/gpe-login-0.73/familiar/dist.gpe-session-scripts/etc/gpe/xsettings.default gpe-login-0.73/familiar/dist.gpe-session-scripts/etc/gpe/xsettings.default
--- ../gpe-login-0.73-r0/gpe-login-0.73/familiar/dist.gpe-session-scripts/etc/gpe/xsettings.default	2003-04-05 20:52:10.000000000 +0100
+++ gpe-login-0.73/familiar/dist.gpe-session-scripts/etc/gpe/xsettings.default	2004-09-16 11:08:06.000000000 +0100
@@ -1,5 +1,5 @@
 Gtk/FontName:S:Sans 8
 Net/ThemeName:S:Angelistic
 Gtk/IconSizes:S:gtk-dialog=32,32:gtk-button=16,16
-Gtk/ToolbarIconSize:I:2
+Gtk/ToolbarIconSize:I:1
 MATCHBOX/Background:S:col-gradient-horizontal:#6a7ff2,#a2f2f2
