diff -urNd ../gpe-login-0.76-r0/gpe-login-0.76/X11/gpe-login.pre-session gpe-login-0.76/X11/gpe-login.pre-session
--- ../gpe-login-0.76-r0/gpe-login-0.76/X11/gpe-login.pre-session	2005-04-01 21:58:46 +01:00
+++ gpe-login-0.76/X11/gpe-login.pre-session	2005-04-01 22:04:04 +01:00
@@ -30,13 +30,23 @@
 chown $USER /dev/touchscreen/0
 fi
 
-if [ -e /dev/collie-fl ]; then
+if [ -e /dev/misc/buzzer ]; then
+chown $USER /dev/misc/buzzer
+fi
+
+# The following added for collie
+if [ -e /dev/sharp_fl ]; then
 chown $USER /dev/sharp_fl
 fi
 
-if [ -e /dev/misc/buzzer ]; then
-chown $USER /dev/misc/buzzer
+if [ -e /dev/fb0 ]; then
+chown $USER /dev/fb0
+fi
+
+if [ -e /dev/tty ]; then
+chown $USER /dev/tty*
 fi
+# collie bits finished
 
 # check for ramdisk location
 if [ -d /mnt/ramfs ]; then
