--- ../libxine-1.0.0-beta12-r0/xine-lib/src/post/visualizations/fftscope.c	2003-04-30 07:58:51.000000000 +0100
+++ xine-lib/src/post/visualizations/fftscope.c	2004-08-17 17:40:56.000000000 +0100
@@ -103,7 +103,7 @@
  *  Bit reverser for unsigned ints
  *  Reverses 'bits' bits.
  */
-inline const unsigned int
+static inline const unsigned int
 reverse (unsigned int val, int bits)
 {
   unsigned int retn = 0;
