
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- xine-lib-1-beta12/src/audio_out/audio_oss_out.c~fix-oss-plugin
+++ xine-lib-1-beta12/src/audio_out/audio_oss_out.c
@@ -46,20 +46,7 @@
 #include <fcntl.h>
 #include <math.h>
 #include <unistd.h>
-#if defined(__OpenBSD__)
-# include <soundcard.h>
-#elif defined (__FreeBSD__)
-#  if __FreeBSD__ < 4
-#   include <machine/soundcard.h>
-#  else
-#   include <sys/soundcard.h>
-#  endif
-#else
-# if defined(__linux__)
-#  include <linux/config.h> /* Check for DEVFS */
-# endif
-# include <sys/soundcard.h>
-#endif
+#include <sys/soundcard.h>
 #include <sys/ioctl.h>
 #include <inttypes.h>
 
