--- configure.ac.orig	2003-02-12 12:43:36.000000000 +0100
+++ configure.ac	2003-02-12 12:44:24.000000000 +0100
@@ -281,10 +281,26 @@ dnl ------------------------------------
 dnl zlib
 dnl ---------------------------------------------
 dnl Test for libz
+AC_ARG_WITH(zlib-path,[  --with-xlib-path=path      Where zlib is installed],
+            zlib_path="$withval", zlib_path="")
+
+save_LDFLAGS="$LDFLAGS"
+save_CPPFLAGS="$CPPFLAGS"
+if test ! -z "$zlib_path"; then
+    LDFLAGS="$LDFLAGS -L$zlib_path/lib"
+    CPPFLAGS="$CPPFLAGS -I$zlib_path/include"
+    CFLAGS="$CFLAGS -I$zlib_path/include"
+fi
 AC_CHECK_LIB(z, gzsetparams,
 	[ AC_CHECK_HEADER(zlib.h,
 	  have_zlib=yes
 	  ZLIB_LIBS="-lz",)], AC_MSG_ERROR(zlib needed))
+LDFLAGS="$save_LDFLAGS"
+CPPFLAGS="$save_CPPFLAGS"
+
+if test ! -z "$zlib_path"; then
+    ZLIB_LIBS="-L$zlib_path/lib $ZLIB_LIBS"
+fi
 
 AM_CONDITIONAL(HAVE_ZLIB, [test x"$have_zlib" = "xyes"])
 AC_SUBST(ZLIB_LIBS)
@@ -437,8 +453,8 @@ AM_CONDITIONAL(HAVE_XINERAMA, test x$ac_
 dnl ---------------------------------------------
 dnl Checks for Ascii-Art library
 dnl ---------------------------------------------
-AM_PATH_AALIB(1.4,, AC_MSG_RESULT([*** All of AALIB dependent parts will be disabled ***]))
-AM_CONDITIONAL(HAVE_AA, test x$no_aalib != "xyes")
+dnl AM_PATH_AALIB(1.4,, AC_MSG_RESULT([*** All of AALIB dependent parts will be disabled ***]))
+AM_CONDITIONAL(HAVE_AA, /bin/false)
 
 dnl ---------------------------------------------
 dnl Check solaris framebuffer device support
@@ -450,7 +466,7 @@ dnl ------------------------------------
 dnl Check linux framebuffer device support
 dnl ---------------------------------------------
 AC_CHECK_HEADER(linux/fb.h, have_fb=yes,)
-AM_CONDITIONAL(HAVE_FB, [test x"$have_fb" = "xyes"])
+AM_CONDITIONAL(HAVE_FB, /bin/false)
 
 dnl ---------------------------------------------
 dnl Find pkg-config
@@ -480,15 +496,16 @@ fi
 
 AC_SUBST(DIRECTFB_CFLAGS)
 AC_SUBST(DIRECTFB_LIBS)
-AM_CONDITIONAL(HAVE_DIRECTFB, test x$have_directfb = "xyes" )
+AM_CONDITIONAL(HAVE_DIRECTFB, /bin/false)
 
 dnl ---------------------------------------------
 dnl check for SDL
 dnl ---------------------------------------------
-AM_PATH_SDL(1.1.5, 
-	AC_DEFINE(HAVE_SDL,1,[Define this if you have SDL library installed]),
-	[])
-AM_CONDITIONAL(HAVE_SDL, [test x"$no_sdl" != x"yes"])
+dnl AM_PATH_SDL(1.1.5, 
+dnl	AC_DEFINE(HAVE_SDL,1,[Define this if you have SDL library installed]),
+dnl	[])
+AC_DEFINE(HAVE_SDL,0,[Define this if you have SDL library installed])
+AM_CONDITIONAL(HAVE_SDL, /bin/false)
 
 
 dnl ---------------------------------------------
@@ -574,15 +591,17 @@ AM_CONDITIONAL(HAVE_VORBIS, [test x"$no_
 dnl ---------------------------------------------
 dnl check for libFLAC
 dnl ---------------------------------------------
-AM_PATH_LIBFLAC([],
-	AC_MSG_RESULT([*** All FLAC dependent parts will be disabled ***]))
-AM_CONDITIONAL(HAVE_FLAC, [test x"$no_libFLAC" != "xyes"])
+dnl AM_PATH_LIBFLAC([],
+dnl	AC_MSG_RESULT([*** All FLAC dependent parts will be disabled ***]))
+dnl AM_CONDITIONAL(HAVE_FLAC, [test x"$no_libFLAC" != "xyes"])
+AM_CONDITIONAL(HAVE_FLAC, /bin/false)
 
 dnl ---------------------------------------------
 dnl XviD libs.
 dnl ---------------------------------------------
-AM_PATH_XVID([], AC_MSG_RESULT([*** All XviD dependent parts will be disabled ***]))
-AM_CONDITIONAL(HAVE_XVID, [test x"$no_xvid" != "xyes"])	
+dnl AM_PATH_XVID([], AC_MSG_RESULT([*** All XviD dependent parts will be disabled ***]))
+dnl AM_CONDITIONAL(HAVE_XVID, [test x"$no_xvid" != "xyes"])	
+AM_CONDITIONAL(HAVE_XVID, /bin/false)
 
 
 dnl ---------------------------------------------
@@ -635,18 +654,21 @@ AM_CONDITIONAL(HAVE_ALSA09, test x"$have
 dnl ---------------------------------------------
 dnl ESD support
 dnl ---------------------------------------------
-AM_PATH_ESD(0.2.8,
-        AC_DEFINE(HAVE_ESD,1,[Define this if you have ESD (libesd) installed]),
-        AC_MSG_RESULT(*** All of ESD dependent parts will be disabled ***))
-AM_CONDITIONAL(HAVE_ESD, test x"$no_esd" != "xyes")
+dnl AM_PATH_ESD(0.2.8,
+dnl        AC_DEFINE(HAVE_ESD,1,[Define this if you have ESD (libesd) installed]),
+dnl        AC_MSG_RESULT(*** All of ESD dependent parts will be disabled ***))
+AC_DEFINE(HAVE_ESD,0,[Define this if you have ESD (libesd) installed]),
+AM_CONDITIONAL(HAVE_ESD, /bin/false)
 
 dnl ---------------------------------------------
 dnl ARTS support
 dnl ---------------------------------------------
-AM_PATH_ARTS(0.9.5,
-        AC_DEFINE(HAVE_ARTS,1,[Define this if you have ARTS (libartsc) installed]),
-        AC_MSG_RESULT(*** All of ARTS dependent parts will be disabled ***))
-AM_CONDITIONAL(HAVE_ARTS, test x"$no_arts" != "xyes")
+dnl AM_PATH_ARTS(0.9.5,
+dnl        AC_DEFINE(HAVE_ARTS,1,[Define this if you have ARTS (libartsc) installed]),
+dnl        AC_MSG_RESULT(*** All of ARTS dependent parts will be disabled ***))
+dnl AM_CONDITIONAL(HAVE_ARTS, test x"$no_arts" != "xyes")
+AC_DEFINE(HAVE_ARTS,0,[blubb],[])
+AM_CONDITIONAL(HAVE_ARTS, /bin/false)
 
 dnl ---------------------------------------------
 dnl gnome-vfs support
@@ -700,26 +722,30 @@ AC_SUBST(KSTAT_LIBS)
 dnl ---------------------------------------------
 dnl cdrom ioctls
 dnl ---------------------------------------------
-AC_CHECK_HEADERS(sys/cdio.h linux/cdrom.h)
-AM_CHECK_CDROM_IOCTLS(
-	[AC_DEFINE(HAVE_CDROM_IOCTLS,1,[Define this if you have CDROM ioctls])],
-	[AC_MSG_RESULT([*** DVD and (S)VCD support will be disabled ***])])
-AM_CONDITIONAL(HAVE_CDROM_IOCTLS, [test x"$have_cdrom_ioctls" = "xyes"])
+dnl AC_CHECK_HEADERS(sys/cdio.h linux/cdrom.h)
+dnl AM_CHECK_CDROM_IOCTLS(
+dnl	[AC_DEFINE(HAVE_CDROM_IOCTLS,1,[Define this if you have CDROM ioctls])],
+dnl	[AC_MSG_RESULT([*** DVD and (S)VCD support will be disabled ***])])
+dnl AM_CONDITIONAL(HAVE_CDROM_IOCTLS, [test x"$have_cdrom_ioctls" = "xyes"])
+AC_DEFINE([HAVE_CDROM_IOCTLS],0,[][blah])
+AM_CONDITIONAL(HAVE_CDROM_IOCTLS, /bin/false)
 
 dnl ---------------------------------------------
 dnl check for a usable version of libdvdnav
 dnl ---------------------------------------------
-AC_ARG_WITH(included-dvdnav,[  --with-included-dvdnav  Use included dvdnav library [recommended]],
-            included_dvdnav="yes"; no_dvdnav="yes", included_dvdnav="no")
-
-if test x"$included_dvdnav" = "xno"; then
-  AM_PATH_DVDNAV(0.1.4,
-	  AC_DEFINE(HAVE_DVDNAV,1,[Define this if you have a suitable version of libdvdnav]),
-	  [AC_MSG_RESULT([*** no usable version of libdvdnav found, using internal copy ***])])
-else
-  AC_MSG_RESULT([Use included DVDNAV/DVDREAD support])
-fi
-AM_CONDITIONAL(HAVE_DVDNAV, [test x"$no_dvdnav" != "xyes"])
+dnl AC_ARG_WITH(included-dvdnav,[  --with-included-dvdnav  Use included dvdnav library [recommended]],
+dnl             included_dvdnav="yes"; no_dvdnav="yes", included_dvdnav="no")
+dnl 
+dnl if test x"$included_dvdnav" = "xno"; then
+dnl   AM_PATH_DVDNAV(0.1.4,
+dnl 	  AC_DEFINE(HAVE_DVDNAV,1,[Define this if you have a suitable version of libdvdnav]),
+dnl 	  [AC_MSG_RESULT([*** no usable version of libdvdnav found, using internal copy ***])])
+dnl else
+dnl   AC_MSG_RESULT([Use included DVDNAV/DVDREAD support])
+dnl fi
+dnl AM_CONDITIONAL(HAVE_DVDNAV, [test x"$no_dvdnav" != "xyes"])
+AC_DEFINE(HAVE_DVDNAV,0,[blah])
+AM_CONDITIONAL(HAVE_DVDNAV, /bin/false)
 
 dnl ---------------------------------------------
 dnl ASF build can be optional
@@ -1050,7 +1076,7 @@ case "$host_or_hostalias" in
     AC_DEFINE_UNQUOTED(FPM_DEFAULT,,[Define to select libmad fixed point arithmetic implementation])
     ;;
 
-  armv4l-*-linux*)
+  arm-*-linux*)
     CFLAGS="$CFLAGS -O2  -fsigned-char  -ffast-math -mcpu=strongarm1100 -fomit-frame-pointer -fthread-jumps  -fregmove"
     dnl    CFLAGS="$CFLAGS -O1 -fforce-mem -fforce-addr -fthread-jumps -fcse-follow-jumps -fcse-skip-blocks -fexpensive-optimizations -fregmove -fschedule-insns2 -finline-functions -fsigned-char -fomit-frame-pointer -march=armv4 -mtune=strongarm"
     DEBUG_CFLAGS="$DEBUG_CFLAGS -O2"
