---
 Makefile.in |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- dist.orig/Makefile.in
+++ dist/Makefile.in
@@ -1116,7 +1116,7 @@ install_docs:
	    ($(MKDIR) -p $(DESTDIR)$(docdir) && \
	    $(CHMOD) $(dmode) $(DESTDIR)$(docdir))
	@cd $(DESTDIR)$(docdir) && $(RM) -r $(DOCLIST)
-	@cd $(srcdir)/docs && $(CP) -pr $(DOCLIST) $(DESTDIR)$(docdir)/
+	@cd $(srcdir)/docs && $(CP) -pR $(DOCLIST) $(DESTDIR)$(docdir)/

 uninstall_docs:
	@cd $(DESTDIR)$(docdir) && $(RM) -r $(DOCLIST)
