diff -urNd ../matchbox-panel-0.8.1-r4/matchbox-panel-0.8.1/applets/mb-applet-battery.c matchbox-panel-0.8.1/applets/mb-applet-battery.c
--- ../matchbox-panel-0.8.1-r4/matchbox-panel-0.8.1/applets/mb-applet-battery.c	2004-02-25 22:58:09.000000000 +0000
+++ matchbox-panel-0.8.1/applets/mb-applet-battery.c	2004-09-16 12:17:58.000000000 +0100
@@ -385,8 +385,7 @@
 	sprintf(tray_msg, _("AC Connected\nFully charged.\n"));
     } else {
       if (apm_vals[PERCENTAGE] > 0 
-	  && apm_vals[PERCENTAGE] < 100 
-	  && apm_vals[TIME_LEFT] > 0)
+	  && apm_vals[PERCENTAGE] < 100)
 	{
 	  sprintf(tray_msg, 
 		  _("Battery Power\nJuice %.2i %%\nTime left: %.2i mins\n"), apm_vals[PERCENTAGE], apm_vals[TIME_LEFT]);
