--- prismstumbler-0.7.3/src/iface-wlan-14.c.orig	2007-04-30 22:22:11.000000000 +0200
+++ prismstumbler-0.7.3/src/iface-wlan-14.c	2007-04-30 22:22:40.000000000 +0200
@@ -26,6 +26,7 @@
 #include <string.h>
 #include <errno.h>
 
+#include <net/if.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <linux/types.h>	/* for "__kernel_caddr_t" et al */
--- prismstumbler-0.7.3/src/cards.c.orig	2007-04-30 22:23:06.000000000 +0200
+++ prismstumbler-0.7.3/src/cards.c	2007-04-30 22:23:33.000000000 +0200
@@ -9,6 +9,8 @@
 
 #include <stdio.h>
 #include <sys/socket.h>
+#include <linux/types.h>
+#include <net/if.h>
 #include <linux/wireless.h>
 #include <glib.h>
 #include <string.h>
