---
 watchdog-5.2/Makefile.am |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: watchdog-5.2/Makefile.am
===================================================================
--- watchdog-5.2.orig/Makefile.am
+++ watchdog-5.2/Makefile.am
@@ -15,7 +15,7 @@
 	        echo "NOT installing $(CONFIG_FILENAME) - exists already"; \
 	else \
 	        echo "installing $(CONFIG_FILENAME)"; \
-		mkinstalldirs `dirname $(DESTDIR)$(CONFIG_FILENAME)`; \
+		$(mkinstalldirs) `dirname $(DESTDIR)$(CONFIG_FILENAME)`; \
 	        $(INSTALL_DATA) watchdog.conf $(DESTDIR)$(CONFIG_FILENAME); \
 	fi
 
