
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- ntop-3.0/plugins/Makefile.am~plugins
+++ ntop-3.0/plugins/Makefile.am
@@ -175,3 +175,12 @@
 	@@AWK@ -f ../utils/xmldump.awk globals-structtypes.xml
 
 .NOTPARALLEL:
+icmpPlugin_so_SOURCES =
+lastSeenPlugin_so_SOURCES =
+netflowPlugin_so_SOURCES =
+nfsPlugin_so_SOURCES =
+pdaPlugin_so_SOURCES =
+rrdPlugin_so_SOURCES =
+sflowPlugin_so_SOURCES =
+xmldumpPlugin_so_SOURCES =
+rmonPlugin_so_SOURCES =
