
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- PyQt-x11-gpl-3.13/sip/qtcanvas/qcanvas.sip~qt2-fix
+++ PyQt-x11-gpl-3.13/sip/qtcanvas/qcanvas.sip
@@ -235,7 +235,7 @@
 		sipClass = sipClass_QCanvasLine;
 		break;
 
-#if QT_VERSION >= 300
+#if QT_VERSION >= 0x030000
 	case 8:
 		sipClass = sipClass_QCanvasSpline;
 		break;
--- PyQt-x11-gpl-3.13/sip/qttable/qtable.sip~qt2-fix
+++ PyQt-x11-gpl-3.13/sip/qttable/qtable.sip
@@ -114,7 +114,7 @@
 %ConvertToSubClassCode
 	Py_BEGIN_ALLOW_THREADS
 
-#if QT_VERSION >= 300
+#if QT_VERSION >= 0x030000
 	int rtti = sipCpp -> rtti();
 
 	if (rtti == QCheckTableItem::RTTI)
--- PyQt-x11-gpl-3.13/sip/qt/qobject.sip~qt2-fix
+++ PyQt-x11-gpl-3.13/sip/qt/qobject.sip
@@ -102,7 +102,7 @@
 #endif
 		{sipName_QComboBox,		&sipClass_QComboBox},
 		{sipName_QCommonStyle,		&sipClass_QCommonStyle},
-#if QT_VERSION >= 300
+#if QT_VERSION >= 0x030000
 		{sipName_QDateEdit,		&sipClass_QDateEdit},
 		{sipName_QDateTimeEdit,		&sipClass_QDateTimeEdit},
 		{sipName_QDesktopWidget,	&sipClass_QDesktopWidget},
@@ -111,13 +111,13 @@
 		{sipName_QDial,			&sipClass_QDial},
 #endif
 		{sipName_QDialog,		&sipClass_QDialog},
-#if QT_VERSION >= 300
+#if QT_VERSION >= 0x030000
 		{sipName_QDockArea,		&sipClass_QDockArea},
 		{sipName_QDockWindow,		&sipClass_QDockWindow},
 #endif
 		{sipName_QDoubleValidator,	&sipClass_QDoubleValidator},
 		{sipName_QDragObject,		&sipClass_QDragObject},
-#if QT_VERSION >= 300
+#if QT_VERSION >= 0x030000
 		{sipName_QErrorMessage,		&sipClass_QErrorMessage},
 #endif
 #if QT_VERSION >= 0x030100
@@ -133,7 +133,7 @@
 		{sipName_QFrame,		&sipClass_QFrame},
 		{sipName_QGrid,			&sipClass_QGrid},
 		{sipName_QGridLayout,		&sipClass_QGridLayout},
-#if QT_VERSION >= 300
+#if QT_VERSION >= 0x030000
 		{sipName_QGridView,		&sipClass_QGridView},
 #endif
 		{sipName_QGroupBox,		&sipClass_QGroupBox},
@@ -175,7 +175,7 @@
 		{sipName_QNetworkOperation,	&sipClass_QNetworkOperation},
 		{sipName_QNetworkProtocol,	&sipClass_QNetworkProtocol},
 #endif
-#if QT_VERSION >= 300
+#if QT_VERSION >= 0x030000
 		{sipName_QObjectCleanupHandler,	&sipClass_QObjectCleanupHandler},
 #endif
 #if defined(SIP_FEATURE_Qt_STYLE_PLATINUM)
@@ -185,7 +185,7 @@
 #if defined(WS_X11) && defined(SIP_FEATURE_Qt_PRINTDIALOG)
 		{sipName_QPrintDialog,		&sipClass_QPrintDialog},
 #endif
-#if QT_VERSION >= 300
+#if QT_VERSION >= 0x030000
 		{sipName_QProcess,		&sipClass_QProcess},
 #endif
 		{sipName_QProgressBar,		&sipClass_QProgressBar},
@@ -194,7 +194,7 @@
 #endif
 		{sipName_QPushButton,		&sipClass_QPushButton},
 		{sipName_QRadioButton,		&sipClass_QRadioButton},
-#if QT_VERSION >= 300
+#if QT_VERSION >= 0x030000
 		{sipName_QRegExpValidator,	&sipClass_QRegExpValidator},
 #endif
 #if QT_VERSION >= 220 && defined(SIP_FEATURE_Qt_STYLE_SGI)
@@ -229,11 +229,11 @@
 		{sipName_QTabWidget,		&sipClass_QTabWidget},
 		{sipName_QTextBrowser,		&sipClass_QTextBrowser},
 		{sipName_QTextDrag,		&sipClass_QTextDrag},
-#if QT_VERSION >= 300
+#if QT_VERSION >= 0x030000
 		{sipName_QTextEdit,		&sipClass_QTextEdit},
 #endif
 		{sipName_QTextView,		&sipClass_QTextView},
-#if QT_VERSION >= 300
+#if QT_VERSION >= 0x030000
 		{sipName_QTimeEdit,		&sipClass_QTimeEdit},
 #endif
 		{sipName_QTimer,		&sipClass_QTimer},
--- PyQt-x11-gpl-3.13/sip/qt/qevent.sip~qt2-fix
+++ PyQt-x11-gpl-3.13/sip/qt/qevent.sip
@@ -528,7 +528,7 @@
 		sipClass = sipClass_QChildEvent;
 		break;
 
-#if QT_VERSION >= 300
+#if QT_VERSION >= 0x030000
 	case QEvent::ContextMenu:
 		sipClass = sipClass_QContextMenuEvent;
 		break;
@@ -539,7 +539,7 @@
 		break;
 #endif
 
-#if QT_VERSION >= 300 && QT_VERSION < 0x030100
+#if QT_VERSION >= 0x030000 && QT_VERSION < 0x030100
 	case QEvent::IMCompose:
 		sipClass = sipClass_QIMEvent;
 		break;
