--- glibc-2.3.1.bak/inet/herrno.c	2003-01-02 10:54:16.000000000 -0500
+++ glibc-2.3.1/inet/herrno.c	2003-01-02 10:55:25.000000000 -0500
@@ -38,8 +38,8 @@
    visible at link time.  Programs must use the accessor functions.  */
 # if defined HAVE_ELF && defined SHARED && defined DO_VERSIONING
 #  include <shlib-compat.h>
-compat_symbol (libc, h_errno, h_errno, GLIBC_2_0);
-compat_symbol (libc, _h_errno, _h_errno, GLIBC_2_0);
+// compat_symbol (libc, h_errno, h_errno, GLIBC_2_0);
+// compat_symbol (libc, _h_errno, _h_errno, GLIBC_2_0);
 # endif
 #endif
 
--- glibc-2.3.1.bak/resolv/res_libc.c	2003-01-02 10:54:16.000000000 -0500
+++ glibc-2.3.1/resolv/res_libc.c	2003-01-02 10:54:45.000000000 -0500
@@ -87,7 +87,7 @@
    visible at link time.  Programs must use the accessor functions.  */
 #if defined HAVE_ELF && defined SHARED && defined DO_VERSIONING
 # include <shlib-compat.h>
-compat_symbol (libc, _res, _res, GLIBC_2_0);
+// compat_symbol (libc, _res, _res, GLIBC_2_0);
 #endif
 
 #include <shlib-compat.h>
--- glibc-2.3.1.bak/sysdeps/generic/errno.c	2003-01-02 10:54:16.000000000 -0500
+++ glibc-2.3.1/sysdeps/generic/errno.c	2003-01-02 10:55:16.000000000 -0500
@@ -36,7 +36,7 @@
    visible at link time.  Programs must use the accessor functions.  */
 # if defined HAVE_ELF && defined SHARED && defined DO_VERSIONING
 #  include <shlib-compat.h>
-compat_symbol (libc, errno, errno, GLIBC_2_0);
-compat_symbol (libc, _errno, _errno, GLIBC_2_0);
+// compat_symbol (libc, errno, errno, GLIBC_2_0);
+// compat_symbol (libc, _errno, _errno, GLIBC_2_0);
 # endif
 #endif
