--- glibc-2.2.5.orig/sysdeps/unix/sysv/linux/m68k/chown.c	2001-07-10 02:57:24.000000000 +0200
+++ glibc-2.2.5/sysdeps/unix/sysv/linux/m68k/chown.c	2002-08-11 15:37:09.000000000 +0200
@@ -72,3 +72,8 @@
 #endif
 }
 weak_alias (__chown, chown)
+
+#include <shlib-compat.h>
+#if SHLIB_COMPAT (libc, GLIBC_2_1, GLIBC_2_2)
+compat_symbol (libc, __chown, chown, GLIBC_2_1);
+#endif
--- glibc-2.2.5.orig/sysdeps/unix/sysv/linux/m68k/lchown.c	2000-01-17 06:20:35.000000000 +0100
+++ glibc-2.2.5/sysdeps/unix/sysv/linux/m68k/lchown.c	2002-08-11 15:38:56.000000000 +0200
@@ -1 +1,6 @@
 #include <sysdeps/unix/sysv/linux/i386/lchown.c>
+
+#include <shlib-compat.h>
+#if SHLIB_COMPAT (libc, GLIBC_2_1, GLIBC_2_2)
+compat_symbol (libc, __lchown, lchown, GLIBC_2_1);
+#endif
