
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- qt-2.3.9-snapshot-20041211/src/tools/qcstring.h~gcc3
+++ qt-2.3.9-snapshot-20041211/src/tools/qcstring.h
@@ -119,7 +119,7 @@
 //                   We want to keep source compatibility for 2.x
 // ### TODO for 4.0: completely remove these and the cstr* functions
 
-#if !defined(QT_GENUINE_STR)
+#if 0
 
 #undef	strlen
 #define strlen qstrlen
--- qt-2.3.9-snapshot-20041211/src/kernel/qwsdecoration_qws.h~gcc3
+++ qt-2.3.9-snapshot-20041211/src/kernel/qwsdecoration_qws.h
@@ -50,7 +50,7 @@
     enum Region { None=0, All=1, Title=2, Top=3, Bottom=4, Left=5, Right=6,
 		TopLeft=7, TopRight=8, BottomLeft=9, BottomRight=10,
 		Close=11, Minimize=12, Maximize=13, Normalize=14,
-		Menu=15, LastRegion=Menu };
+		Menu=15, LastRegion=Menu, UserDefined = 100 };
 
     virtual QRegion region(const QWidget *, const QRect &rect, Region r=All) = 0;
     virtual void close( QWidget * );
