--- R-2.0.0/src/library/Makefile.in~	2004-12-12 11:01:07.840925056 +0000
+++ R-2.0.0/src/library/Makefile.in	2004-12-12 11:02:49.683442640 +0000
@@ -20,11 +20,11 @@
 ## Need to go through Rcmd rather than just call $(PERL) to ensure that
 ## PERL5LIB/PERLLIB are set appropriately (or do this ourselves).
 ## See e.g. @file{src/gnuwin32/help/Makefile}.
-R_CMD_BUILD_HELP = $(top_builddir)/bin/R CMD perl $(top_srcdir)/share/perl/build-help.pl
+R_CMD_BUILD_HELP = $(R_EXE_NATIVE) CMD perl $(top_srcdir)/share/perl/build-help.pl
 ## </NOTE>
 R_CMD_BUILD_HELP_OPTS =
 
-R_EXE = $(top_builddir)/bin/R --vanilla
+R_EXE = $(R_EXE_NATIVE) --vanilla
 
 PDFLATEX = @PDFLATEX@
 
