diff -urNd ../gpe-aerial-0.2.11-r0/gpe-aerial-0.2.11/Makefile gpe-aerial-0.2.11/Makefile
--- ../gpe-aerial-0.2.11-r0/gpe-aerial-0.2.11/Makefile	2004-08-03 15:36:47.000000000 +0100
+++ gpe-aerial-0.2.11/Makefile	2004-08-03 15:49:42.000000000 +0100
@@ -3,7 +3,6 @@
 CVSBUILD = no
 DEBUG = no
 INSTALL = install
-STRIP=strip
 PREFIX = /usr/local
 DESTDIR = /
 
@@ -52,8 +51,7 @@
 
 install-program: $(PACKAGE) all-desktop
 	mkdir -p $(DESTDIR)/$(PREFIX)/bin
-	$(INSTALL) $(PACKAGE) $(DESTDIR)$(PREFIX)/bin/
-	$(STRIP) $(DESTDIR)$(PREFIX)/bin/$(PACKAGE)
+	$(INSTALL) -s $(PACKAGE) $(DESTDIR)$(PREFIX)/bin/
 	mkdir -p $(DESTDIR)$(PREFIX)/share/pixmaps/
 	for i in $(PIXMAPS); do install -m 644 -D pixmaps/$$i $(DESTDIR)$(PREFIX)/share/pixmaps/$$i; done
 	mkdir -p $(DESTDIR)$(PREFIX)/share/applications
