--- nslu2-linksys-ramdisk-2.3r63/etc/passwd~	2006-04-18 21:54:47.000000000 -0500
+++ nslu2-linksys-ramdisk-2.3r63/etc/passwd	2006-04-18 21:56:53.000000000 -0500
@@ -4,7 +4,7 @@
 mail:x:8:12:mail:/var/spool/mail:
 ftp:x:14:50:FTP User:/:
 nobody:x:99:99:Nobody:/:
-ourtelnetrescueuser:sPuRQwXaya5YE:100:100::/home/user:/bin/sh
+ourtelnetrescueuser:x:100:100::/home/user:/dev/null
 guest:xqnMpE/plEnFs:501:501::/home/user/guest:/dev/null
 admin:cgwvsHpJSf6XU:502:501::/home/user/admin:/dev/null
 
--- nslu2-linksys-ramdisk-2.3r63/usr/local/passwd~	2006-04-18 21:54:47.000000000 -0500
+++ nslu2-linksys-ramdisk-2.3r63/usr/local/passwd	2006-04-18 21:57:48.000000000 -0500
@@ -4,7 +4,7 @@
 mail:x:8:12:mail:/var/spool/mail:
 ftp:x:14:50:FTP User:/:
 nobody:x:99:99:Nobody:/:
-ourtelnetrescueuser:sPuRQwXaya5YE:100:100::/home/user:/bin/sh
+ourtelnetrescueuser:x:100:100::/home/user:/dev/null
 guest:xqnMpE/plEnFs:501:501::/home/user/guest:/dev/null
 admin:cgwvsHpJSf6XU:502:501::/home/user/admin:/dev/null
 
