--- inputattach.c.old	2004-10-09 23:17:27.000000000 +0100
+++ inputattach.c	2004-10-09 23:17:37.000000000 +0100
@@ -369,6 +369,7 @@
 { "--ps2serkbd",	"-ps2ser",	B1200, CS8,			SERIO_PS2SER,	0x00,	1,	NULL },
 { "--twiddler",		"-twid",	B2400, CS8,			SERIO_TWIDKBD,	0x00,	0,	twiddler_init },
 { "--twiddler-joy",	"-twidjoy",	B2400, CS8,			SERIO_TWIDJOY,	0x00,	0,	twiddler_init },
+{ "--snes232",          "-pad",         B115200, CS8,                   SERIO_SNES232,  0x00,   0,      NULL },
 { "--dump",		"-dump",	B2400, CS8, 			0,		0x00,	0,	dump_init },
 { "", "", 0, 0 }
 
