Index: antlr-2.7.5/Makefile.in
===================================================================
--- antlr-2.7.5.orig/Makefile.in	2005-01-28 15:45:42.000000000 -0600
+++ antlr-2.7.5/Makefile.in	2005-02-18 12:53:15.000000000 -0600
@@ -123,16 +123,16 @@
 extradir = $(datadir)/$(versioneddir)
 
 install ::
-	$(MKDIR) -p "$(bindir)"
-	$(MKDIR) -p "$(extradir)"
-	$(MKDIR) -p "$(docdir)"
-	$(INSTALL) -m 755 scripts/run-antlr                          "$(bindir)/antlr"
-	$(INSTALL) -m 755 scripts/antlr-config                       "$(bindir)/antlr-config"
-	$(INSTALL) -m 444 @abs_top_srcdir@/extras/antlr-mode.el      "$(extradir)"
-	$(INSTALL) -m 444 @abs_top_srcdir@/extras/antlr-jedit.xml    "$(extradir)"
-	$(INSTALL) -m 444 @abs_top_srcdir@/LICENSE.txt               "$(docdir)"
-	$(INSTALL) -m 444 @abs_top_srcdir@/README.txt                "$(docdir)"
-	$(INSTALL) -m 444 @abs_top_srcdir@/INSTALL.txt               "$(docdir)"
+	$(MKDIR) -p "$(DESTDIR)$(bindir)"
+	$(MKDIR) -p "$(DESTDIR)$(extradir)"
+	$(MKDIR) -p "$(DESTDIR)$(docdir)"
+	$(INSTALL) -m 755 scripts/run-antlr                          "$(DESTDIR)$(bindir)/antlr"
+	$(INSTALL) -m 755 scripts/antlr-config                       "$(DESTDIR)$(bindir)/antlr-config"
+	$(INSTALL) -m 444 @abs_top_srcdir@/extras/antlr-mode.el      "$(DESTDIR)$(extradir)"
+	$(INSTALL) -m 444 @abs_top_srcdir@/extras/antlr-jedit.xml    "$(DESTDIR)$(extradir)"
+	$(INSTALL) -m 444 @abs_top_srcdir@/LICENSE.txt               "$(DESTDIR)$(docdir)"
+	$(INSTALL) -m 444 @abs_top_srcdir@/README.txt                "$(DESTDIR)$(docdir)"
+	$(INSTALL) -m 444 @abs_top_srcdir@/INSTALL.txt               "$(DESTDIR)$(docdir)"
 
 install ::
 	echo "installation done"
Index: antlr-2.7.5/doc/Makefile.in
===================================================================
--- antlr-2.7.5.orig/doc/Makefile.in	2005-01-28 15:13:19.000000000 -0600
+++ antlr-2.7.5/doc/Makefile.in	2005-02-18 12:50:51.000000000 -0600
@@ -43,12 +43,12 @@
 	$(eol)
 
 install ::
-	@$(MKDIR) -p "$(antlr_doc_DIR)"
+	@$(MKDIR) -p "$(DESTDIR)$(antlr_doc_DIR)"
 	@echo "install doc files .. "
 	@for f in $(antlr_doc_FILES) ; do \
 		echo "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(antlr_doc_DIR)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(antlr_doc_DIR)" ; \
 		fi ;\
 	done
 
Index: antlr-2.7.5/lib/csharp/src/Makefile.in
===================================================================
--- antlr-2.7.5.orig/lib/csharp/src/Makefile.in	2005-01-28 15:13:23.000000000 -0600
+++ antlr-2.7.5/lib/csharp/src/Makefile.in	2005-02-18 12:51:54.000000000 -0600
@@ -134,13 +134,13 @@
 distclean :: clean
 
 install :: @ANTLR_NET@ 
-	@$(MKDIR) -p "$(libdir)"
+	@$(MKDIR) -p "$(DESTDIR)$(libdir)"
 	@echo "install C# core files .. "
 	@for f in @ANTLR_NET@  ; do \
 		echo "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(libdir)" ; \
-			$(INSTALL) -m 444 "$${f}" "$(datadir)/$(versioneddir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(libdir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(datadir)/$(versioneddir)" ; \
 		fi ;\
 	done
 
Index: antlr-2.7.5/lib/csharp/ASTFrame/Makefile.in
===================================================================
--- antlr-2.7.5.orig/lib/csharp/ASTFrame/Makefile.in	2005-01-28 15:13:23.000000000 -0600
+++ antlr-2.7.5/lib/csharp/ASTFrame/Makefile.in	2005-02-18 12:51:38.000000000 -0600
@@ -46,13 +46,13 @@
 
 ## install our target ..
 install :: @ASTFRAME_NET@ 
-	@$(MKDIR) -p "$(libdir)"
+	@$(MKDIR) -p "$(DESTDIR)$(libdir)"
 	@echo "install C# core files .. "
 	@for f in @ASTFRAME_NET@  ; do \
 		echo "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(libdir)" ; \
-			$(INSTALL) -m 444 "$${f}" "$(datadir)/$(versioneddir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(libdir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(datadir)/$(versioneddir)" ; \
 		fi ;\
 	done
 
Index: antlr-2.7.5/lib/cpp/antlr/Makefile.in
===================================================================
--- antlr-2.7.5.orig/lib/cpp/antlr/Makefile.in	2005-01-28 15:13:23.000000000 -0600
+++ antlr-2.7.5/lib/cpp/antlr/Makefile.in	2005-02-18 12:51:03.000000000 -0600
@@ -74,11 +74,11 @@
 
 install:
 	@echo "install hpp files .. "
-	@$(MKDIR) -p "$(includedir)/antlr"
+	@$(MKDIR) -p "$(DESTDIR)$(includedir)/antlr"
 	@for f in $(antlr_hpp_FILES) ; do \
 		echo "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(includedir)/antlr" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(includedir)/antlr" ; \
 		fi ;\
 	done
 
Index: antlr-2.7.5/lib/cpp/src/Makefile.in
===================================================================
--- antlr-2.7.5.orig/lib/cpp/src/Makefile.in	2005-01-28 15:13:23.000000000 -0600
+++ antlr-2.7.5/lib/cpp/src/Makefile.in	2005-02-18 12:51:21.000000000 -0600
@@ -113,12 +113,12 @@
   $(eol)
 
 this-install: @ANTLR_LIB@ 
-	@$(MKDIR) -p "$(libdir)"
+	@$(MKDIR) -p "$(DESTDIR)$(libdir)"
 	@echo "install C++ core files .. "
 	@for f in $(antlr_lib_FILES) ; do \
 		echo "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(libdir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(libdir)" ; \
 		fi ;\
 	done
 
Index: antlr-2.7.5/antlr/Makefile.in
===================================================================
--- antlr-2.7.5.orig/antlr/Makefile.in	2005-01-28 15:13:18.000000000 -0600
+++ antlr-2.7.5/antlr/Makefile.in	2005-02-18 12:50:34.000000000 -0600
@@ -314,11 +314,11 @@
 
 ## install antlr.jar in libdirectory if exists
 install ::	@ANTLR_JAR@ 
-	-$(MKDIR) -p "$(libdir)"
-	-$(MKDIR) -p "$(datadir)/$(versioneddir)"
+	-$(MKDIR) -p "$(DESTDIR)$(libdir)"
+	-$(MKDIR) -p "$(DESTDIR)$(datadir)/$(versioneddir)"
 	-test -f @ANTLR_JAR@ && {\
-	 $(INSTALL) -m 444 @ANTLR_JAR@ "$(datadir)/$(versioneddir)" ; \
-	 $(INSTALL) -m 444 @ANTLR_JAR@ "$(libdir)" ; \
+	 $(INSTALL) -m 444 @ANTLR_JAR@ "$(DESTDIR)$(datadir)/$(versioneddir)" ; \
+	 $(INSTALL) -m 444 @ANTLR_JAR@ "$(DESTDIR)$(libdir)" ; \
 	}
 
 ## get configured dependencies
Index: antlr-2.7.5/lib/python/Makefile.in
===================================================================
--- antlr-2.7.5.orig/lib/python/Makefile.in	2005-01-28 15:13:24.000000000 -0600
+++ antlr-2.7.5/lib/python/Makefile.in	2005-02-18 12:52:50.000000000 -0600
@@ -41,26 +41,26 @@
 extradir = $(datadir)/$(versioneddir)
 
 install-this:
-	$(MKDIR) -p "$(antlr_py_DIR)"
-	$(MKDIR) -p "$(libdir)"
+	$(MKDIR) -p "$(DESTDIR)$(antlr_py_DIR)"
+	$(MKDIR) -p "$(DESTDIR)$(libdir)"
 	@echo "install python files .. "
 	@for f in $(antlr_py_FILES) ; do \
 		echo "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(libdir)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(libdir)" ; \
 		fi ;\
 	done
 	@for f in $(antlr_py1_FILES) ; do \
 		echo "install $${f}" ; \
 		if test -f "$${f}" ; then \
-			$(INSTALL) -m 444 "$${f}" "$(antlr_py_DIR)" ; \
+			$(INSTALL) -m 444 "$${f}" "$(DESTDIR)$(antlr_py_DIR)" ; \
 		fi ;\
 	done
 	@f="$(objdir)/scripts/pyantlr.sh"; \
 	if test -f "$${f}" ; then \
 			echo "install Python installer" ; \
-			$(MKDIR) -p "$(sbindir)" ; \
-			$(INSTALL) -m 555 "$${f}" "$(sbindir)" ; \
+			$(MKDIR) -p "$(DESTDIR)$(sbindir)" ; \
+			$(INSTALL) -m 555 "$${f}" "$(DESTDIR)$(sbindir)" ; \
 	fi
 
 
