diff -Naur gstreamer-0.10.21-org/Makefile.am gstreamer-0.10.21/Makefile.am
--- gstreamer-0.10.21-org/Makefile.am	2008-01-09 14:48:03.000000000 +0100
+++ gstreamer-0.10.21/Makefile.am	2008-10-31 20:11:16.000000000 +0100
@@ -12,14 +12,12 @@
 
 SUBDIRS = \
 	gst libs plugins tools tests \
-	docs \
 	pkgconfig po \
 	common
 
 # These are all the possible subdirs
 DIST_SUBDIRS = \
 	gst libs plugins tools tests \
-	docs \
 	pkgconfig po \
 	common
 
