--- configure.ac	2009-10-05 23:29:45.000000000 +0200
+++ configure.ac	2009-10-05 23:30:46.000000000 +0200
@@ -658,8 +658,6 @@
 GST_PLUGIN_LDFLAGS="-module -avoid-version -export-symbols-regex '^[_]*gst_plugin_desc\$\$' $GST_ALL_LDFLAGS"
 AC_SUBST(GST_PLUGIN_LDFLAGS, "$GST_PLUGIN_LDFLAGS")
 
-SHAVE_INIT([common],[enable])
-
 dnl things for our internal libcheck
 if test "x$BUILD_CHECK" = "xyes"; then
   AG_GST_CHECK_CHECKS()
