--- Mesa/bin/mklib.orig	2006-06-04 20:56:22.000000000 -0600
+++ Mesa/bin/mklib	2006-06-04 20:57:04.000000000 -0600
@@ -98,6 +98,9 @@
 	-L*)
 	    DEPS="$DEPS $1"
 	    ;;
+	-Wl*)
+	    DEPS="$DEPS $1"
+	    ;;
 	'-cplusplus')
 	    CPLUSPLUS=1
 	    ;;
--- Mesa-6.5.2/configs/default.old	2007-01-26 11:49:32.000000000 +0000
+++ Mesa-6.5.2/configs/default	2007-01-26 11:49:49.000000000 +0000
@@ -73,7 +73,7 @@
 GLU_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -l$(GL_LIB) -lm
 GLUT_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -l$(GLU_LIB) -l$(GL_LIB) $(EXTRA_LIB_PATH) -lX11 -lXmu -lXt -lXi -lm
 GLW_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -l$(GL_LIB) $(EXTRA_LIB_PATH) -lXt -lX11
-APP_LIB_DEPS = -L$(TOP)/$(LIB_DIR) -l$(GLUT_LIB) -l$(GLU_LIB) -l$(GL_LIB) -lm
+APP_LIB_DEPS = -L$(TOP)/$(LIB_DIR) $(EXTRA_LIB_PATH) -l$(GLUT_LIB) -l$(GLU_LIB) -l$(GL_LIB) -lm
 
 
 
