--- linux-2.4.22/drivers/scsi/scsi_scan.c.orig	2004-11-06 14:01:22.000000000 +1030
+++ linux-2.4.22/drivers/scsi/scsi_scan.c	2004-11-06 15:00:33.000000000 +1030
@@ -704,6 +704,7 @@
         /* restrict device on certain port grant */
 	SDpnt->host->hostt->removable = SDpnt->removable;
 
+#if 0
         if ( SDpnt->removable && (USB_DEVPATH_1 == SDpnt->host->hostt->port) )
         {
 		create_proc_read_entry("usb_err",
@@ -718,8 +719,9 @@
 		shpnt->porttype = 1;
 		return 0;
         }
+#endif
 		// add by super
-        if (SDpnt->removable){
+        if (SDpnt->removable && (USB_DEVPATH_2 == SDpnt->host->hostt->port)){
 		printk("is removable disk \n");
 		*IXP425_GPIO_GPOUTR &= 0xfffb;
 		create_proc_read_entry("usb_conn",
--- linux-2.4.22/drivers/usb/storage/usb.c.orig	2004-11-06 14:01:06.000000000 +1030
+++ linux-2.4.22/drivers/usb/storage/usb.c	2004-11-06 15:01:23.000000000 +1030
@@ -787,6 +787,7 @@
 		ss->htmplt.removable = 1;
 	}
 #endif
+#if 0
                 /* restrict device on certain port */
         if ( ss->htmplt.removable && (USB_DEVPATH_1 ==  dev->devpath[0]))
                 if ( ss->htmplt.removable ^ (USB_DEVPATH_2 == dev->devpath[0]) )
@@ -803,7 +804,8 @@
 			free_us_data( pre_ss, ss );
 			return NULL;
                 }
-                if (ss->htmplt.removable){
+#endif
+                if (ss->htmplt.removable && (USB_DEVPATH_2 == dev->devpath[0])){
 			printk("is removable disk \n");
 			*IXP425_GPIO_GPOUTR &= 0xfffb;
 			create_proc_read_entry("usb_conn",
