diff --git a/wavplugin.pro b/wavplugin.pro
index da89490..d26355b 100644
--- a/wavplugin.pro
+++ b/wavplugin.pro
@@ -3,7 +3,7 @@ CONFIG		+=  qt plugin warn_on
 HEADERS		=   wavplugin.h wavpluginimpl.h
 SOURCES	        =   wavplugin.cpp wavpluginimpl.cpp
 TARGET		=   wavplugin
-DESTDIR		=   $(OPIEDIR)/plugins/codecs
+DESTDIR		=   ${palmtopdir}/plugins/codecs
 INCLUDEPATH += $(OPIEDIR)/include ..
 DEPENDPATH      +=   ..
 LIBS            +=  -lqpe
