--- libc/ports/sysdeps/unix/sysv/linux/arm/nptl/sysdep-cancel.h.org	2008-04-04 18:44:53.000000000 -0700
+++ libc/ports/sysdeps/unix/sysv/linux/arm/nptl/sysdep-cancel.h	2008-04-04 18:45:27.000000000 -0700
@@ -73,6 +73,9 @@
 # define DOCARGS_5	DOCARGS_4
 # define UNDOCARGS_5	UNDOCARGS_4
 
+# define DOCARGS_6	DOCARGS_5
+# define UNDOCARGS_6	UNDOCARGS_5
+
 # ifdef IS_IN_libpthread
 #  define CENABLE	bl PLTJMP(__pthread_enable_asynccancel)
 #  define CDISABLE	bl PLTJMP(__pthread_disable_asynccancel)
