
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- uClibc-0.9.26/libc/sysdeps/linux/arm/ioperm.c~26headers
+++ uClibc-0.9.26/libc/sysdeps/linux/arm/ioperm.c
@@ -47,6 +47,8 @@
 #include <asm/page.h>
 #include <sys/sysctl.h>
 
+#include <linux/version.h>
+
 #define PATH_ARM_SYSTYPE	"/etc/arm_systype"
 #define PATH_CPUINFO		"/proc/cpuinfo"
 
@@ -100,8 +102,13 @@
 {
     char systype[256];
     int i, n;
+#if LINUX_VERSION_CODE < 132119
     static int iobase_name[] = { CTL_BUS, BUS_ISA, BUS_ISA_PORT_BASE };
     static int ioshift_name[] = { CTL_BUS, BUS_ISA, BUS_ISA_PORT_SHIFT };
+#else
+    static int iobase_name[] = { CTL_BUS, CTL_BUS_ISA, BUS_ISA_PORT_BASE };
+    static int ioshift_name[] = { CTL_BUS, CTL_BUS_ISA, BUS_ISA_PORT_SHIFT };
+#endif
     size_t len = sizeof(io.base);
 
     if (! sysctl (iobase_name, 3, &io.io_base, &len, NULL, 0)
