Index: pyreverse-0.5.2/setup.py
===================================================================
--- pyreverse-0.5.2.orig/setup.py
+++ pyreverse-0.5.2/setup.py
@@ -17,10 +17,10 @@
 # this program; if not, write to the Free Software Foundation, Inc.,
 # 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 """ Generic Setup script, takes package info from __pkginfo__.py file """
+from __future__ import nested_scopes
 
 __revision__ = '$Id: setup.py,v 1.17 2005/01/05 11:40:41 syt Exp $'
 
-from __future__ import nested_scopes
 import os
 import sys
 import shutil
