Index: fbreader-0.7.3c/makefiles/target.mk
===================================================================
--- fbreader-0.7.3c.orig/makefiles/target.mk	2006-04-27 23:56:44.000000000 +0200
+++ fbreader-0.7.3c/makefiles/target.mk	2006-04-28 22:59:55.152319760 +0200
@@ -1,6 +1,6 @@
-#TARGET_ARCH = openzaurus
+TARGET_ARCH = openzaurus
 #UI_TYPE = opie
-#UI_TYPE = gpe
+UI_TYPE = gpe
 
 #TARGET_ARCH = opensimpad-0.9.0
 #UI_TYPE = opie
@@ -8,8 +8,8 @@
 #TARGET_ARCH = qvfb
 #UI_TYPE = opie
 
-TARGET_ARCH = maemo
-UI_TYPE = maemo
+#TARGET_ARCH = maemo
+#UI_TYPE = maemo
 
 #TARGET_ARCH = desktop
 #UI_TYPE = gtk
