? atd-startup.patch
Index: ChangeLog
===================================================================
RCS file: /cvs/apps/atd/ChangeLog,v
retrieving revision 1.2
diff -B -b -u -r1.2 ChangeLog
--- ChangeLog	2 Feb 2004 15:30:39 -0000	1.2
+++ ChangeLog	13 Jan 2006 11:54:00 -0000
@@ -1,3 +1,8 @@
+2006-01-13 Florian Boor <florian@kernelconcepts.de>
+
+	* dist/etc/init.d/atd: Remove obsolete --oknodo and add -m to 
+          enable pidfile. 
+
 V0.7
 	- removed waiting for childpid after having spawned an event
 	  this caused trouble with events rescheduling themselves
Index: dist/etc/init.d/atd
===================================================================
RCS file: /cvs/apps/atd/dist/etc/init.d/atd,v
retrieving revision 1.3
diff -B -b -u -r1.3 atd
--- dist/etc/init.d/atd	24 Jan 2003 15:09:03 -0000	1.3
+++ dist/etc/init.d/atd	13 Jan 2006 11:54:00 -0000
@@ -11,12 +11,12 @@
         echo -n "Starting at daemon: atd"
         mkdir -p /var/spool/at
 	chmod a+w /var/spool/at
-        start-stop-daemon --start --quiet --pidfile /var/run/atd.pid --exec /usr/sbin/atd /var/spool/at >/dev/null 2>&1 &
+        start-stop-daemon --start --quiet -m --pidfile /var/run/atd.pid --exec /usr/sbin/atd /var/spool/at >/dev/null 2>&1 &
         echo "."
         ;;
   stop)
         echo -n "Stopping at daemon: atd"
-        start-stop-daemon --stop --quiet --oknodo --pidfile /var/run/atd.pid --exec /usr/sbin/atd
+        start-stop-daemon --stop --quiet --pidfile /var/run/atd.pid --exec /usr/sbin/atd
         rm -f /var/spool/at/trigger
         echo "."
         ;;
