--- SDL-1.2.7/configure.in~fixlibs	2005-10-06 10:27:59.025396872 +0200
+++ SDL-1.2.7/configure.in	2005-10-06 10:30:16.963087411 +0200
@@ -1508,8 +1508,8 @@
         AC_MSG_RESULT($video_qtopia)
         if test x$video_qtopia = xyes; then
             CFLAGS="$CFLAGS -DENABLE_QTOPIA -DQT_QWS_EBX -DQT_QWS_CUSTOM -DQWS -I${QPEDIR}/include -I${QTDIR}/include/ -DNO_DEBUG -fno-rtti -fno-exceptions"
-            SDL_LIBS="$SDL_LIBS -L${QPEDIR}/lib -L${QTDIR}/lib/ -lqpe -lqte"
-            SDL_CFLAGS="$SDL_CFLAGS -DQWS"	  
+            SDL_LIBS="$SDL_LIBS -Wl,-rpath-link -Wl,${QPEDIR}/lib -Wl,-rpath-link -Wl,${QTDIR}/lib"
+            SYSTEM_LIBS="$SYSTEM_LIBS -L${QPEDIR}/lib -L${QTDIR}/lib/ -lqpe -lqte"
             VIDEO_SUBDIRS="$VIDEO_SUBDIRS qtopia"
             VIDEO_DRIVERS="$VIDEO_DRIVERS qtopia/libvideo_qtopia.la"
         else
@@ -1803,11 +1803,6 @@
         CheckPTHREAD
         CheckSIGACTION
         CheckAltivec
-        # Set up files for the main() stub
-        if test "x$video_qtopia" = "xyes"; then
-          SDL_CFLAGS="$SDL_CFLAGS -Dmain=SDL_main"
-
-        fi
         # Set up files for the audio library
         # We use the OSS and ALSA API's, not the Sun audio API
         #if test x$enable_audio = xyes; then
