--- gaim.ori/src/gtkstatusbox.c	2005-09-04 20:42:29.000000000 +0200
+++ gaim/src/gtkstatusbox.c	2005-09-05 08:44:13.000000000 +0200
@@ -173,7 +173,7 @@
 
 	text_rend = gtk_cell_renderer_text_new();
 	icon_rend = gtk_cell_renderer_pixbuf_new();
-	icon_size = gtk_icon_size_from_name(GAIM_ICON_SIZE_STATUS);
+	icon_size = GTK_ICON_SIZE_SMALL_TOOLBAR;
 
 	status_box->imhtml_visible = FALSE;
 	status_box->error_pixbuf = gtk_widget_render_icon (GTK_WIDGET(status_box), GAIM_STOCK_STATUS_OFFLINE,
