Index: gaim/plugins/docklet/docklet-x11.c
===================================================================
--- gaim.orig/plugins/docklet/docklet-x11.c	2005-04-02 17:23:28.000000000 +0200
+++ gaim/plugins/docklet/docklet-x11.c	2005-08-21 02:06:51.869056336 +0200
@@ -112,7 +112,7 @@
 			break;
 	}
 
-	gtk_image_set_from_stock(GTK_IMAGE(image), icon_name, GTK_ICON_SIZE_LARGE_TOOLBAR);
+	gtk_image_set_from_stock(GTK_IMAGE(image), icon_name, GTK_ICON_SIZE_SMALL_TOOLBAR);
 
 #if 0
 	GdkPixbuf *p;
