Index: lxml-1.3.3/setupinfo.py
===================================================================
--- lxml-1.3.3.orig/setupinfo.py	2007-08-20 22:39:09.000000000 +0000
+++ lxml-1.3.3/setupinfo.py	2007-08-20 22:39:46.000000000 +0000
@@ -86,7 +86,7 @@
         return static_library_dirs
     # filter them from xslt-config --libs
     result = []
-    possible_library_dirs = flags('xslt-config --libs')
+    possible_library_dirs = flags('pkg-config libxslt --libs')
     for possible_library_dir in possible_library_dirs:
         if possible_library_dir.startswith('-L'):
             result.append(possible_library_dir[2:])
@@ -100,7 +100,7 @@
         return static_include_dirs
     # filter them from xslt-config --cflags
     result = []
-    possible_include_dirs = flags('xslt-config --cflags')
+    possible_include_dirs = flags('pkg-config libxslt --cflags')
     for possible_include_dir in possible_include_dirs:
         if possible_include_dir.startswith('-I'):
             result.append(possible_include_dir[2:])
@@ -119,7 +119,7 @@
         return result
 
     # anything from xslt-config --cflags that doesn't start with -I
-    possible_cflags = flags('xslt-config --cflags')
+    possible_cflags = flags('pkg-config libxslt --cflags')
     for possible_cflag in possible_cflags:
         if not possible_cflag.startswith('-I'):
             result.append(possible_cflag)
