
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- ecore/src/bin/ecore_test.c~no-x-test.patch
+++ ecore/src/bin/ecore_test.c
@@ -581,7 +581,7 @@
 
 #ifdef BUILD_ECORE_EVAS
 /* choose: TEST_X, TEST_FB */
-#define TEST_X 
+#define TEST_FB 
 
 /**** ECORE_EVAS TEST CODE */
 
--- ecore/src/bin/ecore_evas_test_app.c~no-x-test.patch
+++ ecore/src/bin/ecore_evas_test_app.c
@@ -21,33 +21,10 @@
    
    /* create an evas */
    if (!ecore_evas_init()) return -1;
-   if ((argc > 1) && (!strcmp(argv[1], "-fb")))
+     if ( argv > 1 )
      ee = ecore_evas_fb_new(NULL, 0,  240, 320);
-   else if ((argc > 1) && (!strcmp(argv[1], "-x")))
-     ee = ecore_evas_software_x11_new(NULL, 0,  0, 0, 240, 320);
-#if HAVE_ECORE_EVAS_GL
-   else if ((argc > 1) && (!strcmp(argv[1], "-gl")))
-     ee = ecore_evas_gl_x11_new(NULL, 0,  0, 0, 240, 320);
-#endif
-   else if ((argc > 1) && (!strcmp(argv[1], "-h")))
-     {
-	printf("%s -x         Test ecore_evas in X (default)\n"
-	       "%s -gl        Test ecore_evas in X GL\n"
-	       "%s -fb        Test ecore_evas in the Framebuffer\n"
-	       "%s -h         Display this help\n",
-	       argv[0], argv[0], argv[0], argv[0]);
-	ecore_evas_shutdown();
-	ecore_shutdown();
-	return 0;
-     }
-   else
-#ifdef BUILD_ECORE_X
-     ee = ecore_evas_software_x11_new(NULL, 0,  0, 0, 240, 320);	 
-#else
-#ifdef BUILD_ECORE_FB
+     else
      ee = ecore_evas_fb_new(NULL, 270,  240, 320);
-#endif   
-#endif   
    if (!ee) return -1;
    ecore_evas_callback_delete_request_set(ee, app_delete_request);
    ecore_evas_callback_resize_set(ee, app_resize);
