From a0c063716101ec93dcad30028ba41489926050c9 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Tue, 22 Dec 2009 21:35:00 +0100
Subject: [PATCH 1/2] omapfb-driver: hack out dpms for xorg 1.7.x

---
 src/omapfb-driver.c |   39 ---------------------------------------
 1 files changed, 0 insertions(+), 39 deletions(-)

diff --git a/src/omapfb-driver.c b/src/omapfb-driver.c
index a9d5fbc..36a0d53 100644
--- a/src/omapfb-driver.c
+++ b/src/omapfb-driver.c
@@ -41,9 +41,6 @@
 
 #include "exa.h"
 
-#define DPMS_SERVER
-#include <X11/extensions/dpms.h>
-
 #include <linux/fb.h>
 
 /* TODO: we'd like this to come from kernel headers, but that's not a good
@@ -580,9 +577,6 @@ OMAPFBScreenInit(int scrnIndex, ScreenPtr pScreen, int argc, char **argv)
 	/* NO-OP currently */
 	pScreen->SaveScreen = OMAPFBSaveScreen;
 	
-	/* Setup DPMS support */
-	xf86DPMSInit(pScreen, OMAPFBDPMSSet, 0);
-	
 #ifdef USE_EXA
 	/* EXA init */
 	xf86LoadSubModule(pScrn, "exa");
@@ -723,38 +717,6 @@ static Bool OMAPFBSwitchMode(int scrnIndex, DisplayModePtr mode, int flags)
 	return TRUE;
 }
 
-static void
-OMAPFBDPMSSet(ScrnInfoPtr pScrn, int mode, int flags)
-{
-	OMAPFBPtr ofb = OMAPFB(pScrn);
-
-	switch (mode) {
-		case DPMSModeOn:
-			if (ioctl(ofb->fd, FBIOBLANK, (void *)VESA_NO_BLANKING)) {
-				xf86DrvMsg(pScrn->scrnIndex, X_ERROR,
-				           "FBIOBLANK: %s\n", strerror(errno));
-			}
-			break;
-		case DPMSModeStandby:
-		case DPMSModeSuspend:
-			/* TODO: Maybe we would want to use the above modes for
-			 * dimming the LCD? That'd match the functionality
-			 * (save power)
-			 */
-		case DPMSModeOff:
-			/* OMAPFB only supports on and off */
-			if (ioctl(ofb->fd, FBIOBLANK, (void *)VESA_POWERDOWN)) {
-				xf86DrvMsg(pScrn->scrnIndex, X_ERROR,
-				           "FBIOBLANK: %s\n", strerror(errno));
-			}
-			break;
-			break;
-		default:
-			return;
-	}
-
-}
-
 void
 OMAPFBPrintCapabilities(ScrnInfoPtr pScrn,
                         struct omapfb_caps *caps,
@@ -879,7 +841,6 @@ OMAPFBSetup(pointer module, pointer opts, int *errmaj, int *errmin)
 	if (!setupDone) {
 		setupDone = TRUE;
 		xf86AddDriver(&OMAPFB, module, HaveDriverFuncs);
-		LoaderRefSymLists(fbSymbols, NULL);
 		return (pointer)1;
 	} else {
 		if (errmaj) *errmaj = LDR_ONCEONLY;
-- 
1.6.5

