Index: evas/configure.in
===================================================================
--- evas.orig/configure.in	2007-08-20 16:44:22.000000000 +0000
+++ evas/configure.in	2007-08-20 16:49:08.000000000 +0000
@@ -260,9 +260,7 @@
   AC_CHECK_HEADER(X11/X.h,
     [
       AC_DEFINE(BUILD_ENGINE_SOFTWARE_X11, 1, [Software X11 Rendering Backend])
-      x_dir=${x_dir:-/usr/X11R6}
-      x_cflags=${x_cflags:--I${x_includes:-$x_dir/include}}
-      x_libs="${x_libs:--L${x_libraries:-$x_dir/lib}} -lX11 -lXext"
+      x_libs="${x_libs} -lX11 -lXext"
     ],
     [
       AC_MSG_RESULT(disabling software X11 engine)
@@ -303,9 +301,7 @@
   AC_CHECK_HEADER(X11/X.h,
     [
       AC_DEFINE(BUILD_ENGINE_SOFTWARE_16_X11, 1, [Software 16bit X11 Rendering Backend])
-      x_dir=${x_dir:-/usr/X11R6}
-      x_cflags=${x_cflags:--I${x_includes:-$x_dir/include}}
-      x_libs="${x_libs:--L${x_libraries:-$x_dir/lib}} -lX11 -lXext"
+      x_libs="${x_libs} -lX11 -lXext"
     ],
     [
       AC_MSG_RESULT(disabling software 16bit X11 engine)
@@ -599,10 +595,7 @@
       [
         AC_DEFINE(BUILD_ENGINE_GL_X11, 1, [OpenGL X11 Rendering Backend])
         AC_DEFINE(BUILD_ENGINE_GL_COMMON, 1, [Generic OpenGL Rendering Support])
-        x_dir=${x_dir:-/usr/X11R6};
-        x_cflags=${x_cflags:--I$x_dir/include}
-        x_libs="${x_libs:--L$x_dir/lib -lX11 -lXext}"
-        gl_cflags="-I/usr/include"
+        x_libs="${x_libs} -lX11 -lXext"
         gl_libs="-lGL -lGLU -lpthread"
         gl_dir=""
       ],
@@ -694,9 +687,7 @@
       AC_DEFINE(BUILD_ENGINE_CAIRO_X11, 1, [Cairo X11 Rendering Backend])
       AM_CONDITIONAL(BUILD_ENGINE_CAIRO_COMMON, true)
       AC_DEFINE(BUILD_ENGINE_CAIRO_COMMON, 1, [Generic Cairo Rendering Support])
-      x_dir="/usr/X11R6";
-      x_cflags="-I"$x_dir"/include"
-      x_libs="-L"$x_dir"/lib -lX11 -lXext"
+      x_libs="${x_libs} -lX11 -lXext"
     ], [
       AM_CONDITIONAL(BUILD_ENGINE_CAIRO_X11, false)
       AM_CONDITIONAL(BUILD_ENGINE_CAIRO_COMMON, false)
@@ -738,9 +729,7 @@
       AC_CHECK_HEADER(X11/extensions/Xrender.h,
         [
           AC_DEFINE(BUILD_ENGINE_XRENDER_X11, 1, [XRender X11 Rendering Backend])
-          x_dir=${x_dir:-/usr/X11R6}
-          x_cflags=${x_cflags:--I${x_includes:-$x_dir/include}}
-          x_libs="${x_libs:--L${x_libraries:-$x_dir/lib}} -lX11 -lXext -lXrender"
+          x_libs="${x_libs} -lX11 -lXext -lXrender"
 	],
 	[
           AC_MSG_RESULT(disabling xrender X11 engine)
@@ -826,9 +815,7 @@
     [
       PKG_CHECK_MODULES(GLITZ, glitz glitz-glx,
 	[
-          x_dir=${x_dir:-/usr/X11R6}
-          x_cflags=${x_cflags:--I${x_includes:-$x_dir/include}}
-          x_libs="${x_libs:--L${x_libraries:-$x_dir/lib}} -lX11 -lXext"
+          x_libs="${x_libs} -lX11 -lXext"
           AC_DEFINE(BUILD_ENGINE_GLITZ_X11, 1, [Glitz X11 Rendering Backend])
         ],
         [
