diff --git a/src/modules/engines/fb/evas_fb_main.c b/src/modules/engines/fb/evas_fb_main.c
index e26c171..75e0743 100644
--- a/src/modules/engines/fb/evas_fb_main.c
+++ b/src/modules/engines/fb/evas_fb_main.c
@@ -530,7 +530,7 @@ fb_postinit(FB_Mode *mode)
       fb_cleanup();
       exit(1);
     }
-  mode->mem_offset = (unsigned)(fb_fix.smem_start) & (~PAGE_MASK);
+  mode->mem_offset = (unsigned)(fb_fix.smem_start) & (getpagesize()-1);
   mode->mem = (unsigned char *)mmap(NULL, fb_fix.smem_len + mode->mem_offset,
 				 PROT_WRITE | PROT_READ, MAP_SHARED, fb, 0);
   if ((int)mode->mem == -1)
