diff -uri files/orrery.c files.datadir/orrery.c
--- files/orrery.c	2009-01-07 11:28:32.000000000 +0100
+++ files.datadir/orrery.c	2009-02-18 19:08:44.887262837 +0100
@@ -70,7 +70,7 @@
 
 #define FULL_CIRCLE (23040) /* Full circle in calls to draw_arc */
 
-#define DATA_DIR    "/media/card/orrery"
+#define DATA_DIR    "/usr/share/orrery"
 #define N_PLANETS (9)
 #define N_SOLAR_SYSTEM_OBJECTS (10)
 
