Index: ntp-4.2.0/include/ntp_stdlib.h
===================================================================
--- ntp-4.2.0.orig/include/ntp_stdlib.h	2003-07-17 11:27:16.000000000 +0100
+++ ntp-4.2.0/include/ntp_stdlib.h	2005-11-26 18:39:14.000000000 +0000
@@ -133,10 +133,6 @@
 extern u_char *	cache_key;		/* key pointer */
 extern u_int	cache_keylen;		/* key length */
 
-/* clocktypes.c */
-struct clktype;
-extern struct clktype clktypes[];
-
 /* getopt.c */
 extern char *	ntp_optarg;		/* global argument pointer */
 extern int	ntp_optind;		/* global argv index */
Index: ntp-4.2.0/include/ntpd.h
===================================================================
--- ntp-4.2.0.orig/include/ntpd.h	2003-09-13 04:08:04.000000000 +0100
+++ ntp-4.2.0/include/ntpd.h	2005-11-26 18:38:45.000000000 +0000
@@ -226,8 +226,6 @@
 #endif
 
 /* ntp_control.c */
-struct ctl_trap;
-extern struct ctl_trap ctl_trap[];
 extern int	num_ctl_traps;
 extern keyid_t	ctl_auth_keyid;		/* keyid used for authenticating write requests */
 
Index: ntp-4.2.0/include/ntp_refclock.h
===================================================================
--- ntp-4.2.0.orig/include/ntp_refclock.h	2003-07-17 11:27:16.000000000 +0100
+++ ntp-4.2.0/include/ntp_refclock.h	2005-11-26 18:40:23.000000000 +0000
@@ -65,6 +65,7 @@
 	const char *clocktype;	/* long description */
 	const char *abbrev;	/* short description */
 };
+extern struct clktype clktypes[];
 
 /*
  * Configuration flag values
Index: ntp-4.2.0/include/ntp_control.h
===================================================================
--- ntp-4.2.0.orig/include/ntp_control.h	2003-08-14 08:31:47.000000000 +0100
+++ ntp-4.2.0/include/ntp_control.h	2005-11-26 18:41:22.000000000 +0000
@@ -266,6 +267,7 @@
 	u_char tr_flags;		/* trap flags */
 	u_char tr_version;		/* version number of trapper */
 };
+extern struct ctl_trap ctl_trap[];
 
 /*
  * Flag bits
