--- notez-1.1.0/mainWindow.cpp.orig	2002-11-17 19:43:34 +0100
+++ notez-1.1.0/mainWindow.cpp	2004-11-21 22:55:06 +0100
@@ -7,11 +7,11 @@
 
     // check if the directory plain exists, if not
     // create it
-    QString d1("Documents/text");
+    QString d1(QDir::homeDirPath() + "/Documents/text");
     QDir pd1(d1);
     if (!pd1.exists())
     {
-	QDir pd1("Documents");
+	QDir pd1(QDir::homeDirPath() + "/Documents");
 	if (!pd1.mkdir("text", FALSE))
 	{
 	    QMessageBox::critical( 0, "noteZ",
@@ -22,11 +22,11 @@
 	}
     }
     
-    QString d2("Documents/text/plain");
+    QString d2(QDir::homeDirPath() + "/Documents/text/plain");
     QDir pd2(d2);
     if (!pd2.exists())
     {
-	QDir pd2("Documents/text");
+	QDir pd2(QDir::homeDirPath() + "/Documents/text");
 	if (!pd2.mkdir("plain", FALSE))
 	{
 	    QMessageBox::critical( 0, "noteZ",
@@ -40,14 +40,14 @@
 
     unsigned int item, convertCount = 0;
     QString strippedName;
-    QDir moveDir("Documents/text");
+    QDir moveDir(QDir::homeDirPath() + "/Documents/text");
     moveDir.setFilter(QDir::Files);
     moveDir.setNameFilter("[0-9]*-[0-9]*-[0-9]*-[0-9]*-[0-9]*-[0-9]*.txt");
     
     for (item = 0; item < moveDir.count(); item++) {
 	strippedName = moveDir[item].left(moveDir[item].find("."));
-	QFile oldFile("Documents/text/" + moveDir[item]);
-	QFile newFile("Documents/text/plain/" + strippedName + ".ntz");
+	QFile oldFile(QDir::homeDirPath() + "/Documents/text/" + moveDir[item]);
+	QFile newFile(QDir::homeDirPath() + "/Documents/text/plain/" + strippedName + ".ntz");
 	if (!oldFile.open(IO_ReadOnly))
 	    QMessageBox::warning(0, "File i/o error", "Could not read\nDocuments/text/" + moveDir[item] + "\nfor conversion.");
 
@@ -104,7 +104,7 @@
 
       case QMessageBox::Yes:
 	this->selected = notesList->currentItem();
-	QFile::remove("Documents/text/plain/" + fileList[notesList->currentItem()]);
+	QFile::remove(QDir::homeDirPath() + "/Documents/text/plain/" + fileList[notesList->currentItem()]);
 	refreshList(notesList);
 	break;
 
@@ -146,7 +146,7 @@
   QDate today(QDate::currentDate());
   QTime now(QTime::currentTime());
   QString name;
-  this->editFile(name.sprintf("Documents/text/plain/%4i-%02i-%02i-%02i-%02i-%02i.ntz", today.year(), today.month(), today.day(), now.hour(), now.minute(), now.second()),
+  this->editFile(name.sprintf(QDir::homeDirPath() + "/Documents/text/plain/%4i-%02i-%02i-%02i-%02i-%02i.ntz", today.year(), today.month(), today.day(), now.hour(), now.minute(), now.second()),
 		 true );
 
 }
@@ -159,7 +159,7 @@
   if( notesList->count() > 0 )
     {
       this->selected = number;
-      this->editFile("Documents/text/plain/" + fileList[number], 
+      this->editFile(QDir::homeDirPath() + "/Documents/text/plain/" + fileList[number], 
 		     false);
     }
 
@@ -173,14 +173,14 @@
 
   liste->clear();
 
-  fileList.setPath("Documents/text/plain/");
+  fileList.setPath(QDir::homeDirPath() + "/Documents/text/plain/");
   fileList.setFilter(QDir::Files);
   fileList.setSorting(QDir::Name);
   fileList.setNameFilter("*.ntz");
 
   for (item = 0; item < fileList.count(); item++) {
 
-    QFile datei("Documents/text/plain/" + fileList[item]);
+    QFile datei(QDir::homeDirPath() + "/Documents/text/plain/" + fileList[item]);
 
     if (!datei.open(IO_ReadOnly)) {
 
