From aaf549ecacc05bd5ddd29bb345cae471ce3ab48c Mon Sep 17 00:00:00 2001
From: Marco Cavallini <m.cavallini@koansoftware.com>
Date: Thu, 21 Jan 2010 16:46:18 +0100
Subject: [PATCH] gui_x11.cpp: Load font fixed when 9x15 fails in GuiCalibratorX11


Signed-off-by: Marco Cavallini <m.cavallini@koansoftware.com>
---
 gui_x11.cpp |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/gui_x11.cpp b/gui_x11.cpp
index bc483cd..02ddc73 100644
--- a/gui_x11.cpp
+++ b/gui_x11.cpp
@@ -112,8 +112,11 @@ GuiCalibratorX11::GuiCalibratorX11(Calibrator* calibrator0)
     // Load font and get font information structure
     font_info = XLoadQueryFont(display, "9x15");
     if (font_info == NULL) {
-        XCloseDisplay(display);
-        throw std::runtime_error("Unable to open 9x15 font");
+        font_info = XLoadQueryFont(display, "fixed");
+        if (font_info == NULL) {
+            XCloseDisplay(display);
+            throw std::runtime_error("Unable to open 9x15 neither fixed font");
+        }
     }
 
     // Compute absolute circle centers
-- 
1.6.3.3

