Index: xf86-input-tslib-0.0.5/src/tslib.c
===================================================================
--- xf86-input-tslib-0.0.5.orig/src/tslib.c	2008-11-11 13:48:12.000000000 +0000
+++ xf86-input-tslib-0.0.5/src/tslib.c	2008-11-11 13:51:51.000000000 +0000
@@ -321,10 +321,10 @@
 	priv->screen_num = xf86SetIntOption(pInfo->options, "ScreenNumber", 0 );
 
 	priv->width = xf86SetIntOption(pInfo->options, "Width", 0);
-	if (priv->width <= 0)	priv->width = DEFAULT_WIDTH;
+	if (priv->width <= 0)	priv->width = screenInfo.screens[0]->width;
 
 	priv->height = xf86SetIntOption(pInfo->options, "Height", 0);
-	if (priv->height <= 0)	priv->height = DEFAULT_HEIGHT;
+	if (priv->height <= 0)	priv->height = screenInfo.screens[0]->height;
 
 	s = xf86SetStrOption(pInfo->options, "Rotate", 0);
 	if (s > 0) {
