--- icewm-1.2.26/configure.in.orig	2006-07-18 15:16:37.024817360 -0600
+++ icewm-1.2.26/configure.in	2006-07-18 15:17:32.889324664 -0600
@@ -426,14 +426,11 @@
      AC_MSG_ERROR("xfreetype or core fonts must be enabled")
   fi
   if test "$enable_xfreetype" != "no" -o "$enable_xfreetype" = "implied"; then
-    AC_PATH_PROG(XFT_CONFIG, xft-config,, ${with_xft_arg-${PATH}})
-    if test "${XFT_CONFIG}" == ""; then
-      AC_PATH_PROG(PKG_CONFIG, pkg-config)
-      if test "${PKG_CONFIG}" != ""; then
-        ${PKG_CONFIG} xft 2>/dev/null
-        if test $? -eq 0 ; then
-          XFT_CONFIG='pkg-config xft'
-        fi
+    AC_PATH_PROG(PKG_CONFIG, pkg-config)
+    if test "${PKG_CONFIG}" != ""; then
+      ${PKG_CONFIG} xft 2>/dev/null
+      if test $? -eq 0 ; then
+        XFT_CONFIG='pkg-config xft'
       fi
     fi
     if test "${XFT_CONFIG}" != ""; then
