--- ./src/Makefile.in	2005-02-01 21:56:08.000000000 +0100
+++ ./src/Makefile.in	2005-02-01 21:55:12.000000000 +0100
@@ -206,10 +206,18 @@
 %.o: %.cc
 	@echo "  CXX   " $@
 	@$(CXX) $(CXXFLAGS) $(GCCDEP) -c $<
-	
-genpref $(BINARIES):
+
+genpref.o: genpref.cc
+	@echo "  BUILD_CXX   " $@
+	@$(BUILD_CXX) $(BUILD_CXXFLAGS) $(DEBUG) $(DEFS) $(GCCDEP) -c $<
+
+$(BINARIES):
 	@echo "  LD    " $@
 	@$(LD) -o $@$(EXEEXT) $($@_OBJS) $(LFLAGS) $($@_LFLAGS) $(LIBS) $($@_LIBS)
+	
+genpref:
+	@echo "  BUILD_CXX    " $@
+	@$(BUILD_CXX) $(BUILD_LDFLAGS) -o $@$(EXEEXT) $($@_OBJS) $(LIBS) $($@_LIBS)
 
 ################################################################################
 
