
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- xine-lib-1-rc7/src/libvorbis/xine_decoder.c~fix-syntax
+++ xine-lib-1-rc7/src/libvorbis/xine_decoder.c
@@ -218,7 +218,7 @@
     float **pcm;
     int samples;
 
-    if(vorbis_synthesis(&this->vb,&this->op)==0) 
+    if(vorbis_synthesis(&this->vb,&this->op,1)==0) 
       vorbis_synthesis_blockin(&this->vd,&this->vb);
 
     if (buf->pts!=0)
