--- m4/ticonv.m4~	2007-10-30 13:57:32.000000000 +0100
+++ trunk/m4/ticonv.m4	2007-10-30 13:57:32.000000000 +0100
@@ -76,15 +76,6 @@
 
 CFLAGS="$CFLAGS -I$srcdir"
 
-AC_MSG_CHECKING(preferred charset formats for system iconv)
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#define CONFIGURE_IN
-#include "iconv-detect.c"
-]])],[
-	AC_MSG_RESULT(found)
-],[
-	AC_MSG_RESULT(not found)
-],[])
 
 CFLAGS="$save_CFLAGS"
 LIBS="$save_LIBS"
