--- zd1211-driver-r52/src/zdhci.c.orig	2006-02-09 02:47:21.000000000 +0100
+++ zd1211-driver-r52/src/zdhci.c	2006-02-09 02:48:27.000000000 +0100
@@ -892,6 +892,7 @@
                         //check capability ...
                         if (cap & CAP_PRIVACY) {
                                 if (!mPrivacyInvoked)
+                                if (!mPrivacyInvoked && !mProbeWithSsid)
                                         continue;
                         } else {
                                 if (mPrivacyInvoked)
