--- /tmp/Xserver	2006-01-15 15:13:24.000000000 +0100
+++ xserver-common-1.2/X11/Xserver	2006-01-15 15:14:00.000000000 +0100
@@ -44,12 +44,12 @@
 case `module_id` in
 	"HP iPAQ H3100" | "HP iPAQ H3800")
 		ARGS="$ARGS -dpi 100 -screen 320x240@90 -rgba vrgb" ;;
-	"HP iPAQ H3600" | "HP -dpi 100 iPAQ H3700" | "HP iPAQ H3900")
-		ARGS="$ARGS -screen 320x240@270 -rgba vbgr" ;;
+	"HP iPAQ H3600" | "HP iPAQ H3700" | "HP iPAQ H3900")
+		ARGS="$ARGS -dpi 100 -screen 320x240@270 -rgba vbgr" ;;
 	"HP iPAQ H5400" | "HP iPAQ H2200")
 		ARGS="$ARGS -dpi 100 -rgba rgb" ;;
         "HP iPAQ HX4700")
-                ARGS="$ARGS -dpi 200" ;;
+                ARGS="$ARGS -dpi 100" ;;
 	"Ramses")
 		# What is this "vt2" in aid of?
 		ARGS="$ARGS -dpi 100 -screen 320x240@90 -rgba vrgb vt2" ;;
@@ -63,13 +63,13 @@
 		killall -9 apmd
 		;;
 	"SHARP Shepherd" | "SHARP Husky" | "SHARP Corgi")
-		ARGS="$ARGS -dpi 200 -rgba rgb" ;;
+		ARGS="$ARGS -dpi 100 -rgba rgb" ;;
 	"SHARP Spitz" | "SHARP Akita" | "SHARP Borzoi")
-		ARGS="$ARGS -dpi 200 -rgba rgb -screen 480x640@270" ;;
+		ARGS="$ARGS -dpi 100 -rgba rgb -screen 480x640@270" ;;
 	"Simpad")
 		ARGS="$ARGS -dpi 100 -rgba rgb" ;;
 	"Generic OMAP1510/1610/1710")
-		ARGS="$ARGS -dpi 220 -mouse /dev/input/event0" ;;
+		ARGS="$ARGS -dpi 100 -mouse /dev/input/event0" ;;
 esac
 
 exec $XSERVER $ARGS $*
