diff -Nur xserver-common-1.1~/X11/Xinit.d/11zaurus xserver-common-1.1/X11/Xinit.d/11zaurus
--- xserver-common-1.1~/X11/Xinit.d/11zaurus	2005-08-29 03:52:46.000000000 -0700
+++ xserver-common-1.1/X11/Xinit.d/11zaurus	2005-09-12 11:03:37.000000000 -0700
@@ -5,8 +5,19 @@
    exit 0
 fi
 
+module_id() {
+    # Get model name
+    echo `grep "^Hardware" /proc/cpuinfo | sed -e "s/.*: *//" | tr a-z A-Z`
+}
+
 chkhinge -e
 if [ $? = 12 ]; then
-   xrandr -o right
+   case `module_id` in
+   	*SPITZ | *AKITA | *BORZOI)
+   		DIRECTION="left" ;;
+        default)
+                DIRECTION="right" ;;
+   esac
+   xrandr -o $DIRECTION
 fi
 
