diff -Nur xserver-common-1.1~/Makefile xserver-common-1.1/Makefile
--- xserver-common-1.1~/Makefile	2005-08-31 13:04:40.000000000 -0700
+++ xserver-common-1.1/Makefile	2005-08-31 13:05:33.000000000 -0700
@@ -17,7 +17,6 @@
 install-program: 
 	for i in X11/Xsession.d X11/Xinit.d; do install -d $(DESTDIR)/etc/$$i; FILES=`echo $$i/* | sed "s:$$i/CVS::"`; install -m 755 $$FILES $(DESTDIR)/etc/$$i/; done
 	install -d $(DESTDIR)$(PREFIX)/bin
-	install -m 755 run-calibrate.sh $(DESTDIR)$(PREFIX)/bin/run-calibrate.sh
 	install -m 644 X11/Xdefaults $(DESTDIR)/etc/X11/Xdefaults
 	install -m 755 X11/Xinit $(DESTDIR)/etc/X11/Xinit
 	install -m 755 X11/Xserver $(DESTDIR)/etc/X11/Xserver
diff -Nur xserver-common-1.1~/X11/Xinit.d/30xTs_Calibrate xserver-common-1.1/X11/Xinit.d/30xTs_Calibrate
--- xserver-common-1.1~/X11/Xinit.d/30xTs_Calibrate	2005-08-31 13:04:40.000000000 -0700
+++ xserver-common-1.1/X11/Xinit.d/30xTs_Calibrate	2005-08-31 13:05:18.000000000 -0700
@@ -1,5 +1,5 @@
 #!/bin/sh
 
 if [ ! -f /etc/pointercal ]; then
-    /usr/bin/run-calibrate.sh
+    xtscal
 fi
