diff -Nur -x '*.orig' -x '*~' nautilus-cd-burner-2.21.5/src/nautilus-burn-recorder.c nautilus-cd-burner-2.21.5.new/src/nautilus-burn-recorder.c
--- nautilus-cd-burner-2.21.5/src/nautilus-burn-recorder.c	2008-01-21 16:16:16.000000000 +0000
+++ nautilus-cd-burner-2.21.5.new/src/nautilus-burn-recorder.c	2008-01-22 09:43:03.000000000 +0000
@@ -374,7 +374,7 @@
         int                   tmp;
 
         if (line && process->debug) {
-                g_print ("cdrecord stdout: %s", line);
+                g_print ("wodim stdout: %s", line);
         }
 
         if (sscanf (line, "Track %2u: %d of %d MB written (fifo %d%%) [buf %d%%] %fx.",
@@ -482,7 +482,7 @@
         NautilusBurnRecorder *recorder = data;
 
         if (line && process->debug) {
-                g_print ("cdrecord blank stdout: %s", line);
+                g_print ("wodim blank stdout: %s", line);
         }
 
         if (g_str_has_prefix (line, "Re-load disk and hit <CR>") ||
@@ -674,7 +674,7 @@
         NautilusBurnRecorder *recorder = data;
 
         if (line && process->debug) {
-                g_print ("cdrecord stderr: %s", line);
+                g_print ("wodim stderr: %s", line);
         }
 
         if (strstr (line, "No disk / Wrong disk!") != NULL) {
@@ -1381,7 +1381,7 @@
                 err_line_func = cdrdao_stderr_line;
         } else {
                 argv = g_ptr_array_new ();
-                g_ptr_array_add (argv, "cdrecord");
+                g_ptr_array_add (argv, "wodim");
 
                 g_ptr_array_add (argv, "fs=16m");
 
@@ -1498,7 +1498,7 @@
         }
 
         argv = g_ptr_array_new ();
-        g_ptr_array_add (argv, "cdrecord");
+        g_ptr_array_add (argv, "wodim");
 
         dev_str = g_strdup_printf ("dev=%s", nautilus_burn_drive_get_device (drive));
         g_ptr_array_add (argv, dev_str);
