
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- engage/src/module/Makefile.am~no-local-includes
+++ engage/src/module/Makefile.am
@@ -10,8 +10,7 @@
 
 # the module .so file
 INCLUDES               = -I. \
-			 -I$(top_srcdir) \
-			 -I$(includedir) \
+			 -I$(top_srcdir) \
 			 @e_cflags@
 pkgdir                 = @e_modules@/engage/$(MODULE_ARCH)
 pkg_LTLIBRARIES        = module.la
