--- ./src/Makefile.am.old
+++ ./src/Makefile.am
@@ -8,7 +8,9 @@
 mpeg2dec_SOURCES = mpeg2dec.c dump_state.c getopt.c gettimeofday.c
 mpeg2dec_LDADD = $(libvo) $(libmpeg2) $(libmpeg2convert)
 extract_mpeg2_SOURCES = extract_mpeg2.c getopt.c
+extract_mpeg2_LDADD = $(libvo) $(libmpeg2) $(libmpeg2convert)
 corrupt_mpeg2_SOURCES = corrupt_mpeg2.c getopt.c
+corrupt_mpeg2_LDADD = $(libvo) $(libmpeg2) $(libmpeg2convert)
 
 man_MANS = mpeg2dec.1 extract_mpeg2.1
 
