Index: xf86-video-fbdev-0.4.0/src/fbdev.c
===================================================================
--- xf86-video-fbdev-0.4.0.orig/src/fbdev.c
+++ xf86-video-fbdev-0.4.0/src/fbdev.c
@@ -10,29 +10,29 @@
 #include <string.h>
 
 /* all driver need this */
-#include "xf86.h"
-#include "xf86_OSproc.h"
+#include <xf86.h>
+#include <xf86_OSproc.h>
 
-#include "mipointer.h"
-#include "mibstore.h"
-#include "micmap.h"
-#include "colormapst.h"
-#include "xf86cmap.h"
+#include <mipointer.h>
+#include <mibstore.h>
+#include <micmap.h>
+#include <colormapst.h>
+#include <xf86cmap.h>
 #include "shadow.h"
-#include "dgaproc.h"
+#include <dgaproc.h>
 
 /* for visuals */
-#include "fb.h"
+#include <fb.h>
 #ifdef USE_AFB
 #include "afb.h"
 #endif
 
-#include "xf86Resources.h"
-#include "xf86RAC.h"
+#include <xf86Resources.h>
+#include <xf86RAC.h>
 
-#include "fbdevhw.h"
+#include <fbdevhw.h>
 
-#include "xf86xv.h"
+#include <xf86xv.h>
 
 #ifdef XSERVER_LIBPCIACCESS
 #include <pciaccess.h>

