diff -rdup trunk.oorig/libstdc++-v3/config/locale/generic/c_locale.h trunk/libstdc++-v3/config/locale/generic/c_locale.h
--- trunk.oorig/libstdc++-v3/config/locale/generic/c_locale.h	2011-01-31 11:39:33.000000000 +0100
+++ trunk/libstdc++-v3/config/locale/generic/c_locale.h	2011-01-31 17:40:14.449833043 +0100
@@ -46,8 +46,11 @@
 namespace std _GLIBCXX_VISIBILITY(default)
 {
 _GLIBCXX_BEGIN_NAMESPACE_VERSION
-
+#ifdef __UCLIBC__
+  typedef __ctype_touplow_t*	__c_locale;
+#else
   typedef int*			__c_locale;
+#endif
 
   // Convert numeric value of type double and long double to string and
   // return length of string.  If vsnprintf is available use it, otherwise
diff -rdup trunk.oorig/libstdc++-v3/config/os/gnu-linux/ctype_base.h trunk/libstdc++-v3/config/os/gnu-linux/ctype_base.h
--- trunk.oorig/libstdc++-v3/config/os/gnu-linux/ctype_base.h	2011-01-31 11:39:33.000000000 +0100
+++ trunk/libstdc++-v3/config/os/gnu-linux/ctype_base.h	2011-01-31 17:25:54.959690407 +0100
@@ -42,7 +42,11 @@ _GLIBCXX_BEGIN_NAMESPACE_VERSION
   struct ctype_base
   {
     // Non-standard typedefs.
+#ifdef __UCLIBC__
+    typedef const __ctype_touplow_t*	__to_type;
+#else
     typedef const int* 		__to_type;
+#endif
 
     // NB: Offsets into ctype<char>::_M_table force a particular size
     // on the mask type. Because of this, we don't use an enum.
diff -rdup trunk.oorig/libstdc++-v3/include/c/clocale trunk/libstdc++-v3/include/c/clocale
--- trunk.oorig/libstdc++-v3/include/c/clocale	2011-01-04 13:09:36.000000000 +0100
+++ trunk/libstdc++-v3/include/c/clocale	2011-01-31 17:25:54.959690407 +0100
@@ -32,5 +32,8 @@
 #pragma GCC system_header
 
 #include_next <locale.h>
+#ifdef __UCLIBC__
+#include <ctype.h>
+#endif
 
 #endif
diff -rdup trunk.oorig/libstdc++-v3/include/c_global/clocale trunk/libstdc++-v3/include/c_global/clocale
--- trunk.oorig/libstdc++-v3/include/c_global/clocale	2011-01-31 11:39:26.000000000 +0100
+++ trunk/libstdc++-v3/include/c_global/clocale	2011-01-31 17:25:54.959690407 +0100
@@ -42,6 +42,9 @@
 
 #include <bits/c++config.h>
 #include <locale.h>
+#ifdef __UCLIBC__
+#include <ctype.h>
+#endif
 
 #ifndef _GLIBCXX_CLOCALE
 #define _GLIBCXX_CLOCALE 1
diff -rdup trunk.oorig/libstdc++-v3/include/c_std/clocale trunk/libstdc++-v3/include/c_std/clocale
--- trunk.oorig/libstdc++-v3/include/c_std/clocale	2011-01-31 11:39:24.000000000 +0100
+++ trunk/libstdc++-v3/include/c_std/clocale	2011-01-31 17:25:54.959690407 +0100
@@ -45,6 +45,10 @@
 #include <bits/c++config.h>
 #include <locale.h>
 
+#ifdef __UCLIBC__
+#include <ctype.h>
+#endif
+
 // Get rid of those macros defined in <locale.h> in lieu of real functions.
 #undef setlocale
 #undef localeconv
