
#
# Signed off by Michael 'Mickey' Lauer <mickey@Vanille.de>
#

--- qt-x11-opensource-src-4.1.2/src/corelib/arch/arm/arch/qatomic.h~fix-asm-constraints
+++ qt-x11-opensource-src-4.1.2/src/corelib/arch/arm/arch/qatomic.h
@@ -32,7 +32,7 @@
 {
     register int ret;
     asm volatile("swpb %0,%1,[%2]"
-                 : "=r"(ret)
+                 : "&=r"(ret)
                  : "r"(newval), "r"(ptr)
                  : "cc", "memory");
     return ret;
