---busybox-1.2.2/util-linux/umount.c~	2006-11-04 17:59:20.000000000 -0600
+++ busybox-1.2.2/util-linux/umount.c	2006-11-04 18:03:55.000000000 -0600
@@ -16,13 +16,14 @@
 #include <errno.h>
 #include <getopt.h>
 
-#define OPTION_STRING		"flDnravd"
+#define OPTION_STRING		"flDnravdL"
 #define OPT_FORCE			1
 #define OPT_LAZY			2
 #define OPT_DONTFREELOOP	4
 #define OPT_NO_MTAB			8
 #define OPT_REMOUNT			16
 #define OPT_ALL				(ENABLE_FEATURE_UMOUNT_ALL ? 32 : 0)
+#define OPT_LINKSYS_LAZY		256
 
 int umount_main(int argc, char **argv)
 {
@@ -45,6 +46,8 @@
 	argc -= optind;
 	argv += optind;
 
+	if (opt & OPT_LINKSYS_LAZY) opt |= OPT_LAZY; /* -L same as -l */
+
 	doForce = MAX((opt & OPT_FORCE), (opt & OPT_LAZY));
 
 	/* Get a list of mount points from mtab.  We read them all in now mostly
