--- xserver-common-1.7/X11/Xserver.orig	2006-02-16 12:11:22.316078680 -0800
+++ xserver-common-1.7/X11/Xserver	2006-02-16 12:11:53.681310440 -0800
@@ -49,7 +49,7 @@ case `module_id` in
 	"HP iPAQ H5400" | "HP iPAQ H2200")
 		ARGS="$ARGS -dpi 100 -rgba rgb" ;;
         "HP iPAQ HX4700")
-                ARGS="$ARGS -dpi 200" ;;
+                ARGS="$ARGS -dpi 100" ;;
 	"Ramses")
 		# What is this "vt2" in aid of?
 		ARGS="$ARGS -dpi 100 -rgba vrgb -screen 320x240@90 vt2" ;;
@@ -60,9 +60,9 @@ case `module_id` in
 		ARGS="$ARGS -dpi 100 -rgba vrgb -screen 320x240@270"
 		;;
 	"SHARP Shepherd" | "SHARP Husky" | "SHARP Corgi")
-		ARGS="$ARGS -dpi 200 -rgba rgb" ;;
+		ARGS="$ARGS -dpi 100 -rgba rgb" ;;
 	"SHARP Spitz" | "SHARP Akita" | "SHARP Borzoi")
-		ARGS="$ARGS -dpi 200 -rgba rgb -screen 480x640@270" ;;
+		ARGS="$ARGS -dpi 100 -rgba rgb -screen 480x640@270" ;;
 	"Simpad")
 		ARGS="$ARGS -dpi 100 -rgba rgb" ;;
 	"Generic OMAP1510/1610/1710")
