Index: include/lib/dvb/edvb.h
===================================================================
RCS file: /cvs/tuxbox/apps/tuxbox/enigma/include/lib/dvb/edvb.h,v
retrieving revision 1.37
diff -u -r1.37 edvb.h
--- a/include/lib/dvb/edvb.h	12 Jun 2005 19:55:44 -0000	1.37
+++ b/include/lib/dvb/edvb.h	1 Mar 2006 23:10:04 -0000
@@ -196,6 +196,7 @@
 //////////////////////////////Network Stuff////////////////////////////////////
 public:
 #ifndef DISABLE_NETWORK
+	void applyNetworkFix();
 	void configureNetwork();
 	void UDHCPC_DataAvail(eString);
 	void UDHCPC_Closed(int);
Index: lib/dvb/edvb.cpp
===================================================================
RCS file: /cvs/tuxbox/apps/tuxbox/enigma/lib/dvb/edvb.cpp,v
retrieving revision 1.104
diff -u -r1.104 edvb.cpp
--- a/lib/dvb/edvb.cpp	24 Aug 2005 19:07:40 -0000	1.104
+++ b/lib/dvb/edvb.cpp	1 Mar 2006 23:10:05 -0000
@@ -80,6 +80,7 @@
 {
 #ifndef DISABLE_NETWORK
 	CONNECT(delayTimer.timeout, eDVB::restartSamba);
+	applyNetworkFix();
 #ifndef DISABLE_NFS
 	CONNECT(delayTimer.timeout, eDVB::doMounts);
 #endif
@@ -773,6 +774,28 @@
 	}
 }
 
+void eDVB::applyNetworkFix()
+{
+	if ( eSystemInfo::getInstance()->getHwType() == eSystemInfo::DM7020 )
+	{
+		int networkfix=0;
+		eConfig::getInstance()->getKey("/elitedvb/network/fix", networkfix);
+		struct stat s;
+		if ( !stat("/proc/net/dream_mode", &s) )
+		{
+			FILE *f = fopen("/proc/net/dream_mode", "w");
+			if (f)
+			{
+				const char *tmp = networkfix ? "1" : "0";
+				fwrite(tmp, 2, 1, f);
+				fclose(f);
+			}
+			else
+				eDebug("couldn't open /proc/net/dream_mode");
+		}
+	}
+}
+
 void eDVB::configureNetwork()
 {
 #ifndef USE_IFUPDOWN
Index: po/de.po
===================================================================
RCS file: /cvs/tuxbox/apps/tuxbox/enigma/po/de.po,v
retrieving revision 1.85
diff -u -r1.85 de.po
--- a/po/de.po	25 Feb 2006 21:05:15 -0000	1.85
+++ b/po/de.po	1 Mar 2006 23:10:11 -0000
@@ -4135,6 +4135,14 @@
 msgid "try to reconnect when an external cahandler connection was lost"
 msgstr "Versucht die CAHandler Verbindung, wenn verloren, wiederherzustellen."
 
+#: src/setup_extra.cpp:126
+msgid "Enable network fix"
+msgstr "Netzwerk 'fix' aktivieren"
+
+#: src/setup_extra.cpp:126
+msgid "enable this when your network is very slow(for testing)"
+msgstr "anhaken, wenn das Netzwerk extrem langsam ist(zum testen)"
+
 #: src/setup_extra.cpp:127
 msgid "Enable file sharing"
 msgstr "Dateifreigaben aktivieren"
Index: src/setup_extra.cpp
===================================================================
RCS file: /cvs/tuxbox/apps/tuxbox/enigma/src/setup_extra.cpp,v
retrieving revision 1.61
diff -u -r1.61 setup_extra.cpp
--- a/src/setup_extra.cpp	5 Feb 2006 23:41:01 -0000	1.61
+++ b/src/setup_extra.cpp	1 Mar 2006 23:10:12 -0000
@@ -122,8 +122,9 @@
 	new eListBoxEntryCheck(&list, _("Auto bouquet change"), "/elitedvb/extra/autobouquetchange", _("change into next bouquet when end of current bouquet is reached"));
 	new eListBoxEntryCheck(&list, _("Auto reconnect cahandler"), "/elitedvb/extra/cahandlerReconnect", _("try to reconnect when an external cahandler connection was lost"));
 #ifndef DISABLE_NETWORK
-	if ( eSystemInfo::getInstance()->getHwType() == eSystemInfo::DM7000 ||
-	    eSystemInfo::getInstance()->getHwType() == eSystemInfo::DM7020)
+	if ( eSystemInfo::getInstance()->getHwType() == eSystemInfo::DM7020)
+		CONNECT((new eListBoxEntryCheck(&list, _("Enable network fix"), "/elitedvb/network/fix", _("enable this when your network is very slow(for testing)")))->selected, eExpertSetup::networkFixChanged );
+	if ( eSystemInfo::getInstance()->getHwType() == eSystemInfo::DM7000 )
 		new eListBoxEntryCheck(&list, _("Enable file sharing"), "/elitedvb/network/samba", _("start file sharing(samba) on startup"));
 #endif
 #ifndef TUXTXT_CFG_STANDALONE
@@ -251,6 +252,11 @@
 	eZap::getInstance()->reconfigureHTTPServer();
 }
 
+void eExpertSetup::networkFixChanged(bool b)
+{
+	eDVB::getInstance()->applyNetworkFix();
+}
+
 void eExpertSetup::fastZappingChanged(bool b)
 {
 	Decoder::setFastZap(b);
Index: src/setup_extra.h
===================================================================
RCS file: /cvs/tuxbox/apps/tuxbox/enigma/src/setup_extra.h,v
retrieving revision 1.15
diff -u -r1.15 setup_extra.h
--- a/src/setup_extra.h	3 Nov 2005 12:31:31 -0000	1.15
+++ b/src/setup_extra.h	1 Mar 2006 23:10:12 -0000
@@ -20,6 +20,7 @@
 #ifndef TUXTXT_CFG_STANDALONE
 	void tuxtxtCachingChanged(bool);
 #endif
+	void networkFixChanged(bool);
 	void colorbuttonsChanged(bool);
 	void reinitializeHTTPServer(bool);
 	void fastZappingChanged(bool b);
