--- /codecs/orig-Makefile	2005-11-29 13:24:39.000000000 -0500
+++ /codecs/Makefile	2007-02-28 09:54:42.000000000 -0500
@@ -29,7 +29,7 @@
 endif
 
 UI_SPEEX=$(wildcard $(CROSS_COMPILE_TARGET)/usr/include/speex.h)
-UIS_SPEEX=$(wildcard $(CROSS_COMPILE_TARGET)/usr/include/speex/speex.h)
+UIS_SPEEX=$(wildcard $(CROSS_COMPILE_TARGET)/include/speex/speex.h)
 ULI_SPEEX=$(wildcard $(CROSS_COMPILE_TARGET)/usr/local/include/speex.h)
 ULIS_SPEEX=$(wildcard $(CROSS_COMPILE_TARGET)/usr/local/include/speex/speex.h)
 ifneq (${UI_SPEEX},)
@@ -38,7 +38,8 @@
 endif
 ifneq (${UIS_SPEEX},)
   MODSPEEX=codec_speex.so
-  CFLAGS+=-I$(CROSS_COMPILE_TARGET)/usr/include/speex
+  CFLAGS+=-I$(CROSS_COMPILE_TARGET)/include/speex
+  LIBSPEEX=-L$(CROSS_COMPILE_TARGET)/lib
   LIBSPEEX+=-lspeex -lm
 endif
 ifneq (${ULI_SPEEX},)
