Index: coreutils-6.0/src/install.c
===================================================================
--- coreutils-6.0.orig/src/install.c	2009-03-04 15:37:45.000000000 -0800
+++ coreutils-6.0/src/install.c	2009-03-04 15:38:57.000000000 -0800
@@ -526,7 +526,14 @@
 strip (char const *name)
 {
   int status;
-  pid_t pid = fork ();
+  pid_t pid;
+  char *strip_name;
+
+  strip_name = getenv ("STRIP");
+  if (strip_name == NULL)
+    strip_name = "strip";
+
+  pid = fork ();
 
   switch (pid)
     {
@@ -534,7 +541,7 @@
       error (EXIT_FAILURE, errno, _("fork system call failed"));
       break;
     case 0:			/* Child. */
-      execlp ("strip", "strip", name, NULL);
+      execlp (strip_name, "strip", name, NULL);
       error (EXIT_FAILURE, errno, _("cannot run strip"));
       break;
     default:			/* Parent. */
