
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- xmms-1.2.10/General/ir/ir.c~gcc4
+++ xmms-1.2.10/General/ir/ir.c
@@ -16,10 +16,10 @@
 #include "ir.h"
 
 /* Important stuff to know */
-static gboolean keepGoing = FALSE;
+extern gboolean keepGoing = FALSE;
 
 /* The thread handle */
-static pthread_t irapp_thread;
+extern pthread_t irapp_thread;
 
 /* Declarations for calls that we need to mention in the plugin struct */
 static void init(void);
