---
 drivers/video/sidsafb.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

Index: linux-2.6.18-avr32/drivers/video/sidsafb.c
===================================================================
--- linux-2.6.18-avr32.orig/drivers/video/sidsafb.c	2006-10-11 15:14:06.000000000 +0200
+++ linux-2.6.18-avr32/drivers/video/sidsafb.c	2006-10-18 14:51:31.000000000 +0200
@@ -300,8 +300,11 @@ static int sidsafb_check_var(struct fb_v
 		var->green.length = 5;
 		var->blue.length = 5;
 		break;
-	case 24:
 	case 32:
+		var->transp.offset = 24;
+		var->transp.length = 8;
+		/* fall through */
+	case 24:
 		var->red.offset = 16;
 		var->green.offset = 8;
 		var->blue.offset = 0;
