Index: classpath-0.97.2/configure.ac
===================================================================
--- classpath-0.97.2.orig/configure.ac	2008-06-06 02:58:57.000000000 +0200
+++ classpath-0.97.2/configure.ac	2008-08-02 00:58:16.537364104 +0200
@@ -41,6 +41,8 @@
 AC_CONFIG_HEADERS([include/config.h])
 AC_PREFIX_DEFAULT(/usr/local/classpath)
 
+AC_CONFIG_MACRO_DIR([m4])
+
 dnl -----------------------------------------------------------
 dnl Enable collections.jar (disabled by default)
 dnl -----------------------------------------------------------
