Index: libjingle-0.3.10/talk/base/asynctcpsocket.cc
===================================================================
--- libjingle-0.3.10.orig/talk/base/asynctcpsocket.cc	2006-09-01 04:31:52.000000000 -0700
+++ libjingle-0.3.10/talk/base/asynctcpsocket.cc	2010-09-15 12:57:34.485210001 -0700
@@ -45,6 +45,7 @@ extern "C" {
 }
 #endif // POSIX
 
+#include <cstring>
 namespace cricket {
 
 const size_t MAX_PACKET_SIZE = 64 * 1024;
Index: libjingle-0.3.10/talk/base/bytebuffer.cc
===================================================================
--- libjingle-0.3.10.orig/talk/base/bytebuffer.cc	2006-09-01 04:31:53.000000000 -0700
+++ libjingle-0.3.10/talk/base/bytebuffer.cc	2010-09-15 12:52:40.005210002 -0700
@@ -30,6 +30,7 @@
 #include "talk/base/byteorder.h"
 #include <algorithm>
 #include <cassert>
+#include <cstring>
 
 #if defined(_MSC_VER) && _MSC_VER < 1300
 namespace std {
Index: libjingle-0.3.10/talk/base/host.cc
===================================================================
--- libjingle-0.3.10.orig/talk/base/host.cc	2006-09-01 04:31:53.000000000 -0700
+++ libjingle-0.3.10/talk/base/host.cc	2010-09-15 13:08:51.945210000 -0700
@@ -33,6 +33,7 @@
 #include <iostream>
 #include <cassert>
 #include <errno.h>
+#include <cstdlib>
 
 #if defined(_MSC_VER) && _MSC_VER < 1300
 namespace std {
Index: libjingle-0.3.10/talk/base/logging.cc
===================================================================
--- libjingle-0.3.10.orig/talk/base/logging.cc	2006-09-01 04:31:52.000000000 -0700
+++ libjingle-0.3.10/talk/base/logging.cc	2010-09-15 13:14:46.475210002 -0700
@@ -33,6 +33,8 @@
 
 #include <iostream>
 #include <iomanip>
+#include <cstring>
+#include <cstdio>
 
 #include "talk/base/logging.h"
 #include "talk/base/stream.h"
Index: libjingle-0.3.10/talk/base/messagequeue.h
===================================================================
--- libjingle-0.3.10.orig/talk/base/messagequeue.h	2010-09-15 12:48:43.000000000 -0700
+++ libjingle-0.3.10/talk/base/messagequeue.h	2010-09-15 12:50:06.225210002 -0700
@@ -35,6 +35,7 @@
 #include <vector>
 #include <queue>
 #include <algorithm>
+#include <cstring>
 
 namespace cricket {
 
Index: libjingle-0.3.10/talk/base/physicalsocketserver.cc
===================================================================
--- libjingle-0.3.10.orig/talk/base/physicalsocketserver.cc	2006-09-01 04:31:52.000000000 -0700
+++ libjingle-0.3.10/talk/base/physicalsocketserver.cc	2010-09-15 13:05:45.445210001 -0700
@@ -37,6 +37,7 @@ extern "C" {
 #include <fcntl.h>
 #include <sys/time.h>
 #include <unistd.h>
+#include <string.h>
 }
 #endif
 
@@ -50,6 +51,8 @@ extern "C" {
 
 #ifdef __linux 
 #define IP_MTU 14 // Until this is integrated from linux/in.h to netinet/in.h
+#include <cstdio>
+#include <algorithm>
 #endif  // __linux
 
 #ifdef WIN32
@@ -61,7 +64,6 @@ extern "C" {
 
 #include <algorithm>
 #include <iostream>
-
 class WinsockInitializer {
 public:
   WinsockInitializer() {
Index: libjingle-0.3.10/talk/base/protocolinfo.cc
===================================================================
--- libjingle-0.3.10.orig/talk/base/protocolinfo.cc	2006-09-01 04:31:53.000000000 -0700
+++ libjingle-0.3.10/talk/base/protocolinfo.cc	2010-09-15 13:11:47.815210002 -0700
@@ -28,7 +28,7 @@
  */
 
 #include "talk/base/protocolinfo.h"
-
+#include <cstring>
 namespace cricket {
 
 static const char * const PROTO_NAMES[PROTO_LAST+1] = { "udp", "tcp", "ssltcp" };
Index: libjingle-0.3.10/talk/base/socketadapters.cc
===================================================================
--- libjingle-0.3.10.orig/talk/base/socketadapters.cc	2006-09-01 04:31:53.000000000 -0700
+++ libjingle-0.3.10/talk/base/socketadapters.cc	2010-09-15 13:02:10.245210000 -0700
@@ -42,6 +42,7 @@
 #endif
 
 #include <cassert>
+#include <cstring>
 
 #include "talk/base/base64.h"
 #include "talk/base/basicdefs.h"
Index: libjingle-0.3.10/talk/base/stream.cc
===================================================================
--- libjingle-0.3.10.orig/talk/base/stream.cc	2006-09-01 04:31:53.000000000 -0700
+++ libjingle-0.3.10/talk/base/stream.cc	2010-09-15 13:13:51.255210002 -0700
@@ -29,6 +29,8 @@
 #include <sys/stat.h>
 #include <errno.h>
 #include <string>
+#include <cstdio>
+#include <cstring>
 #include "talk/base/basictypes.h"
 #include "talk/base/common.h"
 #include "talk/base/stream.h"
Index: libjingle-0.3.10/talk/base/stringutils.h
===================================================================
--- libjingle-0.3.10.orig/talk/base/stringutils.h	2006-09-01 04:31:52.000000000 -0700
+++ libjingle-0.3.10/talk/base/stringutils.h	2010-09-15 13:09:40.695210000 -0700
@@ -36,6 +36,7 @@
 #endif  // WIN32
 
 #include <string>
+#include <cstring>
 
 ///////////////////////////////////////////////////////////////////////////////
 // Rename a bunch of common string functions so they are consistent across
Index: libjingle-0.3.10/talk/base/xmpppassword.h
===================================================================
--- libjingle-0.3.10.orig/talk/base/xmpppassword.h	2006-09-01 04:31:52.000000000 -0700
+++ libjingle-0.3.10/talk/base/xmpppassword.h	2010-09-15 13:10:33.105210000 -0700
@@ -30,7 +30,7 @@
 
 #include "talk/base/linked_ptr.h"
 #include "talk/base/scoped_ptr.h"
-
+#include <cstring>
 namespace buzz {
 
 class XmppPasswordImpl {
Index: libjingle-0.3.10/talk/p2p/base/pseudotcp.cc
===================================================================
--- libjingle-0.3.10.orig/talk/p2p/base/pseudotcp.cc	2006-09-01 04:31:51.000000000 -0700
+++ libjingle-0.3.10/talk/p2p/base/pseudotcp.cc	2010-09-15 13:17:33.135210000 -0700
@@ -52,6 +52,8 @@ extern "C" {
 }
 #endif // POSIX
 
+#include <cstring>
+#include <cstdlib>
 namespace cricket {
 
 //////////////////////////////////////////////////////////////////////
Index: libjingle-0.3.10/talk/p2p/base/stun.cc
===================================================================
--- libjingle-0.3.10.orig/talk/p2p/base/stun.cc	2006-09-01 04:31:51.000000000 -0700
+++ libjingle-0.3.10/talk/p2p/base/stun.cc	2010-09-15 13:16:30.845210000 -0700
@@ -29,6 +29,7 @@
 #include "talk/p2p/base/stun.h"
 #include <iostream>
 #include <cassert>
+#include <cstring>
 
 #if defined(_MSC_VER) && _MSC_VER < 1300
 namespace std {
Index: libjingle-0.3.10/talk/p2p/base/stunserver.cc
===================================================================
--- libjingle-0.3.10.orig/talk/p2p/base/stunserver.cc	2006-09-01 04:31:51.000000000 -0700
+++ libjingle-0.3.10/talk/p2p/base/stunserver.cc	2010-09-15 13:18:10.045210001 -0700
@@ -28,6 +28,7 @@
 #include "talk/base/bytebuffer.h"
 #include "talk/p2p/base/stunserver.h"
 #include <iostream>
+#include <cstring>
 
 #ifdef POSIX
 extern "C" {
Index: libjingle-0.3.10/talk/xmpp/xmpplogintask.cc
===================================================================
--- libjingle-0.3.10.orig/talk/xmpp/xmpplogintask.cc	2006-09-01 04:31:52.000000000 -0700
+++ libjingle-0.3.10/talk/xmpp/xmpplogintask.cc	2010-09-15 13:15:44.185209999 -0700
@@ -35,6 +35,7 @@
 
 #include <iostream>
 #include <string>
+#include <cstdio>
 #include <vector>
 #ifdef SSL_USE_OPENSSL
 #include <openssl/sha.h>
