diff -Nru libX11-X11R7.1-1.0.1.org/src/util/makekeys.c libX11-X11R7.1-1.0.1/src/util/makekeys.c
--- libX11-X11R7.1-1.0.1.org/src/util/makekeys.c	2007-03-08 14:34:34.000000000 +0100
+++ libX11-X11R7.1-1.0.1/src/util/makekeys.c	2007-03-08 14:34:58.000000000 +0100
@@ -49,7 +49,7 @@
     KeySym	val;
 } info[KTNUM];
 
-#define MIN_REHASH 10
+#define MIN_REHASH 15
 #define MATCHES 10
 
 char tab[KTNUM];
