--- monotone-0.21/configure.ac.orig	2005-07-17 19:23:29.580829434 -0700
+++ monotone-0.21/configure.ac	2005-07-17 19:28:50.237223287 -0700
@@ -82,9 +82,14 @@
 AC_DEFUN([BOOST_VERSION_CHECK],
 [AC_CACHE_CHECK([boost version 1.32 or newer], ac_cv_version_boost,
 [
-  AC_TRY_RUN(
+  AC_TRY_COMPILE(
   [#include <boost/version.hpp>
-  int main() { return (BOOST_VERSION < 103200); }],
+  #if BOOST_VERSION < 103200
+  int main() { return (BOOST_VERSION < 103200); }
+  #else
+  #error boost version is ok
+  #endif
+  ],
   ac_cv_version_boost=yes,
   ac_cv_version_boost=no)
 ])
@@ -101,9 +106,14 @@
 AC_DEFUN([BOOST_FIX_VERSION],
 [AC_CACHE_CHECK([if boost requires extra flags to compile], ac_fix_boost,
 [
-  AC_TRY_RUN(
+  AC_TRY_COMPILE(
   [#include <boost/version.hpp>
-  int main() { return (BOOST_VERSION != 103200); }],
+  #if BOOST_VERSION != 103200
+  int main() { return (BOOST_VERSION != 103200); }
+  #else
+  #error boost version is not 1.32.0
+  #endif
+  ],
   ac_fix_boost=yes,
   ac_fix_boost=no)
 ])
