diff -uNr git.orig//ffalarms.vala git/ffalarms.vala
--- git.orig//ffalarms.vala	2010-09-26 16:15:04.000000000 +0200
+++ git/ffalarms.vala	2010-10-13 21:31:15.000000000 +0200
@@ -139,7 +139,7 @@
 }
 
 
-unowned TimeZone local_tz()
+unowned ICal.TimeZone local_tz()
 {
     string s = Environment.get_variable("TZ");
     try {
@@ -149,15 +149,15 @@
     } catch (FileError e) {
 	s = "UTC";
     }
-    return TimeZone.get_builtin_timezone(s.strip());
+    return ICal.TimeZone.get_builtin_timezone(s.strip());
 }
 
 
 // XXX may return time_t
 ICal.Time next_alarm_as_utc(Component c)
 {
-    unowned TimeZone tz = local_tz();
-    unowned TimeZone utc = TimeZone.get_utc_timezone();
+    unowned ICal.TimeZone tz = local_tz();
+    unowned ICal.TimeZone utc = ICal.TimeZone.get_utc_timezone();
     var t = time_t();
     var utc_now = ICal.Time.from_timet_with_zone(t, false, utc);
     ICal.Time tz_now = ICal.Time.from_timet_with_zone(t, false, tz);
@@ -173,7 +173,7 @@
     }
     if (!next.is_utc() && next.zone == null)
 	ICal.Time.set_timezone(ref next, tz);
-    TimeZone.convert_time(ref next, tz, utc);
+    ICal.TimeZone.convert_time(ref next, tz, utc);
     next.set_timezone(ref next, utc);
     return (next.compare (utc_now) >= 0) ? next : ICal.Time.null_time();
 }
