Index: a/ffalarms.vala
===================================================================
--- a/ffalarms.vala	(revision 67)
+++ b/ffalarms.vala	(working copy)
@@ -1407,7 +1407,6 @@
 	bx.pack_end(bt);
 	bx.pack_end(frame("pad_small"));
 	w.inwin_content_set(bx);
-	w.inwin_style_set("minimal_vertical");
 	bt.show();
 	w.inwin_activate();
     }
@@ -1476,7 +1475,7 @@
 	this.parent = parent;
     }
 
-    public unowned Button add(string label, Evas.SmartCallback cb)
+    public unowned Button add(string label, Evas.Callback cb)
     {
 	unowned Button b;
 
@@ -1505,7 +1504,7 @@
 	hover.content_set("top", box);
     }
 
-    public new unowned Button add(string label, Evas.SmartCallback cb)
+    public new unowned Button add(string label, Evas.Callback cb)
     {
 	unowned Button b = base.add(label, cb);
 	b.smart_callback_add("clicked", hover.hide);
