--- qtopia-free-1.7.0/src/3rdparty/applications/embeddedkonsole/MyPty.cpp.~1~	Mon May 12 20:18:29 2003
+++ qtopia-free-1.7.0/src/3rdparty/applications/embeddedkonsole/MyPty.cpp	Wed Jul 23 17:35:16 2003
@@ -163,6 +163,7 @@
 	ttmode.c_cc[VINTR] = 3;
 	ttmode.c_cc[VERASE] = 8;
 	tcsetattr( STDIN_FILENO, TCSANOW, &ttmode );
+  if(strlen(getenv("TERM"))<=0)
 	setenv("TERM","vt100",1);
 	setenv("COLORTERM","0",1);
 
--- qtopia-free-1.7.0/src/3rdparty/applications/embeddedkonsole/TEWidget.cpp.~1~	Mon May 12 20:18:32 2003
+++ qtopia-free-1.7.0/src/3rdparty/applications/embeddedkonsole/TEWidget.cpp	Wed Jul 23 18:27:56 2003
@@ -1027,9 +1027,18 @@
   if ( e->type() == QEvent::KeyPress )
   {
     QKeyEvent* ke = (QKeyEvent*)e;
+     if(ke->key() == 4165) return true; //     qDebug("key pressed is 0x%x, ascii is 0x%x, state %d", ke->key(), ke->ascii(), ke->state());
 
     actSel=0; // Key stroke implies a screen update, so TEWidget won't
               // know where the current selection is.
+     if( ke->state() == ShiftButton && ke->key() == Key_Tab) {
+        //lets hardcode this sucker
+        emitText("\\"); // expose
+     }
+     else if( ke->state() == ControlButton && ke->key() == Key_V) {
+        pasteClipboard();
+     }
+     else
 
     emit keyPressedSignal(ke); // expose
     ke->accept();
--- qtopia-free-1.7.0/src/3rdparty/applications/embeddedkonsole/konsole.cpp.~1~	2003-05-12 20:18:38.000000000 -0600
+++ qtopia-free-1.7.0/src/3rdparty/applications/embeddedkonsole/konsole.cpp	2003-07-24 20:43:25.000000000 -0600
@@ -736,7 +736,7 @@
     }
     if( iD  == -3) {
         cfg.setGroup("Tabs");
-        QString tmp=cfg.readEntry("Position","Top");
+        QString tmp=cfg.readEntry("Position","Bottom");
 
         if(tmp=="Top") { // No tr
             tab->setTabPosition(QTabWidget::Bottom);
