--- qtopia-free-1.6.2/src/settings/security/securitybase.ui.orig	Tue Jul 22 10:12:20 2003
+++ qtopia-free-1.7.0/src/settings/security/securitybase.ui	Tue Jul 22 10:08:45 2003
@@ -156,6 +156,12 @@
                     <item>
                         <property>
                             <name>text</name>
+                            <string>192.168.129.0/24</string>
+                        </property>
+                    </item>
+                    <item>
+                        <property>
+                            <name>text</name>
                             <string>172.16.0.0/12</string>
                         </property>
                     </item>
