diff -ur GConf-2.6.4/gconf/gconf-client.c GConf-2.6.4.new/gconf/gconf-client.c
--- GConf-2.6.4/gconf/gconf-client.c	2004-09-19 15:29:35.000000000 +0300
+++ GConf-2.6.4.new/gconf/gconf-client.c	2005-04-08 10:59:46.268506800 +0300
@@ -776,7 +776,6 @@
 static gboolean
 clear_cache_foreach (char* key, GConfEntry* entry, GConfClient* client)
 {
-  g_free (key);
   gconf_entry_free (entry);
 
   return TRUE;
