diff -Naur hdparm-5.4/hdparm.c hdparm-5.4-fix/hdparm.c
--- hdparm-5.4/hdparm.c	2004-07-14 00:36:03.000000000 +0200
+++ hdparm-5.4-fix/hdparm.c	2003-04-18 22:29:38.000000000 +0200
@@ -52,7 +52,7 @@
 static int flagcount = 0, do_flush = 0, is_scsi_hd = 0, is_xt_hd = 0;
 static int do_ctimings, do_timings = 0;
 
-static unsigned long set_readahead= 0, get_readahead= 0, readahead= 0;     
+static unsigned long set_readahead= 0, get_readahead= 0, readahead_state= 0;
 static unsigned long set_readonly = 0, get_readonly = 0, readonly = 0;
 static unsigned long set_unmask   = 0, get_unmask   = 0, unmask   = 0;     
 static unsigned long set_mult     = 0, get_mult     = 0, mult     = 0;     
@@ -785,8 +785,8 @@
 
 	if (set_readahead) {
 		if (get_readahead)
-			printf(" setting fs readahead to %ld\n", readahead);
-		if (ioctl(fd, BLKRASET, readahead)) 
+			printf(" setting fs readahead to %ld\n", readahead_state);
+		if (ioctl(fd, BLKRASET, readahead_state)) 
 			perror(" BLKRASET failed");
 	}
 #ifdef HDIO_UNREGISTER_HWIF
@@ -1830,7 +1830,7 @@
 					case 'a':
 						get_readahead = noisy;
 						noisy = 1;
-						GET_NUMBER(set_readahead,readahead);
+						GET_NUMBER(set_readahead,readahead_state);
 						break;
 					case 'B':
 						get_apmmode = noisy;
