From 41eabd493ccc241ccd52b77c576211759976972d Mon Sep 17 00:00:00 2001
From: Thomas Kunze <thommycheck@gmx.de>
Date: Mon, 20 Oct 2008 17:33:29 +0200
Subject: [PATCH 03/23] enable cpufreq for collie

---
 arch/arm/Kconfig |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 9722f8b..609f0fb 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -1048,7 +1048,7 @@ config CPU_FREQ_SA1100
 
 config CPU_FREQ_SA1110
 	bool
-	depends on CPU_FREQ && (SA1100_ASSABET || SA1100_CERF || SA1100_PT_SYSTEM3)
+	depends on CPU_FREQ && (SA1100_ASSABET || SA1100_CERF || SA1100_PT_SYSTEM3 || SA1100_COLLIE)
 	default y
 
 config CPU_FREQ_INTEGRATOR
-- 
1.5.6.5

