
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- libgaim/configure.ac~dont-look-for-gtk
+++ libgaim/configure.ac
@@ -253,10 +253,6 @@
 *** GLib 2.0 is required to build Gaim; please make sure you have the GLib
 *** development headers installed. The latest version of GLib is
 *** always available at http://www.gtk.org/.]))
-AM_PATH_GTK_2_0(2.0.0,,AC_MSG_ERROR([
-*** GTK+ 2.0 is required to build Gaim; please make sure you have the GTK+
-*** development headers installed. The latest version of GTK+ is
-*** always available at http://www.gtk.org/.]))
 
 AC_PATH_PROG(gaimpath, gaim)
 AC_SUBST(GTK_CFLAGS)
