--- ffmpeg/libavcodec/common.h.orig	2005-02-10 15:10:34.000000000 +0100
+++ ffmpeg/libavcodec/common.h	2005-02-10 15:11:28.000000000 +0100
@@ -501,10 +501,6 @@
 #define srand srand_is_forbidden_due_to_state_trashing
 #define sprintf sprintf_is_forbidden_due_to_security_issues_use_snprintf
 #define strcat strcat_is_forbidden_due_to_security_issues_use_pstrcat
-#if !(defined(LIBAVFORMAT_BUILD) || defined(_FRAMEHOOK_H))
-#define printf please_use_av_log
-#define fprintf please_use_av_log
-#endif
 
 #define CHECKED_ALLOCZ(p, size)\
 {\
