--- configure.ac.old	2005-06-09 17:27:01.000000000 +0200
+++ configure.ac	2005-06-09 17:27:20.000000000 +0200
@@ -29,7 +29,7 @@
 # Set compilation flags
 
 CFLAGS="$CFLAGS -Wall -Wmissing-prototypes -Wmissing-declarations \
-                -pedantic-errors -Wcast-align -std=c99 -Werror"
+                -Wcast-align -std=c99"
 
 # So says the Coding Spec (doesn't -Wall cover these..?!?)
 CFLAGS+="-Wunreachable-code -Wcast-align"
--- helplib/Makefile.am.old	2005-06-09 17:33:08.000000000 +0200
+++ helplib/Makefile.am	2005-06-09 17:33:47.000000000 +0200
@@ -21,7 +21,7 @@
 
 # Disable this only if system libs give warnings:
 #
-libossohelp_la_CFLAGS+=-pedantic-errors -Werror
+#libossohelp_la_CFLAGS+=-pedantic-errors
 
 libossohelp_la_includedir = $(includedir)
 libossohelp_la_include_HEADERS = osso-helplib.h
--- helptest/Makefile.am.old	2005-06-09 17:34:44.000000000 +0200
+++ helptest/Makefile.am	2005-06-09 17:34:55.000000000 +0200
@@ -23,7 +23,7 @@
 
 # Disable this only if system libs give warnings:
 #
-helptest_CFLAGS+=-pedantic-errors -Werror
+#helptest_CFLAGS+=-pedantic-errors
 
 helptest_SOURCES = \
     helptest.c
