Index: e2fsprogs-1.38/MCONFIG.in
===================================================================
--- e2fsprogs-1.38.orig/MCONFIG.in	2005-02-04 16:43:57.000000000 +0100
+++ e2fsprogs-1.38/MCONFIG.in	2008-10-14 00:57:33.988145555 +0200
@@ -59,7 +59,7 @@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
 INSTALL_DATA = @INSTALL_DATA@
 INSTALL_SCRIPT = @INSTALL_SCRIPT@
-MKINSTALLDIRS = @MKINSTALLDIRS@
+MKINSTALLDIRS = @mkdir_p@
 
 #
 # Library definitions
