--- ifupdown/inet.defn.orig	2004-04-29 20:16:07.000000000 +0200
+++ ifupdown/inet.defn	2004-04-29 20:16:13.000000000 +0200
@@ -81,8 +81,7 @@
         elsif (execable("/sbin/dhclient"))
     pump -i %iface% [[-h %hostname%]] [[-l %leasehours%]] \
         elsif (execable("/sbin/pump") && mylinuxver() >= mylinux(2,1,100))
-    udhcpc -n -p /var/run/udhcpc.%iface%.pid -i %iface% [[-H %hostname%]] \
-           [[-c %clientid%]] \
+    udhcpc -b -p /var/run/udhcpc.%iface%.pid -i %iface% -H `cat /etc/hostname` \
         elsif (execable("/sbin/udhcpc") && mylinuxver() >= mylinux(2,2,0))
     dhcpcd [[-h %hostname%]] [[-i %vendor%]] [[-I %clientid%]] \
            [[-l %leasetime%]] %iface% \
@@ -91,11 +90,11 @@
   down
     dhclient3 -r -pf /var/run/dhclient.%iface%.pid -lf /var/run/dhclient.%iface%.leases %iface% \
         if (execable("/sbin/dhclient3"))
-    cat /var/run/dhclient.%iface%.pid | xargs -i kill -TERM {} \
+    kill -TERM `cat /var/run/dhclient.%iface%.pid` \
         elsif (execable("/sbin/dhclient"))
     pump -i %iface% -r \
         elsif (execable("/sbin/pump") && mylinuxver() >= mylinux(2,1,100))
-    cat /var/run/udhcpc.%iface%.pid | xargs -i kill -TERM {} \
+    kill -TERM `cat /var/run/udhcpc.%iface%.pid` \
         elsif (execable("/sbin/udhcpc"))
     dhcpcd -k %iface% \
         elsif (execable("/sbin/dhcpcd"))