
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- postfix-2.0.15/postfix-install~install.patch
+++ postfix-2.0.15/postfix-install
@@ -187,8 +187,8 @@
     exit 1
 }
 
-test -x bin/postconf || {
-    echo $0: Error: no bin/postconf file. Did you forget to run \"make\"? 1>&2
+test -x $POSTCONF || {
+    echo $0: Error: no $POSTCONF file. Did you forget to run \"make\"? 1>&2
     exit 1
 }
 
@@ -336,7 +336,7 @@
 
 : ${install_root=/}
 : ${tempdir=`pwd`}
-: ${config_directory=`bin/postconf -h -d config_directory`}
+: ${config_directory=`$POSTCONF -h -d config_directory`}
 
 # Find out the location of installed configuration files.
 
@@ -399,7 +399,7 @@
 	case "$junk" in
 	"") eval unset $name;;
 	esac
-	eval : \${$name=\`bin/postconf -c $CONFIG_DIRECTORY -h $name\`} ||
+	eval : \${$name=\`$POSTCONF -c $CONFIG_DIRECTORY -h $name\`} ||
 	    exit 1
     done
 }
@@ -414,7 +414,7 @@
     case "$junk" in
     "") eval unset $name;;
     esac
-    eval : \${$name=\`bin/postconf -d -h $name\`} || exit 1
+    eval : \${$name=\`$POSTCONF -d -h $name\`} || exit 1
 done
 
 # Override settings manually.
@@ -521,6 +521,8 @@
 SAMPLE_DIRECTORY=$install_root$sample_directory
 README_DIRECTORY=$install_root$readme_directory
 
+test "x$POSTCONF" != "x" || POSTCONF="bin/postconf"
+
 # Avoid repeated tests for existence of these; default permissions suffice.
 
 test -d $DAEMON_DIRECTORY || mkdir -p $DAEMON_DIRECTORY || exit 1
@@ -639,7 +641,7 @@
 # Postfix releases, and software should not suddenly be installed in
 # the wrong place when Postfix is being upgraded.
 
-bin/postconf -c $CONFIG_DIRECTORY -e \
+$POSTCONF -c $CONFIG_DIRECTORY -e \
     "daemon_directory = $daemon_directory" \
     "command_directory = $command_directory" \
     "queue_directory = $queue_directory" \
