:M:

Index: vala-0.5.1/vapi/vte.vapi
===================================================================
--- vala-0.5.1.orig/vapi/vte.vapi
+++ vala-0.5.1/vapi/vte.vapi
@@ -37,7 +37,7 @@ namespace Vte {
 		public void feed_child (string text, long length);
 		public void feed_child_binary (string data, long length);
 		[NoArrayLength]
-		public int fork_command (string? command, string[]? argv, string[]? envv, string? directory, bool lastlog, bool utmp, bool wtmp);
+		public int fork_command (string command, string[] argv, string[]? envv, string? directory, bool lastlog, bool utmp, bool wtmp);
 		public int forkpty (string[] envv, string directory, bool lastlog, bool utmp, bool wtmp);
 		public weak Gtk.Adjustment get_adjustment ();
 		public bool get_allow_bold ();
