
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- qt-embedded-opensource-4.0.0-b2-snapshot-20050128/mkspecs/qws/linux-arm-g++/qmake.conf~fix-mkspecs
+++ qt-embedded-opensource-4.0.0-b2-snapshot-20050128/mkspecs/qws/linux-arm-g++/qmake.conf
@@ -8,12 +8,12 @@
 QT                      += core gui network
 QMAKE_INCREMENTAL_STYLE = sublib
 
-QMAKE_CC		= arm-linux-gcc
+QMAKE_CC		= $(CCACHE) arm-linux-gcc
 QMAKE_LEX		= flex
 QMAKE_LEXFLAGS		=
 QMAKE_YACC		= yacc
 QMAKE_YACCFLAGS		= -d
-QMAKE_CFLAGS		= -pipe
+QMAKE_CFLAGS		= -pipe $(EXTRA_CFLAGS)
 QMAKE_CFLAGS_WARN_ON	= -Wall -W
 QMAKE_CFLAGS_WARN_OFF	=
 QMAKE_CFLAGS_RELEASE	= -O2
@@ -22,8 +22,8 @@
 QMAKE_CFLAGS_YACC	= -Wno-unused -Wno-parentheses
 QMAKE_CFLAGS_THREAD	= -D_REENTRANT
 
-QMAKE_CXX		= arm-linux-g++
-QMAKE_CXXFLAGS		= $$QMAKE_CFLAGS -fno-exceptions -fno-rtti
+QMAKE_CXX		= $(CCACHE) arm-linux-g++
+QMAKE_CXXFLAGS		= $$QMAKE_CFLAGS -fno-exceptions -fno-rtti $(EXTRA_CXXFLAGS)
 QMAKE_CXXFLAGS_WARN_ON	= $$QMAKE_CFLAGS_WARN_ON
 QMAKE_CXXFLAGS_WARN_OFF	= $$QMAKE_CFLAGS_WARN_OFF
 QMAKE_CXXFLAGS_RELEASE	= $$QMAKE_CFLAGS_RELEASE
@@ -45,14 +45,14 @@
 
 QMAKE_LINK		= arm-linux-g++
 QMAKE_LINK_SHLIB	= arm-linux-g++
-QMAKE_LFLAGS		=
+QMAKE_LFLAGS		= $(EXTRA_LFLAGS)
 QMAKE_LFLAGS_RELEASE	=
 QMAKE_LFLAGS_DEBUG	=
 QMAKE_LFLAGS_SHLIB      = -shared
 QMAKE_LFLAGS_PLUGIN     = $$QMAKE_LFLAGS_SHLIB
 QMAKE_LFLAGS_SONAME     = -Wl,-soname,
 QMAKE_LFLAGS_THREAD     =
-QMAKE_RPATH             = -Wl,-rpath,
+QMAKE_RPATH             = -Wl,-rpath-link,
 
 QMAKE_LIBS		= 
 QMAKE_LIBS_DYNLOAD      = -ldl
@@ -66,6 +66,7 @@
 
 QMAKE_MOC		= $$[QT_INSTALL_BINS]/moc
 QMAKE_UIC		= $$[QT_INSTALL_BINS]/uic
+QMAKE_RCC		= rcc4
 
 QMAKE_AR		= ar cqs
 QMAKE_RANLIB		=
--- qt-embedded-opensource-4.0.0-b2-snapshot-20050128/mkspecs/features/resources.prf~fix-mkspecs
+++ qt-embedded-opensource-4.0.0-b2-snapshot-20050128/mkspecs/features/resources.prf
@@ -1,6 +1,6 @@
 isEmpty(QMAKE_RCC) {
     win32:QMAKE_RCC = $$[QT_INSTALL_BINS]\rcc.exe
-    else:QMAKE_RCC = $$[QT_INSTALL_BINS]/rcc
+    else:QMAKE_RCC = rcc4
 }
 
 isEmpty(RCC_DIR):RCC_DIR = .
