 src/include/IxEthDB.h |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

Index: ixp400_xscale_sw/src/include/IxEthDB.h
===================================================================
--- ixp400_xscale_sw.orig/src/include/IxEthDB.h
+++ ixp400_xscale_sw/src/include/IxEthDB.h
@@ -273,7 +273,7 @@
 typedef struct  
 {
    UINT8 macAddress[IX_IEEE803_MAC_ADDRESS_SIZE];
-} IX_OSAL_ATTRIBUTE_PACKED IxEthDBMacAddr;
+} IxEthDBMacAddr;
 
 /**
  *
