diff -ru klibc.clean/Makefile klibc-1.5/Makefile
--- klibc.clean/Makefile	2008-02-08 22:37:36.000000000 +0100
+++ klibc-1.5/Makefile	2008-02-15 12:46:48.000000000 +0100
@@ -33,7 +33,7 @@
 export PERL       := perl
 
 # Location for installation
-export prefix      = /usr
+export prefix      = $(INST)
 export bindir      = $(prefix)/bin
 export libdir      = $(prefix)/lib
 export mandir      = $(prefix)/man
diff -ru klibc.clean/scripts/Kbuild.install klibc-1.5/scripts/Kbuild.install
--- klibc.clean/scripts/Kbuild.install	2008-02-08 22:37:36.000000000 +0100
+++ klibc-1.5/scripts/Kbuild.install	2008-02-18 22:36:35.434193918 +0100
@@ -88,14 +88,11 @@
 header:
 	$(Q)echo "  INSTALL headers + man pages to $(INSTALLROOT)$(INSTALLDIR)"
 	$(Q)mkdir -p $(INSTALLROOT)$(bindir)
-	$(Q)mkdir -p $(INSTALLROOT)$(mandir)/man1
-	$(Q)mkdir -p $(INSTALLROOT)$(SHLIBDIR)
 	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)
 	$(Q)-rm -rf $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include
 	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include
 	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)lib
-	$(Q)mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)bin
-	$(Q)set -e ; for d in linux scsi asm-$(KLIBCARCH) asm-generic $(ASMKLIBCARCH); do \
+	$(Q)-set -e ; for d in linux scsi asm-$(KLIBCARCH) asm-generic $(ASMKLIBCARCH); do \
 	  mkdir -p $(INSTALLROOT)$(INSTALLDIR)/$(KLIBCCROSS)include/$$d ; \
 	  for r in $(KLIBCKERNELSRC)/include $(KLIBCKERNELOBJ)/include \
 	           $(KLIBCKERNELOBJ)/include2 ; do \
@@ -106,7 +103,6 @@
 	done
 	$(Q)cd $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include && ln -sf asm-$(KLIBCARCH) asm
 	$(Q)cp -rf usr/include/. $(INSTALLROOT)$(INSTALLDIR)/$(KCROSS)include/.
-	$(Q)$(install-data) $(srctree)/klcc/klcc.1 $(INSTALLROOT)$(mandir)/man1/$(KCROSS)klcc.1
 	$(Q)$(install-bin) $(objtree)/klcc/$(KCROSS)klcc $(INSTALLROOT)$(bindir)
 
 footer: header
diff -ru klibc.clean/usr/dash/Kbuild klibc-1.5/usr/dash/Kbuild
--- klibc.clean/usr/dash/Kbuild	2008-02-08 22:37:41.000000000 +0100
+++ klibc-1.5/usr/dash/Kbuild	2008-02-15 12:18:08.000000000 +0100
@@ -107,5 +107,3 @@
 $(obj)/arith.c: $(obj)/arith.h
 	$(Q):
 
-# Targets to install
-install-y := sh.shared
diff -ru klibc.clean/usr/gzip/Kbuild klibc-1.5/usr/gzip/Kbuild
--- klibc.clean/usr/gzip/Kbuild	2008-02-08 22:37:42.000000000 +0100
+++ klibc-1.5/usr/gzip/Kbuild	2008-02-15 12:18:24.000000000 +0100
@@ -21,5 +21,3 @@
 # Cleaning
 targets := gzip gzip.g gunzip zcat
 
-# Targets to install
-install-y := gzip gunzip zcat
diff -ru klibc.clean/usr/kinit/fstype/Kbuild klibc-1.5/usr/kinit/fstype/Kbuild
--- klibc.clean/usr/kinit/fstype/Kbuild	2008-02-08 22:37:40.000000000 +0100
+++ klibc-1.5/usr/kinit/fstype/Kbuild	2008-02-15 12:26:20.000000000 +0100
@@ -21,5 +21,3 @@
 # Cleaning
 clean-dirs := static shared
 
-# install binary
-install-y := $(shared-y)
diff -ru klibc.clean/usr/kinit/ipconfig/Kbuild klibc-1.5/usr/kinit/ipconfig/Kbuild
--- klibc.clean/usr/kinit/ipconfig/Kbuild	2008-02-08 22:37:40.000000000 +0100
+++ klibc-1.5/usr/kinit/ipconfig/Kbuild	2008-02-15 12:26:33.000000000 +0100
@@ -27,5 +27,3 @@
 # Cleaning
 clean-dirs := static shared
 
-# install binary
-install-y := $(shared-y)
diff -ru klibc.clean/usr/kinit/Kbuild klibc-1.5/usr/kinit/Kbuild
--- klibc.clean/usr/kinit/Kbuild	2008-02-08 22:37:40.000000000 +0100
+++ klibc-1.5/usr/kinit/Kbuild	2008-02-15 12:18:43.000000000 +0100
@@ -33,5 +33,3 @@
 subdir- := fstype ipconfig nfsmount resume run-init
 
 
-# install binary
-install-y := kinit kinit.shared
diff -ru klibc.clean/usr/kinit/nfsmount/Kbuild klibc-1.5/usr/kinit/nfsmount/Kbuild
--- klibc.clean/usr/kinit/nfsmount/Kbuild	2008-02-08 22:37:40.000000000 +0100
+++ klibc-1.5/usr/kinit/nfsmount/Kbuild	2008-02-15 12:26:46.000000000 +0100
@@ -23,5 +23,3 @@
 
 clean-dirs := static shared
 
-# Install binary
-install-y := $(shared-y)
diff -ru klibc.clean/usr/kinit/resume/Kbuild klibc-1.5/usr/kinit/resume/Kbuild
--- klibc.clean/usr/kinit/resume/Kbuild	2008-02-08 22:37:39.000000000 +0100
+++ klibc-1.5/usr/kinit/resume/Kbuild	2008-02-15 12:27:01.000000000 +0100
@@ -26,5 +26,3 @@
 # Cleaning
 clean-dirs := static shared
 
-# install binary
-install-y := $(shared-y)
diff -ru klibc.clean/usr/kinit/run-init/Kbuild klibc-1.5/usr/kinit/run-init/Kbuild
--- klibc.clean/usr/kinit/run-init/Kbuild	2008-02-08 22:37:39.000000000 +0100
+++ klibc-1.5/usr/kinit/run-init/Kbuild	2008-02-15 12:27:12.000000000 +0100
@@ -21,5 +21,3 @@
 # Cleaning
 clean-dirs := static shared
 
-# install binary
-install-y := $(shared-y)
Only in klibc-1.5/usr/klibc/arch/arm: klib.list
diff -ru klibc.clean/usr/klibc/Kbuild klibc-1.5/usr/klibc/Kbuild
--- klibc.clean/usr/klibc/Kbuild	2008-02-08 22:37:49.000000000 +0100
+++ klibc-1.5/usr/klibc/Kbuild	2008-02-15 12:17:24.000000000 +0100
@@ -175,5 +175,3 @@
 	          $(INSTALLROOT)$(INSTALLDIR)/$(KLIBCCROSS)lib))
 	$(Q)$(install-lib) $(obj)/klibc-$(SOLIBHASH).so \
 	                      $(INSTALLROOT)$(INSTALLDIR)/$(KLIBCCROSS)lib
-	$(Q)$(install-lib) $(obj)/klibc-$(SOLIBHASH).so \
-	                      $(INSTALLROOT)$(SHLIBDIR)
diff -ru klibc.clean/usr/utils/Kbuild klibc-1.5/usr/utils/Kbuild
--- klibc.clean/usr/utils/Kbuild	2008-02-08 22:37:37.000000000 +0100
+++ klibc-1.5/usr/utils/Kbuild	2008-02-15 12:17:51.000000000 +0100
@@ -64,5 +64,3 @@
 # Clean deletes the static and shared dir
 clean-dirs := static shared
 
-# install only install the shared binaries
-install-y := $(shared-y) shared/reboot shared/poweroff
