Index: Linux-PAM-1.1.0/modules/pam_userdb/pam_userdb.c
===================================================================
--- Linux-PAM-1.1.0.orig/modules/pam_userdb/pam_userdb.c	2010-06-08 14:19:25.147555285 -0700
+++ Linux-PAM-1.1.0/modules/pam_userdb/pam_userdb.c	2010-06-08 14:20:50.707692049 -0700
@@ -30,6 +30,7 @@
 #else
 # ifdef HAVE_DB_H
 #  define DB_DBM_HSEARCH    1 /* use the dbm interface */
+#  define HAVE_DBM /* for BerkDB 5.0 and later */
 #  include <db.h>
 # else
 #  error "failed to find a libdb or equivalent"
