Index: mythplugins-0.22rc1/mythmovies/mythmovies/mythmovies.pro
===================================================================
--- mythplugins-0.22rc1.orig/mythmovies/mythmovies/mythmovies.pro
+++ mythplugins-0.22rc1/mythmovies/mythmovies/mythmovies.pro
@@ -10,7 +10,7 @@ TARGET = mythmovies
 target.path = $${LIBDIR}/mythtv/plugins
 INSTALLS += target
 
-INCLUDEPATH += $${PREFIX}/include/mythtv/libmythui
+INCLUDEPATH +=  $${SYSROOT}$${PREFIX}/include/mythtv/libmythui
 
 # Input
 HEADERS += moviesui.h helperobjects.h moviessettings.h
