Index: mythplugins/programs-libs.pro
===================================================================
--- mythplugins/programs-libs.pro	(revision 22430)
+++ mythplugins/programs-libs.pro	(working copy)
@@ -4,7 +4,7 @@
 INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv/libmyth
 DEPENDPATH *= $${INCLUDEPATH}
 
-LIBS += -L$${LIBDIR} $$EXTRA_LIBS -lmythdb-$$LIBVERSION
+LIBS += -L$${SYSROOT}$${LIBDIR} $$EXTRA_LIBS -lmythdb-$$LIBVERSION
 LIBS += -lmythavutil-$$LIBVERSION -lmythavcodec-$$LIBVERSION
 LIBS += -lmyth-$$LIBVERSION -lmythui-$$LIBVERSION -lmythupnp-$$LIBVERSION
 
Index: mythplugins/configure
===================================================================
--- mythplugins/configure	(revision 22430)
+++ mythplugins/configure	(working copy)
@@ -266,24 +266,25 @@
 
 has_library()
 {
-    if test -f /etc/ld.so.conf ; then
-        LIBPATHS=`cat /etc/ld.so.conf | grep -v "#" | grep -v "include"`
-    else
-        LIBPATHS=''
-    fi
+    LIBPATHS=''
+    if test "x$sysroot" = "x" ; then
+        if test -f /etc/ld.so.conf ; then
+            LIBPATHS=`cat /etc/ld.so.conf | grep -v "#" | grep -v "include"`
+        fi
 
-    LIBPATHS="$LIBPATHS `echo $LD_LIBRARY_PATH | sed s/':'/' '/g` "
-    # Mac OS X has a different variable for this:
-    LIBPATHS="$LIBPATHS `echo $DYLD_LIBRARY_PATH | sed s/':'/' '/g` "
+        LIBPATHS="$LIBPATHS `echo $LD_LIBRARY_PATH | sed s/':'/' '/g` "
+        # Mac OS X has a different variable for this:
+        LIBPATHS="$LIBPATHS `echo $DYLD_LIBRARY_PATH | sed s/':'/' '/g` "
 
-    # recurse down just one level, should be enough in most cases
-    if test -f /etc/ld.so.conf ; then
-        INCL=`cat /etc/ld.so.conf | grep -v "#" | grep  "^include"`
-        SRCH=`echo "$INCL" | sed '{ s/^include \//\//; s/^include /\/etc\//}'`
-        for x in $SRCH ; do
-            LP_APPEND=`cat $x | grep -v "#" | grep -v "^include"`
-            LIBPATHS="$LIBPATHS $LP_APPEND"
-        done
+        # recurse down just one level, should be enough in most cases
+        if test -f /etc/ld.so.conf ; then
+            INCL=`cat /etc/ld.so.conf | grep -v "#" | grep  "^include"`
+            SRCH=`echo "$INCL" | sed '{ s/^include \//\//; s/^include /\/etc\//}'`
+            for x in $SRCH ; do
+                LP_APPEND=`cat $x | grep -v "#" | grep -v "^include"`
+                LIBPATHS="$LIBPATHS $LP_APPEND"
+            done
+        fi
     fi
 
     LIBPATHS="`echo $LIBPATHS $sysroot$prefix/$libdir_name $sysroot/$libdir_name $sysroot/usr/$libdir_name $sysroot/usr/local/$libdir_name $sysroot/usr/X11R6/$libdir_name | sed s/'\/ '/' '/g` "
@@ -300,8 +301,10 @@
 has_header()
 {
     HPATHS="$sysroot$prefix/include $sysroot/usr/local/include $sysroot/usr/include $sysroot/usr/include/g++-v3 $sysroot/usr/X11R6/include $sysroot/"
-    INCL=`echo $DYLD_LIBRARY_PATH $LD_LIBRARY_PATH | sed s/':'/' '/g`
-    HPATHS="$HPATHS `echo $INCL | sed s/$libdir_name/include/g` "
+    if test "x$sysroot" = "x" ; then
+        INCL=`echo $DYLD_LIBRARY_PATH $LD_LIBRARY_PATH | sed s/':'/' '/g`
+        HPATHS="$HPATHS `echo $INCL | sed s/$libdir_name/include/g` "
+    fi
 
     HAS_IT="no"
     for HPATH in $HPATHS ; do
@@ -689,7 +692,7 @@
         if test "$sdl" = "yes" ; then
             echo "        libvisual      support will be included in MythMusic"
             echo "#define LIBVISUAL_SUPPORT 1" >> ./mythmusic/mythmusic/config.h
-            echo "INCLUDEPATH += /usr/include/libvisual-0.4" >> ./mythmusic/mythmusic/config.pro
+            echo "INCLUDEPATH += ${sysroot}/usr/include/libvisual-0.4" >> ./mythmusic/mythmusic/config.pro
             echo "LIBS += -lvisual-0.4" >> ./mythmusic/mythmusic/config.pro
         else
             echo "        libvisual      support will not be included in MythMusic (requires SDL support)"
Index: mythplugins/settings.pro
===================================================================
--- mythplugins/settings.pro	(revision 22430)
+++ mythplugins/settings.pro	(working copy)
@@ -3,8 +3,8 @@
 LIBVERSION = 0.22
 
 INCLUDEPATH += $${SYSROOT}$${PREFIX}/include
-#INCLUDEPATH += /usr/include/cdda
-#INCLUDEPATH += /usr/include/FLAC
+#INCLUDEPATH += $${SYSROOT}/usr/include/cdda
+#INCLUDEPATH += $${SYSROOT}/usr/include/FLAC
 
 LIBS *= -L$${SYSROOT}$${PREFIX}/$${LIBDIRNAME}
 
Index: mythplugins/mythgallery/mythgallery/mythgallery.pro
===================================================================
--- mythplugins/mythgallery/mythgallery/mythgallery.pro	(revision 22430)
+++ mythplugins/mythgallery/mythgallery/mythgallery.pro	(working copy)
@@ -14,7 +14,7 @@
 target.path = $${LIBDIR}/mythtv/plugins
 INSTALLS += target
 
-INCLUDEPATH += $${PREFIX}/include/mythtv
+INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv
 
 # Input
 HEADERS += iconview.h          singleview.h
Index: mythplugins/mythweather/mythweather/mythweather.pro
===================================================================
--- mythplugins/mythweather/mythweather/mythweather.pro	(revision 22430)
+++ mythplugins/mythweather/mythweather/mythweather.pro	(working copy)
@@ -13,8 +13,8 @@
 QMAKE_COPY_DIR = sh ../../cpsvndir
 win32:QMAKE_COPY_DIR = sh ../../cpsimple
 
-INCLUDEPATH += $${PREFIX}/include/mythtv
-INCLUDEPATH += $${PREFIX}/include/mythtv/libmythui
+INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv
+INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv/libmythui
 
 datafiles.path = $${PREFIX}/share/mythtv/mythweather/
 datafiles.files = weather-screens.xml
Index: mythplugins/mythflix/mythflix/mythflix.pro
===================================================================
--- mythplugins/mythflix/mythflix/mythflix.pro	(revision 22430)
+++ mythplugins/mythflix/mythflix/mythflix.pro	(working copy)
@@ -8,7 +8,7 @@
 target.path = $${LIBDIR}/mythtv/plugins
 INSTALLS += target
 
-INCLUDEPATH += $${PREFIX}/include/mythtv
+INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv
 
 installfiles.path = $${PREFIX}/share/mythtv/mythflix
 installfiles.files = netflix-rss.xml
Index: mythplugins/mythmovies/mythmovies/mythmovies.pro
===================================================================
--- mythplugins/mythmovies/mythmovies/mythmovies.pro	(revision 22430)
+++ mythplugins/mythmovies/mythmovies/mythmovies.pro	(working copy)
@@ -10,7 +10,7 @@
 target.path = $${LIBDIR}/mythtv/plugins
 INSTALLS += target
 
-INCLUDEPATH += $${PREFIX}/include/mythtv/libmythui
+INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv/libmythui
 
 # Input
 HEADERS += moviesui.h helperobjects.h moviessettings.h
Index: mythplugins/mythzoneminder/mythzoneminder/mythzoneminder.pro
===================================================================
--- mythplugins/mythzoneminder/mythzoneminder/mythzoneminder.pro	(revision 22430)
+++ mythplugins/mythzoneminder/mythzoneminder/mythzoneminder.pro	(working copy)
@@ -8,7 +8,7 @@
 target.path = $${LIBDIR}/mythtv/plugins
 INSTALLS += target
 
-INCLUDEPATH += $${PREFIX}/include/mythtv/libmythui
+INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv/libmythui
 
 # Input
 HEADERS += zmconsole.h zmplayer.h zmevents.h zmliveplayer.h zmdefines.h 
Index: mythplugins/mythnews/mythnews/mythnews.pro
===================================================================
--- mythplugins/mythnews/mythnews/mythnews.pro	(revision 22430)
+++ mythplugins/mythnews/mythnews/mythnews.pro	(working copy)
@@ -10,9 +10,9 @@
 target.path = $${LIBDIR}/mythtv/plugins
 INSTALLS += target
 
-INCLUDEPATH += $${PREFIX}/include/mythtv
-INCLUDEPATH += $${PREFIX}/include/mythtv/libmythui
-INCLUDEPATH += $${PREFIX}/include/mythtv/libmythdb
+INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv
+INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv/libmythui
+INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv/libmythdb
 
 installfiles.path = $${PREFIX}/share/mythtv/mythnews
 installfiles.files = news-sites.xml
Index: mythplugins/targetdep.pro
===================================================================
--- mythplugins/targetdep.pro	(revision 22430)
+++ mythplugins/targetdep.pro	(working copy)
@@ -10,10 +10,10 @@
 MYTH_LIB_EXT  =$${LIBVERSION}.$${QMAKE_EXTENSION_LIB}
 
 
-DEPLIBS = $${LIBDIR}
+DEPLIBS = $${SYSROOT}$${LIBDIR}
 
 # On Windows, dlls were installed with exes:
-mingw : DEPLIBS = $${PREFIX}/bin
+mingw : DEPLIBS = $${SYSROOT}$${PREFIX}/bin
 
 TARGETDEPS += $${DEPLIBS}/libmyth-$${MYTH_SHLIB_EXT}
 TARGETDEPS += $${DEPLIBS}/libmythdb-$${MYTH_SHLIB_EXT}
