Index: epiphany-2.21.92/lib/ephy-file-helpers.c
===================================================================
--- epiphany-2.21.92.orig/lib/ephy-file-helpers.c	2008-04-19 10:26:48.000000000 +0100
+++ epiphany-2.21.92/lib/ephy-file-helpers.c	2008-04-19 10:27:21.000000000 +0100
@@ -713,7 +713,7 @@
 	desktop = g_file_new_for_path (desktop_dir);
 	
 	/* Don't do anything if destination is the desktop */
-	if (g_file_contains_file (desktop, file))
+	if (g_file_has_prefix (desktop, file))
 	{
 		ret = FALSE;
 	}
