Index: gaim/src/gtkconv.c
===================================================================
--- gaim.orig/src/gtkconv.c	2005-08-25 17:16:50.000000000 +0200
+++ gaim/src/gtkconv.c	2005-08-29 23:56:48.142463528 +0200
@@ -6126,7 +6126,7 @@
 	gaim_prefs_add_bool("/gaim/gtk/conversations/spellcheck", TRUE);
 	gaim_prefs_add_bool("/gaim/gtk/conversations/show_incoming_formatting", TRUE);
 
-	gaim_prefs_add_bool("/gaim/gtk/conversations/show_formatting_toolbar", TRUE);
+	gaim_prefs_add_bool("/gaim/gtk/conversations/show_formatting_toolbar", FALSE);
 	gaim_prefs_add_bool("/gaim/gtk/conversations/passthrough_unknown_commands", FALSE);
 
 	gaim_prefs_add_string("/gaim/gtk/conversations/placement", "last");
