--- linux-2.6.17-vanilla/include/asm-arm/arch-sa1100/jornada720.h	2006-06-18 01:49:35.000000000 +0000
+++ linux-2.6.17/include/asm-arm/arch-sa1100/jornada720.h	2006-07-10 17:06:11.000000000 +0000
@@ -11,6 +11,8 @@
 #error "include <asm/hardware.h> instead"
 #endif
 
+#define jornada720_mcu_read() jornada720_mcu_byte(MCU_TxDummy)
+
 #define SA1111_BASE             (0x40000000)
 
 #define GPIO_JORNADA720_KEYBOARD	GPIO_GPIO(0)
@@ -19,9 +21,38 @@
 #define GPIO_JORNADA720_KEYBOARD_IRQ	IRQ_GPIO0
 #define GPIO_JORNADA720_MOUSE_IRQ		IRQ_GPIO9
 
+/* Epson 1356 LCD controll parameters */
+#define REGISTER_OFFSET ((unsigned char *) 0xf0000000/* 0 */)
+#define DISP_MEM_OFFSET ((unsigned char *) 0xf1000000/* 0 */)
+#define DISP_MEM_OFFSET_PHYS ((unsigned char *) 0x48200000/* 0 filip wtf? */)
+#define ALLOCATED_FB_MEM_SIZE 0x80000 /* display memory size (512kb) */
+
+#define MAX_BPP              16
+#define MAX_XRES            640
+#define MAX_YRES            240
+#define MIN_XRES             64
+#define MIN_YRES             64
+
+/* MCU COMMANDS */
+#define MCU_GetBatteryData  0xc0
+#define MCU_GetScanKeyCode  0x90
+#define MCU_GetTouchSamples 0xa0
+#define MCU_GetContrast     0xD0
+#define MCU_SetContrast     0xD1
+#define MCU_GetBrightness   0xD2
+#define MCU_SetBrightness   0xD3
+#define MCU_ContrastOff     0xD8
+#define MCU_BrightnessOff   0xD9
+#define MCU_PWMOFF          0xDF
+#define MCU_TxDummy         0x11
+#define MCU_ErrorCode       0x00
+
 #ifndef __ASSEMBLY__
 
 void jornada720_mcu_init(void);
+int  jornada720_mcu_byte(int arg_data);
+int  jornada720_mcu_start(int arg_data);
+void jornada720_mcu_end(void);
 void jornada_contrast(int arg_contrast);
 void jornada720_battery(void);
 int jornada720_getkey(unsigned char *data, int size);
