Index: git/src/Makefile.am
===================================================================
--- git.orig/src/Makefile.am	2008-09-18 16:48:44.000000000 +0800
+++ git/src/Makefile.am	2008-09-18 16:49:54.000000000 +0800
@@ -44,7 +44,11 @@
 MAINTAINERCLEANFILES = Makefile.in
 
 connman.exp:
-	nm -B *.o | awk '{ print $$3 }' | sort -u | grep -E -e '^connman_' > $@
+	if test -n $$host ; then \
+		$(host)-nm -B *.o | awk '{ print $$3 }' | sort -u | grep -E -e '^connman_' > $@ ;\
+	else \
+		nm -B *.o | awk '{ print $$3 }' | sort -u | grep -E -e '^connman_' > $@ ;\
+	fi 
 
 connman.ver: connman.exp
 	echo "{ global:" > $@
