
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- guile-1.6.4/configure.in~configure-lossage
+++ guile-1.6.4/configure.in
@@ -462,7 +462,9 @@
 # explicitly to LIBOBJS to make sure that it is translated to
 # `alloca.lo' for libtool later on.  This can and should be done more cleanly.
 AC_FUNC_ALLOCA
-if test "$ALLOCA" = "alloca.o"; then AC_LIBOBJ([alloca]); fi
+if test "$ALLOCA" = "alloca.o"; then 
+   AC_LIBOBJ([alloca])
+fi
 
 AC_CHECK_MEMBERS([struct stat.st_rdev])
 AC_CHECK_MEMBERS([struct stat.st_blksize])
--- guile-1.6.4/guile-readline/configure.in~configure-lossage
+++ guile-1.6.4/guile-readline/configure.in
@@ -2,6 +2,7 @@
 
 AC_INIT
 AC_CONFIG_SRCDIR([readline.c])
+AC_CONFIG_AUX_DIR([..])
 . $srcdir/../GUILE-VERSION
 PACKAGE=guile-readline
 AM_INIT_AUTOMAKE($PACKAGE, $VERSION, no-define)
