--- /tmp/misc.cpp	2009-01-18 21:24:26.000000000 +0100
+++ ekiga-3.0.2/src/gui/misc.cpp	2009-01-18 21:24:47.000000000 +0100
@@ -75,42 +75,6 @@
   return button;
 }
 
-
-
-/* This function overrides from a pwlib function */
-#ifndef STATIC_LIBS_USED
-static gboolean
-assert_error_msg (gpointer data)
-{
-  GtkWidget *main_window = NULL;
-
-  main_window = GnomeMeeting::Process ()->GetMainWindow ();
-
-  
-  //gdk_threads_enter ();
-  std::cout << "FIXME" << std::endl << std::flush;
-  //FIXME Should use a signal in the runtime.
-  gnomemeeting_error_dialog (GTK_WINDOW (main_window), 
-			     _("Error"),
-			     "%s", (gchar *) data);
-  //gdk_threads_leave ();
-
-  return FALSE;
-}
-
-
-void 
-PAssertFunc (G_GNUC_UNUSED const char *file,
-	     G_GNUC_UNUSED int line,
-	     G_GNUC_UNUSED const char *className,
-	     const char *msg)
-{
-  g_idle_add_full (G_PRIORITY_HIGH, assert_error_msg, g_strdup (msg), g_free);
-}
-#endif
-
-
-
 PString
 gnomemeeting_pstring_cut (PString s)
 {
