Index: LVM2.2.01.15/make.tmpl.in
===================================================================
--- LVM2.2.01.15.orig/make.tmpl.in	2008-03-26 11:24:50.000000000 +0000
+++ LVM2.2.01.15/make.tmpl.in	2008-03-26 11:25:02.000000000 +0000
@@ -72,7 +72,7 @@
   CFLAGS += -DINTL_PACKAGE=\"@INTL_PACKAGE@\" -DLOCALEDIR=\"@LOCALEDIR@\"
 endif
 
-LDFLAGS += -L$(top_srcdir)/lib -L$(libdir)
+LDFLAGS += -L$(top_srcdir)/lib
 
 #CFLAGS += -DDEBUG_POOL
 #CFLAGS += -DBOUNDS_CHECK
