Index: rpm-4.4/configure.ac
===================================================================
--- rpm-4.4.orig/configure.ac	2005-01-29 10:54:13.599022712 -0500
+++ rpm-4.4/configure.ac	2005-01-29 10:54:37.698359056 -0500
@@ -230,7 +230,7 @@
 dnl use defaults if cross-compiling, otherwise use the default path.
 dnl
 if test "$cross_compiling" = "yes"; then
-    MYPATH=""
+    MYPATH=":"
 else
 #    MYPATH="/bin:/usr/bin:/usr/local/bin:$PATH:/sbin:/usr/sbin:/usr/local/sbin:/opt/gnu/bin"
     MYPATH=$PATH
