Index: zaurusd/Makefile.am
===================================================================
--- zaurusd.orig/Makefile.am
+++ zaurusd/Makefile.am
@@ -4,5 +4,4 @@ SUBDIRS = apps config scripts
 MAINTAINERCLEANFILES = aclocal.m4 compile config.guess config.sub configure depcomp install-sh ltmain.sh Makefile.in missing
 
 snapshot:
-        $(MAKE) dist distdir=$(PACKAGE)-snap`date +"%Y%m%d"`
-
+	$(MAKE) dist distdir=$(PACKAGE)-snap`date +"%Y%m%d"`
