--- zd1211-4916-r0/src/zd1205.c.cborig	2005-03-08 14:20:22.846179000 -0800
+++ zd1211-4916-r0/src/zd1205.c	2005-03-08 14:21:09.432096896 -0800
@@ -267,7 +267,7 @@
 zd1205_SwTcbQ_t free_txq_buf, active_txq_buf;
 struct net_device *g_dev;
 zd_80211Obj_t dot11Obj = {0};
-#define RX_COPY_BREAK       0//1518 //we do bridge, don't care IP header alignment
+#define RX_COPY_BREAK MAX_WLAN_SIZE //It doesn't appear that unaligned frames work with TCP/UDP beyond a certain size
 #define BEFORE_BEACON       5
 /* Definition of Wireless Extension */
 
