--- gst-plugins-0.8.8.orig/gst-libs/gst/gconf/gconf.c	2005-02-02 09:14:01.000000000 +0100
+++ gst-plugins-0.8.8/gst-libs/gst/gconf/gconf.c	2005-08-13 23:28:36.000000000 +0200
@@ -205,9 +205,13 @@
   if (!ret) {
     ret = gst_element_factory_make (DEFAULT_AUDIOSINK, NULL);
 
-    if (!ret)
-      g_warning ("No GConf default audio sink key and %s doesn't work",
-          DEFAULT_AUDIOSINK);
+    if (!ret) {
+      ret = gst_element_factory_make ("esdsink", NULL);
+
+      if (!ret)
+        g_warning ("No GConf default audio sink key and neither %s nor esdsink work",
+            DEFAULT_AUDIOSINK);
+    }
   }
 
   return ret;
