
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- linux-2.6.9-rc4/arch/arm/Kconfig.debug~schedstats-arm
+++ linux-2.6.9-rc4/arch/arm/Kconfig.debug
@@ -57,6 +57,18 @@
 	  time and disk space needed for compilation of the kernel. If in
 	  doubt say N.
 
+config SCHEDSTATS
+	bool "Collect scheduler statistics"
+	depends on DEBUG_KERNEL && PROC_FS
+	help
+	  If you say Y here, additional code will be inserted into the
+	  scheduler and related routines to collect statistics about
+	  scheduler behavior and provide them in /proc/schedstat.  These
+	  stats may be useful for both tuning and debugging the scheduler
+	  If you aren't debugging the scheduler or trying to tune a specific
+	  application, you can say N to avoid the very slight overhead
+	  this adds.
+
 # These options are only for real kernel hackers who want to get their hands dirty.
 config DEBUG_LL
 	bool "Kernel low-level debugging functions"
