Index: classpath-0.97.2/native/jni/java-io/java_io_VMFile.c
===================================================================
--- classpath-0.97.2.orig/native/jni/java-io/java_io_VMFile.c	2008-10-10 15:24:54.000000000 +0200
+++ classpath-0.97.2/native/jni/java-io/java_io_VMFile.c	2008-10-10 15:25:36.000000000 +0200
@@ -439,7 +439,7 @@
 {
 #ifndef WITHOUT_FILESYSTEM
   const char *filename;
-  int result;
+  volatile int result;
 
   /* Don't use the JCL convert function because it throws an exception
      on failure */
