--- gnuplot-4.0.0/src/term.h.old	2005-03-01 15:17:46.424111687 +0000
+++ gnuplot-4.0.0/src/term.h	2005-03-01 15:18:50.961405665 +0000
@@ -54,6 +54,7 @@
  */
 #ifdef SHORT_TERMLIST
 # include "dumb.trm"		/* dumb terminal */
+# include "qtopia.trm"         /* QTopia terminal */
 # include "post.trm"		/* postscript */
 # include "table.trm"		/* built-in, but used for the documentation */
 # if !(defined(OS2) || defined(MSDOS) || defined(_Windows) || defined(ATARI) || defined(MTOS) || defined(AMIGA))
@@ -432,6 +433,9 @@
 /* TeXDraw drawing package for LaTeX */
 #include "texdraw.trm"
 
+/* Qtopia */
+#include "qtopia.trm"
+
 /* METAFONT */
 #include "metafont.trm"
 
