Index: configure.in
===================================================================
RCS file: /cvsroot/fltk/fltk/configure.in,v
retrieving revision 1.33.2.31.2.91.2.18
diff -u -r1.33.2.31.2.91.2.18 configure.in
--- fltk.orig/configure.in	23 Jul 2004 19:42:50 -0000	1.33.2.31.2.91.2.18
+++ fltk/configure.in	7 Aug 2004 14:11:39 -0000
@@ -276,7 +276,9 @@
 
 dnl How do we make libraries?
 AC_PROG_RANLIB
-AC_PATH_PROG(AR, ar)
+if test "x$AR" = "x:"; then
+    AC_PATH_PROG(AR, ar)
+fi
 
 if test "x$AR" = "x:"; then
     AC_MSG_ERROR(Configure could not find the library archiver, aborting.)
@@ -293,6 +295,9 @@
 dnl Architecture checks...
 AC_C_BIGENDIAN
 
+AH_TEMPLATE([U16], [Define U16 type])
+AH_TEMPLATE([U32], [Define U32 type])
+AH_TEMPLATE([U64], [Define U64 type])
 AC_CHECK_SIZEOF(short, 2)
 AC_CHECK_SIZEOF(int, 4)
 AC_CHECK_SIZEOF(long, 4)
@@ -335,15 +340,18 @@
 fi
 
 dnl Standard headers and functions...
+AH_TEMPLATE([HAVE_SYS_SELECT_H], [Define if system provides select.h])
 AC_HEADER_DIRENT
 AC_CHECK_HEADER(sys/select.h,AC_DEFINE(HAVE_SYS_SELECT_H))
 AC_CHECK_HEADER(sys/stdtypes.h,AC_DEFINE(HAVE_SYS_SELECT_H))
+AH_TEMPLATE([HAVE_SCANDIR], [DOCUMENT_ME])
 AC_CHECK_FUNC(scandir,
     if test "x$uname" = xSunOS -o "x$uname" = xQNX; then
         AC_MSG_WARN(Not using $uname scandir emulation function.)
     else
         AC_DEFINE(HAVE_SCANDIR)
     fi)
+AH_TEMPLATE([HAVE_VSNPRINTF], [Define to use system vsnprintf])
 AC_CHECK_FUNC(vsnprintf,[
     case "$uname" in
     	HP-UX*)
@@ -366,6 +374,7 @@
             AC_DEFINE(HAVE_VSNPRINTF)
 	    ;;
     esac])
+AH_TEMPLATE([HAVE_SNPRINTF], [Define to use system snprintf])
 AC_CHECK_FUNC(snprintf,[
     case "$uname" in
     	HP-UX*)
@@ -433,6 +442,7 @@
 	ac_cv_lib_png_png_set_tRNS_to_alpha=no
     fi])
 
+AH_TEMPLATE([HAVE_PNG_H], [DOCUMENT_ME])
 AC_CHECK_LIB(png,png_set_tRNS_to_alpha, [
     PNGINC=""
     PNG=""
@@ -450,6 +460,9 @@
 AC_SUBST(ZLIB)
 AC_SUBST(ZLIBINC)
 
+AH_TEMPLATE([HAVE_LIBJPEG], [DOCUMENT_ME])
+AH_TEMPLATE([HAVE_LIBPNG], [DOCUMENT_ME])
+AH_TEMPLATE([HAVE_LIBZ], [DOCUMENT_ME])
 AC_DEFINE(HAVE_LIBJPEG)
 AC_DEFINE(HAVE_LIBPNG)
 AC_DEFINE(HAVE_LIBZ)
@@ -461,6 +474,7 @@
 AC_EXEEXT
 
 dnl Check for pthreads for multi-threaded apps...
+AH_TEMPLATE([HAVE_PTHREAD_H], [DOCUMENT_ME])
 have_pthread=no
 
 if test "x$enable_threads" = xyes; then
@@ -493,6 +507,12 @@
 
 AC_ARG_WITH(links, [  --with-links            make header links for common misspellings])
 
+AH_TEMPLATE([HAVE_GL], [DOCUMENT_ME])
+AH_TEMPLATE([HAVE_GL_GLU_H], [DOCUMENT_ME])
+AH_TEMPLATE([HAVE_PTHREAD], [DOCUMENT_ME])
+AH_TEMPLATE([USE_XFT], [DOCUMENT_ME])
+AH_TEMPLATE([HAVE_XDBE], [DOCUMENT_ME])
+AH_TEMPLATE([HAVE_OVERLAY], [DOCUMENT_ME])
 case $uname in
     CYGWIN* | MINGW*)
 	dnl Cygwin environment...
@@ -921,6 +941,7 @@
 OPTIM="$DEBUGFLAG $OPTIM"
 
 dnl Define the FLTK documentation directory...
+AH_TEMPLATE([FLTK_DOCDIR], [Define the FLTK documentation directory])
 if test x$prefix = xNONE; then
     AC_DEFINE_UNQUOTED(FLTK_DOCDIR, "/usr/local/share/doc/fltk")
 else
@@ -928,12 +949,52 @@
 fi
 
 dnl Define the FLTK data directory...
+AH_TEMPLATE([FLTK_DATADIR], [Define the FLTK data directory])
 if test x$prefix = xNONE; then
     AC_DEFINE_UNQUOTED(FLTK_DATADIR, "/usr/local/share/fltk")
 else
     AC_DEFINE_UNQUOTED(FLTK_DATADIR, "$prefix/share/fltk")
 fi
 
+AH_VERBATIM([BORDER_WIDTH],
+[/* Thickness of FL_UP_BOX and FL_DOWN_BOX.  Current 1,2, and 3 are
+ * supported.
+ *
+ * 3 is the historic FLTK look.
+ * 2 is the default and looks like Microsoft Windows, KDE, and Qt.
+ * 1 is a plausible future evolution...
+ *
+ * Note that this may be simulated at runtime by redefining the boxtypes
+ * using Fl::set_boxtype(). */
+#ifndef BORDER_WIDTH
+# define BORDER_WIDTH 2
+#endif])
+
+AH_VERBATIM([USE_COLORMAP],
+[/* Setting this to zero will save a good deal of code (especially for
+ * fl_draw_image), but FLTK will only work on TrueColor visuals. */
+#ifndef USE_COLORMAP
+# define USE_COLORMAP 1
+#endif])
+
+AH_VERBATIM([USE_XDBE],
+[/* Actually try to use the double-buffer extension? */
+#ifndef USE_XDBE
+# define USE_XDBE HAVE_XDBE
+#endif])
+
+AH_VERBATIM([WORDS_BIGENDIAN],
+[/* Byte order of your machine: 1 = big-endian, 0 = little-endian */
+#ifndef WORDS_BIGENDIAN
+# define WORDS_BIGENDIAN 0
+#endif])
+
+AH_VERBATIM([USE_POLL],
+[/* Use the poll() call provided on Linux and Irix instead of select() */
+#ifndef USE_POLL
+# define USE_POLL 0
+#endif])
+
 dnl Write all of the files...
 AC_CONFIG_HEADER(config.h:configh.in)
 AC_OUTPUT(makeinclude fltk.list fltk-config FL/Makefile)
