diff -Nurd linux-2.6.24.orig//drivers/mtd/chips/cfi_cmdset_0002.c linux-2.6.24/drivers/mtd/chips/cfi_cmdset_0002.c
--- linux-2.6.24.orig//drivers/mtd/chips/cfi_cmdset_0002.c	2011-03-10 17:05:22.868634509 +0100
+++ linux-2.6.24/drivers/mtd/chips/cfi_cmdset_0002.c	2011-03-10 17:06:58.723636256 +0100
@@ -219,6 +219,7 @@
 static struct cfi_fixup cfi_fixup_table[] = {
 #ifdef AMD_BOOTLOC_BUG
 	{ CFI_MFR_AMD, CFI_ID_ANY, fixup_amd_bootblock, NULL },
+	{ CFI_MFR_AMIC, CFI_ID_ANY, fixup_amd_bootblock, NULL },
 #endif
 	{ CFI_MFR_AMD, 0x0050, fixup_use_secsi, NULL, },
 	{ CFI_MFR_AMD, 0x0053, fixup_use_secsi, NULL, },
diff -Nurd linux-2.6.24.orig//include/linux/mtd/cfi.h linux-2.6.24/include/linux/mtd/cfi.h
--- linux-2.6.24.orig//include/linux/mtd/cfi.h	2011-03-10 17:05:19.584884614 +0100
+++ linux-2.6.24/include/linux/mtd/cfi.h	2011-03-10 17:08:03.886636402 +0100
@@ -485,6 +485,7 @@
 #define CFI_ID_ANY  0xffff
 
 #define CFI_MFR_AMD 0x0001
+#define CFI_MFR_AMIC 0x0037
 #define CFI_MFR_ATMEL 0x001F
 #define CFI_MFR_ST  0x0020 	/* STMicroelectronics */
 
