diff -urN bzip2-1.0.2.orig/Makefile-libbz2_so bzip2-1.0.2/Makefile-libbz2_so
--- bzip2-1.0.2.orig/Makefile-libbz2_so	2001-12-30 02:15:59.000000000 +0000
+++ bzip2-1.0.2/Makefile-libbz2_so	2009-04-02 12:42:43.000000000 +0000
@@ -20,8 +20,8 @@
       bzlib.o
 
 all: $(OBJS)
-	$(CC) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.2 $(OBJS)
-	$(CC) $(CFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.2
+	$(CC) $(LDFLAGS) -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.2 $(OBJS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2-shared bzip2.c libbz2.so.1.0.2
 	rm -f libbz2.so.1.0
 	ln -s libbz2.so.1.0.2 libbz2.so.1.0
 
