diff -Nurd avahi-0.1.orig/avahi-glib.pc.in avahi-0.1/avahi-glib.pc.in
--- avahi-0.1.orig/avahi-glib.pc.in	2005-08-21 23:27:14 +0200
+++ avahi-0.1/avahi-glib.pc.in	2005-08-28 00:34:50 +0200
@@ -7,5 +7,5 @@
 Description: Avahi Multicast DNS Responder (GLib Support)
 Version: @PACKAGE_VERSION@
 Requires: glib-2.0
-Libs: -L${libdir} -lavahi-commong -lavahi-glib
+Libs: -L${libdir} -lavahi-common -lavahi-glib
 Cflags: -D_REENTRANT -I${includedir}
