diff --git a/board/neo1973/common/lowlevel_init.S b/board/neo1973/common/lowlevel_init.S
index 582aa0e..92a9c7b 100644
--- a/board/neo1973/common/lowlevel_init.S
+++ b/board/neo1973/common/lowlevel_init.S
@@ -67,11 +67,11 @@
 #define B0_PMC		 	0x0	/* normal */
 
 /* BANK1CON */
-#define B1_Tacs		 	0x0	/*  0clk */
-#define B1_Tcos		 	0x1	/*  1clk */
-#define B1_Tacc		 	0x4	/*  4clk */
-#define B1_Tcoh		 	0x1	/*  1clk */
-#define B1_Tah		 	0x0	/*  0clk */
+#define B1_Tacs		 	0x3	/*  4clk */
+#define B1_Tcos		 	0x3	/*  4clk */
+#define B1_Tacc		 	0x7	/*  14clk */
+#define B1_Tcoh		 	0x3	/*  4clk */
+#define B1_Tah		 	0x3	/*  4clk */
 #define B1_Tacp		 	0x0
 #define B1_PMC		 	0x0
 
