From beda5431f747669440b52c9340bda843d6440e69 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Thu, 23 Dec 2010 21:04:25 +0100
Subject: [PATCH] nokia/rx51: use -O0 as work around for gcc-4.5

---
 board/nokia/rx51/Makefile |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/board/nokia/rx51/Makefile b/board/nokia/rx51/Makefile
index 31ad6fd..986af23 100644
--- a/board/nokia/rx51/Makefile
+++ b/board/nokia/rx51/Makefile
@@ -23,6 +23,9 @@
 
 include $(TOPDIR)/config.mk
 
+# gcc-4.5 breaks generated code if -O1+ or -Os is used
+CFLAGS += -O0
+
 LIB	= $(obj)lib$(BOARD).a
 
 COBJS	:= rx51.o
-- 
1.7.3.4

