Index: button/buttonsettings.cpp
===================================================================
RCS file: /cvs/opie/core/settings/button/buttonsettings.cpp,v
retrieving revision 1.13
retrieving revision 1.14
diff -u -u -r1.13 -r1.14
--- button/buttonsettings.cpp	24 Apr 2005 20:24:00 -0000	1.13
+++ button/buttonsettings.cpp	12 Oct 2005 21:08:35 -0000	1.14
@@ -37,7 +37,7 @@
 #include <qlabel.h>
 #include <qtimer.h>
 #include <qscrollview.h>
-
+#include <qcopchannel_qws.h>
 
 using namespace Opie::Core;
 
@@ -157,6 +157,7 @@
 ButtonSettings::~ButtonSettings ( )
 {
     QPEApplication::ungrabKeyboard ( );
+    QCopChannel::send ("QPE/System", "deviceButtonMappingChanged()" );
 }
 
 void ButtonSettings::updateLabels ( )
