--- linux-2.6.17-vanilla/arch/arm/Kconfig	2006-06-18 01:49:35.000000000 +0000
+++ linux-2.6.17/arch/arm/Kconfig	2006-07-10 17:06:11.000000000 +0000
@@ -669,7 +669,7 @@
 
 config CPU_FREQ_SA1110
 	bool
-	depends on CPU_FREQ && (SA1100_ASSABET || SA1100_CERF || SA1100_PT_SYSTEM3)
+	depends on CPU_FREQ && (SA1100_ASSABET || SA1100_CERF || SA1100_PT_SYSTEM3 || SA1100_JORNADA720)
 	default y
 
 config CPU_FREQ_INTEGRATOR
