Index: main.c
===================================================================
RCS file: /cvs/gpe/base/gpe-mininet/main.c,v
retrieving revision 1.7
diff -u -r1.7 main.c
--- main.c	9 Jun 2004 22:52:09 -0000	1.7
+++ main.c	9 Jul 2004 18:46:58 -0000
@@ -329,7 +329,7 @@
 	GdkPixbuf *pixmap;
 
 	/* first check firewall stuff */
-	activate_networking_rules();
+//	activate_networking_rules();
 	
 	if (gpe_application_init (&argc, &argv) == FALSE)
 		exit (1);
