--- radvd-0.7.2/Makefile.am.old	2003-12-24 21:22:03.000000000 +0000
+++ radvd-0.7.2/Makefile.am	2003-12-24 21:21:28.000000000 +0000
@@ -18,7 +18,7 @@
 	-DPATH_RADVD_LOG=\"$(PATH_RADVD_LOG)\" -DLOG_FACILITY=$(LOG_FACILITY) \
 	-DPATH_RADVD_PID=\"$(PATH_RADVD_PID)\" \
 	-DVERSION=\"$(VERSION)\" -DINET6=1
-INCLUDES+=-I$(srcdir) -I.
+INCLUDES=-I$(srcdir) -I.
 
 ########################################################################
 
--- radvd-0.7.2/configure.in~	2002-06-15 11:25:04.000000000 +0100
+++ radvd-0.7.2/configure.in	2003-12-24 22:29:14.000000000 +0000
@@ -124,9 +124,11 @@
 AC_CHECK_HEADERS(sys/time.h)
 AC_HEADER_TIME
 
-AC_CHECK_HEADER(netinet/ipv6.h, hdrfound=yes,
- AC_CHECK_HEADER(netinet/ip6.h, hdrfound=yes, hdrfound=no)
-)
+AC_CHECK_HEADER(netinet/ipv6.h, hdrfound=yes, hdrfound=no)
+if test "$hdrfound" = no
+then
+AC_CHECK_HEADER(netinet/ip6.h, hdrfound=yes, hdrfound=no)
+fi
 if test "$hdrfound" = no
 then
  AC_MSG_CHECKING(for netinet/ip6.h or ipv6.h in /usr/inet6/include)
