From 4f62a6e6738b4563aa1015e6c165a30231363e1f Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Sun, 21 Feb 2010 15:01:45 +0100
Subject: [PATCH] glxdri2: remove flushInvalidate()

---
 glx/glxdri2.c |    3 ---
 1 files changed, 0 insertions(+), 3 deletions(-)

diff --git a/glx/glxdri2.c b/glx/glxdri2.c
index e4f2d60..dae7b42 100644
--- a/glx/glxdri2.c
+++ b/glx/glxdri2.c
@@ -225,9 +225,6 @@ __glXDRIdrawableSwapBuffers(ClientPtr client, __GLXdrawable *drawable)
     if (screen->flush)
 	(*screen->flush->invalidate)(priv->driDrawable);
 
-    if (screen->flush && screen->flush->flushInvalidate)
-	(*screen->flush->flushInvalidate)(priv->driDrawable);
-
     return TRUE;
 }
 
-- 
1.7.0

