diff -urN fbvncserver-0.9.4.old/fbvncserver.c fbvncserver-0.9.4/fbvncserver.c
--- fbvncserver-0.9.4.old/fbvncserver.c	2003-03-03 01:00:09.000000000 -0500
+++ fbvncserver-0.9.4/fbvncserver.c	2004-02-27 11:35:36.000000000 -0500
@@ -31,7 +31,7 @@
 #define FANCY_PANEL_JPG "/usr/local/vnc/ipaq_panel.jpg"
 #define DESKTOP_NAME "iPAQ"
 #else   /* #ifdef ZAURUS */
-#define FANCY_PANEL_JPG "/home/root/Applications/fbvncserver/zaurus_panel.jpg"
+#define FANCY_PANEL_JPG "/usr/share/fbvncserver/zaurus_panel.jpg"
 #define DESKTOP_NAME "Zaurus"
 #endif
 
@@ -578,7 +578,7 @@
   rfbScreen->ptrAddEvent     = ptrevent;
   rfbScreen->kbdAddEvent     = keyevent;
   rfbScreen->newClientHook   = newclient;
-  rfbScreen->httpDir         = "/home/root/Applications/fbvncserver/classes";
+  rfbScreen->httpDir         = "/usr/share/fbvncserver/classes";
   rfbInitServer(rfbScreen);
 
   /* Make sure we catch EPIPE so we don't get killed by closed sockets when
diff -urN fbvncserver-0.9.4.old/launch/subclass.cpp fbvncserver-0.9.4/launch/subclass.cpp
--- fbvncserver-0.9.4.old/launch/subclass.cpp	2003-05-10 19:20:02.000000000 -0400
+++ fbvncserver-0.9.4/launch/subclass.cpp	2004-02-27 11:35:55.000000000 -0500
@@ -17,13 +17,13 @@
 #include <qpe/process.h>
 #include <qpe/global.h>
 
-#define SCRIPT_LOCATION "/usr/local/bin"
+#define SCRIPT_LOCATION "/usr/bin"
 
 
 mainDialog::mainDialog(QWidget *parent, const char *name, WFlags fl)
     : mainDialogSuper(parent, name, fl)
 {
-    QPixmap qi("/home/root/Applications/fbvncserver/sdglogo.png", "PNG");
+    QPixmap qi("/usr/share/fbvncserver/sdglogo.png", "PNG");
     QSizePolicy qsp(QSizePolicy::Fixed, QSizePolicy::Fixed);
     SDG_Pixmap->setSizePolicy(qsp);
     SDG_Pixmap->setPixmap(qi);
diff -urN fbvncserver-0.9.4.old/splash/subclass.cpp fbvncserver-0.9.4/splash/subclass.cpp
--- fbvncserver-0.9.4.old/splash/subclass.cpp	2003-04-29 23:23:41.000000000 -0400
+++ fbvncserver-0.9.4/splash/subclass.cpp	2004-02-27 11:35:36.000000000 -0500
@@ -17,7 +17,7 @@
 fbvncserverSplash::fbvncserverSplash(QWidget *parent, const char *name, bool modal, WFlags fl)
     : fbvncserverSplashSuper(parent, name, modal, fl)
 {
-    QPixmap qi("/home/root/Applications/fbvncserver/sdglogo.png", "PNG");
+    QPixmap qi("/usr/share/fbvncserver/sdglogo.png", "PNG");
     QSizePolicy qsp(QSizePolicy::Fixed, QSizePolicy::Fixed);
     SDG_Pixmap->setSizePolicy(qsp);
     SDG_Pixmap->setPixmap(qi);
