Index: ffmpeg.c
===================================================================
--- trunk/ffmpeg.c	(revision 434)
+++ trunk/ffmpeg.c	(working copy)
@@ -164,7 +164,8 @@
     return write(fd, buf, size);
 }
 
-static offset_t file_seek(URLContext *h, offset_t pos, int whence)
+
+static int64_t file_seek(URLContext *h, int64_t pos, int whence)
 {
     int fd = (size_t)h->priv_data;
     return lseek(fd, pos, whence);
