--- ip/Makefile.o	2004-08-05 12:38:25.836331936 +0200
+++ ip/Makefile	2004-08-05 12:38:42.256835640 +0200
@@ -22,7 +22,7 @@
 rtmon: $(RTMONOBJ) $(LIBNETLINK)
 
 install: all
-	install -m 0755 -s $(TARGETS) $(DESTDIR)$(SBINDIR)
+	install -m 0755 $(TARGETS) $(DESTDIR)$(SBINDIR)
 	install -m 0755 routel routef $(DESTDIR)$(SBINDIR)
 
 clean:
--- tc/Makefile.o	2004-08-05 12:37:33.325314824 +0200
+++ tc/Makefile	2004-08-05 12:38:12.085422392 +0200
@@ -61,8 +61,8 @@
 
 install: all
 	mkdir -p $(DESTDIR)/usr/lib/tc
-	install -m 0755 -s tc $(DESTDIR)$(SBINDIR)
-	for i in $(TCSO); do install -m 755 -s $$i $(DESTDIR)/usr/lib/tc; done
+	install -m 0755 tc $(DESTDIR)$(SBINDIR)
+	for i in $(TCSO); do install -m 755 $$i $(DESTDIR)/usr/lib/tc; done
 
 clean:
 	rm -f $(TCOBJ) $(TCLIB) libtc.a tc
