--- glib-2.2.3/gobject/Makefile.am.old	2004-03-06 20:48:33.000000000 +0000
+++ glib-2.2.3/gobject/Makefile.am	2004-03-06 20:50:58.000000000 +0000
@@ -87,7 +87,7 @@
 gobject_private_h_sources = @STRIP_BEGIN@ \
 @STRIP_END@
 # GObject library C sources to build the library from
-gobject_c_sources = @STRIP_BEGIN@ \
+gobject_c_sources =  \
 	gboxed.c		\
 	gclosure.c		\
 	genums.c		\
@@ -102,8 +102,7 @@
 	gvalue.c		\
 	gvaluearray.c		\
 	gvaluetransform.c	\
-	gvaluetypes.c		\
-@STRIP_END@
+	gvaluetypes.c
 
 # we use our own built_sources variable rules to avoid automake's
 # BUILT_SOURCES oddities
@@ -135,9 +134,9 @@
 #
 gobject_target_headers = $(gobject_public_h_sources) $(gobject_built_public_sources)
 gobject_target_sources = $(gobject_c_sources)
-MAINTAINERCLEANFILES += $(gobject_built_sources)
-EXTRA_HEADERS +=
-EXTRA_DIST += $(gobject_private_h_sources) $(gobject_extra_sources)
+MAINTAINERCLEANFILES = $(gobject_built_sources)
+EXTRA_HEADERS =
+EXTRA_DIST = $(gobject_private_h_sources) $(gobject_extra_sources)
 EXTRA_DIST += $(gobject_built_sources)
 
 #
@@ -145,7 +144,7 @@
 #
 # setup autogeneration dependancies
 gen_sources = xgen-gmh xgen-gmc xgen-gms
-CLEANFILES += $(gen_sources)
+CLEANFILES = $(gen_sources)
 
 $(libgobject_2_0_la_OBJECTS): $(gobject_built_sources)  ${gobject_built_public_sources} # this is our oldest file, used for implicit auto-generation deps
 # initial creation of the real stamp-* files
