Index: klibc-1.5.15/klcc/klcc.in
===================================================================
--- klibc-1.5.15.orig/klcc/klcc.in	2009-12-14 00:32:41.373661102 +0100
+++ klibc-1.5.15/klcc/klcc.in	2009-12-14 00:34:20.855735356 +0100
@@ -147,7 +147,7 @@
     } elsif ( $a =~ /^-([fmwWQdO]|std=|ansi|pedantic|M[GPD]|MMD)/ ) {
	# Options to gcc
	push(@ccopt, $a);
-    } elsif ( $a =~ /^-([DUI]|M[FQT])(.*)$/ ) {
+    } elsif ( $a =~ /^-([DUI]|M[FQT]|isystem)(.*)$/ ) {
	# Options to gcc, which can take either a conjoined argument
	# (-DFOO) or a disjoint argument (-D FOO)
	push(@ccopt, $a);
