--- a/services4/3rdparty/dc_omap3430_linux/omaplfb_linux.c
+++ b/services4/3rdparty/dc_omap3430_linux/omaplfb_linux.c
@@ -171,7 +171,7 @@
 #if !defined (CONFIG_OMAP2_DSS)
 	omap_dispc_free_irq(DISPC_IRQ_VSYNC, OMAPLFBVSyncISR, psSwapChain);
 #else
-	omap_dispc_unregister_isr(OMAPLFBVSyncISR);
+	omap_dispc_unregister_isr(OMAPLFBVSyncISR, psSwapChain, DISPC_IRQ_VSYNC);
 #endif
 	return PVRSRV_OK;		
 } 
