--- /tmp/Cpu.c	2009-09-25 12:41:08.000000000 +0200
+++ davinci_multimedia_application_interface/dmai/packages/ti/sdo/dmai/linux/Cpu.c	2009-09-25 12:41:36.000000000 +0200
@@ -107,6 +107,7 @@
     }
     else if ((strcmp(valBuf,"OMAP3EVM Board") == 0) ||
             (strcmp(valBuf, "OMAP3 EVM") == 0) ||
+            (strcmp(valBuf, "OMAP3 Touch Book") == 0) ||
             (strcmp(valBuf, "OMAP3 Beagle Board") == 0)) {
             *device = Cpu_Device_OMAP3530;
     }
