From 60455ac7f09e4d1c1472eb206e66185d9fd89ef9 Mon Sep 17 00:00:00 2001
From: Steve Sakoman <steve@sakoman.com>
Date: Mon, 18 Aug 2008 11:11:57 -0700
Subject: [PATCH] omap3evm: provide missing udelay function

---
 board/omap3evm/omap3evm.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/board/omap3evm/omap3evm.c b/board/omap3evm/omap3evm.c
index 857516b..d4ef29b 100755
--- a/board/omap3evm/omap3evm.c
+++ b/board/omap3evm/omap3evm.c
@@ -70,6 +70,10 @@ static inline void delay(unsigned long loops)
 			  "bne 1b":"=r" (loops):"0"(loops));
 }
 
+void udelay (unsigned long usecs) {
+	delay(usecs);
+}
+
 /*****************************************
  * Routine: board_init
  * Description: Early hardware init.
-- 
1.5.4.3

