--- src/Makefile.porig	2004-06-10 15:03:25.000000000 -0400
+++ src/Makefile	2004-06-11 20:16:30.701134886 -0400
@@ -6,7 +6,7 @@
 #
 
 CC      = gcc
-STRIP   = strip
+STRIP   = -s
 O       = .o
 SO      = .so
 E       =
@@ -37,17 +37,17 @@
 #
 ifdef MATRIXSSL_DEBUG
 DFLAGS	= -g -Wall -DDEBUG
-STRIP	= test
+STRIP =
 else
-DFLAGS	= -O3
+DFLAGS	= -O2
 endif
 
 #
 #	Compile options
 #
-SHARED	= -shared
-CFLAGS  = $(DFLAGS) -DLINUX
-LDFLAGS = -nostdlib -lc -lpthread
+SHARED = -shared
+CFLAGS = $(DFLAGS) $(STRIP) -DLINUX
+LDLIBS = -nostdlib -lc -lpthread
 
 #
 #	Override variables for compilation on Mac OS X (Darwin)
@@ -57,8 +57,8 @@
 SO	= .dylib
 SHARED	= -dynamiclib
 CFLAGS += -DOSX -isystem -I/usr/include
-LDFLAGS = -flat_namespace -install_name @executable_path/../src/$(LIBRARY)
-STRIP = test
+LDLIBS = -flat_namespace -install_name @executable_path/../src/$(LIBRARY)
+STRIP =
 endif
 
 all: compile
@@ -77,8 +77,7 @@
 # Build the library
 #
 $(LIBRARY): $(OBJECTS)
-	$(CC) $(SHARED) -o $@ $^ $(LDFLAGS)
-	$(STRIP) $(LIBRARY)
+	$(CC) $(SHARED) -o $@ $^ $(LDFLAGS) $(LDLIBS)
 
 #
 #	Clean up all generated files
