diff -urpN busybox-1.13.2/procps/top.c busybox-1.13.2-top24/procps/top.c
--- busybox-1.13.2/procps/top.c	2008-11-09 18:28:20.000000000 +0100
+++ busybox-1.13.2-top24/procps/top.c	2009-03-02 22:24:31.000000000 +0100
@@ -200,7 +200,7 @@ static NOINLINE int read_cpu_jiffy(FILE 
 			&p_jif->usr, &p_jif->nic, &p_jif->sys, &p_jif->idle,
 			&p_jif->iowait, &p_jif->irq, &p_jif->softirq,
 			&p_jif->steal);
-	if (ret > 4) {
+	if (ret >= 4) {
 		p_jif->total = p_jif->usr + p_jif->nic + p_jif->sys + p_jif->idle
 			+ p_jif->iowait + p_jif->irq + p_jif->softirq + p_jif->steal;
 		/* procps 2.x does not count iowait as busy time */
