
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- enscript-1.6.4/configure.in~autotools
+++ enscript-1.6.4/configure.in
@@ -11,6 +11,8 @@
 
 AC_PROG_INSTALL
 
+AC_PROG_CC
+
 AC_ARG_WITH(cc,
 [  --with-cc(=CC)	  use system's native compiler (or compiler CC)],
   if test "X$withval" != "Xno"; then
@@ -25,10 +27,6 @@
   fi
 )
 
-if test "X$CC" = "X"; then
-  AC_PROG_CC
-fi
-
 AC_ISC_POSIX
 AM_C_PROTOTYPES
 
--- enscript-1.6.4/Makefile.am~autotools
+++ enscript-1.6.4/Makefile.am
@@ -31,4 +31,4 @@
 88595.txt 88597.txt 88599.txt 885910.txt ibmpc.txt mac.txt vms.txt	\
 hp8.txt koi8.txt mtr.eps mtr_small.eps mtr_tiny.eps
 
-ACLOCAL_AMFLAGS = -I m4
+#ACLOCAL_AMFLAGS = -I m4
--- enscript-1.6.4/afm/Makefile.am~autotools
+++ enscript-1.6.4/afm/Makefile.am
@@ -36,11 +36,11 @@
 EXTRA_DIST = font.map $(default_afm) $(public_fonts)
 
 install-data-local:
-	$(top_srcdir)/mkinstalldirs $(datadir)/enscript
-	$(top_srcdir)/mkinstalldirs $(datadir)/enscript/afm
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)$(datadir)/enscript
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)$(datadir)/enscript/afm
 	for f in $(EXTRA_DIST); do \
-	  $(INSTALL_DATA) $(srcdir)/$$f $(datadir)/enscript/afm/$$f; \
+	  $(INSTALL_DATA) $(srcdir)/$$f $(DESTDIR)$(datadir)/enscript/afm/$$f; \
 	done
 
 uninstall-local:
-	rm -rf $(datadir)/enscript/afm
+	rm -rf $(DESTDIR)$(datadir)/enscript/afm
--- enscript-1.6.4/lib/Makefile.am~autotools
+++ enscript-1.6.4/lib/Makefile.am
@@ -38,15 +38,15 @@
 all-local: enscript.cfg
 
 install-data-local: enscript.cfg
-	$(top_srcdir)/mkinstalldirs $(sysconfdir)
-	if test -r $(sysconfdir)/enscript.cfg; then \
-	  cp $(sysconfdir)/enscript.cfg $(sysconfdir)/enscript.cfg.old; \
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)$(sysconfdir)
+	if test -r $(DESTDIR)$(sysconfdir)/enscript.cfg; then \
+	  cp $(DESTDIR)$(sysconfdir)/enscript.cfg $(DESTDIR)$(sysconfdir)/enscript.cfg.old; \
 	else :; \
 	fi
-	$(INSTALL_DATA) enscript.cfg $(sysconfdir)/enscript.cfg
+	$(INSTALL_DATA) enscript.cfg $(DESTDIR)$(sysconfdir)/enscript.cfg
 
 uninstall-local:
-	rm -f $(sysconfdir)/enscript.cfg
+	rm -f $(DESTDIR)$(sysconfdir)/enscript.cfg
 
 enscript.cfg: $(srcdir)/enscript.cfg.in Makefile
 	sed 's%@DATADIR@%$(datadir)%g; s%@media@%@MEDIA@%g; s%@BINDIR@%$(bindir)%g; s%@spooler@%@SPOOLER@%g; s%@pslevel@%@PSLEVEL@%g' \
--- enscript-1.6.4/states/hl/Makefile.am~autotools
+++ enscript-1.6.4/states/hl/Makefile.am
@@ -45,11 +45,11 @@
 EXTRA_DIST = $(states)
 
 install-data-local:
-	$(top_srcdir)/mkinstalldirs $(datadir)/enscript
-	$(top_srcdir)/mkinstalldirs $(datadir)/enscript/hl
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)$(datadir)/enscript
+	$(top_srcdir)/mkinstalldirs $(DESTDIR)$(datadir)/enscript/hl
 	for f in $(states); do \
-	  $(INSTALL_DATA) $(srcdir)/$$f $(datadir)/enscript/hl/$$f; \
+	  $(INSTALL_DATA) $(srcdir)/$$f $(DESTDIR)$(datadir)/enscript/hl/$$f; \
 	done
 
 uninstall-local:
-	rm -rf $(datadir)/enscript/hl
+	rm -rf $(DESTDIR)$(datadir)/enscript/hl
