Index: xboard-4.2.7-r0/xboard-4.2.7/Makefile.in
===================================================================
--- xboard-4.2.7-r0.orig/xboard-4.2.7/Makefile.in
+++ xboard-4.2.7-r0/xboard-4.2.7/Makefile.in
@@ -63,8 +63,8 @@ default: xboard zic2xpm
 all: default pseudosource info FAQ html dvi ps
 
 install: installdirs default
-	$(INSTALL_PROGRAM) -s xboard $(bindir)/xboard
-	$(INSTALL_PROGRAM) -s zic2xpm $(bindir)/zic2xpm
+	$(INSTALL_PROGRAM) xboard $(bindir)/xboard
+	$(INSTALL_PROGRAM) zic2xpm $(bindir)/zic2xpm
 	$(INSTALL_PROGRAM) cmail $(bindir)/cmail
 	$(INSTALL_PROGRAM) $(srcdir)/pxboard $(bindir)/pxboard
 	$(INSTALL_DATA) $(srcdir)/xboard.man $(man6dir)/xboard$(man6ext)
