Index: PyQt-x11-gpl-4.3.3/sip/QtAssistant/qassistantclient.sip
===================================================================
--- PyQt-x11-gpl-4.3.3.orig/sip/QtAssistant/qassistantclient.sip	2008-01-07 13:31:35.000000000 +0100
+++ PyQt-x11-gpl-4.3.3/sip/QtAssistant/qassistantclient.sip	2008-01-07 13:32:20.000000000 +0100
@@ -26,7 +26,7 @@
 {
 
 %TypeHeaderCode
-#include <qassistantclient.h>
+#include <QtAssistant/qassistantclient.h>
 %End
 
 %ConvertToSubClassCode
