Description: Install pphs as a user script, not a library
Author: Jonas Smedegaard <dr@jones.dk>
--- a/base/unixinst.mak	2009-05-24 18:40:45.000000000 +0200
+++ b/base/unixinst.mak	2009-05-24 18:43:02.000000000 +0200
@@ -39,7 +39,7 @@
 	$(SH) -c 'for f in \
 gsbj gsdj gsdj500 gslj gslp gsnd \
 bdftops dumphint dvipdf eps2eps font2c \
-pdf2dsc pdf2ps pdfopt pf2afm pfbtopfa printafm \
+pdf2dsc pdf2ps pdfopt pf2afm pfbtopfa pphs printafm \
 ps2ascii ps2epsi ps2pdf ps2pdf12 ps2pdf13 ps2pdf14 ps2pdfwr ps2ps ps2ps2 \
 wftopfa fixmswrd.pl lprsetup.sh pj-gs.sh pv.sh sysvlp.sh unix-lpr.sh ;\
 	do if ( test -f $(PSLIBDIR)/$$f ); then \
@@ -72,7 +72,7 @@
 jispaper.ps landscap.ps level1.ps lines.ps markhint.ps markpath.ps \
 mkcidfm.ps opdfread.ps PDFA_def.ps PDFX_def.ps \
 packfile.ps pcharstr.ps pf2afm.ps pfbtopfa.ps ppath.ps \
-pphs pphs.ps \
+pphs.ps \
 prfont.ps printafm.ps \
 ps2ai.ps ps2ascii.ps ps2epsi.ps quit.ps rollconv.ps \
 showchar.ps showpage.ps stcinfo.ps stcolor.ps stocht.ps \
