diff -Nurd sflphone-common/configure.ac sflphone-common.new//configure.ac
--- sflphone-common/configure.ac	2011-04-05 01:01:39.000000000 +0200
+++ sflphone-common.new//configure.ac	2011-05-28 10:59:32.443337188 +0200
@@ -43,10 +43,6 @@
 		libs/utilspp/Makefile \
 		libs/utilspp/functor/Makefile \
 		libs/utilspp/singleton/Makefile \
-		libs/dbus-c++/Makefile \
-		libs/dbus-c++/src/Makefile \
-		libs/dbus-c++/tools/Makefile \
-		libs/dbus-c++/data/Makefile \
 		libs/iax2/Makefile])
 
 AC_CONFIG_FILES([src/Makefile \
@@ -202,7 +198,7 @@
 LIBCCGNU2_MIN_VERSION=1.3.1
 PKG_CHECK_MODULES(CCGNU2, libccgnu2 >= ${LIBCCGNU2_MIN_VERSION},, AC_MSG_ERROR([Mising common cpp development package: libcommoncpp2-dev]))
 AC_SUBST(CCGNU2_LIBS)
-AC_SUBST(CCGNU2_FLAGS)
+AC_SUBST(CCGNU2_CFLAGS)
 
 LIBCCEXT2_MIN_VERSION=1.3.1
 PKG_CHECK_MODULES(CCEXT2, libccext2 >= ${LIBCCEXT2_MIN_VERSION})
@@ -225,6 +221,10 @@
 AC_SUBST(ZRTPCPP_LIBS)
 AC_SUBST(ZRTPCPP_CFLAGS)
 
+	LIBDBUSCPLUSPLUS_MIN_VERSION=0.6.0-pre1
+	PKG_CHECK_MODULES(DBUSCPLUSPLUS, dbus-c++-1 >= ${LIBDBUSCPLUSPLUS_MIN_VERSION})
+	AC_SUBST(DBUSCPLUSPLUS_LIBS)
+AC_SUBST(DBUSCPLUSPLUS_CFLAGS)
 
 dnl DBus-C++ detection (used to be in library own build system)
 DBUS_REQUIRED_VERSION=0.60
diff -Nurd sflphone-common/globals.mak sflphone-common.new//globals.mak
--- sflphone-common/globals.mak	2011-04-05 01:01:39.000000000 +0200
+++ sflphone-common.new//globals.mak	2011-05-28 11:00:10.586337202 +0200
@@ -33,8 +33,6 @@
 		   -I$(src)/libs/pjproject/pjmedia/include \
 		   -I$(src)/libs/pjproject/pjnath/include
 
-DBUSCPP_CFLAGS=$(top_srcdir)/libs/dbus-c++/include/dbus-c++
-
 if BUILD_SPEEX
 SPEEXCODEC=-DHAVE_SPEEX_CODEC
 else
@@ -50,7 +48,6 @@
 # Preprocessor flags
 AM_CPPFLAGS = \
 	-I$(src)/libs \
-	-I$(src)/libs/dbus-c++/include \
 	-I$(src)/libs/iax2 \
 	-I$(src)/libs/pjproject \
 	-I$(src)/src \
@@ -69,10 +66,3 @@
 	-DENABLE_TRACE \
          $(SPEEXCODEC) \
          $(GSMCODEC)
-
-
-indent:
-	@echo "Indenting code:"
-	if [ -f $(ASTYLERC) ] ; then \
-                find $(top_srcdir)/src/ -name \*.cpp -o -name \*.h | xargs $(indent) --options=$(ASTYLERC) ; \
-	fi
diff -Nurd sflphone-common/libs/Makefile.am sflphone-common.new//libs/Makefile.am
--- sflphone-common/libs/Makefile.am	2011-04-05 01:01:39.000000000 +0200
+++ sflphone-common.new//libs/Makefile.am	2011-05-28 11:00:35.028337210 +0200
@@ -1 +1 @@
-SUBDIRS = utilspp iax2 dbus-c++
+SUBDIRS = utilspp iax2
diff -Nurd sflphone-common/src/audio/alsa/Makefile.am sflphone-common.new//src/audio/alsa/Makefile.am
--- sflphone-common/src/audio/alsa/Makefile.am	2011-04-05 01:01:39.000000000 +0200
+++ sflphone-common.new//src/audio/alsa/Makefile.am	2011-05-28 11:00:59.180337220 +0200
@@ -4,5 +4,7 @@
 
 libalsalayer_la_SOURCES = alsalayer.cpp 
 
+libalsalayer_la_CXXFLAGS = @DBUSCPLUSPLUS_CFLAGS@
+
 noinst_HEADERS = alsalayer.h
 
diff -Nurd sflphone-common/src/audio/audiortp/Makefile.am sflphone-common.new//src/audio/audiortp/Makefile.am
--- sflphone-common/src/audio/audiortp/Makefile.am	2011-04-05 01:01:39.000000000 +0200
+++ sflphone-common.new//src/audio/audiortp/Makefile.am	2011-05-28 11:01:31.196337231 +0200
@@ -2,6 +2,8 @@
 
 noinst_LTLIBRARIES = libaudiortp.la
 
+libaudiortp_la_CXXFLAGS = @DBUSCPLUSPLUS_CFLAGS@
+
 libaudiortp_la_SOURCES = \
 		AudioRtpSession.cpp \
 		AudioRtpRecordHandler.cpp \
diff -Nurd sflphone-common/src/audio/Makefile.am sflphone-common.new//src/audio/Makefile.am
--- sflphone-common/src/audio/Makefile.am	2011-04-05 01:01:39.000000000 +0200
+++ sflphone-common.new//src/audio/Makefile.am	2011-05-28 11:03:13.367337267 +0200
@@ -8,6 +8,8 @@
 # SPEEXDSP=-DHAVE_SPEEXDSP_LIB
 # endif
 
+libaudio_la_CXXFLAGS = @DBUSCPLUSPLUS_CFLAGS@
+
 libaudio_la_SOURCES = \
 		audioloop.cpp \
 		ringbuffer.cpp \
diff -Nurd sflphone-common/src/audio/pulseaudio/Makefile.am sflphone-common.new//src/audio/pulseaudio/Makefile.am
--- sflphone-common/src/audio/pulseaudio/Makefile.am	2011-04-05 01:01:39.000000000 +0200
+++ sflphone-common.new//src/audio/pulseaudio/Makefile.am	2011-05-28 11:02:03.934337242 +0200
@@ -2,6 +2,8 @@
 
 noinst_LTLIBRARIES = libpulselayer.la
 
+libpulselayer_la_CXXFLAGS = @DBUSCPLUSPLUS_CFLAGS@
+
 libpulselayer_la_SOURCES = \
 		audiostream.cpp \
 		pulselayer.cpp
diff -Nurd sflphone-common/src/audio/sound/Makefile.am sflphone-common.new//src/audio/sound/Makefile.am
--- sflphone-common/src/audio/sound/Makefile.am	2011-04-05 01:01:39.000000000 +0200
+++ sflphone-common.new//src/audio/sound/Makefile.am	2011-05-28 11:02:48.411337259 +0200
@@ -2,6 +2,8 @@
 
 noinst_LTLIBRARIES = libsound.la
 
+libsound_la_CXXFLAGS = @DBUSCPLUSPLUS_CFLAGS@
+
 libsound_la_SOURCES = \
 		audiofile.cpp \
 		tone.cpp \
diff -Nurd sflphone-common/src/dbus/Makefile.am sflphone-common.new//src/dbus/Makefile.am
--- sflphone-common/src/dbus/Makefile.am	2011-04-05 01:01:39.000000000 +0200
+++ sflphone-common.new//src/dbus/Makefile.am	2011-05-28 11:04:16.179337289 +0200
@@ -1,6 +1,6 @@
 include $(top_srcdir)/globals.mak
 
-DBUSXX=$(top_srcdir)/libs/dbus-c++/tools/dbusxx-xml2cpp
+DBUSXX=dbusxx-xml2cpp
 
 noinst_LTLIBRARIES = libdbus.la
 
@@ -26,9 +26,7 @@
 NETWORKMANAGER=
 endif
 
-libdbus_la_CXXFLAGS = -DPREFIX=\"$(prefix)\" -DPROGSHAREDIR=\"${datadir}/sflphone\" $(NETWORKMANAGER)
-
-libdbus_la_LIBADD = $(top_srcdir)/libs/dbus-c++/src/libdbus-c++-1.la
+libdbus_la_CXXFLAGS = -DPREFIX=\"$(prefix)\" -DPROGSHAREDIR=\"${datadir}/sflphone\" $(NETWORKMANAGER) @DBUSCPLUSPLUS_CFLAGS@
 
 noinst_HEADERS =            \
     callmanager.h           \
diff -Nurd sflphone-common/src/history/Makefile.am sflphone-common.new//src/history/Makefile.am
--- sflphone-common/src/history/Makefile.am	2011-04-05 01:01:39.000000000 +0200
+++ sflphone-common.new//src/history/Makefile.am	2011-05-28 11:04:37.341337298 +0200
@@ -2,6 +2,8 @@
 
 SUBDIRS = 
 
+libhistory_la_CXXFLAGS = @DBUSCPLUSPLUS_CFLAGS@
+
 noinst_LTLIBRARIES = libhistory.la
 
 libhistory_la_SOURCES =         \
diff -Nurd sflphone-common/src/iax/Makefile.am sflphone-common.new//src/iax/Makefile.am
--- sflphone-common/src/iax/Makefile.am	2011-04-05 01:01:39.000000000 +0200
+++ sflphone-common.new//src/iax/Makefile.am	2011-05-28 11:04:58.246337305 +0200
@@ -10,7 +10,8 @@
 		iaxvoiplink.cpp
 
 libiaxlink_la_CXXFLAGS = \
-		-DUSE_IAX
+		-DUSE_IAX \
+		@DBUSCPLUSPLUS_CFLAGS@
 
 noinst_HEADERS = \
 		iaxaccount.h \
diff -Nurd sflphone-common/src/Makefile.am sflphone-common.new//src/Makefile.am
--- sflphone-common/src/Makefile.am	2011-04-05 01:01:39.000000000 +0200
+++ sflphone-common.new//src/Makefile.am	2011-05-28 11:06:03.630337329 +0200
@@ -40,7 +40,7 @@
 
 sflphoned_CXXFLAGS = \
 		-DPREFIX=\"$(prefix)\" -DPROGSHAREDIR=\"${datadir}/sflphone\" $(IAX_CXXFLAG) $(NETWORKMANAGER) \
-		-DVERSION=\"$(VERSION)\"
+		-DVERSION=\"$(VERSION)\" @DBUSCPLUSPLUS_CFLAGS@
 
 # libsflphone 
 
@@ -66,7 +66,6 @@
 libsflphone_la_LIBADD = \
 	$(src)/libs/utilspp/libutilspp.la \
 	$(src)/libs/iax2/libiax2.la \
-	$(src)/libs/dbus-c++/src/libdbus-c++-1.la \
 	$(IAX_LIB) \
 	./im/libim.la \
 	./sip/libsiplink.la \
@@ -78,6 +77,7 @@
 	./history/libhistory.la
 
 libsflphone_la_LDFLAGS = \
+		@DBUSCPLUSPLUS_LIBS@ \
 		@CCGNU2_LIBS@ \
 		@CCEXT2_LIBS@ \
 		@ZRTPCPP_LIBS@ \
@@ -92,6 +92,7 @@
 		@xml_LIBS@
 
 libsflphone_la_CFLAGS = \
+		@DBUSCPLUSPLUS_CFLAGS@ \
 		@CCGNU2_CFLAGS@ \
 		@CCEXT2_CFLAGS@ \
 		@ZRTPCPP_CFLAGS@ \
diff -Nurd sflphone-common/src/sip/Makefile.am sflphone-common.new//src/sip/Makefile.am
--- sflphone-common/src/sip/Makefile.am	2011-04-05 01:01:39.000000000 +0200
+++ sflphone-common.new//src/sip/Makefile.am	2011-05-28 11:06:37.973337340 +0200
@@ -21,4 +21,9 @@
 		sipvoiplink.h
 
 libsiplink_la_CXXFLAGS = \
-		@PCRE_LIBS@
+		@PCRE_LIBS@ \
+		@DBUSCPLUSPLUS_CFLAGS@
+
+libsiplink_la_LIBADD = \
+		@PCRE_LIBS@ \
+		@DBUSCPLUSPLUS_LIBS@
