From 10e7f7b2132aaac8e3ebc549d3837cc2df888df7 Mon Sep 17 00:00:00 2001
From: Thibaut Girka <thib@sitedethib.com>
Date: Tue, 29 Jun 2010 12:54:03 +0200
Subject: [PATCH] glamo-mci: add MMC_CAP_NONREMOVABLE and move enable_irq to right place

---
 drivers/mmc/host/glamo-mci.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/host/glamo-mci.c b/drivers/mmc/host/glamo-mci.c
index b2442c2..11deb7d 100644
--- a/drivers/mmc/host/glamo-mci.c
+++ b/drivers/mmc/host/glamo-mci.c
@@ -811,6 +811,7 @@ static int glamo_mci_probe(struct platform_device *pdev)
 	mmc->ocr_avail	= mmc_regulator_get_ocrmask(host->regulator);
 	mmc->caps	= MMC_CAP_4_BIT_DATA |
 			    MMC_CAP_MMC_HIGHSPEED |
+			    MMC_CAP_NONREMOVABLE |
 			    MMC_CAP_SD_HIGHSPEED;
 	mmc->f_min	= host->clk_rate / 256;
 	mmc->f_max	= sd_max_clk;
@@ -930,10 +931,10 @@ static int glamo_mci_resume(struct device *dev)
 	glamo_mci_reset(host);
 	mdelay(10);
 
-	ret = mmc_resume_host(host->mmc);
-
 	enable_irq(host->irq);
 
+	ret = mmc_resume_host(host->mmc);
+
 	mmc_host_lazy_disable(host->mmc);
 
 	return ret;
-- 
1.7.1

