From ed36456fa34c2653d4a39c3dbeb66b4e671103fe Mon Sep 17 00:00:00 2001
From: Gennady Kupava <gb@bsdmn.com>
Date: Mon, 2 Aug 2010 14:53:55 +0200
Subject: [PATCH] mach-gta02: fix gsm power_on sysfs node path

---
 arch/arm/mach-s3c2440/mach-gta02.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-s3c2440/mach-gta02.c b/arch/arm/mach-s3c2440/mach-gta02.c
index e7d0b82..08e9dcc 100644
--- a/arch/arm/mach-s3c2440/mach-gta02.c
+++ b/arch/arm/mach-s3c2440/mach-gta02.c
@@ -1341,7 +1341,7 @@ static struct platform_device* gta02_hdq_children[] = {
 
 static struct gta02_device_children gta02_device_children[] = {
 	{
-		.dev_name = "pcf50633-gpio",
+		.dev_name = "pcf50633-gpio.0",
 		.num_children = 2,
 		.children = gta02_pcf50633_gpio_children,
 	},
-- 
1.7.1.1

