--- zeroconf-0.6.1.orig/debian-zeroconf
+++ zeroconf-0.6.1/debian-zeroconf
@@ -19,6 +19,8 @@
 [ -f /etc/default/zeroconf ] &&
     . /etc/default/zeroconf

+[ -n "$DISABLE" ] && exit 0
+
 for BLACK in $IFBLACKLIST; do
     case $IFACE in
        $BLACK)
@@ -31,7 +33,6 @@
 if [ -n "$FALLBACK" ]; then
     /bin/ip addr show $IFACE scope global | grep -q "inet"
     IP=$?
-    [ $IP -eq 0 ] && exit 0
     if [ $IP -eq 0 ]; then
         /bin/ip route add 169.254.0.0/16 dev $IFACE
         exit 0
@@ -39,6 +40,6 @@
 fi

 # otherwise, we are good to go
-/usr/local/sbin/zeroconf -i $IFACE
+/usr/sbin/zeroconf -i $IFACE

 exit 0

