
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- totem-1.0.4/src/totem.c~gtkonly
+++ totem-1.0.4/src/totem.c
@@ -1667,8 +1667,10 @@
 			totem_action_set_mrl_and_play (totem, mrl);
 			g_free (mrl);
 
+#ifndef HAVE_GTK_ONLY
 			gnome_entry_append_history (GNOME_ENTRY (gentry),
 					TRUE, uri);
+#endif
 		}
 	}
 
@@ -3593,7 +3595,9 @@
 	/* Properties */
 	totem->properties = bacon_video_widget_properties_new ();
 
+#ifndef HAVE_GTK_ONLY
 	totem_session_setup (totem, argv);
+#endif
 	totem_setup_recent (totem);
 	totem_callback_connect (totem);
 	totem_setup_window (totem);
@@ -3617,7 +3621,9 @@
 	/* Command-line handling */
 	if (totem_options_process_late (totem, &argc, &argv) != FALSE)
 	{
+#ifndef HAVE_GTK_ONLY
 		totem_session_restore (totem, argv);
+#endif
 	} else if (argc >= 1 && totem_action_open_files (totem, argv)) {
 			totem_action_play_pause (totem);
 	} else {
