Index: glib-2.12.9/glib/gutils.h
===================================================================
--- glib-2.12.9.orig/glib/gutils.h	2007-10-07 19:13:53.000000000 +0200
+++ glib-2.12.9/glib/gutils.h	2007-10-07 19:15:04.000000000 +0200
@@ -97,7 +97,11 @@
 #  define G_INLINE_FUNC
 #  undef  G_CAN_INLINE
 #elif defined (__GNUC__) 
-#  define G_INLINE_FUNC extern inline
+#  if defined (__GNUC_GNU_INLINE__)
+#    define G_INLINE_FUNC extern  __attribute__((gnu_inline)) inline
+#  else
+#    define G_INLINE_FUNC extern inline
+#  endif
 #elif defined (G_CAN_INLINE) 
 #  define G_INLINE_FUNC static inline
 #else /* can't inline */
