Index: main.c
===================================================================
RCS file: /cvs/gpe/base/gpe-contacts/main.c,v
retrieving revision 1.102
diff -u -r1.102 main.c
--- main.c	16 Oct 2004 22:21:17 -0000	1.102
+++ main.c	26 Oct 2004 11:41:39 -0000
@@ -1246,6 +1246,7 @@
   list_view = gtk_tree_view_new_with_model (GTK_TREE_MODEL (list_store));
   gtk_tree_view_set_headers_visible (GTK_TREE_VIEW (list_view), FALSE);
   tree_sel = gtk_tree_view_get_selection (GTK_TREE_VIEW (list_view));
+  gtk_tree_selection_set_mode(tree_sel, GTK_SELECTION_BROWSE);
 
   GTK_WIDGET_UNSET_FLAGS (list_view, GTK_CAN_FOCUS);
   
