Index: etc/codecs.conf
===================================================================
--- etc/codecs.conf	(revision 29308)
+++ etc/codecs.conf	(working copy)
@@ -1942,6 +1942,14 @@
   out YUY2
   out BGR32,BGR24
 
+videocodec fflibvpxvp8
+  info "FFmpeg wrapper for libvpx/VP8"
+  status working
+  fourcc VP80
+  driver ffmpeg
+  dll "libvpx_vp8"
+  out YV12
+
 videocodec mwv1
   info "Motion Wavelets"
   status working
