Index: libmpdemux/demux_mkv.c
===================================================================
--- libmpdemux/demux_mkv.c	(revision 31032)
+++ libmpdemux/demux_mkv.c	(working copy)
@@ -1519,6 +1519,7 @@ static const videocodec_info_t vinfo[] =
   { MKV_V_MPEG4_AP,  mmioFOURCC('m', 'p', '4', 'v'), 1 },
   { MKV_V_MPEG4_AVC, mmioFOURCC('a', 'v', 'c', '1'), 1 },
   { MKV_V_THEORA,    mmioFOURCC('t', 'h', 'e', 'o'), 1 },
+  { MKV_V_VP8,       mmioFOURCC('V', 'P', '8', '0'), 0 },
   { NULL, 0, 0 }
 };
 
@@ -2053,7 +2054,7 @@ demux_mkv_open (demuxer_t *demuxer)
 
   stream_seek(s, s->start_pos);
   str = ebml_read_header (s, &version);
-  if (str == NULL || strcmp (str, "matroska") || version > 2)
+  if (str == NULL || (strcmp (str, "matroska") && strcmp (str, "webm")) || version > 2)
     {
       mp_msg (MSGT_DEMUX, MSGL_DBG2, "[mkv] no head found\n");
       return 0;
Index: libmpdemux/matroska.h
===================================================================
--- libmpdemux/matroska.h	(revision 31032)
+++ libmpdemux/matroska.h	(working copy)
@@ -69,6 +69,7 @@
 #define MKV_V_MPEG4_AP   "V_MPEG4/ISO/AP"
 #define MKV_V_MPEG4_AVC  "V_MPEG4/ISO/AVC"
 #define MKV_V_THEORA     "V_THEORA"
+#define MKV_V_VP8        "V_VP8"
 
 #define MKV_S_TEXTASCII  "S_TEXT/ASCII"
 #define MKV_S_TEXTUTF8   "S_TEXT/UTF8"
