--- /tmp/asm-offsets.h	2010-12-29 18:25:51.000000000 +0100
+++ trunk/ffmpeg/libavcodec/arm/asm-offsets.h	2010-12-29 18:26:21.000000000 +0100
@@ -35,7 +35,7 @@
 #define AC_PRED                  0xa80
 #define BLOCK_LAST_INDEX         0x2160
 #define INTER_SCANTAB_RASTER_END 0x2360
-#define H263_AIC                 0x2610
+#define H263_AIC                 0x2618
 #elif defined(__APPLE__)
 #define Y_DC_SCALE               0xa30
 #define C_DC_SCALE               0xa34
