Index: mplayer-0.0+1.0rc4+gitr0+d1c686f6a63c31276b0811baf7c78aa6f4c6c6a6-r5/vo_omapfb.c
===================================================================
--- mplayer-0.0+1.0rc4+gitr0+d1c686f6a63c31276b0811baf7c78aa6f4c6c6a6-r5.orig/vo_omapfb.c	2010-09-12 22:42:52.589660557 +0200
+++ mplayer-0.0+1.0rc4+gitr0+d1c686f6a63c31276b0811baf7c78aa6f4c6c6a6-r5/vo_omapfb.c	2010-09-12 22:43:01.759065169 +0200
@@ -556,7 +556,7 @@
 }
 
 
-static int control(uint32_t request, void *data, ...)
+static int control(uint32_t request, void *data)
 {
     switch (request) {
         case VOCTRL_QUERY_FORMAT:
