Index: python/Lib/distutils/command/install.py
===================================================================
--- python.orig/Lib/distutils/command/install.py	2007-03-06 17:15:43.000000000 -0300
+++ python/Lib/distutils/command/install.py	2007-03-06 17:16:04.000000000 -0300
@@ -601,7 +601,7 @@
                     ('install_headers', has_headers),
                     ('install_scripts', has_scripts),
                     ('install_data',    has_data),
-                    ('install_egg_info', lambda self:True),
+                    ('install_egg_info', lambda self:False),
                    ]
 
 # class install
