Index: python-2.5.2/debian/rules
===================================================================
--- python-2.5.2.orig/debian/rules	2008-02-26 14:15:36.000000000 -0300
+++ python-2.5.2/debian/rules	2008-02-26 14:17:42.000000000 -0300
@@ -134,6 +134,8 @@
 	)
 	find $(d_dev)/$(scriptdir) -name "*.pyo" -type f | grep -v -f $(only_dev_list) | xargs rm -f
 
+	find $(d_dev)/$(scriptdir)/encodings | grep -f $(only_dev_list) | xargs -i mv '{}' $(d_dev)/$(scriptdir)/encodings_orient
+
 	# move the interpreter
 	mv $(d_dev)/usr/bin/python2.5 $(d_base)/usr/bin/python2.5
 
Index: python-2.5.2/Lib/encodings/__init__.py
===================================================================
--- python-2.5.2.orig/Lib/encodings/__init__.py	2008-02-26 14:15:07.000000000 -0300
+++ python-2.5.2/Lib/encodings/__init__.py	2008-02-26 14:17:42.000000000 -0300
@@ -99,6 +99,14 @@
             pass
         else:
             break
+
+        try:
+            mod = __import__('encodings_orient.' + modname,
+                             globals(), locals(), _import_tail)
+        except ImportError:
+            pass
+        else:
+            break
     else:
         mod = None
 
Index: python-2.5.2/Makefile.pre.in
===================================================================
--- python-2.5.2.orig/Makefile.pre.in	2008-02-26 14:15:07.000000000 -0300
+++ python-2.5.2/Makefile.pre.in	2008-02-26 14:17:42.000000000 -0300
@@ -717,7 +717,7 @@
 PLATMACPATH=:plat-mac:plat-mac/lib-scriptpackages
 LIBSUBDIRS=	lib-tk site-packages test test/output test/data \
 		test/decimaltestdata \
-		encodings compiler hotshot \
+		encodings encodings_orient compiler hotshot \
 		email email/mime email/test email/test/data \
 		sqlite3 sqlite3/test \
 		logging bsddb bsddb/test csv wsgiref \
Index: python-2.5.2/debian/onlysdk
===================================================================
--- python-2.5.2.orig/debian/onlysdk	2008-02-26 14:15:07.000000000 -0300
+++ python-2.5.2/debian/onlysdk	2008-02-26 14:17:42.000000000 -0300
@@ -1,7 +1,6 @@
 distutils
 compile
-encodings/cp
-encodings/mac
+encodings_orient
 doctest
 unittest
 hotshot
Index: python-2.5.2/Lib/encodings_orient/__init__.py
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ python-2.5.2/Lib/encodings_orient/__init__.py	2008-02-26 14:17:42.000000000 -0300
@@ -0,0 +1 @@
+#Dummy
