diff -urb work/arch/mips/zboot/Makefile mtx/arch/mips/zboot/Makefile
--- work/arch/mips/zboot/Makefile	2003-07-01 20:09:20.000000000 +0200
+++ mtx/arch/mips/zboot/Makefile	2003-07-01 19:20:29.000000000 +0200
@@ -64,6 +64,9 @@
 ifdef CONFIG_MIPS_XXS1500
 BOOT_DIR = xxs1500
 endif
+ifdef CONFIG_MIPS_MTX1
+	$(MAKE) -C pb1xxx $@
+endif
 
 # Do the dirs
 clean:
diff -urb work/arch/mips/zboot/pb1xxx/Makefile mtx/arch/mips/zboot/pb1xxx/Makefile
--- work/arch/mips/zboot/pb1xxx/Makefile	2003-07-01 20:09:20.000000000 +0200
+++ mtx/arch/mips/zboot/pb1xxx/Makefile	2003-07-01 19:20:37.000000000 +0200
@@ -57,6 +57,10 @@
 BNAME=bosporus
 endif
 
+ifdef CONFIG_MIPS_MTX1
+BNAME=mtx-1
+endif
+
 # These two variables control where the zImage is stored
 # in flash and loaded in memory.  It only controls how the srec
 # file is generated, the code is the same.
