--- /dev/null	Thu Jan  1 01:00:00 1970
+++ localedata/locales/nb_NO	Sun Jun 15 15:32:24 2003
@@ -0,0 +1,98 @@
+comment_char    %
+escape_char     /
+%
+% Norwegian Bokm<U00E5>l language locale for Norway.
+%
+% Contact: Petter Reinholdtsen
+% Email: pere@hungry.com
+% Language: nb
+% Territory: NO
+% Revision: 1.0
+% Date: 2003-05-20
+% Application: general
+% Users: general
+% Charset: ISO-8859-1
+%
+% License is GNU GPL.
+
+LC_IDENTIFICATION
+title      "Norwegian Bokm<U00E5>l language locale for Norway"
+source     ""
+address    ""
+contact    "GNU libc maintainers"
+email      "bug-glibc@gnu.org"
+tel        ""
+fax        ""
+language   "Norwegian Bokm<U00E5>l"
+territory  "Norway"
+revision   "1.0"
+date       "2003-06-15"
+%
+category  "nb_NO:2000";LC_IDENTIFICATION
+category  "nb_NO:2000";LC_CTYPE
+category  "nb_NO:2000";LC_COLLATE
+category  "nb_NO:2000";LC_TIME
+category  "nb_NO:2000";LC_NUMERIC
+category  "nb_NO:2000";LC_MONETARY
+category  "nb_NO:2000";LC_MESSAGES
+category  "nb_NO:2000";LC_PAPER
+category  "nb_NO:2000";LC_NAME
+category  "nb_NO:2000";LC_ADDRESS
+category  "nb_NO:2000";LC_TELEPHONE
+
+END LC_IDENTIFICATION
+
+
+LC_CTYPE
+copy "no_NO"
+END LC_CTYPE
+
+
+LC_COLLATE
+copy "no_NO"
+END LC_COLLATE
+
+
+LC_MONETARY
+copy "no_NO"
+END LC_MONETARY
+
+
+LC_NUMERIC
+copy "no_NO"
+END LC_NUMERIC
+
+
+LC_TIME
+copy "no_NO"
+END LC_TIME
+
+
+LC_MESSAGES
+copy "no_NO"
+END LC_MESSAGES
+
+
+LC_PAPER
+copy "no_NO"
+END LC_PAPER
+
+
+LC_TELEPHONE
+copy "no_NO"
+END LC_TELEPHONE
+
+
+LC_MEASUREMENT
+copy "no_NO"
+END LC_MEASUREMENT
+
+
+LC_NAME
+copy "no_NO"
+END LC_NAME
+
+
+LC_ADDRESS
+copy "no_NO"
+END LC_ADDRESS
--- intl/locale.alias.old	2003-10-22 13:41:40.000000000 -0400
+++ intl/locale.alias	2003-10-22 13:41:54.000000000 -0400
@@ -58,8 +58,6 @@
 korean.euc 	ko_KR.eucKR
 ko_KR		ko_KR.eucKR
 lithuanian      lt_LT.ISO-8859-13
-nb_NO		no_NO.ISO-8859-1
-nb_NO.ISO-8859-1 no_NO.ISO-8859-1
 norwegian       no_NO.ISO-8859-1
 nynorsk		nn_NO.ISO-8859-1
 polish          pl_PL.ISO-8859-2
