--- glibc-2.3.2/sysdeps/unix/sysv/linux/dl-osinfo.h.orig	2003-11-01 14:20:45.000000000 -0500
+++ glibc-2.3.2/sysdeps/unix/sysv/linux/dl-osinfo.h	2003-11-01 14:21:32.000000000 -0500
@@ -91,7 +91,7 @@
 	    version <<= 8;						      \
 	    version |= here;						      \
 									      \
-	    if (*cp++ != '.')						      \
+	    if (*cp++ != '.' || parts == 3)				      \
 	      /* Another part following?  */				      \
 	      break;							      \
 	  }								      \
