diff -Nur ecore-0.9.9.032~/configure.in ecore-0.9.9.032/configure.in
--- ecore-0.9.9.032~/configure.in	2006-08-03 12:12:43.000000000 -0700
+++ ecore-0.9.9.032/configure.in	2006-08-03 12:13:30.000000000 -0700
@@ -16,7 +16,6 @@
 AC_CHECK_SIZEOF(long, 4)
 AM_ENABLE_SHARED
 AM_PROG_LIBTOOL
-AC_C___ATTRIBUTE__
 							
 dnl Set PACKAGE_DATA_DIR in config.h.
 if test "x${prefix}" = "xNONE"; then
@@ -205,10 +204,8 @@
       AM_CONDITIONAL(BUILD_ECORE_X, true)
       AC_DEFINE(BUILD_ECORE_X, 1, [Build Ecore_X Module])
       have_ecore_x="yes"
-      x_dir=${x_dir:-/usr/X11R6}
-      x_cflags=${x_cflags:--I${x_includes:-$x_dir/include}}
-      x_libs="${x_libs:--L${x_libraries:-$x_dir/lib}} -lX11 -lXext"
       x_ldflags=" "$ecore_x_libs;
+      x_libs="-lX11 -lXext"
       ecore_x_libs="-lecore_x";
     ],[
       AM_CONDITIONAL(BUILD_ECORE_X, false)
