diff -Nur ecore-0.9.9.037~/configure.in ecore-0.9.9.037/configure.in
--- ecore-0.9.9.037~/configure.in	2007-01-14 20:21:42.000000000 -0800
+++ ecore-0.9.9.037/configure.in	2007-01-14 20:22:18.000000000 -0800
@@ -578,7 +578,7 @@
 if test "x$want_ecore_fb" = "xyes"; then
   AC_CHECK_HEADER(tslib.h,
     [
-      tslib_libs="-ltslib"
+      tslib_libs="-lts"
       AC_DEFINE(HAVE_TSLIB, 1, [Build Ecore_FB Touchscreen Code])
     ]
   )
