From 59c95610fe9601091807a089a463607bd0a01f24 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Tue, 10 May 2011 20:27:29 +0200
Subject: [PATCH 2/2] systemd-analyze: print hostname, kernelversion and arch at the top of the plot

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---
 src/systemd-analyze |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/src/systemd-analyze b/src/systemd-analyze
index 4f3e478..ae7dcfb 100755
--- a/src/systemd-analyze
+++ b/src/systemd-analyze
@@ -110,7 +110,7 @@ elif sys.argv[1] == 'blame':
                 sys.stdout.write("%6lums %s\n" % ((aet - ixt) / 1000, name))
 
 elif sys.argv[1] == 'plot':
-        import cairo
+        import cairo, os
 
         initrd_time, start_time, finish_time = acquire_start_time()
         data = acquire_time_data()
@@ -160,6 +160,9 @@ elif sys.argv[1] == 'plot':
         context.stroke()
         context.restore()
 
+        banner = "Running on %s (%s %s) %s" % (os.uname()[1], os.uname()[2], os.uname()[3], os.uname()[4])
+        draw_text(context, 0, -15, banner, hcenter = 0, vcenter = 1)
+
         for x in range(0, (finish_time - start_time)/10000, 100):
                 draw_text(context, x, -5, "%lus" % (x/100), vcenter = 0, hcenter = 0)
 
-- 
1.6.6.1

