From 1a032dae3a9877287157003ec7e8986b76f2a644 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Mon, 4 Apr 2011 17:14:30 +0200
Subject: [PATCH] elmdentica: adapt to azy changes r58286

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/Makefile.am |    2 +-
 src/statusnet.c |    1 +
 2 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index e729027..2bd1622 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -19,7 +19,7 @@ AM_LDFLAGS =	@PTHREAD_LIBS@		\
 		`curl-config --libs`	\
 		@ELEMENTARY_LIBS@
 
-BUILT_SOURCES = statusnet_Common.h statusnet_Common.c
+BUILT_SOURCES = statusnet_Common.h statusnet_Common.c statusnet_Common_Azy.h statusnet_Common_Azy.c
 CLEANFILES = $(BUILT_SOURCES) .sources-ts
 $(BUILT_SOURCES): .sources-ts
 .sources-ts: statusnet.azy
diff --git a/src/statusnet.c b/src/statusnet.c
index b1c4775..44e8d5f 100644
--- a/src/statusnet.c
+++ b/src/statusnet.c
@@ -35,6 +35,7 @@
 
 #include <Azy.h>
 #include "statusnet_Common.h"
+#include "statusnet_Common_Azy.h"
 
 #include <sqlite3.h>
 
-- 
1.7.4.1

