--- TraceToolkit-0.9.5/LibLTT/EventDB.c~gcc34.patch	2002-04-15 12:10:13.000000000 -0400
+++ TraceToolkit-0.9.5/LibLTT/EventDB.c	2004-05-11 16:09:39.000000000 -0400
@@ -2801,7 +2801,7 @@
       lLastCtrlEventTime = lEventTime;
       }
 
-NoControlEvent:
+NoControlEvent: ;
     } while(DBEventNext(pmTraceDB, &lEvent) == TRUE);
 
   /* Remember the last event and its time */
--- TraceToolkit-0.9.5/LibLTT/RTAIDB.c~gcc34.patch	2002-04-15 23:00:05.000000000 -0400
+++ TraceToolkit-0.9.5/LibLTT/RTAIDB.c	2004-05-11 16:11:04.000000000 -0400
@@ -2937,7 +2937,7 @@
       lLastCtrlEventTime = lEventTime;
       }
 
-NoControlEvent:
+NoControlEvent: ;
     } while(DBEventNext(pmTraceDB, &lEvent) == TRUE);
 
   /* Remember the last event and it's time */
