Index: linux-2.6.16.5-ezx/arch/arm/kernel/irq.c
===================================================================
--- linux-2.6.16.5-ezx.orig/arch/arm/kernel/irq.c	2006-04-12 17:27:57.000000000 -0300
+++ linux-2.6.16.5-ezx/arch/arm/kernel/irq.c	2006-04-22 19:00:53.000000000 -0300
@@ -305,6 +305,8 @@
 	static int count = 100;
 	struct irqaction *action;
 
+	return;
+
 	if (!count || noirqdebug)
 		return;
 
