Index: linux-2.6.16.5-ezx/drivers/mmc/pxamci.c
===================================================================
--- linux-2.6.16.5-ezx.orig/drivers/mmc/pxamci.c	2006-04-17 09:35:07.000000000 -0300
+++ linux-2.6.16.5-ezx/drivers/mmc/pxamci.c	2006-04-27 13:26:48.000000000 -0300
@@ -332,6 +332,10 @@
 		if (ireg & DATA_TRAN_DONE)
 			handled |= pxamci_data_done(host, stat);
 	}
+	if (!handled) {
+		unsigned int stat = readl(host->base + MMC_STAT);
+		printk("PXAMCI: stat %08x\n", stat);
+	}
 
 	return IRQ_RETVAL(handled);
 }
