Index: gnuradio/trunk/gnuradio-core/src/lib/filter/gr_fft_filter_ccc.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/filter/gr_fft_filter_ccc.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/filter/gr_fft_filter_ccc.cc (revision 8292)
@@ -39,4 +39,5 @@
 
 #include <iostream>
+#include <string.h>
 
 gr_fft_filter_ccc_sptr gr_make_fft_filter_ccc (int decimation, const std::vector<gr_complex> &taps)
Index: gnuradio/trunk/gnuradio-core/src/lib/filter/gr_fft_filter_fff.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/filter/gr_fft_filter_fff.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/filter/gr_fft_filter_fff.cc (revision 8292)
@@ -34,4 +34,5 @@
 
 #include <iostream>
+#include <string.h>
 
 gr_fft_filter_fff_sptr gr_make_fft_filter_fff (int decimation, const std::vector<float> &taps)
Index: gnuradio/trunk/gnuradio-core/src/lib/filter/qa_gr_fir_ccc.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/filter/qa_gr_fir_ccc.cc (revision 6574)
+++ gnuradio/trunk/gnuradio-core/src/lib/filter/qa_gr_fir_ccc.cc (revision 8292)
@@ -48,4 +48,5 @@
 #include <random.h>
 #include <malloc16.h>
+#include <string.h>
 
 using std::vector;
Index: gnuradio/trunk/gnuradio-core/src/lib/filter/qa_gr_fir_ccf.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/filter/qa_gr_fir_ccf.cc (revision 6574)
+++ gnuradio/trunk/gnuradio-core/src/lib/filter/qa_gr_fir_ccf.cc (revision 8292)
@@ -48,4 +48,5 @@
 #include <random.h>
 #include <malloc16.h>
+#include <string.h>
 
 using std::vector;
Index: gnuradio/trunk/gnuradio-core/src/lib/filter/qa_gr_fir_fcc.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/filter/qa_gr_fir_fcc.cc (revision 6574)
+++ gnuradio/trunk/gnuradio-core/src/lib/filter/qa_gr_fir_fcc.cc (revision 8292)
@@ -47,4 +47,5 @@
 #include <cppunit/TestAssert.h>
 #include <random.h>
+#include <string.h>
 
 using std::vector;
Index: gnuradio/trunk/gnuradio-core/src/lib/filter/qa_gr_fir_fff.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/filter/qa_gr_fir_fff.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/filter/qa_gr_fir_fff.cc (revision 8292)
@@ -33,4 +33,5 @@
 #include <cppunit/TestAssert.h>
 #include <random.h>
+#include <string.h>
 
 using std::vector;
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_align_on_samplenumbers_ss.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_align_on_samplenumbers_ss.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_align_on_samplenumbers_ss.cc (revision 8292)
@@ -29,4 +29,5 @@
 #include <assert.h>
 #include <stdexcept>
+#include <string.h>
 
 //define ALIGN_ADVANCED_IMPLEMENTATION to have an alternative implementation of the align algoritm which exactly follows the align_interval spec.
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_bin_statistics_f.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_bin_statistics_f.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_bin_statistics_f.cc (revision 8292)
@@ -27,4 +27,5 @@
 #include <gr_bin_statistics_f.h>
 #include <gr_io_signature.h>
+#include <string.h>
 
 gr_bin_statistics_f_sptr
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_circular_file.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_circular_file.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_circular_file.cc (revision 8292)
@@ -40,4 +40,6 @@
 
 #include <algorithm>
+#include <stdio.h>
+#include <string.h>
 
 static const int HEADER_SIZE = 4096;
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_delay.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_delay.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_delay.cc (revision 8292)
@@ -27,4 +27,5 @@
 #include <gr_delay.h>
 #include <gr_io_signature.h>
+#include <string.h>
 
 gr_delay_sptr
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_fake_channel_coder_pp.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_fake_channel_coder_pp.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_fake_channel_coder_pp.cc (revision 8292)
@@ -29,4 +29,5 @@
 #include <string.h>
 #include <stdexcept>
+#include <string.h>
 
 static const int PAD_VAL = 0xAA;
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_fft_vcc.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_fft_vcc.cc (revision 8244)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_fft_vcc.cc (revision 8292)
@@ -30,4 +30,5 @@
 #include <gri_fft.h>
 #include <math.h>
+#include <string.h>
 
 gr_fft_vcc_sptr
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_fft_vfc.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_fft_vfc.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_fft_vfc.cc (revision 8292)
@@ -30,4 +30,5 @@
 #include <math.h>
 #include <stdexcept>
+#include <string.h>
 
 
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_framer_sink_1.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_framer_sink_1.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_framer_sink_1.cc (revision 8292)
@@ -29,4 +29,5 @@
 #include <cstdio>
 #include <stdexcept>
+#include <string.h>
 
 #define VERBOSE 0
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_head.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_head.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_head.cc (revision 8292)
@@ -26,4 +26,5 @@
 #include <gr_head.h>
 #include <gr_io_signature.h>
+#include <string.h>
 
 gr_head::gr_head (size_t sizeof_stream_item, int nitems)
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_keep_one_in_n.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_keep_one_in_n.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_keep_one_in_n.cc (revision 8292)
@@ -27,4 +27,5 @@
 #include <gr_keep_one_in_n.h>
 #include <gr_io_signature.h>
+#include <string.h>
 
 gr_keep_one_in_n_sptr
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_log2_const.h
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_log2_const.h (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_log2_const.h (revision 8292)
@@ -32,15 +32,15 @@
 template<unsigned int k> static inline int gr_log2_const() { assert(0); return 0; }
 
-template<> static inline int gr_log2_const<1>()   { return 0; }
-template<> static inline int gr_log2_const<2>()   { return 1; }
-template<> static inline int gr_log2_const<4>()   { return 2; }
-template<> static inline int gr_log2_const<8>()   { return 3; }
-template<> static inline int gr_log2_const<16>()  { return 4; }
-template<> static inline int gr_log2_const<32>()  { return 5; }
-template<> static inline int gr_log2_const<64>()  { return 6; }
-template<> static inline int gr_log2_const<128>() { return 7; }
-template<> static inline int gr_log2_const<256>() { return 8; }
-template<> static inline int gr_log2_const<512>() { return 9; }
-template<> static inline int gr_log2_const<1024>(){ return 10; }
+template<> inline int gr_log2_const<1>()   { return 0; }
+template<> inline int gr_log2_const<2>()   { return 1; }
+template<> inline int gr_log2_const<4>()   { return 2; }
+template<> inline int gr_log2_const<8>()   { return 3; }
+template<> inline int gr_log2_const<16>()  { return 4; }
+template<> inline int gr_log2_const<32>()  { return 5; }
+template<> inline int gr_log2_const<64>()  { return 6; }
+template<> inline int gr_log2_const<128>() { return 7; }
+template<> inline int gr_log2_const<256>() { return 8; }
+template<> inline int gr_log2_const<512>() { return 9; }
+template<> inline int gr_log2_const<1024>(){ return 10; }
 
 #endif /* INCLUDED_GR_LOG2_CONST_H */
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_null_source.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_null_source.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_null_source.cc (revision 8292)
@@ -27,4 +27,5 @@
 #include <gr_null_source.h>
 #include <gr_io_signature.h>
+#include <string.h>
 
 gr_null_source::gr_null_source (size_t sizeof_stream_item)
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_packet_sink.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_packet_sink.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_packet_sink.cc (revision 8292)
@@ -34,4 +34,5 @@
 #include <stdexcept>
 #include <gr_count_bits.h>
+#include <string.h>
 
 #define VERBOSE 0
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_peak_detector2_fb.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_peak_detector2_fb.cc (revision 7324)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_peak_detector2_fb.cc (revision 8292)
@@ -27,4 +27,5 @@
 #include <gr_peak_detector2_fb.h>
 #include <gr_io_signature.h>
+#include <string.h>
 
 gr_peak_detector2_fb_sptr
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_simple_correlator.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_simple_correlator.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_simple_correlator.cc (revision 8292)
@@ -31,4 +31,5 @@
 #include <stdexcept>
 #include <gr_count_bits.h>
+#include <string.h>
 
 
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_simple_framer.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_simple_framer.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_simple_framer.cc (revision 8292)
@@ -30,4 +30,5 @@
 #include <assert.h>
 #include <stdexcept>
+#include <string.h>
 
 
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_skiphead.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_skiphead.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_skiphead.cc (revision 8292)
@@ -26,4 +26,5 @@
 #include <gr_skiphead.h>
 #include <gr_io_signature.h>
+#include <string.h>
 
 gr_skiphead::gr_skiphead (size_t itemsize, size_t nitems_to_skip)
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_stream_mux.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_stream_mux.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_stream_mux.cc (revision 8292)
@@ -27,4 +27,5 @@
 #include <gr_stream_mux.h>
 #include <gr_io_signature.h>
+#include <string.h>
 
 #define VERBOSE 0
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_stream_to_streams.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_stream_to_streams.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_stream_to_streams.cc (revision 8292)
@@ -27,4 +27,5 @@
 #include <gr_stream_to_streams.h>
 #include <gr_io_signature.h>
+#include <string.h>
 
 gr_stream_to_streams_sptr
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_stream_to_vector.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_stream_to_vector.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_stream_to_vector.cc (revision 8292)
@@ -27,4 +27,5 @@
 #include <gr_stream_to_vector.h>
 #include <gr_io_signature.h>
+#include <string.h>
 
 gr_stream_to_vector_sptr
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_streams_to_stream.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_streams_to_stream.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_streams_to_stream.cc (revision 8292)
@@ -27,4 +27,5 @@
 #include <gr_streams_to_stream.h>
 #include <gr_io_signature.h>
+#include <string.h>
 
 gr_streams_to_stream_sptr
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_streams_to_vector.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_streams_to_vector.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_streams_to_vector.cc (revision 8292)
@@ -27,4 +27,5 @@
 #include <gr_streams_to_vector.h>
 #include <gr_io_signature.h>
+#include <string.h>
 
 gr_streams_to_vector_sptr
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_test.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_test.cc (revision 7847)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_test.cc (revision 8292)
@@ -29,4 +29,5 @@
 #include <stdexcept>
 #include <iostream>
+#include <string.h>
 
 gr_test_sptr gr_make_test (const std::string &name,
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_throttle.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_throttle.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_throttle.cc (revision 8292)
@@ -30,4 +30,5 @@
 #include <stdio.h>
 #include <math.h>
+#include <string.h>
 #ifdef HAVE_TIME_H
 #include <time.h>
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_vector_to_stream.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_vector_to_stream.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_vector_to_stream.cc (revision 8292)
@@ -27,4 +27,5 @@
 #include <gr_vector_to_stream.h>
 #include <gr_io_signature.h>
+#include <string.h>
 
 gr_vector_to_stream_sptr
Index: gnuradio/trunk/gnuradio-core/src/lib/general/gr_vector_to_streams.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/general/gr_vector_to_streams.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/gr_vector_to_streams.cc (revision 8292)
@@ -27,4 +27,5 @@
 #include <gr_vector_to_streams.h>
 #include <gr_io_signature.h>
+#include <string.h>
 
 gr_vector_to_streams_sptr
Index: gnuradio/trunk/gnuradio-core/src/lib/general/malloc16.c
===================================================================
--- import/gnuradio-savannah-20060726/trunk/gnuradio-core/src/lib/general/malloc16.c (revision 1313)
+++ gnuradio/trunk/gnuradio-core/src/lib/general/malloc16.c (revision 8292)
@@ -8,4 +8,5 @@
 #include "malloc16.h"
 #include <string.h>
+#include <strings.h>
 
 void *malloc16Align(int size){
Index: gnuradio/trunk/gnuradio-core/src/lib/gengen/gr_chunks_to_symbols_XX.cc.t
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/gengen/gr_chunks_to_symbols_XX.cc.t (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/gengen/gr_chunks_to_symbols_XX.cc.t (revision 8292)
@@ -31,4 +31,5 @@
 #include <assert.h>
 #include <iostream>
+#include <string.h>
 
 @SPTR_NAME@
Index: gnuradio/trunk/gnuradio-core/src/lib/gengen/gr_peak_detector_XX.cc.t
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/gengen/gr_peak_detector_XX.cc.t (revision 7587)
+++ gnuradio/trunk/gnuradio-core/src/lib/gengen/gr_peak_detector_XX.cc.t (revision 8292)
@@ -29,4 +29,5 @@
 #include <@NAME@.h>
 #include <gr_io_signature.h>
+#include <string.h>
 
 @SPTR_NAME@
Index: gnuradio/trunk/gnuradio-core/src/lib/io/gr_file_descriptor_sink.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/io/gr_file_descriptor_sink.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/io/gr_file_descriptor_sink.cc (revision 8292)
@@ -33,4 +33,5 @@
 #include <fcntl.h>
 #include <stdexcept>
+#include <stdio.h>
 
 
Index: gnuradio/trunk/gnuradio-core/src/lib/io/gr_file_descriptor_source.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/io/gr_file_descriptor_source.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/io/gr_file_descriptor_source.cc (revision 8292)
@@ -33,4 +33,6 @@
 #include <fcntl.h>
 #include <stdexcept>
+#include <stdio.h>
+#include <string.h>
 
 
Index: gnuradio/trunk/gnuradio-core/src/lib/io/gr_file_sink_base.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/io/gr_file_sink_base.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/io/gr_file_sink_base.cc (revision 8292)
@@ -31,4 +31,5 @@
 #include <fcntl.h>
 #include <stdexcept>
+#include <stdio.h>
 
 // win32 (mingw/msvc) specific
Index: gnuradio/trunk/gnuradio-core/src/lib/io/gr_file_source.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/io/gr_file_source.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/io/gr_file_source.cc (revision 8292)
@@ -32,4 +32,5 @@
 #include <fcntl.h>
 #include <stdexcept>
+#include <stdio.h>
 
 // win32 (mingw/msvc) specific
Index: gnuradio/trunk/gnuradio-core/src/lib/io/gr_message_sink.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/io/gr_message_sink.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/io/gr_message_sink.cc (revision 8292)
@@ -33,4 +33,5 @@
 #include <fcntl.h>
 #include <stdexcept>
+#include <string.h>
 
 
Index: gnuradio/trunk/gnuradio-core/src/lib/io/gr_message_source.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/io/gr_message_source.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/io/gr_message_source.cc (revision 8292)
@@ -33,4 +33,5 @@
 #include <fcntl.h>
 #include <stdexcept>
+#include <string.h>
 
 
Index: gnuradio/trunk/gnuradio-core/src/lib/io/gr_udp_sink.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/io/gr_udp_sink.cc (revision 8097)
+++ gnuradio/trunk/gnuradio-core/src/lib/io/gr_udp_sink.cc (revision 8292)
@@ -29,4 +29,5 @@
 #if defined(HAVE_SOCKET)
 #include <netdb.h>
+#include <stdio.h>
 typedef void* optval_t;
 #else
Index: gnuradio/trunk/gnuradio-core/src/lib/io/gr_udp_source.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/io/gr_udp_source.cc (revision 8097)
+++ gnuradio/trunk/gnuradio-core/src/lib/io/gr_udp_source.cc (revision 8292)
@@ -28,4 +28,6 @@
 #include <stdexcept>
 #include <errno.h>
+#include <stdio.h>
+#include <string.h>
 #if defined(HAVE_SOCKET)
 #include <netdb.h>
Index: gnuradio/trunk/gnuradio-core/src/lib/io/gri_logger.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/io/gri_logger.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/io/gri_logger.cc (revision 8292)
@@ -30,4 +30,5 @@
 #include <stdexcept>
 #include <boost/weak_ptr.hpp>
+#include <string.h>
 
 
Index: gnuradio/trunk/gnuradio-core/src/lib/reed-solomon/decode_rs.c
===================================================================
--- import/gnuradio-savannah-20060726/trunk/gnuradio-core/src/lib/reed-solomon/decode_rs.c (revision 605)
+++ gnuradio/trunk/gnuradio-core/src/lib/reed-solomon/decode_rs.c (revision 8292)
@@ -9,4 +9,5 @@
 
 #include <string.h>
+#include <strings.h>
 
 #define NULL ((void *)0)
Index: gnuradio/trunk/gnuradio-core/src/lib/reed-solomon/encode_rs.c
===================================================================
--- import/gnuradio-savannah-20060726/trunk/gnuradio-core/src/lib/reed-solomon/encode_rs.c (revision 605)
+++ gnuradio/trunk/gnuradio-core/src/lib/reed-solomon/encode_rs.c (revision 8292)
@@ -4,4 +4,5 @@
  */
 #include <string.h>
+#include <strings.h>
 
 #ifdef FIXED
Index: gnuradio/trunk/gnuradio-core/src/lib/reed-solomon/exercise.c
===================================================================
--- import/gnuradio-savannah-20060726/trunk/gnuradio-core/src/lib/reed-solomon/exercise.c (revision 605)
+++ gnuradio/trunk/gnuradio-core/src/lib/reed-solomon/exercise.c (revision 8292)
@@ -14,4 +14,5 @@
 #include <stdlib.h>
 #include <string.h>
+#include <strings.h>
 
 #ifdef FIXED
Index: gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_dispatcher.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_dispatcher.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_dispatcher.cc (revision 8292)
@@ -28,4 +28,5 @@
 #include <math.h>
 #include <errno.h>
+#include <stdio.h>
 
 #ifdef HAVE_SELECT
Index: gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_error_handler.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_error_handler.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_error_handler.cc (revision 8292)
@@ -49,4 +49,5 @@
 #include <stdexcept>
 #include <unistd.h>
+#include <stdio.h>
 
 #ifdef HAVE_IO_H
Index: gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_local_sighandler.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_local_sighandler.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_local_sighandler.cc (revision 8292)
@@ -28,4 +28,5 @@
 #include <stdexcept>
 #include <stdio.h>
+#include <string.h>
 
 
Index: gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_message.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_message.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_message.cc (revision 8292)
@@ -26,4 +26,5 @@
 #include <gr_message.h>
 #include <assert.h>
+#include <string.h>
 
 static long s_ncurrently_allocated = 0;
Index: gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_preferences.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_preferences.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_preferences.cc (revision 8292)
@@ -32,4 +32,5 @@
 #include <sys/stat.h>
 #include <unistd.h>
+#include <string.h>
 
 
Index: gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_realtime.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_realtime.cc (revision 6427)
+++ gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_realtime.cc (revision 8292)
@@ -62,4 +62,5 @@
 
 #include <pthread.h>
+#include <stdio.h>
 
 gr_rt_status_t
Index: gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_scheduler_thread.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_scheduler_thread.cc (revision 6201)
+++ gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_scheduler_thread.cc (revision 8292)
@@ -27,4 +27,5 @@
 #include <gr_scheduler_thread.h>
 #include <iostream>
+#include <stdio.h>
 
 #ifdef HAVE_SIGNAL_H
Index: gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_vmcircbuf.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_vmcircbuf.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_vmcircbuf.cc (revision 8292)
@@ -29,4 +29,5 @@
 #include <gr_preferences.h>
 #include <stdio.h>
+#include <string.h>
 #include <gr_local_sighandler.h>
 
Index: gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_vmcircbuf_mmap_tmpfile.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_vmcircbuf_mmap_tmpfile.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/lib/runtime/gr_vmcircbuf_mmap_tmpfile.cc (revision 8292)
@@ -41,5 +41,4 @@
 #include <gr_pagesize.h>
 #include <gr_tmp_path.h>
-
 
 gr_vmcircbuf_mmap_tmpfile::gr_vmcircbuf_mmap_tmpfile (int size)
Index: gnuradio/trunk/gnuradio-core/src/tests/benchmark_nco.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/tests/benchmark_nco.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/tests/benchmark_nco.cc (revision 8292)
@@ -31,4 +31,5 @@
 #include <gr_nco.h>
 #include <gr_fxpt_nco.h>
+#include <string.h>
 
 #define ITERATIONS	20000000
Index: gnuradio/trunk/gnuradio-core/src/tests/benchmark_vco.cc
===================================================================
--- gnuradio/trunk/gnuradio-core/src/tests/benchmark_vco.cc (revision 6044)
+++ gnuradio/trunk/gnuradio-core/src/tests/benchmark_vco.cc (revision 8292)
@@ -31,4 +31,5 @@
 #include <gr_vco.h>
 #include <gr_fxpt_vco.h>
+#include <string.h>
 
 #define ITERATIONS	5000000
Index: gnuradio/trunk/gr-atsc/src/lib/atsc_bit_timing_loop.cc
===================================================================
--- gnuradio/trunk/gr-atsc/src/lib/atsc_bit_timing_loop.cc (revision 6044)
+++ gnuradio/trunk/gr-atsc/src/lib/atsc_bit_timing_loop.cc (revision 8292)
@@ -28,4 +28,5 @@
 #include <gr_io_signature.h>
 #include <atsc_consts.h>
+#include <string.h>
 
 // Input rate changed from 20MHz to 19.2 to support usrp at 3 * 6.4MHz
Index: gnuradio/trunk/gr-atsc/src/lib/atsci_reed_solomon.cc
===================================================================
--- gnuradio/trunk/gr-atsc/src/lib/atsci_reed_solomon.cc (revision 6044)
+++ gnuradio/trunk/gr-atsc/src/lib/atsci_reed_solomon.cc (revision 8292)
@@ -23,4 +23,5 @@
 #include <atsci_reed_solomon.h>
 #include <assert.h>
+#include <string.h>
 
 extern "C" {
Index: gnuradio/trunk/gr-atsc/src/lib/atsci_sliding_correlator.h
===================================================================
--- gnuradio/trunk/gr-atsc/src/lib/atsci_sliding_correlator.h (revision 6044)
+++ gnuradio/trunk/gr-atsc/src/lib/atsci_sliding_correlator.h (revision 8292)
@@ -20,8 +20,9 @@
  * Boston, MA 02110-1301, USA.
  */
+#include <string>
+#include <string.h>
+
 #ifndef _ATSC_SLIDING_CORRELATOR_H_
 #define _ATSC_SLIDING_CORRELATOR_H_
-
-#include <string>
 
 extern const unsigned char atsc_pn511[511];
Index: gnuradio/trunk/gr-atsc/src/lib/atsci_sssr.cc
===================================================================
--- gnuradio/trunk/gr-atsc/src/lib/atsci_sssr.cc (revision 6044)
+++ gnuradio/trunk/gr-atsc/src/lib/atsci_sssr.cc (revision 8292)
@@ -28,4 +28,5 @@
 #include <atsci_diag_output.h>
 #include <gr_math.h>
+#include <stdio.h>
 
 /*
Index: gnuradio/trunk/gr-atsc/src/lib/atsci_trellis_encoder.cc
===================================================================
--- gnuradio/trunk/gr-atsc/src/lib/atsci_trellis_encoder.cc (revision 6044)
+++ gnuradio/trunk/gr-atsc/src/lib/atsci_trellis_encoder.cc (revision 8292)
@@ -24,4 +24,5 @@
 #include <assert.h>
 #include <stdio.h>
+#include <string.h>
 
 static const int DIBITS_PER_BYTE = 4;
Index: gnuradio/trunk/gr-atsc/src/lib/atsci_viterbi_decoder.cc
===================================================================
--- gnuradio/trunk/gr-atsc/src/lib/atsci_viterbi_decoder.cc (revision 6044)
+++ gnuradio/trunk/gr-atsc/src/lib/atsci_viterbi_decoder.cc (revision 8292)
@@ -26,4 +26,5 @@
 #include <cmath>
 #include "atsci_viterbi_mux.cc"
+#include <string.h>
 
 
Index: gnuradio/trunk/gr-atsc/src/lib/interleaver_fifo.h
===================================================================
--- gnuradio/trunk/gr-atsc/src/lib/interleaver_fifo.h (revision 6044)
+++ gnuradio/trunk/gr-atsc/src/lib/interleaver_fifo.h (revision 8292)
@@ -27,4 +27,5 @@
 #include <interleaver_fifo.h>
 #include <string.h>
+#include <strings.h>
 
 /*!
Index: gnuradio/trunk/gr-atsc/src/lib/qa_atsci_data_interleaver.cc
===================================================================
--- gnuradio/trunk/gr-atsc/src/lib/qa_atsci_data_interleaver.cc (revision 6044)
+++ gnuradio/trunk/gr-atsc/src/lib/qa_atsci_data_interleaver.cc (revision 8292)
@@ -23,4 +23,5 @@
 #include <cppunit/TestAssert.h>
 #include <qa_atsci_data_interleaver.h>
+#include <string.h>
 
 
Index: gnuradio/trunk/gr-atsc/src/lib/qa_atsci_equalizer_nop.cc
===================================================================
--- gnuradio/trunk/gr-atsc/src/lib/qa_atsci_equalizer_nop.cc (revision 6044)
+++ gnuradio/trunk/gr-atsc/src/lib/qa_atsci_equalizer_nop.cc (revision 8292)
@@ -29,4 +29,5 @@
 #include <assert.h>
 #include <iostream>
+#include <string.h>
 
 using std::cerr;
Index: gnuradio/trunk/gr-atsc/src/lib/qa_atsci_fake_single_viterbi.cc
===================================================================
--- gnuradio/trunk/gr-atsc/src/lib/qa_atsci_fake_single_viterbi.cc (revision 6044)
+++ gnuradio/trunk/gr-atsc/src/lib/qa_atsci_fake_single_viterbi.cc (revision 8292)
@@ -30,4 +30,5 @@
 #include <qa_atsci_fake_single_viterbi.h>
 #include <random.h>
+#include <string.h>
 
 
Index: gnuradio/trunk/gr-atsc/src/lib/qa_atsci_randomizer.cc
===================================================================
--- gnuradio/trunk/gr-atsc/src/lib/qa_atsci_randomizer.cc (revision 6044)
+++ gnuradio/trunk/gr-atsc/src/lib/qa_atsci_randomizer.cc (revision 8292)
@@ -24,4 +24,5 @@
 
 #include <cppunit/TestAssert.h>
+#include <string.h>
 
 static unsigned int expected_initial_states[] = {
Index: gnuradio/trunk/gr-atsc/src/lib/qa_atsci_reed_solomon.cc
===================================================================
--- gnuradio/trunk/gr-atsc/src/lib/qa_atsci_reed_solomon.cc (revision 6044)
+++ gnuradio/trunk/gr-atsc/src/lib/qa_atsci_reed_solomon.cc (revision 8292)
@@ -30,4 +30,5 @@
 #include <atsci_reed_solomon.h>
 #include <qa_atsci_reed_solomon.h>
+#include <string.h>
 
 
Index: gnuradio/trunk/gr-atsc/src/lib/qa_atsci_single_viterbi.cc
===================================================================
--- gnuradio/trunk/gr-atsc/src/lib/qa_atsci_single_viterbi.cc (revision 6044)
+++ gnuradio/trunk/gr-atsc/src/lib/qa_atsci_single_viterbi.cc (revision 8292)
@@ -31,4 +31,5 @@
 #include <qa_atsci_single_viterbi.h>
 #include <random.h>
+#include <string.h>
 
 
Index: gnuradio/trunk/gr-atsc/src/lib/qa_atsci_trellis_encoder.cc
===================================================================
--- gnuradio/trunk/gr-atsc/src/lib/qa_atsci_trellis_encoder.cc (revision 6044)
+++ gnuradio/trunk/gr-atsc/src/lib/qa_atsci_trellis_encoder.cc (revision 8292)
@@ -31,4 +31,5 @@
 #include <stdlib.h>
 #include <time.h>
+#include <string.h>
 
 #define	NELEM(x) (sizeof (x) / sizeof (x[0]))
Index: gnuradio/trunk/gr-audio-portaudio/src/audio_portaudio_sink.cc
===================================================================
--- gnuradio/trunk/gr-audio-portaudio/src/audio_portaudio_sink.cc (revision 6044)
+++ gnuradio/trunk/gr-audio-portaudio/src/audio_portaudio_sink.cc (revision 8292)
@@ -34,4 +34,5 @@
 #include <gri_portaudio.h>
 #include <omnithread.h>
+#include <string.h>
 
 #define	LOGGING   0		// define to 0 or 1
Index: gnuradio/trunk/gr-audio-portaudio/src/audio_portaudio_source.cc
===================================================================
--- gnuradio/trunk/gr-audio-portaudio/src/audio_portaudio_source.cc (revision 6044)
+++ gnuradio/trunk/gr-audio-portaudio/src/audio_portaudio_source.cc (revision 8292)
@@ -34,4 +34,5 @@
 #include <gri_portaudio.h>
 #include <omnithread.h>
+#include <string.h>
 
 #define	LOGGING 0		// define to 0 or 1
Index: gnuradio/trunk/gr-gsm-fr-vocoder/src/lib/gsm/code.c
===================================================================
--- import/gnuradio-savannah-20060726/trunk/gr-gsm-fr-vocoder/src/lib/gsm/code.c (revision 968)
+++ gnuradio/trunk/gr-gsm-fr-vocoder/src/lib/gsm/code.c (revision 8292)
@@ -20,4 +20,5 @@
 #include	"gsm.h"
 #include	"proto.h"
+#include <string.h>
 
 /* 
Index: gnuradio/trunk/gr-gsm-fr-vocoder/src/lib/gsm/gsm_create.c
===================================================================
--- import/gnuradio-savannah-20060726/trunk/gr-gsm-fr-vocoder/src/lib/gsm/gsm_create.c (revision 968)
+++ gnuradio/trunk/gr-gsm-fr-vocoder/src/lib/gsm/gsm_create.c (revision 8292)
@@ -31,4 +31,5 @@
 #include "private.h"
 #include "proto.h"
+#include <strings.h>
 
 gsm gsm_create P0()
Index: gnuradio/trunk/gr-trellis/src/lib/fsm.cc
===================================================================
--- gnuradio/trunk/gr-trellis/src/lib/fsm.cc (revision 6044)
+++ gnuradio/trunk/gr-trellis/src/lib/fsm.cc (revision 8292)
@@ -27,4 +27,5 @@
 #include <stdexcept>
 #include <cmath>
+#include <stdlib.h>
 #include "base.h"
 #include "fsm.h"
Index: gnuradio/trunk/gr-trellis/src/lib/fsm.h
===================================================================
--- gnuradio/trunk/gr-trellis/src/lib/fsm.h (revision 6044)
+++ gnuradio/trunk/gr-trellis/src/lib/fsm.h (revision 8292)
@@ -25,4 +25,5 @@
 
 #include <vector>
+#include <iosfwd>
 
 /*!
Index: gnuradio/trunk/gr-trellis/src/lib/trellis_permutation.cc
===================================================================
--- gnuradio/trunk/gr-trellis/src/lib/trellis_permutation.cc (revision 6044)
+++ gnuradio/trunk/gr-trellis/src/lib/trellis_permutation.cc (revision 8292)
@@ -28,4 +28,5 @@
 #include <gr_io_signature.h>
 #include <iostream>
+#include <string.h>
 
 trellis_permutation_sptr 
Index: gnuradio/trunk/gr-video-sdl/src/video_sdl_sink_s.cc
===================================================================
--- gnuradio/trunk/gr-video-sdl/src/video_sdl_sink_s.cc (revision 6044)
+++ gnuradio/trunk/gr-video-sdl/src/video_sdl_sink_s.cc (revision 8292)
@@ -36,4 +36,5 @@
 #include <iostream>
 #include <stdexcept>
+#include <string.h>
 
 
Index: gnuradio/trunk/gr-video-sdl/src/video_sdl_sink_uc.cc
===================================================================
--- gnuradio/trunk/gr-video-sdl/src/video_sdl_sink_uc.cc (revision 6044)
+++ gnuradio/trunk/gr-video-sdl/src/video_sdl_sink_uc.cc (revision 8292)
@@ -36,4 +36,5 @@
 #include <iostream>
 #include <stdexcept>
+#include <string.h>
 
 
Index: gnuradio/trunk/usrp/host/apps/usrper.cc
===================================================================
--- gnuradio/trunk/usrp/host/apps/usrper.cc (revision 6044)
+++ gnuradio/trunk/usrp/host/apps/usrper.cc (revision 8292)
@@ -31,4 +31,5 @@
 #include "usrp_prims.h"
 #include "usrp_spi_defs.h"
+#include <string.h>
 
 char *prog_name;
Index: gnuradio/trunk/usrp/host/lib/legacy/fusb_win32.cc
===================================================================
--- gnuradio/trunk/usrp/host/lib/legacy/fusb_win32.cc (revision 6044)
+++ gnuradio/trunk/usrp/host/lib/legacy/fusb_win32.cc (revision 8292)
@@ -29,4 +29,5 @@
 #include <assert.h>
 #include <stdexcept>
+#include <string.h>
 
 static const int MAX_BLOCK_SIZE = fusb_sysconfig::max_block_size();
Index: gnuradio/trunk/usrp/host/lib/legacy/usrp_basic.cc
===================================================================
--- gnuradio/trunk/usrp/host/lib/legacy/usrp_basic.cc (revision 6044)
+++ gnuradio/trunk/usrp/host/lib/legacy/usrp_basic.cc (revision 8292)
@@ -35,4 +35,5 @@
 #include <math.h>
 #include <ad9862.h>
+#include <string.h>
 
 using namespace ad9862;
Index: gnuradio/trunk/usrp/host/lib/legacy/usrp_local_sighandler.cc
===================================================================
--- gnuradio/trunk/usrp/host/lib/legacy/usrp_local_sighandler.cc (revision 6044)
+++ gnuradio/trunk/usrp/host/lib/legacy/usrp_local_sighandler.cc (revision 8292)
@@ -33,4 +33,5 @@
 #include <stdexcept>
 #include <stdio.h>
+#include <string.h>
 
 usrp_local_sighandler::usrp_local_sighandler (int signum,
Index: gnuradio/trunk/usrp/host/lib/legacy/usrp_prims.cc
===================================================================
--- gnuradio/trunk/usrp/host/lib/legacy/usrp_prims.cc (revision 6044)
+++ gnuradio/trunk/usrp/host/lib/legacy/usrp_prims.cc (revision 8292)
@@ -63,4 +63,5 @@
 
 #include "std_paths.h"
+#include <stdio.h>
 
 static char *
Index: gnuradio/trunk/usrp/host/misc/getopt.c
===================================================================
--- gnuradio/trunk/usrp/host/misc/getopt.c (revision 6044)
+++ gnuradio/trunk/usrp/host/misc/getopt.c (revision 8292)
@@ -53,4 +53,5 @@
 
 #include <stdio.h>
+#include <string.h>
 
 /* Comment out all this code if we are using the GNU C Library, and are not
