--- gtk+-2.10.14/gdk/directfb/Makefile.am.orig	2007-07-16 15:46:15.000000000 -0400
+++ gtk+-2.10.14/gdk/directfb/Makefile.am	2007-09-24 15:12:14.000000000 -0400
@@ -8,7 +8,6 @@
 	-I$(top_srcdir)			\
 	-I$(top_srcdir)/gdk		\
 	-I$(top_builddir)/gdk		\
-	-DG_DISABLE_DEPRECATED		\
 	@GTK_DEBUG_FLAGS@		\
 	@GDK_DEP_CFLAGS@		
 
--- gtk+-2.10.14/gdk/quartz/Makefile.am.orig	2007-07-16 15:46:18.000000000 -0400
+++ gtk+-2.10.14/gdk/quartz/Makefile.am	2007-09-24 15:12:28.000000000 -0400
@@ -7,7 +7,6 @@
 	-I$(top_srcdir)/gdk		\
 	-I$(top_builddir)/gdk		\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED	\
-	-DGDK_DISABLE_DEPRECATED	\
 	$(GTK_DEBUG_FLAGS) 		\
 	$(GDK_DEP_CFLAGS)		\
 	"-xobjective-c"

diff -ur gtk+-2.10.9~org/gdk-pixbuf/pixops/Makefile.am gtk+-2.10.9/gdk-pixbuf/pixops/Makefile.am
--- gtk+-2.10.9~org/gdk-pixbuf/pixops/Makefile.am   2007-03-18 00:02:40.000000000 -0400
+++ gtk+-2.10.9/gdk-pixbuf/pixops/Makefile.am   2007-03-18 10:25:06.000000000 -0400
@@@ -3,8 +3,7 @@
 INCLUDES = \
 	-I$(top_srcdir) -I$(top_builddir) 	\
 	$(GTK_DEBUG_FLAGS)			\
-	$(GDK_PIXBUF_DEP_CFLAGS)		\
-	-DGDK_PIXBUF_DISABLE_DEPRECATED
+	$(GDK_PIXBUF_DEP_CFLAGS)		
 
 noinst_PROGRAMS = timescale
 
