--- atd/dist/etc/init.d/atd.orig	2003-01-24 07:09:03.000000000 -0800
+++ atd/dist/etc/init.d/atd	2005-09-22 14:50:55.000000000 -0700
@@ -16,7 +16,7 @@ case "$1" in
         ;;
   stop)
         echo -n "Stopping at daemon: atd"
-        start-stop-daemon --stop --quiet --oknodo --pidfile /var/run/atd.pid --exec /usr/sbin/atd
+        start-stop-daemon --stop --quiet --pidfile /var/run/atd.pid --exec /usr/sbin/atd
         rm -f /var/spool/at/trigger
         echo "."
         ;;
