
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- gnuplot-4.0.0/Makefile.am~nodocs
+++ gnuplot-4.0.0/Makefile.am
@@ -1,7 +1,7 @@
 ## Process this file with automake to produce Makefile.in -*-Makefile-*-
 AUTOMAKE_OPTIONS = foreign 1.2h
 
-SUBDIRS = config m4 term src docs lisp man demo tutorial
+SUBDIRS = config m4 term src man tutorial
 
 EXTRA_DIST = BUGS CodeStyle Copyright FAQ GNUmakefile INSTALL INSTALL.gnu \
 Makefile.maint PATCHLEVEL PGPKEYS PORTING README README.1ST README.exp \
