diff --git node-v0.2.6/deps/libev/wscript.orig node-v0.2.6/deps/libev/wscript
index 19e7bb2..5ec9143 100644
--- node-v0.2.6/deps/libev/wscript.orig
+++ node-v0.2.6/deps/libev/wscript
@@ -52,7 +52,7 @@ def configure(conf):
           return 0;
       }
   """
-  conf.check_cc(fragment=code, define_name="HAVE_CLOCK_SYSCALL", execute=True,
+  conf.check_cc(fragment=code, define_name="HAVE_CLOCK_SYSCALL", execute=False,
                 msg="Checking for SYS_clock_gettime")
 
   have_librt = conf.check(lib='rt', uselib_store='RT')
