--- abi/src/wp/ap/xp/GNUmakefile.am.orig	2005-05-09 21:33:34.915201456 +0000
+++ abi/src/wp/ap/xp/GNUmakefile.am	2005-05-09 21:33:49.341008400 +0000
@@ -107,7 +107,7 @@
 		ap_wp_sidebar.cpp
 
 CDUMP=$(top_srcdir)/src/tools/cdump/xp/cdump.c
-CDUMPAPP=$(top_builddir)/src/tools/cdump/xp/cdump
+CDUMPAPP=cdump
 
 ap_wp_splash.cpp: $(CDUMP) $(srcdir)/splash.png
 	$(CDUMPAPP) $(srcdir)/splash.png g_pngSplash > ap_wp_splash.cpp
