diff -ur maki_V1.0.4ern.orig/maki.desktop maki_V1.0.4ern/maki.desktop
--- maki_V1.0.4ern.orig/maki.desktop	2004-06-17 00:26:00.000000000 +0200
+++ maki_V1.0.4ern/maki.desktop	2004-06-17 00:26:07.000000000 +0200
@@ -2,7 +2,7 @@
 Comment=The Game Of Maki / The Same Game
 Comment[de]=Das Maki Spiel / The Same Game
 Exec=maki
-Icon=Maki
+Icon=maki/Maki
 Type=Application
 Type[de]=Anwendung
 Name=Maki
