Backport from CVS

Index: widget.ui
===================================================================
RCS file: /cvs/opie/core/applets/aboutapplet/widget.ui,v
retrieving revision 1.6
retrieving revision 1.7
diff -u -u -r1.6 -r1.7
--- aboutapplet/widget.ui	22 Apr 2005 13:01:47 -0000	1.6
+++ aboutapplet/widget.ui	13 Nov 2005 14:40:31 -0000	1.7
@@ -134,6 +134,7 @@
 &lt;b&gt;Holger 'Zecke' Freyther&lt;/b&gt; (zecke@handhelds.org)&lt;br/&gt;
 &lt;b&gt;Robert 'Sandman' Griebl&lt;/b&gt; (sandman@handhelds.org)&lt;br/&gt;
 &lt;b&gt;Alberto 'Skyhusker' Hierro&lt;/b&gt; (skyhusker@handhelds.org)&lt;br/&gt;
+&lt;b&gt;Marcin 'Hrw' Juszkiewicz&lt;/b&gt; (hrw@handhelds.org)&lt;br/&gt;
 &lt;b&gt;Chris 'Kergoth' Larson&lt;/b&gt; (kergoth@handhelds.org)&lt;br/&gt;
 &lt;b&gt;Michael 'Mickey' Lauer&lt;/b&gt; (mickeyl@handhelds.org)&lt;br/&gt;
 &lt;b&gt;Lorn 'ljp' Potter&lt;/b&gt; (ljp@handhelds.org)&lt;br/&gt;
