From a15dcc1ce77bd298dd472458e0cc836b9b4484c4 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Tue, 20 Apr 2010 20:59:27 +0200
Subject: [PATCH 44/46] ARM: OMAP: modify mpurate logic to accept anything upto 4GHz, instead of the previous 999MHz

---
 arch/arm/plat-omap/clock.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm/plat-omap/clock.c b/arch/arm/plat-omap/clock.c
index 89cafc9..42788ab 100644
--- a/arch/arm/plat-omap/clock.c
+++ b/arch/arm/plat-omap/clock.c
@@ -190,7 +190,7 @@ static int __init omap_clk_setup(char *str)
 	if (!mpurate)
 		return 1;
 
-	if (mpurate < 1000)
+	if (mpurate < 4000)
 		mpurate *= 1000000;
 
 	return 1;
-- 
1.6.6.1

