--- xserver-common-1.34.orig/X11/xserver-common
+++ xserver-common-1.34/X11/xserver-common
@@ -186,6 +186,8 @@
             # this is to rotate X display properly, tested with display TX09D71VM1CCA
             ronetix*pm9261*  | ronetix*pm9263*)
                  ARGS="$ARGS -screen 240/54x320/81@90x16" ;;
+        "atmel_at91sam9x5-ek")
+            DPI="188" ;;
         # This is a fallback for PCs
         "")
             if [ -f $BINDIR/Xvesa ]; then
