diff -bBdurN hannes/ixp400_xscale_sw/src/include/IxQMgr.h merged/ixp400_xscale_sw/src/include/IxQMgr.h
--- ixp400_xscale_sw/src/include/IxQMgr.h	2005-04-17 20:55:31.000000000 -0700
+++ ixp400_xscale_sw/src/include/IxQMgr.h	2005-10-07 15:08:42.723183066 -0700
@@ -96,8 +96,13 @@
 
 #else
 
+#ifndef __ARMEB__
+#define IX_QMGR_INLINE_READ_LONG IX_OSAL_READ_LONG_LE_AC
+#define IX_QMGR_INLINE_WRITE_LONG IX_OSAL_WRITE_LONG_LE_AC
+#else /* __ARMEB__ */
 #define IX_QMGR_INLINE_READ_LONG IX_OSAL_READ_LONG_LE_DC
 #define IX_QMGR_INLINE_WRITE_LONG IX_OSAL_WRITE_LONG_LE_DC
+#endif /* __ARMEB__ */
 
 #endif
 
