diff -Naur ixp400_xscale_sw.orig/Makefile ixp400_xscale_sw/Makefile
--- ixp400_xscale_sw/.pc/Makefile.patch/Makefile	2005-04-17 20:54:58.000000000 -0700
+++ ixp400_xscale_sw/Makefile	2005-10-01 18:15:08.422924549 -0700
@@ -312,9 +312,9 @@
 else
 # linux compiler flags 
 
-LINUX_MACH_CFLAGS := -D__LINUX_ARM_ARCH__=5 -mcpu=xscale -mtune=xscale
+LINUX_MACH_CFLAGS := -D__LINUX_ARM_ARCH__=5 -mtune=xscale
 
-CFLAGS := -D__KERNEL__ -I$(LINUX_SRC)/include -Wall -Wno-trigraphs -fno-common -pipe -mapcs-32 -mshort-load-bytes -msoft-float -DMODULE -Isrc/include -D__linux -DCPU=33 -DXSCALE=33 $(LINUX_MACH_CFLAGS)
+CFLAGS := -D__KERNEL__ -I$(LINUX_SRC)/include -Wall -Wno-trigraphs -fno-common -pipe -mapcs-32 -msoft-float -DMODULE -Isrc/include -D__linux -DCPU=33 -DXSCALE=33 $(LINUX_MACH_CFLAGS)
 ifndef IX_INCLUDE_MICROCODE
 CFLAGS += -DIX_NPEDL_READ_MICROCODE_FROM_FILE
 endif
@@ -806,9 +806,11 @@
 
 ifeq ($(IX_HOST_OS),linux)
 ifndef IX_INCLUDE_MICROCODE
+ifdef IX_BUILD_MICROCODE
 Makefile: $(NPE_CONVERTER) $(NPE_DAT)
 endif
 endif
+endif
 
 ################################################################
 # Rules to check that macros are defined.
@@ -1253,10 +1255,18 @@
 
 ifndef IX_INCLUDE_MICROCODE
 $(OBJ_DIR)/ixp400.o: $(COMPONENTS:%=$(OBJ_DIR)/ixp400_%.o) $(OSAL_MODULE)
+	touch $(OBJ_DIR)/ixp400.c
+	cp Makefile.kmod26 $(OBJ_DIR)/Makefile
+	make -C $(OBJ_DIR)
 	$(LD) $(LDFLAGS) $^ -o $@
+	make -C $(OBJ_DIR)
 else
 $(OBJ_DIR)/ixp400.o: $(COMPONENTS:%=$(OBJ_DIR)/ixp400_%.o) $(NPE_PRODUCTION_HEADER_OBJ) $(OSAL_MODULE)
+	touch $(OBJ_DIR)/ixp400.c
+	cp Makefile.kmod26 $(OBJ_DIR)/Makefile
+	make -C $(OBJ_DIR)
 	$(LD) $(LDFLAGS) $^ -o $@
+	make -C $(OBJ_DIR)
 endif
 
 
diff -Naur ixp400_xscale_sw.orig/Makefile.kmod26 ixp400_xscale_sw/Makefile.kmod26
--- ixp400_xscale_sw.orig/Makefile.kmod26	1970-01-01 01:00:00.000000000 +0100
+++ ixp400_xscale_sw/Makefile.kmod26	2005-08-24 22:33:12.000000000 +0200
@@ -0,0 +1,11 @@
+obj-m := ixp400.o
+
+PWD         := $(shell pwd)
+
+LINUX_SRC := $($(IX_TARGET)_KERNEL_DIR)
+
+default:
+	$(MAKE) ARCH=arm CROSS_COMPILE=$(LINUX_CROSS_COMPILE) V=1 -C $(LINUX_SRC) SUBDIRS=$(PWD) modules
+
+clean:
+	rm -f ixp400.ko
