
#
# Signed off by Michael 'Mickey' Lauer <mickey@Vanille.de>
#

--- src/qt/qt_main.cc~qt2-fixes-r1.patch
+++ src/qt/qt_main.cc
@@ -137,7 +137,7 @@
       switch (orientation[0]) {
          case 't':
          case 'T': break;
-
+#ifndef QWS
          case 'b':
          case 'B': moveDockWindow (toolBar, DockBottom); break;
 
@@ -146,7 +146,7 @@
 
          case 'r':
          case 'R': moveDockWindow (toolBar, DockRight); break;
-
+#endif
          default: roadmap_log (ROADMAP_FATAL,
                         "Invalid toolbar orientation %s", orientation);
       }
--- src/qt/Makefile~qt2-fixes-r1.patch
+++ src/qt/Makefile
@@ -19,7 +19,7 @@
 
 ifeq ($(DESKTOP),QPE)
    CFLAGS += -I$(QTDIR)/include -DQWS
-   LIBS += -lqte -lqpe
+   LIBS += -lqpe
    INSTALLDIR=$(QTDIR)
 endif
 
