Index: qt-x11-opensource-desktop-4.0.0-rc1/mkspecs/qws/linux-arm-g++/qmake.conf
===================================================================
--- qt-x11-opensource-desktop-4.0.0-rc1.orig/mkspecs/qws/linux-arm-g++/qmake.conf	2005-06-06 16:35:19.000000000 +0200
+++ qt-x11-opensource-desktop-4.0.0-rc1/mkspecs/qws/linux-arm-g++/qmake.conf	2005-06-15 21:26:15.000000000 +0200
@@ -8,12 +8,12 @@
 QT                      += core gui network
 QMAKE_INCREMENTAL_STYLE = sublib
 
-QMAKE_CC		= arm-linux-gcc
+QMAKE_CC		= $(CCACHE) arm-linux-gcc
 QMAKE_LEX		= flex
 QMAKE_LEXFLAGS		=
 QMAKE_YACC		= yacc
 QMAKE_YACCFLAGS		= -d
-QMAKE_CFLAGS		= -pipe
+QMAKE_CFLAGS		= -pipe $(EXTRA_CFLAGS)
 QMAKE_CFLAGS_WARN_ON	= -Wall -W
 QMAKE_CFLAGS_WARN_OFF	=
 QMAKE_CFLAGS_RELEASE	= -O2
@@ -22,8 +22,8 @@
 QMAKE_CFLAGS_YACC	= -Wno-unused -Wno-parentheses
 QMAKE_CFLAGS_THREAD	= -D_REENTRANT
 
-QMAKE_CXX		= arm-linux-g++
-QMAKE_CXXFLAGS		= $$QMAKE_CFLAGS -fno-exceptions -fno-rtti
+QMAKE_CXX		= $(CCACHE) arm-linux-g++
+QMAKE_CXXFLAGS		= $$QMAKE_CFLAGS -fno-exceptions -fno-rtti $(EXTRA_CXXFLAGS)
 QMAKE_CXXFLAGS_WARN_ON	= $$QMAKE_CFLAGS_WARN_ON
 QMAKE_CXXFLAGS_WARN_OFF	= $$QMAKE_CFLAGS_WARN_OFF
 QMAKE_CXXFLAGS_RELEASE	= $$QMAKE_CFLAGS_RELEASE
@@ -45,14 +45,14 @@
 
 QMAKE_LINK		= arm-linux-g++
 QMAKE_LINK_SHLIB	= arm-linux-g++
-QMAKE_LFLAGS		=
+QMAKE_LFLAGS		= $(EXTRA_LFLAGS)
 QMAKE_LFLAGS_RELEASE	=
 QMAKE_LFLAGS_DEBUG	=
 QMAKE_LFLAGS_SHLIB      = -shared
 QMAKE_LFLAGS_PLUGIN     = $$QMAKE_LFLAGS_SHLIB
 QMAKE_LFLAGS_SONAME     = -Wl,-soname,
 QMAKE_LFLAGS_THREAD     =
-QMAKE_RPATH             = -Wl,-rpath,
+QMAKE_RPATH             = -Wl,-rpath-link,
 
 QMAKE_LIBS		= 
 QMAKE_LIBS_DYNLOAD      = -ldl
Index: qt-x11-opensource-desktop-4.0.0-rc1/mkspecs/freebsd-g++/qmake.conf
===================================================================
--- qt-x11-opensource-desktop-4.0.0-rc1.orig/mkspecs/freebsd-g++/qmake.conf	2005-06-06 16:34:59.000000000 +0200
+++ qt-x11-opensource-desktop-4.0.0-rc1/mkspecs/freebsd-g++/qmake.conf	2005-06-15 21:41:52.000000000 +0200
@@ -7,7 +7,7 @@
 CONFIG			+= qt warn_on release link_prl
 QT			+= core gui
 
-QMAKE_CC		= gcc
+QMAKE_CC		= $(CCACHE) gcc
 QMAKE_LEX		= flex
 QMAKE_LEXFLAGS		= 
 QMAKE_YACC		= yacc
@@ -22,7 +22,7 @@
 QMAKE_CFLAGS_YACC	= -Wno-unused -Wno-parentheses
 QMAKE_CFLAGS_THREAD	= -pthread -D_THREAD_SAFE
 
-QMAKE_CXX		= g++
+QMAKE_CXX		= $(CCACHE) g++
 QMAKE_CXXFLAGS		= $$QMAKE_CFLAGS
 QMAKE_CXXFLAGS_DEPS	= $$QMAKE_CFLAGS_DEPS
 QMAKE_CXXFLAGS_WARN_ON	= $$QMAKE_CFLAGS_WARN_ON
Index: qt-x11-opensource-desktop-4.0.0-rc1/mkspecs/freebsd-g++34/qmake.conf
===================================================================
--- qt-x11-opensource-desktop-4.0.0-rc1.orig/mkspecs/freebsd-g++34/qmake.conf	2005-06-06 16:34:59.000000000 +0200
+++ qt-x11-opensource-desktop-4.0.0-rc1/mkspecs/freebsd-g++34/qmake.conf	2005-06-15 21:42:13.000000000 +0200
@@ -7,7 +7,7 @@
 CONFIG			+= qt warn_on release link_prl
 QT			+= core gui
 
-QMAKE_CC		= gcc34
+QMAKE_CC		= $(CCACHE) gcc34
 QMAKE_LEX		= flex
 QMAKE_LEXFLAGS		= 
 QMAKE_YACC		= yacc
@@ -22,7 +22,7 @@
 QMAKE_CFLAGS_YACC	= -Wno-unused -Wno-parentheses
 QMAKE_CFLAGS_THREAD	= -pthread -D_THREAD_SAFE
 
-QMAKE_CXX		= g++34
+QMAKE_CXX		= $(CCACHE) g++34
 QMAKE_CXXFLAGS		= $$QMAKE_CFLAGS
 QMAKE_CXXFLAGS_DEPS	= $$QMAKE_CFLAGS_DEPS
 QMAKE_CXXFLAGS_WARN_ON	= $$QMAKE_CFLAGS_WARN_ON
Index: qt-x11-opensource-desktop-4.0.0-rc1/mkspecs/freebsd-g++40/qmake.conf
===================================================================
--- qt-x11-opensource-desktop-4.0.0-rc1.orig/mkspecs/freebsd-g++40/qmake.conf	2005-06-06 16:34:59.000000000 +0200
+++ qt-x11-opensource-desktop-4.0.0-rc1/mkspecs/freebsd-g++40/qmake.conf	2005-06-15 21:42:44.000000000 +0200
@@ -7,7 +7,7 @@
 CONFIG			+= qt warn_on release link_prl
 QT			+= core gui
 
-QMAKE_CC		= gcc40
+QMAKE_CC		= $(CCACHE) gcc40
 QMAKE_LEX		= flex
 QMAKE_LEXFLAGS		= 
 QMAKE_YACC		= yacc
@@ -22,7 +22,7 @@
 QMAKE_CFLAGS_YACC	= -Wno-unused -Wno-parentheses
 QMAKE_CFLAGS_THREAD	= -pthread -D_THREAD_SAFE
 
-QMAKE_CXX		= g++40
+QMAKE_CXX		= $(CCACHE) g++40
 QMAKE_CXXFLAGS		= $$QMAKE_CFLAGS
 QMAKE_CXXFLAGS_DEPS	= $$QMAKE_CFLAGS_DEPS
 QMAKE_CXXFLAGS_WARN_ON	= $$QMAKE_CFLAGS_WARN_ON
Index: qt-x11-opensource-desktop-4.0.0-rc1/mkspecs/linux-g++-32/qmake.conf
===================================================================
--- qt-x11-opensource-desktop-4.0.0-rc1.orig/mkspecs/linux-g++-32/qmake.conf	2005-06-06 16:35:00.000000000 +0200
+++ qt-x11-opensource-desktop-4.0.0-rc1/mkspecs/linux-g++-32/qmake.conf	2005-06-15 21:40:50.000000000 +0200
@@ -8,7 +8,7 @@
 QT			+= core gui
 QMAKE_INCREMENTAL_STYLE = sublib
 
-QMAKE_CC		= gcc
+QMAKE_CC		= $(CCACHE) gcc
 QMAKE_LEX		= flex
 QMAKE_LEXFLAGS		= 
 QMAKE_YACC		= yacc
@@ -27,7 +27,7 @@
 QMAKE_CFLAGS_THREAD	= -D_REENTRANT
 QMAKE_CFLAGS_HIDESYMS   = -fvisibility=hidden
 
-QMAKE_CXX		= g++
+QMAKE_CXX		= $(CCACHE) g++
 QMAKE_CXXFLAGS		= $$QMAKE_CFLAGS
 QMAKE_CXXFLAGS_DEPS	= $$QMAKE_CFLAGS_DEPS
 QMAKE_CXXFLAGS_WARN_ON	= $$QMAKE_CFLAGS_WARN_ON
Index: qt-x11-opensource-desktop-4.0.0-rc1/mkspecs/linux-g++-64/qmake.conf
===================================================================
--- qt-x11-opensource-desktop-4.0.0-rc1.orig/mkspecs/linux-g++-64/qmake.conf	2005-06-06 16:35:00.000000000 +0200
+++ qt-x11-opensource-desktop-4.0.0-rc1/mkspecs/linux-g++-64/qmake.conf	2005-06-15 21:41:10.000000000 +0200
@@ -11,7 +11,7 @@
 QT			+= core gui
 QMAKE_INCREMENTAL_STYLE = sublib
 
-QMAKE_CC		= gcc
+QMAKE_CC		= $(CCACHE) gcc
 QMAKE_LEX		= flex
 QMAKE_LEXFLAGS		=
 QMAKE_YACC		= yacc
@@ -26,7 +26,7 @@
 QMAKE_CFLAGS_YACC	= -Wno-unused -Wno-parentheses
 QMAKE_CFLAGS_THREAD	= -D_REENTRANT
 
-QMAKE_CXX		= g++
+QMAKE_CXX		= $(CCACHE) g++
 QMAKE_CXXFLAGS		= $$QMAKE_CFLAGS
 QMAKE_CXXFLAGS_DEPS	= $$QMAKE_CFLAGS_DEPS
 QMAKE_CXXFLAGS_WARN_ON	= $$QMAKE_CFLAGS_WARN_ON
Index: qt-x11-opensource-desktop-4.0.0-rc1/mkspecs/linux-g++/qmake.conf
===================================================================
--- qt-x11-opensource-desktop-4.0.0-rc1.orig/mkspecs/linux-g++/qmake.conf	2005-06-06 16:35:00.000000000 +0200
+++ qt-x11-opensource-desktop-4.0.0-rc1/mkspecs/linux-g++/qmake.conf	2005-06-15 21:40:24.000000000 +0200
@@ -8,7 +8,7 @@
 QT			+= core gui
 QMAKE_INCREMENTAL_STYLE = sublib
 
-QMAKE_CC		= gcc
+QMAKE_CC		= $(CCACHE) gcc
 QMAKE_LEX		= flex
 QMAKE_LEXFLAGS		= 
 QMAKE_YACC		= yacc
@@ -27,7 +27,7 @@
 QMAKE_CFLAGS_THREAD	= -D_REENTRANT
 QMAKE_CFLAGS_HIDESYMS   = -fvisibility=hidden
 
-QMAKE_CXX		= g++
+QMAKE_CXX		= $(CCACHE) g++
 QMAKE_CXXFLAGS		= $$QMAKE_CFLAGS
 QMAKE_CXXFLAGS_DEPS	= $$QMAKE_CFLAGS_DEPS
 QMAKE_CXXFLAGS_WARN_ON	= $$QMAKE_CFLAGS_WARN_ON
