Index: linux-wlan-ng-0.2.3/src/prism2/driver/prism2_usb.c
===================================================================
--- linux-wlan-ng-0.2.3.orig/src/prism2/driver/prism2_usb.c	2006-04-18 20:09:17.403457000 +0200
+++ linux-wlan-ng-0.2.3/src/prism2/driver/prism2_usb.c	2006-04-18 20:11:21.064657640 +0200
@@ -313,7 +313,7 @@
 
 
 struct usb_driver prism2_usb_driver = {
-#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,4,19))
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,4,19) && (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,16)))
 	.owner = THIS_MODULE,
 #endif
 	.name = "prism2_usb",
