Index: trunk/src/prism2/driver/prism2_usb.c
===================================================================
--- trunk.orig/src/prism2/driver/prism2_usb.c	2006-04-02 11:06:32.045530320 +0200
+++ trunk/src/prism2/driver/prism2_usb.c	2006-04-02 11:09:31.080312896 +0200
@@ -313,7 +313,7 @@
 
 
 struct usb_driver prism2_usb_driver = {
-#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,4,19) && (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,16)) 
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,4,19) && (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,16))) 
 	.owner = THIS_MODULE,
 #endif
 	.name = "prism2_usb",
