Index: zziplib-0.13.49/zzip/Makefile.am
===================================================================
--- zziplib-0.13.49.orig/zzip/Makefile.am	2010-06-07 02:16:47.749129902 +0200
+++ zziplib-0.13.49/zzip/Makefile.am	2010-06-07 02:17:08.932466913 +0200
@@ -44,14 +44,14 @@
                     $(THREAD_SAFE)
 libzzip_la_LIBADD= -lz @RESOLVES@
 
-libzzipmmapped_la_SOURCES =  mmapped.c memdisk.c fetch.c
-libzzipmmapped_la_headers =  mmapped.h memdisk.h fetch.h
+libzzipmmapped_la_SOURCES =  mmapped.c memdisk.c fetch.c fetch.h
+libzzipmmapped_la_headers =  mmapped.h memdisk.h
 libzzipmmapped_la_LDFLAGS= @ZZIPLIB_LDFLAGS@ $(RELEASE_INFO) $(VERSION_INFO) \
                     $(THREAD_SAFE)
 libzzipmmapped_la_LIBADD= -lz @RESOLVES@
 
-libzzipfseeko_la_SOURCES = fseeko.c fetch.c
-libzzipfseeko_la_headers = fseeko.h fetch.h
+libzzipfseeko_la_SOURCES = fseeko.c fetch.c fetch.h
+libzzipfseeko_la_headers = fseeko.h
 libzzipfseeko_la_LDFLAGS= @ZZIPLIB_LDFLAGS@ $(RELEASE_INFO) $(VERSION_INFO) \
                     $(THREAD_SAFE)
 libzzipfseeko_la_LIBADD= -lz @RESOLVES@
