Index: zziplib-0.10.82/bins/zziptest.c
===================================================================
--- zziplib-0.10.82.orig/bins/zziptest.c	2008-02-20 12:10:35.000000000 +0100
+++ zziplib-0.10.82/bins/zziptest.c	2008-02-20 12:10:44.000000000 +0100
@@ -52,7 +52,7 @@
         }
     } printf("OK.\n");
   
-#if 1
+#if 0
     printf("{check...\n");
     { struct zzip_dir_hdr * hdr = dir->hdr0;
     
