--- linux-2.6.20.4.orig/include/asm-arm/kexec.h	2007-04-10 19:16:39.000000000 +0200
+++ linux-2.6.20.4/include/asm-arm/kexec.h	2007-04-03 05:01:09.000000000 +0200
@@ -8,8 +8,8 @@
 /* Maximum address we can reach in physical address mode */
 #define KEXEC_DESTINATION_MEMORY_LIMIT (-1UL)
 /* Maximum address we can use for the control code buffer */
-#define KEXEC_CONTROL_MEMORY_LIMIT TASK_SIZE
-
+//#define KEXEC_CONTROL_MEMORY_LIMIT TASK_SIZE
+#define KEXEC_CONTROL_MEMORY_LIMIT (-1UL)
 #define KEXEC_CONTROL_CODE_SIZE	4096
 
 #define KEXEC_ARCH KEXEC_ARCH_ARM
