diff -uNr linux-2.6.21.vanilla/drivers/pcmcia/sa1100_generic.c linux-2.6.21/drivers/pcmcia/sa1100_generic.c
--- linux-2.6.21.vanilla/drivers/pcmcia/sa1100_generic.c	2007-05-01 16:40:46.000000000 +0200
+++ linux-2.6.21/drivers/pcmcia/sa1100_generic.c	2007-05-01 19:38:05.000000000 +0200
@@ -28,6 +28,9 @@
     the provisions above, a recipient may use your version of this
     file under either the MPL or the GPL.
     
+    2007 mrnice: added thesings changes from device_driver
+            to platform_driver - many thx to thesing
+
 ======================================================================*/
 
 #include <linux/module.h>
@@ -81,13 +84,15 @@
 	return ret;
 }
 
-static struct device_driver sa11x0_pcmcia_driver = {
-	.probe		= sa11x0_drv_pcmcia_probe,
-	.remove		= soc_common_drv_pcmcia_remove,
-	.name		= "sa11x0-pcmcia",
-	.bus		= &platform_bus_type,
-	.suspend 	= pcmcia_socket_dev_suspend,
-	.resume 	= pcmcia_socket_dev_resume,
+static struct platform_driver sa11x0_pcmcia_driver = {
+       .driver        = {
+              .name   = "sa11x0-pcmcia",
+              .probe = sa11x0_drv_pcmcia_probe,
+              .remove        = soc_common_drv_pcmcia_remove,
+              .suspend= pcmcia_socket_dev_suspend,
+              .resume        = pcmcia_socket_dev_resume,
+              //.bus = &platform_bus_type,
+       },
 };
 
 /* sa11x0_pcmcia_init()
@@ -100,7 +105,7 @@
  */
 static int __init sa11x0_pcmcia_init(void)
 {
-	return driver_register(&sa11x0_pcmcia_driver);
+        return platform_driver_register(&sa11x0_pcmcia_driver);
 }
 
 /* sa11x0_pcmcia_exit()
@@ -110,7 +115,7 @@
  */
 static void __exit sa11x0_pcmcia_exit(void)
 {
-	driver_unregister(&sa11x0_pcmcia_driver);
+        platform_driver_unregister(&sa11x0_pcmcia_driver);
 }
 
 MODULE_AUTHOR("John Dorsey <john+@cs.cmu.edu>");
