diff -urNd ../gpe-go-0.05-r0/gpe-go-0.05/build/Makefile.dpkg_ipkg gpe-go-0.05/build/Makefile.dpkg_ipkg
--- ../gpe-go-0.05-r0/gpe-go-0.05/build/Makefile.dpkg_ipkg	2003-11-09 20:28:40.000000000 +0000
+++ gpe-go-0.05/build/Makefile.dpkg_ipkg	2004-09-18 17:09:01.000000000 +0100
@@ -18,8 +18,10 @@
 LIBGPEWIDGET_PC = libgpewidget
 endif
 
+ifeq ($(IN_LIBGPEWIDGET),)
 GPECFLAGS = $(shell $(PC_EXTRA) pkg-config --cflags $(LIBGPEWIDGET_PC))
 GPELIBS = $(shell $(PC_EXTRA) pkg-config --libs $(LIBGPEWIDGET_PC))
+endif
 
 GTKCFLAGS = $(shell pkg-config --cflags gtk+-2.0)
 GTKLIBS = $(shell pkg-config --libs gtk+-2.0)
@@ -28,16 +30,16 @@
 STANDARD_CFLAGS = -MD -Wall
 
 ifeq ($(DEBUG),yes)
-STANDARD_CFLAGS += -O2 -g
-STANDARD_LDFLAGS = -g
+CFLAGS += -O2 -g
+LDFLAGS = -g
 else
-STANDARD_CFLAGS += -Os -fomit-frame-pointer
+CFLAGS += -Os -fomit-frame-pointer
 endif
 
 dist: check-source clean dist-prep
 	rm -rf ../$(PACKAGE)-$(VERSION)
 	mkdir ../$(PACKAGE)-$(VERSION)
-	( tar cf - --exclude "*/CVS" --exclude CVS --exclude "*~" --exclude "#*#" --exclude "debian" --exclude ".*" --exclude "*.ipk" --exclude "*.ipk.*" --exclude "*.mo" --exclude "*.d" --exclude "*.batch" --exclude "translation-ipkgs.txt" * ) | (cd ../$(PACKAGE)-$(VERSION); tar xf -)
+	( tar cf - --exclude "*/CVS" --exclude CVS --exclude "*~" --exclude "#*#" --exclude "debian" --exclude ".*" --exclude "*.ipk" --exclude "*.ipk.*" --exclude "*.mo" --exclude "*.batch" --exclude "translation-ipkgs.txt" * ) | (cd ../$(PACKAGE)-$(VERSION); tar xf -)
 	( cd ../$(PACKAGE)-$(VERSION); mkdir build; cp $(BUILD)/Makefile.dpkg_ipkg $(BUILD)/Makefile.translation build/ ; sed 's:^CVSBUILD.*:CVSBUILD = no:' < Makefile > Makefile.new; mv Makefile.new Makefile )
 	( cd .. ; tar cf - $(PACKAGE)-$(VERSION) | gzip -9 >$(PACKAGE)-$(VERSION).tar.gz )
 	rm -rf ../$(PACKAGE)-$(VERSION)
@@ -53,8 +55,13 @@
 
 install: install-program install-mo
 
+clean-dist:
+	rm -rf familiar/dist familiar/dist.list
+
+clean: clean-dist
+
 check-source:
-	@if ! grep -q '^Source:' familiar/$(CONTROL); then echo -e "\nNo Source: field in control file.  Aborting.\n"; exit 1; fi
+	@if [ -e familiar ] && ! grep -q '^Source:' familiar/$(CONTROL); then echo -e "\nNo Source: field in control file.  Aborting.\n"; exit 1; fi
 
 ipkg: check-source ipkg-prep clean
 	rm -rf familiar/dist
@@ -68,9 +75,9 @@
 	$(MAKE) DESTDIR=`pwd`/familiar/dist PREFIX=/usr prefix=/usr DEBUG=no install-program
 	rm -rf familiar/dist.list
 	ipkg-build -o 0 -g 0 familiar/dist | sed 's/^Packaged .*into //; t 1; d; : 1; s:.*/::' >> familiar/dist.list
-	rm -f 
 	if [ "x$(LINGUAS)" != "x" ]; then make translation-ipkg; tr ' ' '\n' < translation-ipkgs.txt >> familiar/dist.list; fi
 	md5sum `cat familiar/dist.list` > $(PACKAGE)_$(VERSION).batch
+	rm -rf familiar/dist familiar/dist.list
 	$(MAKE) printinfo
 
 dpkg: dist
@@ -102,3 +109,6 @@
 
 %.pc: %.pc.in
 	sed 's:PREFIX:$(PREFIX):;s:BUILDDIR:$(shell pwd):;s:VERSION:$(VERSION):' < $< > $@
+
+.c.o:;
+	$(CC) $(CFLAGS) $(CPPFLAGS) $(PACKAGE_CFLAGS) $(PACKAGE_CPPFLAGS) -c $< -o $@
diff -urNd ../gpe-go-0.05-r0/gpe-go-0.05/build/Makefile.translation gpe-go-0.05/build/Makefile.translation
--- ../gpe-go-0.05-r0/gpe-go-0.05/build/Makefile.translation	2003-11-09 20:28:40.000000000 +0000
+++ gpe-go-0.05/build/Makefile.translation	2004-09-18 17:09:01.000000000 +0100
@@ -5,7 +5,7 @@
 # use ipkg-build or ipkg-deb-build
 IPKG_BUILD := ipkg-build
 
-TRANSLATION_SITE := http://www.iro.umontreal.ca/~gnutra/maint/
+TRANSLATION_SITE := http://www.iro.umontreal.ca/~gnutra/maint
 
 ifeq ($(DIR_PO),)
 DIR_PO := po
@@ -56,7 +56,7 @@
 	( SOURCES="$(SOURCES)"; for DESK in $(PACKAGE).desktop.in $(EXTRA_DESKTOPS); do if [ -f $$DESK ]; then intltool-extract --type=gettext/ini $$DESK; SOURCES="$$SOURCES $${DESK}.h"; fi; done; if [ "x$$SOURCES" != "x" ]; then xgettext --add-comments=TRANSLATORS: -k_ -kN_ -o $(DIR_PO)/$(PACKAGE).pot.new $$SOURCES; fi )
 	if [ -f $(DIR_PO)/$(PACKAGE).pot.new ]; then if cmp -s $(DIR_PO)/$(PACKAGE).pot.new $(PACKAGE).pot; then rm $(DIR_PO)/$(PACKAGE).pot.new; else mv $(DIR_PO)/$(PACKAGE).pot.new $(DIR_PO)/$(PACKAGE).pot; fi; fi
 
-clean: clean-po
+clean: clean-po clean-dist-translation
 
 clean-po:
 	rm -rf $(DIR_PO)/*.mo
@@ -76,10 +76,17 @@
 
 MAINTAINER = $(shell grep 'Maintainer: ' familiar/$(CONTROL) | cut -d ' ' -f 2-)
 
+ifndef BUILD
+BUILD = ../build
+endif
+
 transdist := familiar/dist-translation
-templates := ../build/familiar
+templates := $(BUILD)/familiar
 ipkglist  := translation-ipkgs.txt
 
+clean-dist-translation:
+	rm -rf $(transdist) $(ipkglist)
+
 real-translation-package: all-mo
 	rm -rf $(transdist) $(ipkglist)
 	for LINGUA in $(LINGUAS); do \
@@ -89,8 +96,7 @@
 		install -m 644 po/$$LINGUA.mo $(transdist)/$$i$(PREFIX)/share/locale/$$LINGUA/LC_MESSAGES/$(PACKAGE).mo; \
 		sed -e "s/<maintainer>/$(MAINTAINER)/;s/<package>/$(BINPACKAGE)/;s/<version>/$(VERSION)/;s/<language>/$$i/;s!<source>!$(SOURCE)!" $(templates)/control.translation > $(transdist)/$$i/CONTROL/control; \
 		install $(templates)/postinst.translation $(transdist)/$$i/CONTROL/postinst; \
-		chown -R root.root $(transdist)/$$i; \
-		$(IPKG_BUILD) $(transdist)/$$i | sed 's/^Packaged .*into //; t 1; d; : 1; s:.*/::' >> $(ipkglist); \
+		$(IPKG_BUILD) -g 0 -o 0 $(transdist)/$$i | sed 's/^Packaged .*into //; t 1; d; : 1; s:.*/::' >> $(ipkglist); \
 	done
 
 translation-ipkg:
diff -urNd ../gpe-go-0.05-r0/gpe-go-0.05/gpe-go.c gpe-go-0.05/gpe-go.c
--- ../gpe-go-0.05-r0/gpe-go-0.05/gpe-go.c	2003-11-09 19:38:50.000000000 +0000
+++ gpe-go-0.05/gpe-go.c	2004-09-18 17:04:24.000000000 +0100
@@ -16,12 +16,12 @@
 #include "sgf.h"
 
 //--GPE libs
-#include "gpe/init.h"
-#include "gpe/pixmaps.h"
-#include "gpe/errorbox.h"
-#include "gpe/question.h"
-#include "gpe/popup_menu.h"
-#include "gpe/picturebutton.h"
+#include <gpe/init.h>
+#include <gpe/pixmaps.h>
+#include <gpe/errorbox.h>
+#include <gpe/question.h>
+#include <gpe/popup_menu.h>
+#include <gpe/picturebutton.h>
 
 //--i18n
 #include <libintl.h>
diff -urNd ../gpe-go-0.05-r0/gpe-go-0.05/Makefile gpe-go-0.05/Makefile
--- ../gpe-go-0.05-r0/gpe-go-0.05/Makefile	2003-11-09 20:28:40.000000000 +0000
+++ gpe-go-0.05/Makefile	2004-09-18 17:11:06.000000000 +0100
@@ -1,19 +1,8 @@
 VERSION = 0.05
-
 PACKAGE = gpe-go
-
-ifndef PREFIX
-PREFIX = /usr/local
-endif
-
-ifndef DEBUG
+PREFIX = /usr
 DEBUG = no
-endif
-
-ifndef CVSBUILD
 CVSBUILD = no
-endif
-
 
 LINGUAS = cs de pl ro sv nl fr pt ru
 
@@ -21,40 +10,34 @@
 
 PIXMAPS = board white.png black.png
 
-GTKCFLAGS   = `pkg-config --cflags gtk+-2.0`
-GTKLDFLAGS += `pkg-config --libs gtk+-2.0`
+GTKCFLAGS = `pkg-config --cflags gtk+-2.0`
+GTKLDFLAGS = `pkg-config --libs gtk+-2.0 gdk-2.0`
 
-CPPFLAGS = $(GTKCFLAGS) -D_GNU_SOURCE 
-ifeq ($(DEBUG),yes)
-CFLAGS  += -O2 -g -DDEBUG=yes
-LDFLAGS += -g
+PACKAGE_CFLAGS += $(STANDARD_CFLAGS) $(GTKCFLAGS) $(GPECFLAGS)
+PACKAGE_CPPFLAGS += $(STANDARD_CPPFLAGS) -DPACKAGE=\"$(PACKAGE)\" -DPREFIX=\"$(PREFIX)\" -DPACKAGE_LOCALE_DIR=\"$(PREFIX)/share/locale\"
+PACKAGE_LDFLAGS += $(STANDARD_LDFLAGS) $(GTKLDFLAGS) $(GPELIBS)
+
+ifeq ($(CVSBUILD),yes)
+BUILD = ../build
 else
-CFLAGS += -Os -fomit-frame-pointer
+BUILD = build
 endif
-CFLAGS += -Wall -I../../base/libgpewidget
-CPPFLAGS += -DPACKAGE=\"$(PACKAGE)\" -DPREFIX=\"$(PREFIX)\" -DPACKAGE_LOCALE_DIR=\"$(PREFIX)/share/locale\"
-CFLAGS += -MD
-
-LDFLAGS += -L../../base/libgpewidget -lgpewidget
-
-.SUFFIXES: .d
 
 OBJS = $(patsubst %,%.o,$(MEMBERS))
-DEPS = $(patsubst %,%.d,$(MEMBERS))
 SOURCES = $(patsubst %,%.c,$(MEMBERS))
+DEPS = $(patsubst %,%.d,$(MEMBERS))
 
 ifdef DESKTOP #environement variable to set if you want extra code included.
-CPPFLAGS += -DDESKTOP
+PACKAGE_CPPFLAGS += -DDESKTOP
 endif
 
 all: $(PACKAGE)
 
 $(PACKAGE): $(OBJS)
-	$(CC) -o $@ $^  $(GTKLDFLAGS) $(LDFLAGS)
+	$(CC) -o $@ $^ $(LDFLAGS) $(PACKAGE_LDFLAGS)
 
 install-program: $(PACKAGE).desktop all 
-	install -D $(PACKAGE) $(DESTDIR)$(PREFIX)/bin/$(PACKAGE)
-	strip $(DESTDIR)$(PREFIX)/bin/$(PACKAGE)
+	install -s -D $(PACKAGE) $(DESTDIR)$(PREFIX)/bin/$(PACKAGE)
 	install -d $(DESTDIR)$(PREFIX)/share/pixmaps
 	install -m 644 $(PACKAGE).png $(DESTDIR)$(PREFIX)/share/pixmaps/
 	install -d $(DESTDIR)$(PREFIX)/share/applications
@@ -69,15 +52,6 @@
 	find . -name "*~" -exec rm -f {} \;
 
 
-#--include GPE generic Makefiles
-ifeq ($(CVSBUILD),yes)
-BUILD = ../../base/build
-else
-BUILD = build
-endif
-
 include $(BUILD)/Makefile.dpkg_ipkg
 include $(BUILD)/Makefile.translation
-
-
 -include $(DEPS)
