diff -urN wireless_tools.29.orig/Makefile wireless_tools.29/Makefile
--- wireless_tools.29.orig/Makefile	2007-09-17 23:56:46.000000000 +0000
+++ wireless_tools.29/Makefile	2009-04-02 09:22:14.000000000 +0000
@@ -145,7 +145,7 @@
 
 # Compilation of the dynamic library
 $(DYNAMIC): $(OBJS:.o=.so)
-	$(CC) -shared -o $@ -Wl,-soname,$@ $(STRIPFLAGS) $(LIBS) -lc $^
+	$(CC) $(LDFLAGS) -shared -o $@ -Wl,-soname,$@ $(STRIPFLAGS) $(LIBS) -lc $^
 
 # Compilation of the static library
 $(STATIC): $(OBJS:.o=.so)
