From 03cbda241413e4ae727b2a92ff452162aa7ad440 Mon Sep 17 00:00:00 2001
From: Michael Krelin <hacker@klever.net>
Date: Sat, 2 Jun 2007 16:06:01 +0200
Subject: [PATCH] no tools

---
 src/src.pro |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/src.pro b/src/src.pro
index 6e08180..957d3dc 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -3,7 +3,7 @@ TEMPLATE = subdirs
 # this order is important
 unset(SRC_SUBDIRS)
 win32:SRC_SUBDIRS += src_winmain
-SRC_SUBDIRS += src_tools_moc src_tools_rcc src_tools_uic src_corelib src_xml src_gui src_sql src_network src_svg src_script
+SRC_SUBDIRS += src_corelib src_xml src_gui src_sql src_network src_svg src_script
 contains(QT_CONFIG, opengl): SRC_SUBDIRS += src_opengl
 contains(QT_CONFIG, qt3support): SRC_SUBDIRS += src_qt3support
 !cross_compile {
-- 
1.5.0.7

