Index: kobodeluxe-0.5.1/graphics/gfxengine.cpp
===================================================================
--- kobodeluxe-0.5.1.orig/graphics/gfxengine.cpp	2008-02-10 03:02:29.000000000 +0100
+++ kobodeluxe-0.5.1/graphics/gfxengine.cpp	2008-02-10 03:03:45.000000000 +0100
@@ -25,6 +25,7 @@
 
 #include <string.h>
 #include <math.h>
+#include <stdlib.h>
 
 #include "gfxengine.h"
 #include "filters.h"
@@ -765,7 +766,7 @@
 		return 0;
 
 	if(_centered && !_fullscreen)
-		SDL_putenv((char *)"SDL_VIDEO_CENTERED=1");
+		putenv((char *)"SDL_VIDEO_CENTERED=1");
 
 	log_printf(DLOG, "Opening screen...\n");
 	if(!SDL_WasInit(SDL_INIT_VIDEO))
