--- obexftp-0.20/obexftp/Makefile.am	2007-06-27 20:33:20.000000000 +0200
+++ obexftp-0.20/obexftp/Makefile.am	2007-06-27 20:35:05.000000000 +0200
@@ -4,12 +4,6 @@
 				-I$(top_srcdir) \
 				-I$(top_srcdir)/includes
 
-LDADD =                         @OPENOBEX_LIBS@
-#				@BLUETOOTH_LIBS@ \
-#				@USB_LIBS@ \
-#				../multicobex/libmulticobex.la \
-#				../bfb/libbfb.la
-
 obexftpincludedir =		$(includedir)/obexftp
 
 libobexftp_la_SOURCES =		object.c object.h \
@@ -23,6 +17,8 @@
 				uuid.h \
 				object.h
 	
+libobexftp_la_LIBADD =          @OPENOBEX_LIBS@ @LIBICONV@
+
 libobexftp_la_LDFLAGS =		-version-info $(OBEXFTP_VERSION)
 
 EXTRA_DIST =			client.i
