 drivers/input/misc/ixp4xx-beeper.c |    1 +
 1 file changed, 1 insertion(+)

--- linux-nslu2.orig/drivers/input/misc/ixp4xx-beeper.c	2006-02-06 20:37:19.000000000 +0100
+++ linux-nslu2/drivers/input/misc/ixp4xx-beeper.c	2006-02-06 22:04:59.000000000 +0100
@@ -19,6 +19,7 @@
 #include <linux/input.h>
 #include <linux/delay.h>
 #include <linux/platform_device.h>
+#include <linux/interrupt.h>
 #include <asm/hardware.h>
 
 MODULE_AUTHOR("Alessandro Zummo <a.zummo@towertech.it>");
