diff -ruN linux-2.6.15.orig/include/asm-arm/arch-ixp4xx/hardware.h linux-2.6.15.new/include/asm-arm/arch-ixp4xx/hardware.h
--- linux-2.6.15.orig/include/asm-arm/arch-ixp4xx/hardware.h    2006-02-11 02:58:47.000000000 +0100
+++ linux-2.6.15.new/include/asm-arm/arch-ixp4xx/hardware.h     2006-02-10 22:26:54.000000000 +0100
@@ -46,5 +46,6 @@
 #include "prpmc1100.h"
 #include "nslu2.h"
 #include "nas100d.h"
+#include "ds101.h"

 #endif  /* _ASM_ARCH_HARDWARE_H */
diff -ruN linux-2.6.15.orig/include/asm-arm/arch-ixp4xx/irqs.h linux-2.6.15.new/include/asm-arm/arch-ixp4xx/irqs.h
--- linux-2.6.15.orig/include/asm-arm/arch-ixp4xx/irqs.h        2006-02-11 02:58:47.000000000 +0100
+++ linux-2.6.15.new/include/asm-arm/arch-ixp4xx/irqs.h 2006-02-10 22:56:11.000000000 +0100
@@ -108,5 +108,12 @@
 #define        IRQ_NAS100D_PCI_INTC    IRQ_IXP4XX_GPIO9
 #define        IRQ_NAS100D_PCI_INTD    IRQ_IXP4XX_GPIO8
 #define        IRQ_NAS100D_PCI_INTE    IRQ_IXP4XX_GPIO7
+/*
+ * DS101 board IRQs
+ */
+
+#define        IRQ_DS101_PCI_INTA    IRQ_IXP4XX_GPIO11
+#define        IRQ_DS101_PCI_INTB    IRQ_IXP4XX_GPIO10
+#define        IRQ_DS101_PCI_INTC    IRQ_IXP4XX_GPIO9

 #endif
diff -ruN linux-2.6.15.orig/include/asm-arm/arch/ds101.h linux-2.6.15.new/include/asm-arm/arch/ds101.h
--- linux-2.6.15.orig/include/asm-arm/arch-ixp4xx/ds101.h      1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.15.new/include/asm-arm/arch-ixp4xx/ds101.h       2006-02-10 22:16:24.000000000 +0100
@@ -0,0 +1,72 @@
+/*
+ * include/asm-arm/arch-ixp4xx/ds101.h
+ *
+ * DS101 platform specific definitions
+ *
+ * Copyright (c) 2005 Tower Technologies
+ *
+ * Author: Alessandro Zummo <a.zummo@towertech.it>
+ *
+ * based on ixdp425.h:
+ *     Copyright 2004 (c) MontaVista, Software, Inc.
+ *
+ * This file is licensed under  the terms of the GNU General Public
+ * License version 2. This program is licensed "as is" without any
+ * warranty of any kind, whether express or implied.
+ */
+
+#ifndef __ASM_ARCH_HARDWARE_H__
+#error "Do not include this directly, instead #include <asm/hardware.h>"
+#endif
+
+#define DS101_SDA_PIN          1
+#define DS101_SCL_PIN          0
+
+/*
+ * DS101 PCI IRQs
+ */
+#define DS101_PCI_MAX_DEV      3
+#define DS101_PCI_IRQ_LINES    3
+
+
+/* PCI controller GPIO to IRQ pin mappings */
+#define DS101_PCI_INTA_PIN     11
+#define DS101_PCI_INTB_PIN     10
+#define DS101_PCI_INTC_PIN     9
+// #define DS101_PCI_INTD_PIN  8
+// #define DS101_PCI_INTE_PIN  7
+
+/* GPIO */
+
+#define DS101_GPIO0           0
+#define DS101_GPIO1           1
+#define DS101_GPIO2           2
+#define DS101_GPIO3           3
+#define DS101_GPIO4           4
+#define DS101_GPIO5           5
+#define DS101_GPIO6           6
+#define DS101_GPIO7           7
+#define DS101_GPIO8           8
+#define DS101_GPIO9           9
+#define DS101_GPIO10          10
+#define DS101_GPIO11          11
+#define DS101_GPIO12          12
+#define DS101_GPIO13          13
+#define DS101_GPIO14          14
+#define DS101_GPIO15          15
+
+
+/* Buttons */
+
+#define DS101_PB_GPIO         DS101_GPIO8
+#define DS101_RB_GPIO         DS101_GPIO12
+#define DS101_PO_GPIO         DS101_GPIO7   /* power off */
+
+#define DS101_PB_IRQ          IRQ_IXP4XX_GPIO8
+#define DS101_RB_IRQ          IRQ_IXP4XX_GPIO12
+
+/*
+#define DS101_PB_BM           (1L << DS101_PB_GPIO)
+#define DS101_PO_BM           (1L << DS101_PO_GPIO)
+#define DS101_RB_BM           (1L << DS101_RB_GPIO)
+*/
