diff -ruN linux-2.6.15.orig/arch/arm/mach-ixp4xx/Makefile linux-2.6.15.new/arch/arm/mach-ixp4xx/Makefile
--- linux-2.6.15.orig/arch/arm/mach-ixp4xx/Makefile     2006-02-11 02:57:58.000000000 +0100
+++ linux-2.6.15.new/arch/arm/mach-ixp4xx/Makefile      2006-02-10 22:13:27.000000000 +0100
@@ -10,4 +10,4 @@
 obj-$(CONFIG_MACH_GTWX5715)	+= gtwx5715-pci.o gtwx5715-setup.o
 obj-$(CONFIG_MACH_NSLU2)	+= nslu2-pci.o nslu2-setup.o nslu2-power.o
 obj-$(CONFIG_MACH_NAS100D)	+= nas100d-pci.o nas100d-setup.o nas100d-power.o
-
+obj-$(CONFIG_MACH_DS101)	+= ds101-pci.o ds101-setup.o ds101-power.o
diff -ruN linux-2.6.15.orig/arch/arm/mach-ixp4xx/Kconfig linux-2.6.15.new/arch/arm/mach-ixp4xx/Kconfig
--- linux-2.6.15.orig/arch/arm/mach-ixp4xx/Kconfig      2006-02-11 03:00:07.000000000 +0100
+++ linux-2.6.15.new/arch/arm/mach-ixp4xx/Kconfig       2006-02-10 22:21:09.000000000 +0100
@@ -86,6 +86,14 @@
          NAS 100d device. For more information on this platform,
          see http://www.nslu2-linux.org/wiki/NAS100d/HomePage

+config MACH_DS101
+       bool
+       prompt "DS101"
+       help
+         Say 'Y' here if you want your kernel to support Synology's
+         DiskStation DS101(j) device. For more information on this
+         platform see http://www.nslu2-linux.org/wiki/DS101/HomePage
+
 #
 # Avila and IXDP share the same source for now. Will change in future
 #
