*** linux-2.6.16/arch/arm/mm/copypage-xscale.c-	Mon Mar 27 19:10:25 2006
--- linux-2.6.16/arch/arm/mm/copypage-xscale.c	Sun Mar 26 19:32:53 2006
***************
*** 15,20 ****
--- 15,21 ----
   */
  #include <linux/init.h>
  #include <linux/mm.h>
+ #include <linux/module.h>
  
  #include <asm/page.h>
  #include <asm/pgtable.h>
***************
*** 129,131 ****
--- 130,135 ----
  	.cpu_clear_user_page	= xscale_mc_clear_user_page, 
  	.cpu_copy_user_page	= xscale_mc_copy_user_page,
  };
+ 
+ EXPORT_SYMBOL(xscale_mc_clear_user_page);
+ EXPORT_SYMBOL(xscale_mc_copy_user_page);
