--- totem-0.99.12/omf.make	2003-11-08 18:04:41.000000000 -0700
+++ totem-0.99.12.new/omf.make	2004-06-30 09:39:58.765495480 -0700
@@ -41,7 +41,9 @@
 install-data-hook-omf:
 	$(mkinstalldirs) $(DESTDIR)$(omf_dest_dir)
 	for file in $(omffile); do \
-		$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
+		if [ -f $$file.out ]; then \
+			$(INSTALL_DATA) $$file.out $(DESTDIR)$(omf_dest_dir)/$$file; \
+		fi; \
 	done
 	-scrollkeeper-update -o $(DESTDIR)$(omf_dest_dir)
 
