Index: mythplugins-0.22rc1/mythnews/mythnews/mythnews.pro
===================================================================
--- mythplugins-0.22rc1.orig/mythnews/mythnews/mythnews.pro
+++ mythplugins-0.22rc1/mythnews/mythnews/mythnews.pro
@@ -10,9 +10,9 @@ TARGET = mythnews
 target.path = $${LIBDIR}/mythtv/plugins
 INSTALLS += target
 
-INCLUDEPATH += $${PREFIX}/include/mythtv
-INCLUDEPATH += $${PREFIX}/include/mythtv/libmythui
-INCLUDEPATH += $${PREFIX}/include/mythtv/libmythdb
+INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv
+INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv/libmythui
+INCLUDEPATH += $${SYSROOT}$${PREFIX}/include/mythtv/libmythdb
 
 installfiles.path = $${PREFIX}/share/mythtv/mythnews
 installfiles.files = news-sites.xml
