--- /tmp/libgpewidget-0.100/doc/gtk-doc.make	2005-04-21 23:56:55.000000000 +0200
+++ libgpewidget-0.100/doc/gtk-doc.make	2005-04-22 12:36:23.628751472 +0200
@@ -27,7 +27,6 @@
 	$(DOC_MAIN_SGML_FILE)		\
 	$(DOC_MODULE)-sections.txt	\
 	$(DOC_MODULE)-overrides.txt
-#	$(DOC_MODULE).types		\
 
 DOC_STAMPS=scan-build.stamp tmpl-build.stamp sgml-build.stamp html-build.stamp \
 	   $(srcdir)/tmpl.stamp $(srcdir)/sgml.stamp $(srcdir)/html.stamp
