*** openpbx.org-1.2_rc3.old/bootstrap.sh	Mon Feb 26 16:02:40 2007
--- openpbx.org-1.2_rc3/bootstrap.sh	Mon Feb 26 16:04:11 2007
***************
*** 89,104 ****
  # Check for required version and die if unhappy
  
  if [ "x$UNAME" = "xFreeBSD" ]; then
! version_compare libtoolize 1 5 20 || exit 1
! version_compare automake19 1 9 6 || exit 1
  version_compare autoconf259 2 59 || exit 1
  ACLOCAL=aclocal19
  AUTOHEADER=autoheader259
  AUTOMAKE=automake19
  AUTOCONF=autoconf259
  else
! version_compare libtoolize 1 5 20 || exit 1
! version_compare automake 1 9 6 || exit 1
  version_compare autoconf 2 59 || exit 1
  ACLOCAL=aclocal
  AUTOHEADER=autoheader
--- 89,104 ----
  # Check for required version and die if unhappy
  
  if [ "x$UNAME" = "xFreeBSD" ]; then
! version_compare libtoolize 1 5 10 || exit 1
! version_compare automake19 1 9 3 || exit 1
  version_compare autoconf259 2 59 || exit 1
  ACLOCAL=aclocal19
  AUTOHEADER=autoheader259
  AUTOMAKE=automake19
  AUTOCONF=autoconf259
  else
! version_compare libtoolize 1 5 10 || exit 1
! version_compare automake 1 9 3 || exit 1
  version_compare autoconf 2 59 || exit 1
  ACLOCAL=aclocal
  AUTOHEADER=autoheader
