From c7f749b564b473f0160a201c769153cfcaf3bc0b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Matthias=20G=C3=BCnther?= <matgnt@gmail.com>
Date: Thu, 13 May 2010 14:27:22 +0200
Subject: [PATCH 5/9] Disable examples/help

---
 examples/examples.pro |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/examples/examples.pro b/examples/examples.pro
index 785c35c..e37f580 100644
--- a/examples/examples.pro
+++ b/examples/examples.pro
@@ -53,7 +53,7 @@ contains(QT_CONFIG, script): SUBDIRS += script
 contains(QT_CONFIG, phonon):!static: SUBDIRS += phonon
 embedded:SUBDIRS += qws
 !wince*:!symbian: {
-    contains(QT_BUILD_PARTS, tools):!contains(QT_CONFIG, no-gui):SUBDIRS += qtestlib help
+    contains(QT_BUILD_PARTS, tools):!contains(QT_CONFIG, no-gui):SUBDIRS += qtestlib
 } else {
     contains(QT_BUILD_PARTS, tools):!contains(QT_CONFIG, no-gui):SUBDIRS += qtestlib
 }
-- 
1.7.0.4

