diff --git gpe-plucker-0.4/Makefile-orig gpe-plucker-0.4/Makefile
index c68b5ab..abf97bb 100644
--- gpe-plucker-0.4/Makefile-orig
+++ gpe-plucker-0.4/Makefile
@@ -43,7 +43,7 @@ $(PACKAGE): $(OBJS) $(LIB)
 
 install-program: all $(PACKAGE).desktop
 	mkdir -p $(DESTDIR)$(PREFIX)/bin
-	install -s $(PACKAGE) $(DESTDIR)$(PREFIX)/bin/$(PACKAGE)
+	install $(PACKAGE) $(DESTDIR)$(PREFIX)/bin/$(PACKAGE)
 	mkdir -p $(DESTDIR)$(PREFIX)/share/pixmaps
 	install -m 644 $(PACKAGE).png $(DESTDIR)$(PREFIX)/share/pixmaps/$(PACKAGE).png
 	#mkdir -p $(DESTDIR)$(PREFIX)/share/$(PACKAGE)
