diff -Nurd busybox-1.11.3.orig//networking/ifupdown.c busybox-1.11.3/networking/ifupdown.c
--- busybox-1.11.3.orig//networking/ifupdown.c	2010-06-15 08:45:44.000000000 +0200
+++ busybox-1.11.3/networking/ifupdown.c	2010-06-15 08:46:09.000000000 +0200
@@ -476,7 +476,7 @@
 		"pump -i %iface% -k",
 	},
 	{ "udhcpc",
-		"udhcpc -R -n -p /var/run/udhcpc.%iface%.pid -i %iface%[[ -H %hostname%]][[ -c %clientid%]]"
+		"udhcpc -R -b -p /var/run/udhcpc.%iface%.pid -i %iface%[[ -H %hostname%]][[ -c %clientid%]]"
 				"[[ -s %script%]][[ %udhcpc_opts%]]",
 		"kill `cat /var/run/udhcpc.%iface%.pid` 2>/dev/null",
 	},
@@ -507,7 +507,7 @@
 	if (!execute("ip link set %iface% up", ifd, exec))
 		return 0;
 #endif
-	return execute("udhcpc -R -n -p /var/run/udhcpc.%iface%.pid "
+	return execute("udhcpc -R -b -p /var/run/udhcpc.%iface%.pid "
 			"-i %iface%[[ -H %hostname%]][[ -c %clientid%]][[ -s %script%]][[ %udhcpc_opts%]]",
 			ifd, exec);
 }
