diff -urNd ../gpe-dm-0.44-r1/gpe-dm-0.44/Xserver gpe-dm-0.44/Xserver
--- ../gpe-dm-0.44-r1/gpe-dm-0.44/Xserver	2005-02-20 17:39:31 +00:00
+++ gpe-dm-0.44/Xserver	2005-04-01 21:47:12 +01:00
@@ -45,8 +45,14 @@
 	"HP iPAQ H5400" | "HP iPAQ H2200")
 		ARGS="$ARGS -rgba rgb" ;;
 	# both 'Sharp-Collie' and just 'Collie' have been reported
-	*Collie | *Poodle)
+	*Poodle)
 		ARGS="$ARGS -screen 320x240@270 -rgba vrgb" ;;
+	*Collie)
+		ARGS="$ARGS -screen 320x240@270 -rgba vrgb"
+		# Horrible hack required to enable resuming after suspend
+		rm -f /dev/apm_bios
+		killall -9 apmd
+		;;
 	"SHARP Shepherd" | "SHARP Husky" | "SHARP Corgi")
 		ARGS="$ARGS -rgba rgb" ;;
 	"Simpad")
