
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- gpe-dm-0.44/gpe-dm.init~ramses
+++ gpe-dm-0.44/gpe-dm.init
@@ -19,7 +19,7 @@
   stop)
         echo "Stopping GPE display manager: gpe-dm"
         killproc /usr/bin/gpe-dm
-	dd if=/dev/zero of=/dev/fb0 2>/dev/null
+	dd if=/dev/zero of=/dev/fb0 2>/dev/null || dd if=/dev/zero of=/dev/fb/0 2>/dev/null
   ;;
 
   *)
--- gpe-dm-0.44/Xserver~ramses
+++ gpe-dm-0.44/Xserver
@@ -44,6 +44,8 @@
 		ARGS="$ARGS -screen 320x240@270 -rgba vbgr" ;;
 	"HP iPAQ H5400" | "HP iPAQ H2200")
 		ARGS="$ARGS -rgba rgb" ;;
+	"Ramses")
+		ARGS="$ARGS -screen 320x240@90 -rgba vrgb vt2" ;;
 	# both 'Sharp-Collie' and just 'Collie' have been reported
 	*Poodle)
 		ARGS="$ARGS -screen 320x240@270 -rgba vrgb" ;;
