--- gpe-dm-0.48/Makefile.orig	2005-08-29 12:16:35.000000000 +0200
+++ gpe-dm-0.48/Makefile	2006-02-09 05:00:22.000000000 +0100
@@ -30,7 +30,8 @@
 	install -s gpe-dm $(DESTDIR)$(PREFIX)/bin/
 	install -d $(DESTDIR)/etc/init.d
 	install -d $(DESTDIR)/etc/rc2.d
-	install gpe-dm.init $(DESTDIR)/etc/init.d/gpe-dm
+	install -d $(DESTDIR)/etc/X11//login-managers	
+	install gpe-dm.init $(DESTDIR)/etc/X11/login-managers/gpe-dm
 	install -d $(DESTDIR)/etc/X11
 
 clean: clean-po
