diff -Nur gpe-dm-0.47~/Makefile gpe-dm-0.47/Makefile
--- gpe-dm-0.47~/Makefile	2005-08-19 18:27:54.000000000 -0700
+++ gpe-dm-0.47/Makefile	2005-08-19 18:28:37.000000000 -0700
@@ -31,10 +31,6 @@
 	install -d $(DESTDIR)/etc/init.d
 	install -d $(DESTDIR)/etc/rc2.d
 	install gpe-dm.init $(DESTDIR)/etc/init.d/gpe-dm
-	install -d $(DESTDIR)/etc/X11
-	install Xsession $(DESTDIR)/etc/X11/Xsession
-	install Xinit $(DESTDIR)/etc/X11/Xinit
-	install Xserver $(DESTDIR)/etc/X11/Xserver
 
 clean: clean-po
 	rm -f gpe-dm gpe-dm.o
