--- qpPhoto_1.0.2/draw.cpp	2002-10-29 22:19:05.000000000 +0100
+++ qpPhoto_1.0.2/draw.cpp.new	2006-05-12 18:42:08.000000000 +0200
@@ -16,6 +16,7 @@
 #include <qspinbox.h>
 #include <qpainter.h>
 #include <qlabel.h>
+#include <qlayout.h>
 #include <qdialog.h>
 #include <qpe/fontdatabase.h>
 #include <qstringlist.h>
@@ -781,7 +782,7 @@
 
 
 
-Draw::Draw( QWidget* parent = 0, const char* name = 0, WFlags fl = 0 )
+Draw::Draw( QWidget* parent, const char* name, WFlags fl )
 :DrawWidget(parent, name, fl)
 {
     /*currentTool = dtNull;
@@ -799,9 +800,12 @@
     FontDatabase fdb;
     availFonts = fdb.families();
 
-
-    dv = new DrawView(GrFrame, "dv");
-    dv->setGeometry( QRect( 1, 1, 210, 210) );
+    dv = new DrawView(this, "dv");
+    dv->setSizePolicy( QSizePolicy( (QSizePolicy::SizeType)7, (QSizePolicy::SizeType)7, dv->sizePolicy().hasHeightForWidth() ) );
+    dv->setMinimumSize( QSize( 210, 210 ) );
+    dv->setFrameShape( QFrame::WinPanel );
+    dv->setFrameShadow( QFrame::Sunken );
+    Layout7->addWidget( dv );
 
     dv->font.setFamily(availFonts[0]);
     dv->font.setPointSize(12);
