Index: mozilla/nsprpub/pr/src/misc/prdtoa.c
===================================================================
--- mozilla.orig/nsprpub/pr/src/misc/prdtoa.c	2004-03-12 07:30:50.000000000 +0000
+++ mozilla/nsprpub/pr/src/misc/prdtoa.c	2007-05-07 11:35:10.000000000 +0100
@@ -132,8 +132,8 @@
  *	is not strictly legal and can cause trouble with aggressively
  *	optimizing compilers (e.g., gcc 2.95.1 under -O2).
  */
-#if defined(__arm) || defined(__arm__) || defined(__arm26__) \
-    || defined(__arm32__)
+#if (defined(__arm) || defined(__arm__) || defined(__arm26__) \
+     || defined(__arm32__)) && !defined(__ARM_EABI__) && !defined(__ARMEB__)
 #define IEEE_ARM
 #elif defined(IS_LITTLE_ENDIAN)
 #define IEEE_8087
Index: mozilla/nsprpub/pr/include/md/_linux.h
===================================================================
--- mozilla.orig/nsprpub/pr/include/md/_linux.h	2005-05-04 19:06:17.000000000 +0100
+++ mozilla/nsprpub/pr/include/md/_linux.h	2007-05-07 11:34:16.000000000 +0100
@@ -285,7 +285,7 @@
 #error "Linux/MIPS pre-glibc2 not supported yet"
 #endif /* defined(__GLIBC__) && __GLIBC__ >= 2 */
 
-#elif defined(__arm__)
+#elif defined(__arm__) && !defined(__ARM_EABI__)
 /* ARM/Linux */
 #if defined(__GLIBC__) && __GLIBC__ >= 2
 #define _MD_GET_SP(_t) (_t)->md.context[0].__jmpbuf[20]
