--- ../libxine-1.0.0-beta12-r0/xine-lib/src/libvorbis/xine_decoder.c	2004-08-17 16:07:31.000000000 +0100
+++ xine-lib/src/libvorbis/xine_decoder.c	2004-08-17 17:08:21.000000000 +0100
@@ -208,7 +208,7 @@
     int **pcm;
     int samples;
 
-    if(vorbis_synthesis(&this->vb,op)==0) 
+    if(vorbis_synthesis(&this->vb,op,1)==0) 
       vorbis_synthesis_blockin(&this->vd,&this->vb);
 
     if (buf->pts!=0)
