Copyright (C) 2002, Fred Boulay <dilb@handhelds.org>
Licensed under GPL v2

--- src/xine-engine/audio_out.c.orig	2002-12-18 02:42:43.000000000 +0100
+++ src/xine-engine/audio_out.c	2002-12-18 02:43:19.000000000 +0100
@@ -75,10 +75,10 @@
 #define LOG
 */
 
-#define NUM_AUDIO_BUFFERS       32
-#define AUDIO_BUF_SIZE       32768
+#define NUM_AUDIO_BUFFERS       16
+#define AUDIO_BUF_SIZE        8192
 
-#define ZERO_BUF_SIZE         5000
+#define ZERO_BUF_SIZE         2048
 
 /* By adding gap errors (difference between reported and expected
  * sound card clock) into metronom's vpts_offset we can use its 
--- src/xine-engine/video_decoder.c.orig	2003-01-14 01:10:29.000000000 +0100
+++ src/xine-engine/video_decoder.c	2003-02-12 12:10:33.000000000 +0100
@@ -373,7 +373,7 @@ void video_decoder_init (xine_stream_t *
    * We provide buffers of 8k size instead of 2k for demuxers sending
    * larger chunks.
    */
-  stream->video_fifo = fifo_buffer_new (500, 8192);
+  stream->video_fifo = fifo_buffer_new (250, 4096);
   stream->spu_track_map_entries = 0;
 
   pthread_attr_init(&pth_attrs);
--- src/xine-engine/video_overlay.h.orig	2002-12-18 02:44:08.000000000 +0100
+++ src/xine-engine/video_overlay.h	2002-12-18 02:44:21.000000000 +0100
@@ -32,7 +32,7 @@
 #define CLUT_Y_CR_CB_INIT(_y,_cr,_cb)	{ (_cb), (_cr), (_y) }
 #endif
 
-#define MAX_OBJECTS   50
+#define MAX_OBJECTS    5
 #define MAX_EVENTS    50
 #define MAX_SHOWING    5
 
--- src/xine-engine/audio_decoder.c.orig	2002-12-19 17:16:11.000000000 +0100
+++ src/xine-engine/audio_decoder.c	2002-12-19 17:16:25.000000000 +0100
@@ -322,7 +322,7 @@ void audio_decoder_init (xine_stream_t *
    * We provide buffers of 8k size instead of 2k for demuxers sending
    * larger chunks.
    */
-  stream->audio_fifo = fifo_buffer_new (230, 8192);
+  stream->audio_fifo = fifo_buffer_new (100, 8192);
   stream->audio_channel_user = -1;
   stream->audio_channel_auto = 0;
   stream->audio_track_map_entries = 0;
