Copyright (C) 2002, Fred Boulay <dilb@handhelds.org>
Licensed under GPL v2

--- src/libffmpeg/libavcodec/armv4l/Makefile.am.orig	2002-12-17 10:27:00.000000000 +0100
+++ src/libffmpeg/libavcodec/armv4l/Makefile.am	2002-12-17 10:27:08.000000000 +0100
@@ -13,10 +13,9 @@ libavcodec_armv4l_src = dsputil_arm.c jr
 libavcodec_armv4l_dummy =  libavcodec_armv4l_dummy.c
 EXTRA_DIST =  $(libavcodec_armv4l_src) $(libavcodec_armv4l_dummy)
 
-#if HAVE_ARMV4L
-#armv4l_modules = $(libavcodec_armv4l_src)
-#endif
-armv4l_modules = 
+if HAVE_ARMV4L
+armv4l_modules = $(libavcodec_armv4l_src)
+endif
 
 libavcodec_armv4l_la_SOURCES = $(armv4l_modules) $(libavcodec_armv4l_dummy)
 
