--- ./alsa-kernel/arm/sa11xx-uda1341.c	2005-06-27 12:59:53.000000000 +0200
+++ ./alsa-kernel/arm/sa11xx-uda1341.c.orig	2005-06-27 12:59:29.000000000 +0200
@@ -366,7 +366,7 @@
 	DEBUG("\t request id <%s>\n", s->id);
 	DEBUG("\t  request dma_dev = 0x%x \n", s->dma_dev);
 	if((ret = sa1100_request_dma((s)->dma_dev, (s)->id, callback, s, &((s)->dma_regs))))
-		printk(__FUNCTION__ ": sa1100_request_dma returned %d\n",ret);
+		printk("%s: sa1100_request_dma returned %d\n",__FUNCTION__,ret);
 	
 
 
@@ -978,7 +978,7 @@
 	switch (req) {
 	case PM_SUSPEND: /* enter D1-D3 */
 		if (card->power_state == SNDRV_CTL_POWER_D3hot){
-			printk(__FUNCTION__ ": exiting eraly from suspend\n");			
+			printk("%s: exiting eraly from suspend\n",__FUNCTION__);			
 			return 0;
 		}
 		snd_power_lock(card);
@@ -1002,7 +1002,7 @@
 	case PM_RESUME:  /* enter D0 */
 
 		if (card->power_state == SNDRV_CTL_POWER_D0){
-			printk(__FUNCTION__ ": exiting eraly from resume\n");
+			printk("%s: exiting eraly from resume\n",__FUNCTION__);
 			return 0;
 		}
 		snd_power_lock(card);
@@ -1020,7 +1020,7 @@
 
 		break;
 	}
-	printk(KERN_DEBUG __FUNCTION__": exiting...\n");	
+	printk(KERN_DEBUG "%s: exiting...\n",__FUNCTION__);	
         return 0;
 	
 }
