--- bash-2.05b.orig/lib/readline/nls.c	2001-10-16 03:32:29.000000000 +0900
+++ bash-2.05b/lib/readline/nls.c	2002-10-29 19:07:36.000000000 +0900
@@ -87,7 +87,8 @@
   char *t;
 
   /* Set the LC_CTYPE locale category from environment variables. */
-  t = setlocale (LC_CTYPE, "");
+  // t = setlocale (LC_CTYPE, "");
+  t = setlocale (LC_CTYPE, NULL);
   if (t && *t && (t[0] != 'C' || t[1]) && (STREQ (t, "POSIX") == 0))
     {
       _rl_meta_flag = 1;
