--- ./lib/readline/examples/Makefile.orig	Fri Nov 26 23:13:35 1999
+++ ./lib/readline/examples/Makefile	Fri Nov 26 23:15:19 1999
@@ -1,8 +1,8 @@
 # This is the Makefile for the examples subdirectory of readline. -*- text -*-
 #
 EXECUTABLES = fileman rltest rl
-CFLAGS  = -g -I../.. -I.. -DREADLINE_LIBRARY
-LDFLAGS = -g -L..
+CFLAGS  = -g
+LDFLAGS = -g
 
 .c.o:
 	$(CC) $(CFLAGS) -c $<
--- ./lib/readline/examples/rl.c~	Mon Nov  8 22:22:03 1999
+++ ./lib/readline/examples/rl.c	Fri Dec 31 17:34:30 1999
@@ -11,7 +11,7 @@
 
 #include <stdio.h>
 #include <sys/types.h>
-#include "posixstat.h"
+#include <sys/stat.h>
 
 #if defined (READLINE_LIBRARY)
 #  include "readline.h"
