Index: Makefile
===================================================================
RCS file: /cvs/gpe/base/libgpewidget/Makefile,v
retrieving revision 1.133
diff -u -r1.133 Makefile
--- libgpewidget/Makefile	16 May 2004 18:41:05 -0000	1.133
+++ libgpewidget/Makefile	16 May 2004 18:58:51 -0000
@@ -45,7 +45,7 @@
 	ln -sf $^ $@
 
 $(SONAME): $(OBJS)
-	$(CC) -nostartfiles -shared -o $@ $^ -Wl,-soname -Wl,$(SONAME) $(LDFLAGS)
+	$(CC) -nostartfiles -shared -o $@ $^ -Wl,-soname -Wl,$(SONAME) $(LDFLAGS) $(PACKAGE_LDFLAGS)
 
 gpe-iconlist.c: gpeiconlist.gob
 	gob2 --no-private-header gpeiconlist.gob
