--- binutils-2.15.90.0.3/configure.build_modules	2004-04-22 21:58:37.000000000 -0400
+++ binutils-2.15.90.0.3/configure	2004-04-22 22:05:19.000000000 -0400
@@ -936,6 +936,11 @@
 fi
 
 
+case "$target" in
+  *-*-*-uclibc*)
+    build_modules=
+    ;;
+esac
 ################################################################################
 
 srcname="gnu development package"
--- binutils-2.15.90.0.3/configure.in.build_modules	2004-04-22 21:58:37.000000000 -0400
+++ binutils-2.15.90.0.3/configure.in	2004-04-22 22:05:19.000000000 -0400
@@ -184,6 +184,11 @@
 fi
 
 
+case "$target" in
+  *-*-*-uclibc*)
+    build_modules=
+    ;;
+esac
 ################################################################################
 
 srcname="gnu development package"
