
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- netkit-rpc-0.17/rpcgen/Makefile~install.patch
+++ netkit-rpc-0.17/rpcgen/Makefile
@@ -13,7 +13,9 @@
 rpc_main.o: ../version.h
 
 install: rpcgen
+	install -d $(INSTALLROOT)$(BINDIR)
 	install -s -m$(BINMODE) rpcgen $(INSTALLROOT)$(BINDIR)
+	install -d $(INSTALLROOT)$(MANDIR)
 	install -m$(MANMODE) rpcgen.1 $(INSTALLROOT)$(MANDIR)/man1
 
 clean:
--- netkit-rpc-0.17/rpcinfo/Makefile~install.patch
+++ netkit-rpc-0.17/rpcinfo/Makefile
@@ -8,7 +8,9 @@
 rpcinfo.o: ../version.h
 
 install: rpcinfo
+	install -d $(INSTALLROOT)$(BINDIR)
 	install -s -m$(BINMODE) rpcinfo $(INSTALLROOT)$(BINDIR)
+	install -d $(INSTALLROOT)$(MANDIR)/man8
 	install -m$(MANMODE) rpcinfo.8 $(INSTALLROOT)$(MANDIR)/man8
 
 clean:
