From afb5e0c9c3a9421c6e05b5a0f61c7e8ab67d4cc6 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Fri, 19 Feb 2010 00:24:06 +0100
Subject: [PATCH] glxdri2: flushInvalidate was renamed to invalidate in http://cgit.freedesktop.org/mesa/mesa/commit/?id=debf00e5fc3828f63e0f99d72c7fa6cd6ce012c5

---
 glx/glxdri2.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/glx/glxdri2.c b/glx/glxdri2.c
index 0f998de..a98b522 100644
--- a/glx/glxdri2.c
+++ b/glx/glxdri2.c
@@ -218,7 +218,7 @@ __glXDRIdrawableSwapBuffers(ClientPtr client, __GLXdrawable *drawable)
     CARD64 unused;
 
     if (screen->flush)
-	(*screen->flush->flushInvalidate)(priv->driDrawable);
+	(*screen->flush->invalidate)(priv->driDrawable);
 
     if (DRI2SwapBuffers(client, drawable->pDraw, 0, 0, 0, &unused,
 			__glXdriSwapEvent, drawable->pDraw) != Success)
-- 
1.7.0

