--- a/rdpdr.c	(revisão 1505)
+++ b/rdpdr.c	(cópia de trabalho)
@@ -841,14 +841,11 @@
 RD_BOOL
 rdpdr_init()
 {
-	if (g_num_devices > 0)
-	{
-		rdpdr_channel =
-			channel_register("rdpdr",
-					 CHANNEL_OPTION_INITIALIZED | CHANNEL_OPTION_COMPRESS_RDP,
-					 rdpdr_process);
-	}
-
+	rdpdr_channel =
+		channel_register("rdpdr",
+				 CHANNEL_OPTION_INITIALIZED | CHANNEL_OPTION_COMPRESS_RDP,
+				 rdpdr_process);
+	
 	return (rdpdr_channel != NULL);
 }
 
