
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- roadmap-1.0.12/src/qt/qt_main.cc~qt2-fixes
+++ roadmap-1.0.12/src/qt/qt_main.cc
@@ -123,7 +123,7 @@
       switch (orientation[0]) {
          case 't':
          case 'T': break;
-
+#ifndef QWS
          case 'b':
          case 'B': moveDockWindow (toolBar, DockBottom); break;
 
@@ -132,7 +132,7 @@
 
          case 'r':
          case 'R': moveDockWindow (toolBar, DockRight); break;
-
+#endif
          default: roadmap_log (ROADMAP_FATAL,
                         "Invalid toolbar orientation %s", orientation);
       }
