Index: uClibc-nptl/Rules.mak
===================================================================
--- uClibc-nptl/Rules.mak	(revision 24450)
+++ uClibc-nptl/Rules.mak	(working copy)
@@ -522,7 +522,8 @@
 
 ifeq ($(LDSO_GNU_HASH_SUPPORT),y)
 # Be sure that binutils support it
-LDFLAGS_GNUHASH:=$(call check_ld,--hash-style=gnu)
+#LDFLAGS_GNUHASH:=$(call check_ld,--hash-style=gnu)
+LDFLAGS_GNUHASH:=--hash-style=gnu
 ifeq ($(LDFLAGS_GNUHASH),)
 $(error Your binutils don't support --hash-style option, while you want to use it)
 else
