---
 configure.in |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

Index: DirectFB-1.4.1/configure.in
===================================================================
--- DirectFB-1.4.1.orig/configure.in	2009-06-26 23:30:03.000000000 +0200
+++ DirectFB-1.4.1/configure.in	2009-07-13 00:23:19.558669631 +0200
@@ -1334,8 +1334,11 @@
 
 enable_tslib=no
 if test "$checkfor_tslib" = "yes"; then
-  PKG_CHECK_MODULES([TSLIB], [tslib-0.0], enable_tslib=yes, enable_tslib=no 
-    AC_MSG_WARN([*** no tslib -- tslib driver will not be built.]))
+  PKG_CHECK_MODULES([TSLIB], [tslib-1.0 >= 1.0.0], [enable_tslib=yes], [enable_tslib=no])
+  if test "$enable_tslib" = "no"; then
+     PKG_CHECK_MODULES([TSLIB], [tslib-0.0], [enable_tslib=yes], [enable_tslib=no,
+       AC_MSG_WARN([*** no tslib -- tslib driver will not be built.])])
+  fi
 fi
 
 enable_ucb1x00_ts=no
