--- linux-2.6.17-vanilla/arch/arm/mach-sa1100/cpu-sa1110.c	2006-06-18 01:49:35.000000000 +0000
+++ linux-2.6.17/arch/arm/mach-sa1100/cpu-sa1110.c	2006-07-10 17:06:11.000000000 +0000
@@ -70,6 +70,17 @@
 	.cas_latency		= 3,
 };
 
+static struct sdram_params samsung_k4s281632b_1h __initdata = {
+        .rows                   = 12,
+	.tck                    = 10,
+	.trcd                   = 20,
+	.trp                    = 20,
+	.twr                    = 10,
+	.refresh                = 64000,
+	.cas_latency            = 3,
+	};
+
+
 static struct sdram_params samsung_k4s641632d_tc75 __initdata = {
 	.rows			= 14,
 	.tck			= 9,
@@ -349,6 +360,9 @@
 	if (machine_is_h3100())
 		sdram = &samsung_km416s4030ct;
 
+   if (machine_is_jornada720())
+		sdram = &samsung_k4s281632b_1h;
+
 	if (sdram) {
 		printk(KERN_DEBUG "SDRAM: tck: %d trcd: %d trp: %d"
 			" twr: %d refresh: %d cas_latency: %d\n",
