--- include/linux/soc/hamcop_base.h.orig	2007-07-20 13:18:06.338543484 +0300
+++ include/linux/soc/hamcop_base.h	2007-07-20 13:19:02.853764103 +0300
@@ -24,6 +24,8 @@
 #ifndef HAMCOP_BASE_H
 #define HAMCOP_BASE_H
 
+#include <linux/gpiodev.h>
+
 struct hamcop_platform_data
 {
         struct gpiodev_ops gpiodev_ops;
@@ -33,7 +35,7 @@
 
         u16 clocksleep;
         u16 pllcontrol;
-}
+};
 
 extern void hamcop_set_gpio_a (struct device *dev, u32 mask, u16 bits);
 extern void hamcop_set_gpio_b (struct device *dev, u32 mask, u16 bits);
