Index: gst-ffmpeg-0.10.7/configure.ac
===================================================================
--- gst-ffmpeg-0.10.7.orig/configure.ac	2009-03-21 00:44:01.000000000 +0000
+++ gst-ffmpeg-0.10.7/configure.ac	2009-03-30 20:45:05.000000000 +0000
@@ -337,7 +337,6 @@
   AC_SUBST(FFMPEG_SVN)
   AC_SUBST(FFMPEG_REVISION)
   AC_SUBST(FFMPEG_EXTERNALS_REVISION)
-  AC_CONFIG_SUBDIRS(gst-libs/ext/ffmpeg)
   AC_MSG_NOTICE([Using included FFMpeg code])
 fi
 
