--- portmap_6.0/Makefile.orig	2007-11-21 00:35:52.000000000 -0600
+++ portmap_6.0/Makefile	2007-11-21 00:37:23.000000000 -0600
@@ -125,9 +125,9 @@
 all:	portmap pmap_dump pmap_set portmap.man
 
 CPPFLAGS += $(HOSTS_ACCESS)
-portmap: CFLAGS   += -fpie
+#portmap: CFLAGS   += -fpie
 portmap: LDLIBS   += $(WRAP_LIB)
-portmap: LDFLAGS  += -pie
+#portmap: LDFLAGS  += -pie
 portmap: portmap.o pmap_check.o from_local.o
 
 from_local: CPPFLAGS += -DTEST
