diff -urNd ../libxine-1.0-r1/xine-lib-1.0/configure.ac xine-lib-1.0/configure.ac
--- ../libxine-1.0-r1/xine-lib-1.0/configure.ac	2005-04-09 02:22:48 +01:00
+++ xine-lib-1.0/configure.ac	2005-04-09 02:24:00 +01:00
@@ -710,16 +710,16 @@
 dnl Checks for Ascii-Art library
 dnl ---------------------------------------------
 
-AM_PATH_AALIB(1.4,, AC_MSG_RESULT([*** All of AALIB dependent parts will be disabled ***]))
-AM_CONDITIONAL(HAVE_AA,  [test x"$enable_aalib" = "xyes" && test x$no_aalib != "xyes"])
+dnl AM_PATH_AALIB(1.4,, AC_MSG_RESULT([*** All of AALIB dependent parts will be disabled ***]))
+AM_CONDITIONAL(HAVE_AA, 0)
 
 
 dnl ---------------------------------------------
 dnl Checks for Color AsCii Art library
 dnl ---------------------------------------------
 
-AM_PATH_CACA(0.3,, AC_MSG_RESULT([*** All CACA-dependent parts will be disabled ***]))
-AM_CONDITIONAL(HAVE_CACA, test x$no_caca != "xyes")
+dnl AM_PATH_CACA(0.3,, AC_MSG_RESULT([*** All CACA-dependent parts will be disabled ***]))
+AM_CONDITIONAL(HAVE_CACA, 0)
 
 
 dnl ---------------------------------------------
