---
 soapcpp2/wsdl/Makefile.am |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gsoap-2.7.orig/soapcpp2/wsdl/Makefile.am
+++ gsoap-2.7/soapcpp2/wsdl/Makefile.am
@@ -13,11 +13,11 @@ SOAP_CPP_SRC=wsdlC.cpp
 SOAPHEADER=wsdl.h
 SOAP_FLAGS=-SC -pwsdl
 BUILT_SOURCES=$(SOAP_CPP_SRC)
 
 $(SOAP_CPP_SRC) : $(SOAPHEADER)
-	$(SOAP) $(SOAP_FLAGS) $(SOAPHEADER)
+	../src/soapcpp2 $(SOAP_FLAGS) $(SOAPHEADER)
 
 #LIBS=
 
 bin_PROGRAMS=wsdl2h
 
