Index: backup/backuprestore.cpp
===================================================================
RCS file: /cvs/opie/noncore/settings/backup/backuprestore.cpp,v
retrieving revision 1.42
retrieving revision 1.43
diff -u -u -r1.42 -r1.43
--- backup/backuprestore.cpp	9 Jun 2005 21:04:56 -0000	1.42
+++ backup/backuprestore.cpp	16 Oct 2005 19:00:43 -0000	1.43
@@ -102,6 +102,7 @@
     scanForApplicationSettings();
     refreshLocations();
     refreshBackupLocations();
+    sourceDirChanged(restoreSource->currentItem());
 
     // Read the list of items to ignore.
     QList<QString> dontBackupList;
