Index: curl-7.12.2/lib/mprintf.c
===================================================================
--- curl-7.12.2.orig/lib/mprintf.c	2004-06-24 07:54:11.000000000 -0400
+++ curl-7.12.2/lib/mprintf.c	2005-02-02 17:15:45.435748600 -0500
@@ -1134,8 +1134,10 @@
 }
 
 #ifndef WIN32 /* not needed on win32 */
+#ifndef __UCLIBC__ /* not needed for uclibc, and breaks the build */
 extern int fputc(int, FILE *);
 #endif
+#endif
 
 int curl_mprintf(const char *format, ...)
 {
