Index: llvm-2.7/tools/llvm-shlib/Makefile
===================================================================
--- llvm-2.7/tools/llvm-shlib/Makefile (revision 103711)
+++ llvm-2.7/tools/llvm-shlib/Makefile (working copy)
@@ -56,5 +56,5 @@
     LLVMLibsOptions := -Wl,--whole-archive $(LLVMLibsOptions) 
                        -Wl,--no-whole-archive
     # Don't allow unresolved symbols.
-    LLVMLibsOptions += -Wl,--no-undefined
+    LLVMLibsOptions += -lgcc -Wl,--no-undefined
 endif

