--- /tmp/board-overo.c	2009-04-23 12:46:21.000000000 +0200
+++ git/arch/arm/mach-omap2/board-overo.c	2009-04-23 12:46:31.000000000 +0200
@@ -48,6 +48,12 @@
 #include <mach/nand.h>
 #include <mach/usb.h>
 
+#include <mach/omap-pm.h>
+#include <mach/clock.h>
+
+#include "pm.h"
+#include "omap3-opp.h"
+
 #include "sdram-micron-mt46h32m32lf-6.h"
 #include "twl4030-generic-scripts.h"
 #include "mmc-twl4030.h"
@@ -359,7 +365,8 @@
 
 static void __init overo_init_irq(void)
 {
-	omap2_init_common_hw(mt46h32m32lf6_sdrc_params, NULL, NULL, NULL);
+    omap2_init_common_hw(mt46h32m32lf6_sdrc_params, omap3_mpu_rate_table,
+                 omap3_dsp_rate_table, omap3_l3_rate_table);
 	omap_init_irq();
 	omap_gpio_init();
 }
