--- gtk+-2.6.10/gtk/gtkstatusbar.h	2005/03/20 07:01:22	1.17
+++ gtk+-2.6.10/gtk/gtkstatusbar.h	2005/12/06 14:25:30	1.17.2.1
@@ -65,7 +65,7 @@
 {
   GtkHBoxClass parent_class;
 
-  GMemChunk *messages_mem_chunk;
+  gpointer messages_mem_chunk;
 
   void	(*text_pushed)	(GtkStatusbar	*statusbar,
 			 guint		 context_id,
--- gtk+-2.6.10/gtk/gtkstatusbar.c	2005/07/21 19:10:48	1.56
+++ gtk+-2.6.10/gtk/gtkstatusbar.c	2005/12/06 14:25:30	1.56.2.1
@@ -302,7 +302,7 @@
   g_return_val_if_fail (text != NULL, 0);
 
   class = GTK_STATUSBAR_GET_CLASS (statusbar);
-  msg = g_chunk_new (GtkStatusbarMsg, class->messages_mem_chunk);
+  msg = g_chunk_new (GtkStatusbarMsg, (GMemChunk *)class->messages_mem_chunk);
   msg->text = g_strdup (text);
   msg->context_id = context_id;
   msg->message_id = statusbar->seq_message_id++;
@@ -343,7 +343,7 @@
 	      statusbar->messages = g_slist_remove_link (statusbar->messages,
 							 list);
 	      g_free (msg->text);
-	      g_mem_chunk_free (class->messages_mem_chunk, msg);
+	      g_mem_chunk_free ((GMemChunk *)class->messages_mem_chunk, msg);
 	      g_slist_free_1 (list);
 	      break;
 	    }
@@ -394,7 +394,7 @@
 	      class = GTK_STATUSBAR_GET_CLASS (statusbar);
 	      statusbar->messages = g_slist_remove_link (statusbar->messages, list);
 	      g_free (msg->text);
-	      g_mem_chunk_free (class->messages_mem_chunk, msg);
+	      g_mem_chunk_free ((GMemChunk *)class->messages_mem_chunk, msg);
 	      g_slist_free_1 (list);
 	      
 	      break;
@@ -459,7 +459,7 @@
 
       msg = list->data;
       g_free (msg->text);
-      g_mem_chunk_free (class->messages_mem_chunk, msg);
+      g_mem_chunk_free ((GMemChunk *)class->messages_mem_chunk, msg);
     }
   g_slist_free (statusbar->messages);
   statusbar->messages = NULL;
