Index: base/blueprobe/blueprobe.init
===================================================================
--- base/blueprobe/blueprobe.init	(revision 8939)
+++ base/blueprobe/blueprobe.init	(working copy)
@@ -46,14 +46,14 @@
 	;;
     "HP iPAQ H5400")
 	BLUETOOTH=yes
-	PORT=/dev/tts/1
+	PORT=/dev/ttyS3
 	SPEED=921600
         PROTO=any
 	PROBE=yes
 	;;
     "HP iPAQ H3900")
 	BLUETOOTH=yes
-	PORT=/dev/tts/1
+	PORT=/dev/ttyS1
 	SPEED=921600
         PROTO=bcsp
 	PROBE=yes
