--- tmp.x86_64-linux.arm-linux/base/rxvt-unicode-1.9-r0/rxvt-unicode-1.9/src/defaultfont.C	2004-02-24 17:50:38.000000000 -0500
+++ rxvt-unicode-1.9/src/defaultfont.C	2004-02-25 19:57:16.000000000 -0500
@@ -23,6 +23,7 @@
 #include "rxvt.h"
 #include "defaultfont.h"
 
+#include <stdlib.h>
 #include <cstdlib>
 
 #define DISPLAY  r->display->display
--- tmp.x86_64-linux.arm-linux/base/rxvt-unicode-1.9-r0/rxvt-unicode-1.9/src/rxvtlib.h.in	2004-02-24 07:18:21.000000000 -0500
+++ rxvt-unicode-1.9/src/rxvtlib.h.in	2004-02-25 20:07:51.000000000 -0500
@@ -23,7 +23,12 @@
 #include <sys/stat.h>
 
 extern "C" {
-#include <X11/Intrinsic.h>      /* Xlib, Xutil, Xresource, Xfuncproto */
+//#include <X11/Intrinsic.h>      /* Xlib, Xutil, Xresource, Xfuncproto */
+#include <X11/Xlib.h>
+#include <X11/Xutil.h>
+#include <X11/Xresource.h>
+#include <X11/Xproto.h>
+#include <X11/region.h>
 }
 
 /*
