Index: xf86-input-evtouch-0.8.8/calibrate.sh
===================================================================
--- xf86-input-evtouch-0.8.8.orig/calibrate.sh	2009-01-14 14:14:10.000000000 +0100
+++ xf86-input-evtouch-0.8.8/calibrate.sh	2009-01-14 14:14:51.000000000 +0100
@@ -25,7 +25,7 @@
 FAILINFO="No evtouch capable device found, if you are sure you have \n\
 an evtouch capable touchscreen, please mail the \
 resulting file of the command \"lshal > evtouch_hal.out\" to the\n\
-ubuntu-mobile@lists.ubuntu.com mailing list, so support for your \
+bug url https://bugs.launchpad.net/bugs/317094, so support for your \
 device can be added."
 
 if [ -z "$UDI" ];then
