diff -u xf86-input-evtouch-0.8.8/evtouch.c xf86-input-evtouch-0.8.8/evtouch.c
--- xf86-input-evtouch-0.8.8/evtouch.c
+++ xf86-input-evtouch-0.8.8/evtouch.c
@@ -305,7 +305,7 @@
 	}
 
         if (pos_changed == 1) {
-#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) == 2
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) >= 2
                 ConvertProc(priv->local, 0, 2, 
                             priv->raw_x, priv->raw_y,
                             0, 0, 0, 0, 
@@ -369,7 +369,7 @@
                         priv->raw_y = priv->min_y;
         }
 
-#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) == 2
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) >= 2
         ConvertProc(priv->local, 0, 2,
                     priv->raw_x, priv->raw_y,
                     0, 0, 0, 0,
diff -u xf86-input-evtouch-0.8.8/debian/changelog xf86-input-evtouch-0.8.8/debian/changelog
--- xf86-input-evtouch-0.8.8.orig/ev_calibrate.c
+++ xf86-input-evtouch-0.8.8/ev_calibrate.c
@@ -218,7 +218,7 @@
         int cap_style = CapButt;        /* style of the line's edje and */
         int join_style = JoinBevel;        /*  joined lines.                */
 
-        int event_mask = ExposureMask | ButtonReleaseMask | PointerMotionMask | KeyPressMask;
+        int event_mask = ExposureMask | ButtonPressMask | ButtonReleaseMask | PointerMotionMask | KeyPressMask;
 
         int depth;
         int screen_num;
