http://www.koka-in.org/~kensyu/zaurus/diary/20030618.html

--- /tmp/Xlib.h	2006-12-01 14:40:27.000000000 +0100
+++ libX11-X11R7.1-1.0.1/include/X11/Xlib.h	2006-12-01 14:40:56.948254000 +0100
@@ -1067,7 +1067,7 @@
 typedef struct {		/* normal 16 bit characters are two bytes */
     unsigned char byte1;
     unsigned char byte2;
-} XChar2b;
+} __attribute__ ((packed)) XChar2b;
 
 typedef struct {
     XChar2b *chars;		/* two byte characters */
